// @adyen/api-library/lib/src/services/legalEntityManagement/index.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const businessLinesApi_1 = require("./businessLinesApi");
const documentsApi_1 = require("./documentsApi");
const hostedOnboardingApi_1 = require("./hostedOnboardingApi");
const legalEntitiesApi_1 = require("./legalEntitiesApi");
const pCIQuestionnairesApi_1 = require("./pCIQuestionnairesApi");
const termsOfServiceApi_1 = require("./termsOfServiceApi");
const transferInstrumentsApi_1 = require("./transferInstrumentsApi");
const service_1 = __importDefault(require("../../service"));
class LegalEntityManagementAPI extends service_1.default {
    constructor(client) {
        super(client);
    }
    get BusinessLinesApi() {
        return new businessLinesApi_1.BusinessLinesApi(this.client);
    }
    get DocumentsApi() {
        return new documentsApi_1.DocumentsApi(this.client);
    }
    get HostedOnboardingApi() {
        return new hostedOnboardingApi_1.HostedOnboardingApi(this.client);
    }
    get LegalEntitiesApi() {
        return new legalEntitiesApi_1.LegalEntitiesApi(this.client);
    }
    get PCIQuestionnairesApi() {
        return new pCIQuestionnairesApi_1.PCIQuestionnairesApi(this.client);
    }
    get TermsOfServiceApi() {
        return new termsOfServiceApi_1.TermsOfServiceApi(this.client);
    }
    get TransferInstrumentsApi() {
        return new transferInstrumentsApi_1.TransferInstrumentsApi(this.client);
    }
}
exports.default = LegalEntityManagementAPI;
