// @adyen/api-library/lib/src/typings/platformsNotifications/payoutScheduleResponse.js
"use strict";
/**
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen for Platforms: Notifications
 * The Notification API sends notifications to the endpoints specified in a given subscription. Subscriptions are managed through the Notification Configuration API. The API specifications listed here detail the format of each notification.  For more information, refer to our [documentation](https://docs.adyen.com/platforms/notifications).
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayoutScheduleResponse = void 0;
class PayoutScheduleResponse {
}
exports.PayoutScheduleResponse = PayoutScheduleResponse;
(function (PayoutScheduleResponse) {
    let ScheduleEnum;
    (function (ScheduleEnum) {
        ScheduleEnum[ScheduleEnum["BiweeklyOn1StAnd15ThAtMidnight"] = 'BIWEEKLY_ON_1ST_AND_15TH_AT_MIDNIGHT'] = "BiweeklyOn1StAnd15ThAtMidnight";
        ScheduleEnum[ScheduleEnum["BiweeklyOn1StAnd15ThAtNoon"] = 'BIWEEKLY_ON_1ST_AND_15TH_AT_NOON'] = "BiweeklyOn1StAnd15ThAtNoon";
        ScheduleEnum[ScheduleEnum["BiDailyAu"] = 'BI_DAILY_AU'] = "BiDailyAu";
        ScheduleEnum[ScheduleEnum["BiDailyEu"] = 'BI_DAILY_EU'] = "BiDailyEu";
        ScheduleEnum[ScheduleEnum["BiDailyUs"] = 'BI_DAILY_US'] = "BiDailyUs";
        ScheduleEnum[ScheduleEnum["Daily"] = 'DAILY'] = "Daily";
        ScheduleEnum[ScheduleEnum["Daily6Pm"] = 'DAILY_6PM'] = "Daily6Pm";
        ScheduleEnum[ScheduleEnum["DailyAu"] = 'DAILY_AU'] = "DailyAu";
        ScheduleEnum[ScheduleEnum["DailyEu"] = 'DAILY_EU'] = "DailyEu";
        ScheduleEnum[ScheduleEnum["DailySg"] = 'DAILY_SG'] = "DailySg";
        ScheduleEnum[ScheduleEnum["DailyUs"] = 'DAILY_US'] = "DailyUs";
        ScheduleEnum[ScheduleEnum["Default"] = 'DEFAULT'] = "Default";
        ScheduleEnum[ScheduleEnum["Every6HoursFromMidnight"] = 'EVERY_6_HOURS_FROM_MIDNIGHT'] = "Every6HoursFromMidnight";
        ScheduleEnum[ScheduleEnum["Hold"] = 'HOLD'] = "Hold";
        ScheduleEnum[ScheduleEnum["Monthly"] = 'MONTHLY'] = "Monthly";
        ScheduleEnum[ScheduleEnum["MonthlyOn15ThAtMidnight"] = 'MONTHLY_ON_15TH_AT_MIDNIGHT'] = "MonthlyOn15ThAtMidnight";
        ScheduleEnum[ScheduleEnum["Weekly"] = 'WEEKLY'] = "Weekly";
        ScheduleEnum[ScheduleEnum["WeeklyMonToFriAu"] = 'WEEKLY_MON_TO_FRI_AU'] = "WeeklyMonToFriAu";
        ScheduleEnum[ScheduleEnum["WeeklyMonToFriEu"] = 'WEEKLY_MON_TO_FRI_EU'] = "WeeklyMonToFriEu";
        ScheduleEnum[ScheduleEnum["WeeklyMonToFriUs"] = 'WEEKLY_MON_TO_FRI_US'] = "WeeklyMonToFriUs";
        ScheduleEnum[ScheduleEnum["WeeklyOnTueFriMidnight"] = 'WEEKLY_ON_TUE_FRI_MIDNIGHT'] = "WeeklyOnTueFriMidnight";
        ScheduleEnum[ScheduleEnum["Yearly"] = 'YEARLY'] = "Yearly";
    })(ScheduleEnum = PayoutScheduleResponse.ScheduleEnum || (PayoutScheduleResponse.ScheduleEnum = {}));
})(PayoutScheduleResponse = exports.PayoutScheduleResponse || (exports.PayoutScheduleResponse = {}));
