// @adyen/api-library/lib/src/typings/platformsNotificationConfiguration/notificationEventConfiguration.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationEventConfiguration = void 0;
class NotificationEventConfiguration {
    static getAttributeTypeMap() {
        return NotificationEventConfiguration.attributeTypeMap;
    }
}
exports.NotificationEventConfiguration = NotificationEventConfiguration;
NotificationEventConfiguration.discriminator = undefined;
NotificationEventConfiguration.attributeTypeMap = [
    {
        "name": "eventType",
        "baseName": "eventType",
        "type": "NotificationEventConfiguration.EventTypeEnum"
    },
    {
        "name": "includeMode",
        "baseName": "includeMode",
        "type": "NotificationEventConfiguration.IncludeModeEnum"
    }
];
(function (NotificationEventConfiguration) {
    let EventTypeEnum;
    (function (EventTypeEnum) {
        EventTypeEnum["AccountClosed"] = "ACCOUNT_CLOSED";
        EventTypeEnum["AccountCreated"] = "ACCOUNT_CREATED";
        EventTypeEnum["AccountFundsBelowThreshold"] = "ACCOUNT_FUNDS_BELOW_THRESHOLD";
        EventTypeEnum["AccountHolderCreated"] = "ACCOUNT_HOLDER_CREATED";
        EventTypeEnum["AccountHolderLimitReached"] = "ACCOUNT_HOLDER_LIMIT_REACHED";
        EventTypeEnum["AccountHolderMigrated"] = "ACCOUNT_HOLDER_MIGRATED";
        EventTypeEnum["AccountHolderPayout"] = "ACCOUNT_HOLDER_PAYOUT";
        EventTypeEnum["AccountHolderStatusChange"] = "ACCOUNT_HOLDER_STATUS_CHANGE";
        EventTypeEnum["AccountHolderStoreStatusChange"] = "ACCOUNT_HOLDER_STORE_STATUS_CHANGE";
        EventTypeEnum["AccountHolderUpcomingDeadline"] = "ACCOUNT_HOLDER_UPCOMING_DEADLINE";
        EventTypeEnum["AccountHolderUpdated"] = "ACCOUNT_HOLDER_UPDATED";
        EventTypeEnum["AccountHolderVerification"] = "ACCOUNT_HOLDER_VERIFICATION";
        EventTypeEnum["AccountUpdated"] = "ACCOUNT_UPDATED";
        EventTypeEnum["BeneficiarySetup"] = "BENEFICIARY_SETUP";
        EventTypeEnum["CompensateNegativeBalance"] = "COMPENSATE_NEGATIVE_BALANCE";
        EventTypeEnum["DirectDebitInitiated"] = "DIRECT_DEBIT_INITIATED";
        EventTypeEnum["FundsMigrated"] = "FUNDS_MIGRATED";
        EventTypeEnum["PaymentFailure"] = "PAYMENT_FAILURE";
        EventTypeEnum["PendingCredit"] = "PENDING_CREDIT";
        EventTypeEnum["RefundFundsTransfer"] = "REFUND_FUNDS_TRANSFER";
        EventTypeEnum["ReportAvailable"] = "REPORT_AVAILABLE";
        EventTypeEnum["ScheduledRefunds"] = "SCHEDULED_REFUNDS";
        EventTypeEnum["ScoreSignalTriggered"] = "SCORE_SIGNAL_TRIGGERED";
        EventTypeEnum["TransferFunds"] = "TRANSFER_FUNDS";
        EventTypeEnum["TransferNotPaidoutTransfers"] = "TRANSFER_NOT_PAIDOUT_TRANSFERS";
    })(EventTypeEnum = NotificationEventConfiguration.EventTypeEnum || (NotificationEventConfiguration.EventTypeEnum = {}));
    let IncludeModeEnum;
    (function (IncludeModeEnum) {
        IncludeModeEnum["Exclude"] = "EXCLUDE";
        IncludeModeEnum["Include"] = "INCLUDE";
    })(IncludeModeEnum = NotificationEventConfiguration.IncludeModeEnum || (NotificationEventConfiguration.IncludeModeEnum = {}));
})(NotificationEventConfiguration = exports.NotificationEventConfiguration || (exports.NotificationEventConfiguration = {}));
