// @adyen/api-library/lib/src/typings/platformsNotificationConfiguration/testNotificationConfigurationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TestNotificationConfigurationRequest = void 0;
class TestNotificationConfigurationRequest {
    static getAttributeTypeMap() {
        return TestNotificationConfigurationRequest.attributeTypeMap;
    }
}
exports.TestNotificationConfigurationRequest = TestNotificationConfigurationRequest;
TestNotificationConfigurationRequest.discriminator = undefined;
TestNotificationConfigurationRequest.attributeTypeMap = [
    {
        "name": "eventTypes",
        "baseName": "eventTypes",
        "type": "Array<TestNotificationConfigurationRequest.EventTypesEnum>"
    },
    {
        "name": "notificationId",
        "baseName": "notificationId",
        "type": "number"
    }
];
(function (TestNotificationConfigurationRequest) {
    let EventTypesEnum;
    (function (EventTypesEnum) {
        EventTypesEnum["AccountClosed"] = "ACCOUNT_CLOSED";
        EventTypesEnum["AccountCreated"] = "ACCOUNT_CREATED";
        EventTypesEnum["AccountFundsBelowThreshold"] = "ACCOUNT_FUNDS_BELOW_THRESHOLD";
        EventTypesEnum["AccountHolderCreated"] = "ACCOUNT_HOLDER_CREATED";
        EventTypesEnum["AccountHolderLimitReached"] = "ACCOUNT_HOLDER_LIMIT_REACHED";
        EventTypesEnum["AccountHolderMigrated"] = "ACCOUNT_HOLDER_MIGRATED";
        EventTypesEnum["AccountHolderPayout"] = "ACCOUNT_HOLDER_PAYOUT";
        EventTypesEnum["AccountHolderStatusChange"] = "ACCOUNT_HOLDER_STATUS_CHANGE";
        EventTypesEnum["AccountHolderStoreStatusChange"] = "ACCOUNT_HOLDER_STORE_STATUS_CHANGE";
        EventTypesEnum["AccountHolderUpcomingDeadline"] = "ACCOUNT_HOLDER_UPCOMING_DEADLINE";
        EventTypesEnum["AccountHolderUpdated"] = "ACCOUNT_HOLDER_UPDATED";
        EventTypesEnum["AccountHolderVerification"] = "ACCOUNT_HOLDER_VERIFICATION";
        EventTypesEnum["AccountUpdated"] = "ACCOUNT_UPDATED";
        EventTypesEnum["BeneficiarySetup"] = "BENEFICIARY_SETUP";
        EventTypesEnum["CompensateNegativeBalance"] = "COMPENSATE_NEGATIVE_BALANCE";
        EventTypesEnum["DirectDebitInitiated"] = "DIRECT_DEBIT_INITIATED";
        EventTypesEnum["FundsMigrated"] = "FUNDS_MIGRATED";
        EventTypesEnum["PaymentFailure"] = "PAYMENT_FAILURE";
        EventTypesEnum["PendingCredit"] = "PENDING_CREDIT";
        EventTypesEnum["RefundFundsTransfer"] = "REFUND_FUNDS_TRANSFER";
        EventTypesEnum["ReportAvailable"] = "REPORT_AVAILABLE";
        EventTypesEnum["ScheduledRefunds"] = "SCHEDULED_REFUNDS";
        EventTypesEnum["ScoreSignalTriggered"] = "SCORE_SIGNAL_TRIGGERED";
        EventTypesEnum["TransferFunds"] = "TRANSFER_FUNDS";
        EventTypesEnum["TransferNotPaidoutTransfers"] = "TRANSFER_NOT_PAIDOUT_TRANSFERS";
    })(EventTypesEnum = TestNotificationConfigurationRequest.EventTypesEnum || (TestNotificationConfigurationRequest.EventTypesEnum = {}));
})(TestNotificationConfigurationRequest = exports.TestNotificationConfigurationRequest || (exports.TestNotificationConfigurationRequest = {}));
