// @adyen/api-library/lib/src/typings/platformsNotifications/legalArrangementDetail.js
"use strict";
/**
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen for Platforms: Notifications
 * The Notification API sends notifications to the endpoints specified in a given subscription. Subscriptions are managed through the Notification Configuration API. The API specifications listed here detail the format of each notification.  For more information, refer to our [documentation](https://docs.adyen.com/platforms/notifications).
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalArrangementDetail = void 0;
class LegalArrangementDetail {
}
exports.LegalArrangementDetail = LegalArrangementDetail;
(function (LegalArrangementDetail) {
    let LegalFormEnum;
    (function (LegalFormEnum) {
        LegalFormEnum[LegalFormEnum["CashManagementTrust"] = 'CashManagementTrust'] = "CashManagementTrust";
        LegalFormEnum[LegalFormEnum["CorporateUnitTrust"] = 'CorporateUnitTrust'] = "CorporateUnitTrust";
        LegalFormEnum[LegalFormEnum["DeceasedEstate"] = 'DeceasedEstate'] = "DeceasedEstate";
        LegalFormEnum[LegalFormEnum["DiscretionaryInvestmentTrust"] = 'DiscretionaryInvestmentTrust'] = "DiscretionaryInvestmentTrust";
        LegalFormEnum[LegalFormEnum["DiscretionaryServicesManagementTrust"] = 'DiscretionaryServicesManagementTrust'] = "DiscretionaryServicesManagementTrust";
        LegalFormEnum[LegalFormEnum["DiscretionaryTradingTrust"] = 'DiscretionaryTradingTrust'] = "DiscretionaryTradingTrust";
        LegalFormEnum[LegalFormEnum["FirstHomeSaverAccountsTrust"] = 'FirstHomeSaverAccountsTrust'] = "FirstHomeSaverAccountsTrust";
        LegalFormEnum[LegalFormEnum["FixedTrust"] = 'FixedTrust'] = "FixedTrust";
        LegalFormEnum[LegalFormEnum["FixedUnitTrust"] = 'FixedUnitTrust'] = "FixedUnitTrust";
        LegalFormEnum[LegalFormEnum["HybridTrust"] = 'HybridTrust'] = "HybridTrust";
        LegalFormEnum[LegalFormEnum["ListedPublicUnitTrust"] = 'ListedPublicUnitTrust'] = "ListedPublicUnitTrust";
        LegalFormEnum[LegalFormEnum["OtherTrust"] = 'OtherTrust'] = "OtherTrust";
        LegalFormEnum[LegalFormEnum["PooledSuperannuationTrust"] = 'PooledSuperannuationTrust'] = "PooledSuperannuationTrust";
        LegalFormEnum[LegalFormEnum["PublicTradingTrust"] = 'PublicTradingTrust'] = "PublicTradingTrust";
        LegalFormEnum[LegalFormEnum["UnlistedPublicUnitTrust"] = 'UnlistedPublicUnitTrust'] = "UnlistedPublicUnitTrust";
        LegalFormEnum[LegalFormEnum["LimitedPartnership"] = 'LimitedPartnership'] = "LimitedPartnership";
        LegalFormEnum[LegalFormEnum["FamilyPartnership"] = 'FamilyPartnership'] = "FamilyPartnership";
        LegalFormEnum[LegalFormEnum["OtherPartnership"] = 'OtherPartnership'] = "OtherPartnership";
    })(LegalFormEnum = LegalArrangementDetail.LegalFormEnum || (LegalArrangementDetail.LegalFormEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Association"] = 'Association'] = "Association";
        TypeEnum[TypeEnum["Partnership"] = 'Partnership'] = "Partnership";
        TypeEnum[TypeEnum["SoleProprietorship"] = 'SoleProprietorship'] = "SoleProprietorship";
        TypeEnum[TypeEnum["Trust"] = 'Trust'] = "Trust";
    })(TypeEnum = LegalArrangementDetail.TypeEnum || (LegalArrangementDetail.TypeEnum = {}));
})(LegalArrangementDetail = exports.LegalArrangementDetail || (exports.LegalArrangementDetail = {}));
