// @adyen/api-library/lib/src/typings/platformsNotifications/viasPhoneNumber.js
"use strict";
/**
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen for Platforms: Notifications
 * The Notification API sends notifications to the endpoints specified in a given subscription. Subscriptions are managed through the Notification Configuration API. The API specifications listed here detail the format of each notification.  For more information, refer to our [documentation](https://docs.adyen.com/platforms/notifications).
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViasPhoneNumber = void 0;
class ViasPhoneNumber {
}
exports.ViasPhoneNumber = ViasPhoneNumber;
(function (ViasPhoneNumber) {
    let PhoneTypeEnum;
    (function (PhoneTypeEnum) {
        PhoneTypeEnum[PhoneTypeEnum["Fax"] = 'Fax'] = "Fax";
        PhoneTypeEnum[PhoneTypeEnum["Landline"] = 'Landline'] = "Landline";
        PhoneTypeEnum[PhoneTypeEnum["Mobile"] = 'Mobile'] = "Mobile";
        PhoneTypeEnum[PhoneTypeEnum["Sip"] = 'SIP'] = "Sip";
    })(PhoneTypeEnum = ViasPhoneNumber.PhoneTypeEnum || (ViasPhoneNumber.PhoneTypeEnum = {}));
})(ViasPhoneNumber = exports.ViasPhoneNumber || (exports.ViasPhoneNumber = {}));
