// @adyen/api-library/lib/src/typings/storedValue/storedValueBalanceCheckResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueBalanceCheckResponse = void 0;
class StoredValueBalanceCheckResponse {
    static getAttributeTypeMap() {
        return StoredValueBalanceCheckResponse.attributeTypeMap;
    }
}
exports.StoredValueBalanceCheckResponse = StoredValueBalanceCheckResponse;
StoredValueBalanceCheckResponse.discriminator = undefined;
StoredValueBalanceCheckResponse.attributeTypeMap = [
    {
        "name": "currentBalance",
        "baseName": "currentBalance",
        "type": "Amount"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "StoredValueBalanceCheckResponse.ResultCodeEnum"
    },
    {
        "name": "thirdPartyRefusalReason",
        "baseName": "thirdPartyRefusalReason",
        "type": "string"
    }
];
(function (StoredValueBalanceCheckResponse) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["Success"] = "Success";
        ResultCodeEnum["Refused"] = "Refused";
        ResultCodeEnum["Error"] = "Error";
        ResultCodeEnum["NotEnoughBalance"] = "NotEnoughBalance";
    })(ResultCodeEnum = StoredValueBalanceCheckResponse.ResultCodeEnum || (StoredValueBalanceCheckResponse.ResultCodeEnum = {}));
})(StoredValueBalanceCheckResponse = exports.StoredValueBalanceCheckResponse || (exports.StoredValueBalanceCheckResponse = {}));
