// @adyen/api-library/lib/src/typings/storedValue/storedValueBalanceCheckRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueBalanceCheckRequest = void 0;
class StoredValueBalanceCheckRequest {
    static getAttributeTypeMap() {
        return StoredValueBalanceCheckRequest.attributeTypeMap;
    }
}
exports.StoredValueBalanceCheckRequest = StoredValueBalanceCheckRequest;
StoredValueBalanceCheckRequest.discriminator = undefined;
StoredValueBalanceCheckRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "StoredValueBalanceCheckRequest.ShopperInteractionEnum"
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string"
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string"
    }
];
(function (StoredValueBalanceCheckRequest) {
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = StoredValueBalanceCheckRequest.ShopperInteractionEnum || (StoredValueBalanceCheckRequest.ShopperInteractionEnum = {}));
})(StoredValueBalanceCheckRequest = exports.StoredValueBalanceCheckRequest || (exports.StoredValueBalanceCheckRequest = {}));
