// @adyen/api-library/lib/src/typings/storedValue/storedValueStatusChangeResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueStatusChangeResponse = void 0;
class StoredValueStatusChangeResponse {
    static getAttributeTypeMap() {
        return StoredValueStatusChangeResponse.attributeTypeMap;
    }
}
exports.StoredValueStatusChangeResponse = StoredValueStatusChangeResponse;
StoredValueStatusChangeResponse.discriminator = undefined;
StoredValueStatusChangeResponse.attributeTypeMap = [
    {
        "name": "authCode",
        "baseName": "authCode",
        "type": "string"
    },
    {
        "name": "currentBalance",
        "baseName": "currentBalance",
        "type": "Amount"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "StoredValueStatusChangeResponse.ResultCodeEnum"
    },
    {
        "name": "thirdPartyRefusalReason",
        "baseName": "thirdPartyRefusalReason",
        "type": "string"
    }
];
(function (StoredValueStatusChangeResponse) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["Success"] = "Success";
        ResultCodeEnum["Refused"] = "Refused";
        ResultCodeEnum["Error"] = "Error";
        ResultCodeEnum["NotEnoughBalance"] = "NotEnoughBalance";
    })(ResultCodeEnum = StoredValueStatusChangeResponse.ResultCodeEnum || (StoredValueStatusChangeResponse.ResultCodeEnum = {}));
})(StoredValueStatusChangeResponse = exports.StoredValueStatusChangeResponse || (exports.StoredValueStatusChangeResponse = {}));
