// @adyen/api-library/lib/src/typings/platformsFund/accountHolderTransactionListRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountHolderTransactionListRequest = void 0;
class AccountHolderTransactionListRequest {
    static getAttributeTypeMap() {
        return AccountHolderTransactionListRequest.attributeTypeMap;
    }
}
exports.AccountHolderTransactionListRequest = AccountHolderTransactionListRequest;
AccountHolderTransactionListRequest.discriminator = undefined;
AccountHolderTransactionListRequest.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "transactionListsPerAccount",
        "baseName": "transactionListsPerAccount",
        "type": "Array<TransactionListForAccount>"
    },
    {
        "name": "transactionStatuses",
        "baseName": "transactionStatuses",
        "type": "Array<AccountHolderTransactionListRequest.TransactionStatusesEnum>"
    }
];
(function (AccountHolderTransactionListRequest) {
    let TransactionStatusesEnum;
    (function (TransactionStatusesEnum) {
        TransactionStatusesEnum["BalanceNotPaidOutTransfer"] = "BalanceNotPaidOutTransfer";
        TransactionStatusesEnum["BalancePlatformSweep"] = "BalancePlatformSweep";
        TransactionStatusesEnum["BalancePlatformSweepReturned"] = "BalancePlatformSweepReturned";
        TransactionStatusesEnum["Chargeback"] = "Chargeback";
        TransactionStatusesEnum["ChargebackCorrection"] = "ChargebackCorrection";
        TransactionStatusesEnum["ChargebackCorrectionReceived"] = "ChargebackCorrectionReceived";
        TransactionStatusesEnum["ChargebackReceived"] = "ChargebackReceived";
        TransactionStatusesEnum["ChargebackReversed"] = "ChargebackReversed";
        TransactionStatusesEnum["ChargebackReversedCorrection"] = "ChargebackReversedCorrection";
        TransactionStatusesEnum["ChargebackReversedCorrectionReceived"] = "ChargebackReversedCorrectionReceived";
        TransactionStatusesEnum["ChargebackReversedReceived"] = "ChargebackReversedReceived";
        TransactionStatusesEnum["Converted"] = "Converted";
        TransactionStatusesEnum["CreditClosed"] = "CreditClosed";
        TransactionStatusesEnum["CreditFailed"] = "CreditFailed";
        TransactionStatusesEnum["CreditReversed"] = "CreditReversed";
        TransactionStatusesEnum["CreditReversedReceived"] = "CreditReversedReceived";
        TransactionStatusesEnum["CreditSuspended"] = "CreditSuspended";
        TransactionStatusesEnum["Credited"] = "Credited";
        TransactionStatusesEnum["DebitFailed"] = "DebitFailed";
        TransactionStatusesEnum["DebitReversedReceived"] = "DebitReversedReceived";
        TransactionStatusesEnum["Debited"] = "Debited";
        TransactionStatusesEnum["DebitedReversed"] = "DebitedReversed";
        TransactionStatusesEnum["DepositCorrectionCredited"] = "DepositCorrectionCredited";
        TransactionStatusesEnum["DepositCorrectionDebited"] = "DepositCorrectionDebited";
        TransactionStatusesEnum["Fee"] = "Fee";
        TransactionStatusesEnum["FundTransfer"] = "FundTransfer";
        TransactionStatusesEnum["FundTransferReversed"] = "FundTransferReversed";
        TransactionStatusesEnum["InvoiceDeductionCredited"] = "InvoiceDeductionCredited";
        TransactionStatusesEnum["InvoiceDeductionDebited"] = "InvoiceDeductionDebited";
        TransactionStatusesEnum["ManualCorrected"] = "ManualCorrected";
        TransactionStatusesEnum["ManualCorrectionCredited"] = "ManualCorrectionCredited";
        TransactionStatusesEnum["ManualCorrectionDebited"] = "ManualCorrectionDebited";
        TransactionStatusesEnum["MerchantPayin"] = "MerchantPayin";
        TransactionStatusesEnum["MerchantPayinReversed"] = "MerchantPayinReversed";
        TransactionStatusesEnum["Payout"] = "Payout";
        TransactionStatusesEnum["PayoutReversed"] = "PayoutReversed";
        TransactionStatusesEnum["PendingCredit"] = "PendingCredit";
        TransactionStatusesEnum["PendingDebit"] = "PendingDebit";
        TransactionStatusesEnum["PendingFundTransfer"] = "PendingFundTransfer";
        TransactionStatusesEnum["ReCredited"] = "ReCredited";
        TransactionStatusesEnum["ReCreditedReceived"] = "ReCreditedReceived";
        TransactionStatusesEnum["SecondChargeback"] = "SecondChargeback";
        TransactionStatusesEnum["SecondChargebackCorrection"] = "SecondChargebackCorrection";
        TransactionStatusesEnum["SecondChargebackCorrectionReceived"] = "SecondChargebackCorrectionReceived";
        TransactionStatusesEnum["SecondChargebackReceived"] = "SecondChargebackReceived";
    })(TransactionStatusesEnum = AccountHolderTransactionListRequest.TransactionStatusesEnum || (AccountHolderTransactionListRequest.TransactionStatusesEnum = {}));
})(AccountHolderTransactionListRequest = exports.AccountHolderTransactionListRequest || (exports.AccountHolderTransactionListRequest = {}));
