// @adyen/api-library/lib/src/typings/platformsFund/payoutAccountHolderResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayoutAccountHolderResponse = void 0;
class PayoutAccountHolderResponse {
    static getAttributeTypeMap() {
        return PayoutAccountHolderResponse.attributeTypeMap;
    }
}
exports.PayoutAccountHolderResponse = PayoutAccountHolderResponse;
PayoutAccountHolderResponse.discriminator = undefined;
PayoutAccountHolderResponse.attributeTypeMap = [
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "invalidFields",
        "baseName": "invalidFields",
        "type": "Array<ErrorFieldType>"
    },
    {
        "name": "merchantReference",
        "baseName": "merchantReference",
        "type": "string"
    },
    {
        "name": "payoutSpeed",
        "baseName": "payoutSpeed",
        "type": "PayoutAccountHolderResponse.PayoutSpeedEnum"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "string"
    }
];
(function (PayoutAccountHolderResponse) {
    let PayoutSpeedEnum;
    (function (PayoutSpeedEnum) {
        PayoutSpeedEnum["Instant"] = "INSTANT";
        PayoutSpeedEnum["SameDay"] = "SAME_DAY";
        PayoutSpeedEnum["Standard"] = "STANDARD";
    })(PayoutSpeedEnum = PayoutAccountHolderResponse.PayoutSpeedEnum || (PayoutAccountHolderResponse.PayoutSpeedEnum = {}));
})(PayoutAccountHolderResponse = exports.PayoutAccountHolderResponse || (exports.PayoutAccountHolderResponse = {}));
