// @adyen/api-library/lib/src/typings/platformsNotifications/fieldType.js
"use strict";
/**
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen for Platforms: Notifications
 * The Notification API sends notifications to the endpoints specified in a given subscription. Subscriptions are managed through the Notification Configuration API. The API specifications listed here detail the format of each notification.  For more information, refer to our [documentation](https://docs.adyen.com/platforms/notifications).
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FieldType = void 0;
class FieldType {
}
exports.FieldType = FieldType;
(function (FieldType) {
    let FieldNameEnum;
    (function (FieldNameEnum) {
        FieldNameEnum[FieldNameEnum["AccountCode"] = 'accountCode'] = "AccountCode";
        FieldNameEnum[FieldNameEnum["AccountHolderCode"] = 'accountHolderCode'] = "AccountHolderCode";
        FieldNameEnum[FieldNameEnum["AccountHolderDetails"] = 'accountHolderDetails'] = "AccountHolderDetails";
        FieldNameEnum[FieldNameEnum["AccountNumber"] = 'accountNumber'] = "AccountNumber";
        FieldNameEnum[FieldNameEnum["AccountStateType"] = 'accountStateType'] = "AccountStateType";
        FieldNameEnum[FieldNameEnum["AccountStatus"] = 'accountStatus'] = "AccountStatus";
        FieldNameEnum[FieldNameEnum["AccountType"] = 'accountType'] = "AccountType";
        FieldNameEnum[FieldNameEnum["Address"] = 'address'] = "Address";
        FieldNameEnum[FieldNameEnum["BalanceAccount"] = 'balanceAccount'] = "BalanceAccount";
        FieldNameEnum[FieldNameEnum["BalanceAccountActive"] = 'balanceAccountActive'] = "BalanceAccountActive";
        FieldNameEnum[FieldNameEnum["BalanceAccountCode"] = 'balanceAccountCode'] = "BalanceAccountCode";
        FieldNameEnum[FieldNameEnum["BalanceAccountId"] = 'balanceAccountId'] = "BalanceAccountId";
        FieldNameEnum[FieldNameEnum["BankAccount"] = 'bankAccount'] = "BankAccount";
        FieldNameEnum[FieldNameEnum["BankAccountCode"] = 'bankAccountCode'] = "BankAccountCode";
        FieldNameEnum[FieldNameEnum["BankAccountName"] = 'bankAccountName'] = "BankAccountName";
        FieldNameEnum[FieldNameEnum["BankAccountUuid"] = 'bankAccountUUID'] = "BankAccountUuid";
        FieldNameEnum[FieldNameEnum["BankBicSwift"] = 'bankBicSwift'] = "BankBicSwift";
        FieldNameEnum[FieldNameEnum["BankCity"] = 'bankCity'] = "BankCity";
        FieldNameEnum[FieldNameEnum["BankCode"] = 'bankCode'] = "BankCode";
        FieldNameEnum[FieldNameEnum["BankName"] = 'bankName'] = "BankName";
        FieldNameEnum[FieldNameEnum["BankStatement"] = 'bankStatement'] = "BankStatement";
        FieldNameEnum[FieldNameEnum["BranchCode"] = 'branchCode'] = "BranchCode";
        FieldNameEnum[FieldNameEnum["BusinessContact"] = 'businessContact'] = "BusinessContact";
        FieldNameEnum[FieldNameEnum["CardToken"] = 'cardToken'] = "CardToken";
        FieldNameEnum[FieldNameEnum["CheckCode"] = 'checkCode'] = "CheckCode";
        FieldNameEnum[FieldNameEnum["City"] = 'city'] = "City";
        FieldNameEnum[FieldNameEnum["CompanyRegistration"] = 'companyRegistration'] = "CompanyRegistration";
        FieldNameEnum[FieldNameEnum["ConstitutionalDocument"] = 'constitutionalDocument'] = "ConstitutionalDocument";
        FieldNameEnum[FieldNameEnum["Country"] = 'country'] = "Country";
        FieldNameEnum[FieldNameEnum["CountryCode"] = 'countryCode'] = "CountryCode";
        FieldNameEnum[FieldNameEnum["Currency"] = 'currency'] = "Currency";
        FieldNameEnum[FieldNameEnum["CurrencyCode"] = 'currencyCode'] = "CurrencyCode";
        FieldNameEnum[FieldNameEnum["DateOfBirth"] = 'dateOfBirth'] = "DateOfBirth";
        FieldNameEnum[FieldNameEnum["Description"] = 'description'] = "Description";
        FieldNameEnum[FieldNameEnum["DestinationAccountCode"] = 'destinationAccountCode'] = "DestinationAccountCode";
        FieldNameEnum[FieldNameEnum["Document"] = 'document'] = "Document";
        FieldNameEnum[FieldNameEnum["DocumentExpirationDate"] = 'documentExpirationDate'] = "DocumentExpirationDate";
        FieldNameEnum[FieldNameEnum["DocumentIssuerCountry"] = 'documentIssuerCountry'] = "DocumentIssuerCountry";
        FieldNameEnum[FieldNameEnum["DocumentIssuerState"] = 'documentIssuerState'] = "DocumentIssuerState";
        FieldNameEnum[FieldNameEnum["DocumentName"] = 'documentName'] = "DocumentName";
        FieldNameEnum[FieldNameEnum["DocumentNumber"] = 'documentNumber'] = "DocumentNumber";
        FieldNameEnum[FieldNameEnum["DocumentType"] = 'documentType'] = "DocumentType";
        FieldNameEnum[FieldNameEnum["DoingBusinessAs"] = 'doingBusinessAs'] = "DoingBusinessAs";
        FieldNameEnum[FieldNameEnum["DrivingLicence"] = 'drivingLicence'] = "DrivingLicence";
        FieldNameEnum[FieldNameEnum["DrivingLicenceBack"] = 'drivingLicenceBack'] = "DrivingLicenceBack";
        FieldNameEnum[FieldNameEnum["DrivingLicense"] = 'drivingLicense'] = "DrivingLicense";
        FieldNameEnum[FieldNameEnum["Email"] = 'email'] = "Email";
        FieldNameEnum[FieldNameEnum["FirstName"] = 'firstName'] = "FirstName";
        FieldNameEnum[FieldNameEnum["FormType"] = 'formType'] = "FormType";
        FieldNameEnum[FieldNameEnum["FullPhoneNumber"] = 'fullPhoneNumber'] = "FullPhoneNumber";
        FieldNameEnum[FieldNameEnum["Gender"] = 'gender'] = "Gender";
        FieldNameEnum[FieldNameEnum["HopWebserviceUser"] = 'hopWebserviceUser'] = "HopWebserviceUser";
        FieldNameEnum[FieldNameEnum["HouseNumberOrName"] = 'houseNumberOrName'] = "HouseNumberOrName";
        FieldNameEnum[FieldNameEnum["Iban"] = 'iban'] = "Iban";
        FieldNameEnum[FieldNameEnum["IdCard"] = 'idCard'] = "IdCard";
        FieldNameEnum[FieldNameEnum["IdCardBack"] = 'idCardBack'] = "IdCardBack";
        FieldNameEnum[FieldNameEnum["IdCardFront"] = 'idCardFront'] = "IdCardFront";
        FieldNameEnum[FieldNameEnum["IdNumber"] = 'idNumber'] = "IdNumber";
        FieldNameEnum[FieldNameEnum["IdentityDocument"] = 'identityDocument'] = "IdentityDocument";
        FieldNameEnum[FieldNameEnum["IndividualDetails"] = 'individualDetails'] = "IndividualDetails";
        FieldNameEnum[FieldNameEnum["JobTitle"] = 'jobTitle'] = "JobTitle";
        FieldNameEnum[FieldNameEnum["LastName"] = 'lastName'] = "LastName";
        FieldNameEnum[FieldNameEnum["LastReviewDate"] = 'lastReviewDate'] = "LastReviewDate";
        FieldNameEnum[FieldNameEnum["LegalArrangement"] = 'legalArrangement'] = "LegalArrangement";
        FieldNameEnum[FieldNameEnum["LegalArrangementCode"] = 'legalArrangementCode'] = "LegalArrangementCode";
        FieldNameEnum[FieldNameEnum["LegalArrangementEntity"] = 'legalArrangementEntity'] = "LegalArrangementEntity";
        FieldNameEnum[FieldNameEnum["LegalArrangementEntityCode"] = 'legalArrangementEntityCode'] = "LegalArrangementEntityCode";
        FieldNameEnum[FieldNameEnum["LegalArrangementLegalForm"] = 'legalArrangementLegalForm'] = "LegalArrangementLegalForm";
        FieldNameEnum[FieldNameEnum["LegalArrangementMember"] = 'legalArrangementMember'] = "LegalArrangementMember";
        FieldNameEnum[FieldNameEnum["LegalArrangementMembers"] = 'legalArrangementMembers'] = "LegalArrangementMembers";
        FieldNameEnum[FieldNameEnum["LegalArrangementName"] = 'legalArrangementName'] = "LegalArrangementName";
        FieldNameEnum[FieldNameEnum["LegalArrangementReference"] = 'legalArrangementReference'] = "LegalArrangementReference";
        FieldNameEnum[FieldNameEnum["LegalArrangementRegistrationNumber"] = 'legalArrangementRegistrationNumber'] = "LegalArrangementRegistrationNumber";
        FieldNameEnum[FieldNameEnum["LegalArrangementTaxNumber"] = 'legalArrangementTaxNumber'] = "LegalArrangementTaxNumber";
        FieldNameEnum[FieldNameEnum["LegalArrangementType"] = 'legalArrangementType'] = "LegalArrangementType";
        FieldNameEnum[FieldNameEnum["LegalBusinessName"] = 'legalBusinessName'] = "LegalBusinessName";
        FieldNameEnum[FieldNameEnum["LegalEntity"] = 'legalEntity'] = "LegalEntity";
        FieldNameEnum[FieldNameEnum["LegalEntityType"] = 'legalEntityType'] = "LegalEntityType";
        FieldNameEnum[FieldNameEnum["MerchantAccount"] = 'merchantAccount'] = "MerchantAccount";
        FieldNameEnum[FieldNameEnum["MerchantCategoryCode"] = 'merchantCategoryCode'] = "MerchantCategoryCode";
        FieldNameEnum[FieldNameEnum["MerchantReference"] = 'merchantReference'] = "MerchantReference";
        FieldNameEnum[FieldNameEnum["MicroDeposit"] = 'microDeposit'] = "MicroDeposit";
        FieldNameEnum[FieldNameEnum["Name"] = 'name'] = "Name";
        FieldNameEnum[FieldNameEnum["Nationality"] = 'nationality'] = "Nationality";
        FieldNameEnum[FieldNameEnum["OriginalReference"] = 'originalReference'] = "OriginalReference";
        FieldNameEnum[FieldNameEnum["OwnerCity"] = 'ownerCity'] = "OwnerCity";
        FieldNameEnum[FieldNameEnum["OwnerCountryCode"] = 'ownerCountryCode'] = "OwnerCountryCode";
        FieldNameEnum[FieldNameEnum["OwnerHouseNumberOrName"] = 'ownerHouseNumberOrName'] = "OwnerHouseNumberOrName";
        FieldNameEnum[FieldNameEnum["OwnerName"] = 'ownerName'] = "OwnerName";
        FieldNameEnum[FieldNameEnum["OwnerPostalCode"] = 'ownerPostalCode'] = "OwnerPostalCode";
        FieldNameEnum[FieldNameEnum["OwnerState"] = 'ownerState'] = "OwnerState";
        FieldNameEnum[FieldNameEnum["OwnerStreet"] = 'ownerStreet'] = "OwnerStreet";
        FieldNameEnum[FieldNameEnum["Passport"] = 'passport'] = "Passport";
        FieldNameEnum[FieldNameEnum["PassportNumber"] = 'passportNumber'] = "PassportNumber";
        FieldNameEnum[FieldNameEnum["PayoutMethodCode"] = 'payoutMethodCode'] = "PayoutMethodCode";
        FieldNameEnum[FieldNameEnum["PayoutSchedule"] = 'payoutSchedule'] = "PayoutSchedule";
        FieldNameEnum[FieldNameEnum["PciSelfAssessment"] = 'pciSelfAssessment'] = "PciSelfAssessment";
        FieldNameEnum[FieldNameEnum["PersonalData"] = 'personalData'] = "PersonalData";
        FieldNameEnum[FieldNameEnum["PhoneCountryCode"] = 'phoneCountryCode'] = "PhoneCountryCode";
        FieldNameEnum[FieldNameEnum["PhoneNumber"] = 'phoneNumber'] = "PhoneNumber";
        FieldNameEnum[FieldNameEnum["PostalCode"] = 'postalCode'] = "PostalCode";
        FieldNameEnum[FieldNameEnum["PrimaryCurrency"] = 'primaryCurrency'] = "PrimaryCurrency";
        FieldNameEnum[FieldNameEnum["Reason"] = 'reason'] = "Reason";
        FieldNameEnum[FieldNameEnum["RegistrationNumber"] = 'registrationNumber'] = "RegistrationNumber";
        FieldNameEnum[FieldNameEnum["ReturnUrl"] = 'returnUrl'] = "ReturnUrl";
        FieldNameEnum[FieldNameEnum["Schedule"] = 'schedule'] = "Schedule";
        FieldNameEnum[FieldNameEnum["Shareholder"] = 'shareholder'] = "Shareholder";
        FieldNameEnum[FieldNameEnum["ShareholderCode"] = 'shareholderCode'] = "ShareholderCode";
        FieldNameEnum[FieldNameEnum["ShareholderCodeAndSignatoryCode"] = 'shareholderCodeAndSignatoryCode'] = "ShareholderCodeAndSignatoryCode";
        FieldNameEnum[FieldNameEnum["ShareholderCodeOrSignatoryCode"] = 'shareholderCodeOrSignatoryCode'] = "ShareholderCodeOrSignatoryCode";
        FieldNameEnum[FieldNameEnum["ShareholderType"] = 'shareholderType'] = "ShareholderType";
        FieldNameEnum[FieldNameEnum["ShopperInteraction"] = 'shopperInteraction'] = "ShopperInteraction";
        FieldNameEnum[FieldNameEnum["Signatory"] = 'signatory'] = "Signatory";
        FieldNameEnum[FieldNameEnum["SignatoryCode"] = 'signatoryCode'] = "SignatoryCode";
        FieldNameEnum[FieldNameEnum["SocialSecurityNumber"] = 'socialSecurityNumber'] = "SocialSecurityNumber";
        FieldNameEnum[FieldNameEnum["SourceAccountCode"] = 'sourceAccountCode'] = "SourceAccountCode";
        FieldNameEnum[FieldNameEnum["SplitAccount"] = 'splitAccount'] = "SplitAccount";
        FieldNameEnum[FieldNameEnum["SplitConfigurationUuid"] = 'splitConfigurationUUID'] = "SplitConfigurationUuid";
        FieldNameEnum[FieldNameEnum["SplitCurrency"] = 'splitCurrency'] = "SplitCurrency";
        FieldNameEnum[FieldNameEnum["SplitValue"] = 'splitValue'] = "SplitValue";
        FieldNameEnum[FieldNameEnum["Splits"] = 'splits'] = "Splits";
        FieldNameEnum[FieldNameEnum["StateOrProvince"] = 'stateOrProvince'] = "StateOrProvince";
        FieldNameEnum[FieldNameEnum["Status"] = 'status'] = "Status";
        FieldNameEnum[FieldNameEnum["StockExchange"] = 'stockExchange'] = "StockExchange";
        FieldNameEnum[FieldNameEnum["StockNumber"] = 'stockNumber'] = "StockNumber";
        FieldNameEnum[FieldNameEnum["StockTicker"] = 'stockTicker'] = "StockTicker";
        FieldNameEnum[FieldNameEnum["Store"] = 'store'] = "Store";
        FieldNameEnum[FieldNameEnum["StoreDetail"] = 'storeDetail'] = "StoreDetail";
        FieldNameEnum[FieldNameEnum["StoreName"] = 'storeName'] = "StoreName";
        FieldNameEnum[FieldNameEnum["StoreReference"] = 'storeReference'] = "StoreReference";
        FieldNameEnum[FieldNameEnum["Street"] = 'street'] = "Street";
        FieldNameEnum[FieldNameEnum["TaxId"] = 'taxId'] = "TaxId";
        FieldNameEnum[FieldNameEnum["Tier"] = 'tier'] = "Tier";
        FieldNameEnum[FieldNameEnum["TierNumber"] = 'tierNumber'] = "TierNumber";
        FieldNameEnum[FieldNameEnum["TransferCode"] = 'transferCode'] = "TransferCode";
        FieldNameEnum[FieldNameEnum["UltimateParentCompany"] = 'ultimateParentCompany'] = "UltimateParentCompany";
        FieldNameEnum[FieldNameEnum["UltimateParentCompanyAddressDetails"] = 'ultimateParentCompanyAddressDetails'] = "UltimateParentCompanyAddressDetails";
        FieldNameEnum[FieldNameEnum["UltimateParentCompanyAddressDetailsCountry"] = 'ultimateParentCompanyAddressDetailsCountry'] = "UltimateParentCompanyAddressDetailsCountry";
        FieldNameEnum[FieldNameEnum["UltimateParentCompanyBusinessDetails"] = 'ultimateParentCompanyBusinessDetails'] = "UltimateParentCompanyBusinessDetails";
        FieldNameEnum[FieldNameEnum["UltimateParentCompanyBusinessDetailsLegalBusinessName"] = 'ultimateParentCompanyBusinessDetailsLegalBusinessName'] = "UltimateParentCompanyBusinessDetailsLegalBusinessName";
        FieldNameEnum[FieldNameEnum["UltimateParentCompanyBusinessDetailsRegistrationNumber"] = 'ultimateParentCompanyBusinessDetailsRegistrationNumber'] = "UltimateParentCompanyBusinessDetailsRegistrationNumber";
        FieldNameEnum[FieldNameEnum["UltimateParentCompanyCode"] = 'ultimateParentCompanyCode'] = "UltimateParentCompanyCode";
        FieldNameEnum[FieldNameEnum["UltimateParentCompanyStockExchange"] = 'ultimateParentCompanyStockExchange'] = "UltimateParentCompanyStockExchange";
        FieldNameEnum[FieldNameEnum["UltimateParentCompanyStockNumber"] = 'ultimateParentCompanyStockNumber'] = "UltimateParentCompanyStockNumber";
        FieldNameEnum[FieldNameEnum["UltimateParentCompanyStockNumberOrStockTicker"] = 'ultimateParentCompanyStockNumberOrStockTicker'] = "UltimateParentCompanyStockNumberOrStockTicker";
        FieldNameEnum[FieldNameEnum["UltimateParentCompanyStockTicker"] = 'ultimateParentCompanyStockTicker'] = "UltimateParentCompanyStockTicker";
        FieldNameEnum[FieldNameEnum["Unknown"] = 'unknown'] = "Unknown";
        FieldNameEnum[FieldNameEnum["Value"] = 'value'] = "Value";
        FieldNameEnum[FieldNameEnum["VerificationType"] = 'verificationType'] = "VerificationType";
        FieldNameEnum[FieldNameEnum["VirtualAccount"] = 'virtualAccount'] = "VirtualAccount";
        FieldNameEnum[FieldNameEnum["VisaNumber"] = 'visaNumber'] = "VisaNumber";
        FieldNameEnum[FieldNameEnum["WebAddress"] = 'webAddress'] = "WebAddress";
        FieldNameEnum[FieldNameEnum["Year"] = 'year'] = "Year";
    })(FieldNameEnum = FieldType.FieldNameEnum || (FieldType.FieldNameEnum = {}));
})(FieldType = exports.FieldType || (exports.FieldType = {}));
