// @adyen/api-library/lib/src/typings/platformsNotifications/legalArrangementEntityDetail.js
"use strict";
/**
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen for Platforms: Notifications
 * The Notification API sends notifications to the endpoints specified in a given subscription. Subscriptions are managed through the Notification Configuration API. The API specifications listed here detail the format of each notification.  For more information, refer to our [documentation](https://docs.adyen.com/platforms/notifications).
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalArrangementEntityDetail = void 0;
class LegalArrangementEntityDetail {
}
exports.LegalArrangementEntityDetail = LegalArrangementEntityDetail;
(function (LegalArrangementEntityDetail) {
    let LegalArrangementMembersEnum;
    (function (LegalArrangementMembersEnum) {
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Beneficiary"] = 'Beneficiary'] = "Beneficiary";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["ControllingPerson"] = 'ControllingPerson'] = "ControllingPerson";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Partner"] = 'Partner'] = "Partner";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Protector"] = 'Protector'] = "Protector";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Settlor"] = 'Settlor'] = "Settlor";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Shareholder"] = 'Shareholder'] = "Shareholder";
        LegalArrangementMembersEnum[LegalArrangementMembersEnum["Trustee"] = 'Trustee'] = "Trustee";
    })(LegalArrangementMembersEnum = LegalArrangementEntityDetail.LegalArrangementMembersEnum || (LegalArrangementEntityDetail.LegalArrangementMembersEnum = {}));
    let LegalEntityTypeEnum;
    (function (LegalEntityTypeEnum) {
        LegalEntityTypeEnum[LegalEntityTypeEnum["Business"] = 'Business'] = "Business";
        LegalEntityTypeEnum[LegalEntityTypeEnum["Individual"] = 'Individual'] = "Individual";
        LegalEntityTypeEnum[LegalEntityTypeEnum["NonProfit"] = 'NonProfit'] = "NonProfit";
        LegalEntityTypeEnum[LegalEntityTypeEnum["Partnership"] = 'Partnership'] = "Partnership";
        LegalEntityTypeEnum[LegalEntityTypeEnum["PublicCompany"] = 'PublicCompany'] = "PublicCompany";
    })(LegalEntityTypeEnum = LegalArrangementEntityDetail.LegalEntityTypeEnum || (LegalArrangementEntityDetail.LegalEntityTypeEnum = {}));
})(LegalArrangementEntityDetail = exports.LegalArrangementEntityDetail || (exports.LegalArrangementEntityDetail = {}));
