// @adyen/api-library/lib/src/typings/platformsNotifications/personalDocumentData.js
"use strict";
/**
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen for Platforms: Notifications
 * The Notification API sends notifications to the endpoints specified in a given subscription. Subscriptions are managed through the Notification Configuration API. The API specifications listed here detail the format of each notification.  For more information, refer to our [documentation](https://docs.adyen.com/platforms/notifications).
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PersonalDocumentData = void 0;
class PersonalDocumentData {
}
exports.PersonalDocumentData = PersonalDocumentData;
(function (PersonalDocumentData) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Drivinglicense"] = 'DRIVINGLICENSE'] = "Drivinglicense";
        TypeEnum[TypeEnum["Id"] = 'ID'] = "Id";
        TypeEnum[TypeEnum["Passport"] = 'PASSPORT'] = "Passport";
        TypeEnum[TypeEnum["Socialsecurity"] = 'SOCIALSECURITY'] = "Socialsecurity";
        TypeEnum[TypeEnum["Visa"] = 'VISA'] = "Visa";
    })(TypeEnum = PersonalDocumentData.TypeEnum || (PersonalDocumentData.TypeEnum = {}));
})(PersonalDocumentData = exports.PersonalDocumentData || (exports.PersonalDocumentData = {}));
