// @adyen/api-library/lib/src/typings/platformsNotifications/transaction.js
"use strict";
/**
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 *
 * Adyen for Platforms: Notifications
 * The Notification API sends notifications to the endpoints specified in a given subscription. Subscriptions are managed through the Notification Configuration API. The API specifications listed here detail the format of each notification.  For more information, refer to our [documentation](https://docs.adyen.com/platforms/notifications).
 *
 * The version of the OpenAPI document: 6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transaction = void 0;
class Transaction {
}
exports.Transaction = Transaction;
(function (Transaction) {
    let TransactionStatusEnum;
    (function (TransactionStatusEnum) {
        TransactionStatusEnum[TransactionStatusEnum["BalanceNotPaidOutTransfer"] = 'BalanceNotPaidOutTransfer'] = "BalanceNotPaidOutTransfer";
        TransactionStatusEnum[TransactionStatusEnum["Chargeback"] = 'Chargeback'] = "Chargeback";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackCorrection"] = 'ChargebackCorrection'] = "ChargebackCorrection";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackCorrectionReceived"] = 'ChargebackCorrectionReceived'] = "ChargebackCorrectionReceived";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackReceived"] = 'ChargebackReceived'] = "ChargebackReceived";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackReversed"] = 'ChargebackReversed'] = "ChargebackReversed";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackReversedCorrection"] = 'ChargebackReversedCorrection'] = "ChargebackReversedCorrection";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackReversedCorrectionReceived"] = 'ChargebackReversedCorrectionReceived'] = "ChargebackReversedCorrectionReceived";
        TransactionStatusEnum[TransactionStatusEnum["ChargebackReversedReceived"] = 'ChargebackReversedReceived'] = "ChargebackReversedReceived";
        TransactionStatusEnum[TransactionStatusEnum["Converted"] = 'Converted'] = "Converted";
        TransactionStatusEnum[TransactionStatusEnum["CreditClosed"] = 'CreditClosed'] = "CreditClosed";
        TransactionStatusEnum[TransactionStatusEnum["CreditFailed"] = 'CreditFailed'] = "CreditFailed";
        TransactionStatusEnum[TransactionStatusEnum["CreditReversed"] = 'CreditReversed'] = "CreditReversed";
        TransactionStatusEnum[TransactionStatusEnum["CreditReversedReceived"] = 'CreditReversedReceived'] = "CreditReversedReceived";
        TransactionStatusEnum[TransactionStatusEnum["CreditSuspended"] = 'CreditSuspended'] = "CreditSuspended";
        TransactionStatusEnum[TransactionStatusEnum["Credited"] = 'Credited'] = "Credited";
        TransactionStatusEnum[TransactionStatusEnum["DebitFailed"] = 'DebitFailed'] = "DebitFailed";
        TransactionStatusEnum[TransactionStatusEnum["DebitReversedReceived"] = 'DebitReversedReceived'] = "DebitReversedReceived";
        TransactionStatusEnum[TransactionStatusEnum["Debited"] = 'Debited'] = "Debited";
        TransactionStatusEnum[TransactionStatusEnum["DebitedReversed"] = 'DebitedReversed'] = "DebitedReversed";
        TransactionStatusEnum[TransactionStatusEnum["DepositCorrectionCredited"] = 'DepositCorrectionCredited'] = "DepositCorrectionCredited";
        TransactionStatusEnum[TransactionStatusEnum["DepositCorrectionDebited"] = 'DepositCorrectionDebited'] = "DepositCorrectionDebited";
        TransactionStatusEnum[TransactionStatusEnum["Fee"] = 'Fee'] = "Fee";
        TransactionStatusEnum[TransactionStatusEnum["FundTransfer"] = 'FundTransfer'] = "FundTransfer";
        TransactionStatusEnum[TransactionStatusEnum["FundTransferReversed"] = 'FundTransferReversed'] = "FundTransferReversed";
        TransactionStatusEnum[TransactionStatusEnum["InvoiceDeductionCredited"] = 'InvoiceDeductionCredited'] = "InvoiceDeductionCredited";
        TransactionStatusEnum[TransactionStatusEnum["InvoiceDeductionDebited"] = 'InvoiceDeductionDebited'] = "InvoiceDeductionDebited";
        TransactionStatusEnum[TransactionStatusEnum["ManualCorrected"] = 'ManualCorrected'] = "ManualCorrected";
        TransactionStatusEnum[TransactionStatusEnum["ManualCorrectionCredited"] = 'ManualCorrectionCredited'] = "ManualCorrectionCredited";
        TransactionStatusEnum[TransactionStatusEnum["ManualCorrectionDebited"] = 'ManualCorrectionDebited'] = "ManualCorrectionDebited";
        TransactionStatusEnum[TransactionStatusEnum["MerchantPayin"] = 'MerchantPayin'] = "MerchantPayin";
        TransactionStatusEnum[TransactionStatusEnum["MerchantPayinReversed"] = 'MerchantPayinReversed'] = "MerchantPayinReversed";
        TransactionStatusEnum[TransactionStatusEnum["Payout"] = 'Payout'] = "Payout";
        TransactionStatusEnum[TransactionStatusEnum["PayoutReversed"] = 'PayoutReversed'] = "PayoutReversed";
        TransactionStatusEnum[TransactionStatusEnum["PendingCredit"] = 'PendingCredit'] = "PendingCredit";
        TransactionStatusEnum[TransactionStatusEnum["PendingDebit"] = 'PendingDebit'] = "PendingDebit";
        TransactionStatusEnum[TransactionStatusEnum["PendingFundTransfer"] = 'PendingFundTransfer'] = "PendingFundTransfer";
        TransactionStatusEnum[TransactionStatusEnum["SecondChargeback"] = 'SecondChargeback'] = "SecondChargeback";
        TransactionStatusEnum[TransactionStatusEnum["SecondChargebackCorrection"] = 'SecondChargebackCorrection'] = "SecondChargebackCorrection";
        TransactionStatusEnum[TransactionStatusEnum["SecondChargebackCorrectionReceived"] = 'SecondChargebackCorrectionReceived'] = "SecondChargebackCorrectionReceived";
        TransactionStatusEnum[TransactionStatusEnum["SecondChargebackReceived"] = 'SecondChargebackReceived'] = "SecondChargebackReceived";
    })(TransactionStatusEnum = Transaction.TransactionStatusEnum || (Transaction.TransactionStatusEnum = {}));
})(Transaction = exports.Transaction || (exports.Transaction = {}));
