// @adyen/api-library/lib/src/services/legalEntityManagement/legalEntitiesApi.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalEntitiesApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/legalEntityManagement/models");
class LegalEntitiesApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://kyc-test.adyen.com/lem/v3";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get a legal entity
    * @param id {@link string } The unique identifier of the legal entity.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link LegalEntity }
    */
    async getLegalEntity(id, requestOptions) {
        const endpoint = `${this.baseUrl}/legalEntities/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "LegalEntity");
    }
    /**
    * @summary Get all business lines under a legal entity
    * @param id {@link string } The unique identifier of the legal entity.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link BusinessLines }
    */
    async getAllBusinessLinesUnderLegalEntity(id, requestOptions) {
        const endpoint = `${this.baseUrl}/legalEntities/{id}/businessLines`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "BusinessLines");
    }
    /**
    * @summary Update a legal entity
    * @param id {@link string } The unique identifier of the legal entity.
    * @param xRequestedVerificationCode {@link string } Use the requested verification code 0_0001 to resolve any suberrors associated with the legal entity. Requested verification codes can only be used in your test environment.
    * @param legalEntityInfo {@link LegalEntityInfo }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link LegalEntity }
    */
    async updateLegalEntity(id, legalEntityInfo, requestOptions) {
        const endpoint = `${this.baseUrl}/legalEntities/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(legalEntityInfo, "LegalEntityInfo");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "LegalEntity");
    }
    /**
    * @summary Create a legal entity
    * @param xRequestedVerificationCode {@link string } Use a suberror code as your requested verification code. You can include one code at a time in your request header. Requested verification codes can only be used in your test environment.
    * @param legalEntityInfoRequiredType {@link LegalEntityInfoRequiredType }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link LegalEntity }
    */
    async createLegalEntity(legalEntityInfoRequiredType, requestOptions) {
        const endpoint = `${this.baseUrl}/legalEntities`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(legalEntityInfoRequiredType, "LegalEntityInfoRequiredType");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "LegalEntity");
    }
    /**
    * @summary Check a legal entity\'s verification errors
    * @param id {@link string } The unique identifier of the legal entity.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link VerificationErrors }
    */
    async checkLegalEntitysVerificationErrors(id, requestOptions) {
        const endpoint = `${this.baseUrl}/legalEntities/{id}/checkVerificationErrors`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "VerificationErrors");
    }
}
exports.LegalEntitiesApi = LegalEntitiesApi;
