// @adyen/api-library/lib/src/services/checkout/classicCheckoutSDKApi.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClassicCheckoutSDKApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/checkout/models");
class ClassicCheckoutSDKApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://checkout-test.adyen.com/v70";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Create a payment session
    * @param idempotencyKey {@link string } A unique identifier for the message with a maximum of 64 characters (we recommend a UUID).
    * @param paymentSetupRequest {@link PaymentSetupRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link PaymentSetupResponse }
    */
    async paymentSession(paymentSetupRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/paymentSession`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(paymentSetupRequest, "PaymentSetupRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "PaymentSetupResponse");
    }
    /**
    * @summary Verify a payment result
    * @param idempotencyKey {@link string } A unique identifier for the message with a maximum of 64 characters (we recommend a UUID).
    * @param paymentVerificationRequest {@link PaymentVerificationRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link PaymentVerificationResponse }
    */
    async verifyPaymentResult(paymentVerificationRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/payments/result`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(paymentVerificationRequest, "PaymentVerificationRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "PaymentVerificationResponse");
    }
}
exports.ClassicCheckoutSDKApi = ClassicCheckoutSDKApi;
