// @adyen/api-library/lib/src/services/checkout/index.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const classicCheckoutSDKApi_1 = require("./classicCheckoutSDKApi");
const modificationsApi_1 = require("./modificationsApi");
const ordersApi_1 = require("./ordersApi");
const paymentLinksApi_1 = require("./paymentLinksApi");
const paymentsApi_1 = require("./paymentsApi");
const recurringApi_1 = require("./recurringApi");
const utilityApi_1 = require("./utilityApi");
const service_1 = __importDefault(require("../../service"));
class CheckoutAPI extends service_1.default {
    constructor(client) {
        super(client);
    }
    get ClassicCheckoutSDKApi() {
        return new classicCheckoutSDKApi_1.ClassicCheckoutSDKApi(this.client);
    }
    get ModificationsApi() {
        return new modificationsApi_1.ModificationsApi(this.client);
    }
    get OrdersApi() {
        return new ordersApi_1.OrdersApi(this.client);
    }
    get PaymentLinksApi() {
        return new paymentLinksApi_1.PaymentLinksApi(this.client);
    }
    get PaymentsApi() {
        return new paymentsApi_1.PaymentsApi(this.client);
    }
    get RecurringApi() {
        return new recurringApi_1.RecurringApi(this.client);
    }
    get UtilityApi() {
        return new utilityApi_1.UtilityApi(this.client);
    }
}
exports.default = CheckoutAPI;
