// @adyen/api-library/lib/src/services/checkout/ordersApi.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrdersApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/checkout/models");
class OrdersApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://checkout-test.adyen.com/v70";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Create an order
    * @param idempotencyKey {@link string } A unique identifier for the message with a maximum of 64 characters (we recommend a UUID).
    * @param createOrderRequest {@link CreateOrderRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link CreateOrderResponse }
    */
    async orders(createOrderRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/orders`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(createOrderRequest, "CreateOrderRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "CreateOrderResponse");
    }
    /**
    * @summary Cancel an order
    * @param idempotencyKey {@link string } A unique identifier for the message with a maximum of 64 characters (we recommend a UUID).
    * @param cancelOrderRequest {@link CancelOrderRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link CancelOrderResponse }
    */
    async cancelOrder(cancelOrderRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/orders/cancel`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(cancelOrderRequest, "CancelOrderRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "CancelOrderResponse");
    }
    /**
    * @summary Get the balance of a gift card
    * @param idempotencyKey {@link string } A unique identifier for the message with a maximum of 64 characters (we recommend a UUID).
    * @param balanceCheckRequest {@link BalanceCheckRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link BalanceCheckResponse }
    */
    async getBalanceOfGiftCard(balanceCheckRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/paymentMethods/balance`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(balanceCheckRequest, "BalanceCheckRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "BalanceCheckResponse");
    }
}
exports.OrdersApi = OrdersApi;
