// @adyen/api-library/lib/src/services/checkout/paymentLinksApi.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentLinksApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/checkout/models");
class PaymentLinksApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://checkout-test.adyen.com/v70";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get a payment link
    * @param linkId {@link string } Unique identifier of the payment link.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link PaymentLinkResponse }
    */
    async getPaymentLink(linkId, requestOptions) {
        const endpoint = `${this.baseUrl}/paymentLinks/{linkId}`
            .replace("{" + "linkId" + "}", encodeURIComponent(String(linkId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "PaymentLinkResponse");
    }
    /**
    * @summary Update the status of a payment link
    * @param linkId {@link string } Unique identifier of the payment link.
    * @param updatePaymentLinkRequest {@link UpdatePaymentLinkRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link PaymentLinkResponse }
    */
    async updatePaymentLink(linkId, updatePaymentLinkRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/paymentLinks/{linkId}`
            .replace("{" + "linkId" + "}", encodeURIComponent(String(linkId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(updatePaymentLinkRequest, "UpdatePaymentLinkRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "PaymentLinkResponse");
    }
    /**
    * @summary Create a payment link
    * @param idempotencyKey {@link string } A unique identifier for the message with a maximum of 64 characters (we recommend a UUID).
    * @param paymentLinkRequest {@link PaymentLinkRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link PaymentLinkResponse }
    */
    async paymentLinks(paymentLinkRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/paymentLinks`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(paymentLinkRequest, "PaymentLinkRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "PaymentLinkResponse");
    }
}
exports.PaymentLinksApi = PaymentLinksApi;
