// @adyen/api-library/lib/src/services/checkout/recurringApi.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecurringApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/checkout/models");
class RecurringApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://checkout-test.adyen.com/v70";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Delete a token for stored payment details
    * @param storedPaymentMethodId {@link string } The unique identifier of the token.
    * @param shopperReference {@link string } Your reference to uniquely identify this shopper, for example user ID or account ID. Minimum length: 3 characters. &gt; Your reference must not include personally identifiable information (PII), for example name or email address.
    * @param merchantAccount {@link string } Your merchant account.
    * @param requestOptions {@link IRequest.Options}
    */
    async deleteTokenForStoredPaymentDetails(storedPaymentMethodId, requestOptions) {
        const endpoint = `${this.baseUrl}/storedPaymentMethods/{storedPaymentMethodId}`
            .replace("{" + "storedPaymentMethodId" + "}", encodeURIComponent(String(storedPaymentMethodId)));
        const resource = new resource_1.default(this, endpoint);
        await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "DELETE" });
    }
    /**
    * @summary Get tokens for stored payment details
    * @param shopperReference {@link string } Your reference to uniquely identify this shopper, for example user ID or account ID. Minimum length: 3 characters. &gt; Your reference must not include personally identifiable information (PII), for example name or email address.
    * @param merchantAccount {@link string } Your merchant account.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ListStoredPaymentMethodsResponse }
    */
    async getTokensForStoredPaymentDetails(requestOptions) {
        const endpoint = `${this.baseUrl}/storedPaymentMethods`;
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ListStoredPaymentMethodsResponse");
    }
}
exports.RecurringApi = RecurringApi;
