// @adyen/api-library/lib/src/services/checkout/utilityApi.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UtilityApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/checkout/models");
class UtilityApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://checkout-test.adyen.com/v70";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get an Apple Pay session
    * @param idempotencyKey {@link string } A unique identifier for the message with a maximum of 64 characters (we recommend a UUID).
    * @param applePaySessionRequest {@link ApplePaySessionRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ApplePaySessionResponse }
    */
    async getApplePaySession(applePaySessionRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/applePay/sessions`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(applePaySessionRequest, "ApplePaySessionRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "ApplePaySessionResponse");
    }
    /**
    * @summary Create originKey values for domains
    * @param idempotencyKey {@link string } A unique identifier for the message with a maximum of 64 characters (we recommend a UUID).
    * @param utilityRequest {@link UtilityRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link UtilityResponse }
    */
    async originKeys(utilityRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/originKeys`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(utilityRequest, "UtilityRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "UtilityResponse");
    }
}
exports.UtilityApi = UtilityApi;
