// @adyen/api-library/lib/src/typings/payment/merchantRiskIndicator.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MerchantRiskIndicator = void 0;
class MerchantRiskIndicator {
    static getAttributeTypeMap() {
        return MerchantRiskIndicator.attributeTypeMap;
    }
}
exports.MerchantRiskIndicator = MerchantRiskIndicator;
MerchantRiskIndicator.discriminator = undefined;
MerchantRiskIndicator.attributeTypeMap = [
    {
        "name": "addressMatch",
        "baseName": "addressMatch",
        "type": "boolean"
    },
    {
        "name": "deliveryAddressIndicator",
        "baseName": "deliveryAddressIndicator",
        "type": "MerchantRiskIndicator.DeliveryAddressIndicatorEnum"
    },
    {
        "name": "deliveryEmail",
        "baseName": "deliveryEmail",
        "type": "string"
    },
    {
        "name": "deliveryEmailAddress",
        "baseName": "deliveryEmailAddress",
        "type": "string"
    },
    {
        "name": "deliveryTimeframe",
        "baseName": "deliveryTimeframe",
        "type": "MerchantRiskIndicator.DeliveryTimeframeEnum"
    },
    {
        "name": "giftCardAmount",
        "baseName": "giftCardAmount",
        "type": "Amount"
    },
    {
        "name": "giftCardCount",
        "baseName": "giftCardCount",
        "type": "number"
    },
    {
        "name": "giftCardCurr",
        "baseName": "giftCardCurr",
        "type": "string"
    },
    {
        "name": "preOrderDate",
        "baseName": "preOrderDate",
        "type": "Date"
    },
    {
        "name": "preOrderPurchase",
        "baseName": "preOrderPurchase",
        "type": "boolean"
    },
    {
        "name": "preOrderPurchaseInd",
        "baseName": "preOrderPurchaseInd",
        "type": "string"
    },
    {
        "name": "reorderItems",
        "baseName": "reorderItems",
        "type": "boolean"
    },
    {
        "name": "reorderItemsInd",
        "baseName": "reorderItemsInd",
        "type": "string"
    },
    {
        "name": "shipIndicator",
        "baseName": "shipIndicator",
        "type": "string"
    }
];
(function (MerchantRiskIndicator) {
    let DeliveryAddressIndicatorEnum;
    (function (DeliveryAddressIndicatorEnum) {
        DeliveryAddressIndicatorEnum["ShipToBillingAddress"] = "shipToBillingAddress";
        DeliveryAddressIndicatorEnum["ShipToVerifiedAddress"] = "shipToVerifiedAddress";
        DeliveryAddressIndicatorEnum["ShipToNewAddress"] = "shipToNewAddress";
        DeliveryAddressIndicatorEnum["ShipToStore"] = "shipToStore";
        DeliveryAddressIndicatorEnum["DigitalGoods"] = "digitalGoods";
        DeliveryAddressIndicatorEnum["GoodsNotShipped"] = "goodsNotShipped";
        DeliveryAddressIndicatorEnum["Other"] = "other";
    })(DeliveryAddressIndicatorEnum = MerchantRiskIndicator.DeliveryAddressIndicatorEnum || (MerchantRiskIndicator.DeliveryAddressIndicatorEnum = {}));
    let DeliveryTimeframeEnum;
    (function (DeliveryTimeframeEnum) {
        DeliveryTimeframeEnum["ElectronicDelivery"] = "electronicDelivery";
        DeliveryTimeframeEnum["SameDayShipping"] = "sameDayShipping";
        DeliveryTimeframeEnum["OvernightShipping"] = "overnightShipping";
        DeliveryTimeframeEnum["TwoOrMoreDaysShipping"] = "twoOrMoreDaysShipping";
    })(DeliveryTimeframeEnum = MerchantRiskIndicator.DeliveryTimeframeEnum || (MerchantRiskIndicator.DeliveryTimeframeEnum = {}));
})(MerchantRiskIndicator = exports.MerchantRiskIndicator || (exports.MerchantRiskIndicator = {}));
