// @adyen/api-library/lib/src/typings/payment/paymentResult.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentResult = void 0;
class PaymentResult {
    static getAttributeTypeMap() {
        return PaymentResult.attributeTypeMap;
    }
}
exports.PaymentResult = PaymentResult;
PaymentResult.discriminator = undefined;
PaymentResult.attributeTypeMap = [
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "authCode",
        "baseName": "authCode",
        "type": "string"
    },
    {
        "name": "dccAmount",
        "baseName": "dccAmount",
        "type": "Amount"
    },
    {
        "name": "dccSignature",
        "baseName": "dccSignature",
        "type": "string"
    },
    {
        "name": "fraudResult",
        "baseName": "fraudResult",
        "type": "FraudResult"
    },
    {
        "name": "issuerUrl",
        "baseName": "issuerUrl",
        "type": "string"
    },
    {
        "name": "md",
        "baseName": "md",
        "type": "string"
    },
    {
        "name": "paRequest",
        "baseName": "paRequest",
        "type": "string"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "PaymentResult.ResultCodeEnum"
    }
];
(function (PaymentResult) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["AuthenticationFinished"] = "AuthenticationFinished";
        ResultCodeEnum["AuthenticationNotRequired"] = "AuthenticationNotRequired";
        ResultCodeEnum["Authorised"] = "Authorised";
        ResultCodeEnum["Cancelled"] = "Cancelled";
        ResultCodeEnum["ChallengeShopper"] = "ChallengeShopper";
        ResultCodeEnum["Error"] = "Error";
        ResultCodeEnum["IdentifyShopper"] = "IdentifyShopper";
        ResultCodeEnum["PartiallyAuthorised"] = "PartiallyAuthorised";
        ResultCodeEnum["Pending"] = "Pending";
        ResultCodeEnum["PresentToShopper"] = "PresentToShopper";
        ResultCodeEnum["Received"] = "Received";
        ResultCodeEnum["RedirectShopper"] = "RedirectShopper";
        ResultCodeEnum["Refused"] = "Refused";
        ResultCodeEnum["Success"] = "Success";
    })(ResultCodeEnum = PaymentResult.ResultCodeEnum || (PaymentResult.ResultCodeEnum = {}));
})(PaymentResult = exports.PaymentResult || (exports.PaymentResult = {}));
