// @adyen/api-library/lib/src/typings/payment/paymentRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRequest = void 0;
class PaymentRequest {
    static getAttributeTypeMap() {
        return PaymentRequest.attributeTypeMap;
    }
}
exports.PaymentRequest = PaymentRequest;
PaymentRequest.discriminator = undefined;
PaymentRequest.attributeTypeMap = [
    {
        "name": "accountInfo",
        "baseName": "accountInfo",
        "type": "AccountInfo"
    },
    {
        "name": "additionalAmount",
        "baseName": "additionalAmount",
        "type": "Amount"
    },
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "applicationInfo",
        "baseName": "applicationInfo",
        "type": "ApplicationInfo"
    },
    {
        "name": "bankAccount",
        "baseName": "bankAccount",
        "type": "BankAccount"
    },
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "Address"
    },
    {
        "name": "browserInfo",
        "baseName": "browserInfo",
        "type": "BrowserInfo"
    },
    {
        "name": "captureDelayHours",
        "baseName": "captureDelayHours",
        "type": "number"
    },
    {
        "name": "card",
        "baseName": "card",
        "type": "Card"
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "string"
    },
    {
        "name": "dccQuote",
        "baseName": "dccQuote",
        "type": "ForexQuote"
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "Address"
    },
    {
        "name": "deliveryDate",
        "baseName": "deliveryDate",
        "type": "Date"
    },
    {
        "name": "deviceFingerprint",
        "baseName": "deviceFingerprint",
        "type": "string"
    },
    {
        "name": "entityType",
        "baseName": "entityType",
        "type": "PaymentRequest.EntityTypeEnum"
    },
    {
        "name": "fraudOffset",
        "baseName": "fraudOffset",
        "type": "number"
    },
    {
        "name": "fundDestination",
        "baseName": "fundDestination",
        "type": "FundDestination"
    },
    {
        "name": "fundSource",
        "baseName": "fundSource",
        "type": "FundSource"
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "PaymentRequest.FundingSourceEnum"
    },
    {
        "name": "installments",
        "baseName": "installments",
        "type": "Installments"
    },
    {
        "name": "localizedShopperStatement",
        "baseName": "localizedShopperStatement",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "mandate",
        "baseName": "mandate",
        "type": "Mandate"
    },
    {
        "name": "mcc",
        "baseName": "mcc",
        "type": "string"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "merchantOrderReference",
        "baseName": "merchantOrderReference",
        "type": "string"
    },
    {
        "name": "merchantRiskIndicator",
        "baseName": "merchantRiskIndicator",
        "type": "MerchantRiskIndicator"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "mpiData",
        "baseName": "mpiData",
        "type": "ThreeDSecureData"
    },
    {
        "name": "nationality",
        "baseName": "nationality",
        "type": "string"
    },
    {
        "name": "orderReference",
        "baseName": "orderReference",
        "type": "string"
    },
    {
        "name": "platformChargebackLogic",
        "baseName": "platformChargebackLogic",
        "type": "PlatformChargebackLogic"
    },
    {
        "name": "recurring",
        "baseName": "recurring",
        "type": "Recurring"
    },
    {
        "name": "recurringProcessingModel",
        "baseName": "recurringProcessingModel",
        "type": "PaymentRequest.RecurringProcessingModelEnum"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "selectedBrand",
        "baseName": "selectedBrand",
        "type": "string"
    },
    {
        "name": "selectedRecurringDetailReference",
        "baseName": "selectedRecurringDetailReference",
        "type": "string"
    },
    {
        "name": "sessionId",
        "baseName": "sessionId",
        "type": "string"
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string"
    },
    {
        "name": "shopperIP",
        "baseName": "shopperIP",
        "type": "string"
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "PaymentRequest.ShopperInteractionEnum"
    },
    {
        "name": "shopperLocale",
        "baseName": "shopperLocale",
        "type": "string"
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name"
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string"
    },
    {
        "name": "shopperStatement",
        "baseName": "shopperStatement",
        "type": "string"
    },
    {
        "name": "socialSecurityNumber",
        "baseName": "socialSecurityNumber",
        "type": "string"
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>"
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string"
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string"
    },
    {
        "name": "threeDS2RequestData",
        "baseName": "threeDS2RequestData",
        "type": "ThreeDS2RequestData"
    },
    {
        "name": "threeDSAuthenticationOnly",
        "baseName": "threeDSAuthenticationOnly",
        "type": "boolean"
    },
    {
        "name": "totalsGroup",
        "baseName": "totalsGroup",
        "type": "string"
    },
    {
        "name": "trustedShopper",
        "baseName": "trustedShopper",
        "type": "boolean"
    }
];
(function (PaymentRequest) {
    let EntityTypeEnum;
    (function (EntityTypeEnum) {
        EntityTypeEnum["NaturalPerson"] = "NaturalPerson";
        EntityTypeEnum["CompanyName"] = "CompanyName";
    })(EntityTypeEnum = PaymentRequest.EntityTypeEnum || (PaymentRequest.EntityTypeEnum = {}));
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = PaymentRequest.FundingSourceEnum || (PaymentRequest.FundingSourceEnum = {}));
    let RecurringProcessingModelEnum;
    (function (RecurringProcessingModelEnum) {
        RecurringProcessingModelEnum["CardOnFile"] = "CardOnFile";
        RecurringProcessingModelEnum["Subscription"] = "Subscription";
        RecurringProcessingModelEnum["UnscheduledCardOnFile"] = "UnscheduledCardOnFile";
    })(RecurringProcessingModelEnum = PaymentRequest.RecurringProcessingModelEnum || (PaymentRequest.RecurringProcessingModelEnum = {}));
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = PaymentRequest.ShopperInteractionEnum || (PaymentRequest.ShopperInteractionEnum = {}));
})(PaymentRequest = exports.PaymentRequest || (exports.PaymentRequest = {}));
