// @adyen/api-library/lib/src/typings/dataProtection/subjectErasureResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubjectErasureResponse = void 0;
class SubjectErasureResponse {
    static getAttributeTypeMap() {
        return SubjectErasureResponse.attributeTypeMap;
    }
}
exports.SubjectErasureResponse = SubjectErasureResponse;
SubjectErasureResponse.discriminator = undefined;
SubjectErasureResponse.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "result",
        "type": "SubjectErasureResponse.ResultEnum"
    }
];
(function (SubjectErasureResponse) {
    let ResultEnum;
    (function (ResultEnum) {
        ResultEnum["ActiveRecurringTokenExists"] = "ACTIVE_RECURRING_TOKEN_EXISTS";
        ResultEnum["AlreadyProcessed"] = "ALREADY_PROCESSED";
        ResultEnum["PaymentNotFound"] = "PAYMENT_NOT_FOUND";
        ResultEnum["Success"] = "SUCCESS";
    })(ResultEnum = SubjectErasureResponse.ResultEnum || (SubjectErasureResponse.ResultEnum = {}));
})(SubjectErasureResponse = exports.SubjectErasureResponse || (exports.SubjectErasureResponse = {}));
