// @adyen/api-library/lib/src/typings/managementWebhooks/midServiceNotificationData.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MidServiceNotificationData = void 0;
class MidServiceNotificationData {
    static getAttributeTypeMap() {
        return MidServiceNotificationData.attributeTypeMap;
    }
}
exports.MidServiceNotificationData = MidServiceNotificationData;
MidServiceNotificationData.discriminator = undefined;
MidServiceNotificationData.attributeTypeMap = [
    {
        "name": "allowed",
        "baseName": "allowed",
        "type": "boolean"
    },
    {
        "name": "enabled",
        "baseName": "enabled",
        "type": "boolean"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "merchantId",
        "baseName": "merchantId",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "MidServiceNotificationData.StatusEnum"
    },
    {
        "name": "storeId",
        "baseName": "storeId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "string"
    },
    {
        "name": "verificationStatus",
        "baseName": "verificationStatus",
        "type": "MidServiceNotificationData.VerificationStatusEnum"
    }
];
(function (MidServiceNotificationData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Success"] = "success";
        StatusEnum["Failure"] = "failure";
        StatusEnum["CapabilityPending"] = "capabilityPending";
        StatusEnum["DataRequired"] = "dataRequired";
        StatusEnum["UpdatesExpected"] = "updatesExpected";
    })(StatusEnum = MidServiceNotificationData.StatusEnum || (MidServiceNotificationData.StatusEnum = {}));
    let VerificationStatusEnum;
    (function (VerificationStatusEnum) {
        VerificationStatusEnum["Valid"] = "valid";
        VerificationStatusEnum["Pending"] = "pending";
        VerificationStatusEnum["Invalid"] = "invalid";
        VerificationStatusEnum["Rejected"] = "rejected";
    })(VerificationStatusEnum = MidServiceNotificationData.VerificationStatusEnum || (MidServiceNotificationData.VerificationStatusEnum = {}));
})(MidServiceNotificationData = exports.MidServiceNotificationData || (exports.MidServiceNotificationData = {}));
