// @adyen/api-library/lib/src/typings/legalEntityManagement/acceptTermsOfServiceResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AcceptTermsOfServiceResponse = void 0;
class AcceptTermsOfServiceResponse {
    static getAttributeTypeMap() {
        return AcceptTermsOfServiceResponse.attributeTypeMap;
    }
}
exports.AcceptTermsOfServiceResponse = AcceptTermsOfServiceResponse;
AcceptTermsOfServiceResponse.discriminator = undefined;
AcceptTermsOfServiceResponse.attributeTypeMap = [
    {
        "name": "acceptedBy",
        "baseName": "acceptedBy",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "ipAddress",
        "baseName": "ipAddress",
        "type": "string"
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "string"
    },
    {
        "name": "termsOfServiceDocumentId",
        "baseName": "termsOfServiceDocumentId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "AcceptTermsOfServiceResponse.TypeEnum"
    }
];
(function (AcceptTermsOfServiceResponse) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AdyenAccount"] = "adyenAccount";
        TypeEnum["AdyenCapital"] = "adyenCapital";
        TypeEnum["AdyenCard"] = "adyenCard";
        TypeEnum["AdyenForPlatformsAdvanced"] = "adyenForPlatformsAdvanced";
        TypeEnum["AdyenForPlatformsManage"] = "adyenForPlatformsManage";
        TypeEnum["AdyenFranchisee"] = "adyenFranchisee";
        TypeEnum["AdyenIssuing"] = "adyenIssuing";
    })(TypeEnum = AcceptTermsOfServiceResponse.TypeEnum || (AcceptTermsOfServiceResponse.TypeEnum = {}));
})(AcceptTermsOfServiceResponse = exports.AcceptTermsOfServiceResponse || (exports.AcceptTermsOfServiceResponse = {}));
