// @adyen/api-library/lib/src/typings/legalEntityManagement/businessLine.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BusinessLine = void 0;
class BusinessLine {
    static getAttributeTypeMap() {
        return BusinessLine.attributeTypeMap;
    }
}
exports.BusinessLine = BusinessLine;
BusinessLine.discriminator = undefined;
BusinessLine.attributeTypeMap = [
    {
        "name": "capability",
        "baseName": "capability",
        "type": "BusinessLine.CapabilityEnum"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "industryCode",
        "baseName": "industryCode",
        "type": "string"
    },
    {
        "name": "legalEntityId",
        "baseName": "legalEntityId",
        "type": "string"
    },
    {
        "name": "problems",
        "baseName": "problems",
        "type": "Array<CapabilityProblem>"
    },
    {
        "name": "salesChannels",
        "baseName": "salesChannels",
        "type": "Array<string>"
    },
    {
        "name": "service",
        "baseName": "service",
        "type": "BusinessLine.ServiceEnum"
    },
    {
        "name": "sourceOfFunds",
        "baseName": "sourceOfFunds",
        "type": "SourceOfFunds"
    },
    {
        "name": "webData",
        "baseName": "webData",
        "type": "Array<WebData>"
    },
    {
        "name": "webDataExemption",
        "baseName": "webDataExemption",
        "type": "WebDataExemption"
    }
];
(function (BusinessLine) {
    let CapabilityEnum;
    (function (CapabilityEnum) {
        CapabilityEnum["ReceivePayments"] = "receivePayments";
        CapabilityEnum["ReceiveFromPlatformPayments"] = "receiveFromPlatformPayments";
        CapabilityEnum["IssueBankAccount"] = "issueBankAccount";
    })(CapabilityEnum = BusinessLine.CapabilityEnum || (BusinessLine.CapabilityEnum = {}));
    let ServiceEnum;
    (function (ServiceEnum) {
        ServiceEnum["PaymentProcessing"] = "paymentProcessing";
        ServiceEnum["Issuing"] = "issuing";
        ServiceEnum["Banking"] = "banking";
    })(ServiceEnum = BusinessLine.ServiceEnum || (BusinessLine.ServiceEnum = {}));
})(BusinessLine = exports.BusinessLine || (exports.BusinessLine = {}));
