// @adyen/api-library/lib/src/typings/legalEntityManagement/documentPage.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentPage = void 0;
class DocumentPage {
    static getAttributeTypeMap() {
        return DocumentPage.attributeTypeMap;
    }
}
exports.DocumentPage = DocumentPage;
DocumentPage.discriminator = undefined;
DocumentPage.attributeTypeMap = [
    {
        "name": "pageName",
        "baseName": "pageName",
        "type": "string"
    },
    {
        "name": "pageNumber",
        "baseName": "pageNumber",
        "type": "number"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "DocumentPage.TypeEnum"
    }
];
(function (DocumentPage) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Back"] = "BACK";
        TypeEnum["Front"] = "FRONT";
        TypeEnum["Undefined"] = "UNDEFINED";
    })(TypeEnum = DocumentPage.TypeEnum || (DocumentPage.TypeEnum = {}));
})(DocumentPage = exports.DocumentPage || (exports.DocumentPage = {}));
