// @adyen/api-library/lib/src/typings/legalEntityManagement/document.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Document = void 0;
class Document {
    static getAttributeTypeMap() {
        return Document.attributeTypeMap;
    }
}
exports.Document = Document;
Document.discriminator = undefined;
Document.attributeTypeMap = [
    {
        "name": "attachment",
        "baseName": "attachment",
        "type": "Attachment"
    },
    {
        "name": "attachments",
        "baseName": "attachments",
        "type": "Array<Attachment>"
    },
    {
        "name": "creationDate",
        "baseName": "creationDate",
        "type": "Date"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "expiryDate",
        "baseName": "expiryDate",
        "type": "string"
    },
    {
        "name": "fileName",
        "baseName": "fileName",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "issuerCountry",
        "baseName": "issuerCountry",
        "type": "string"
    },
    {
        "name": "issuerState",
        "baseName": "issuerState",
        "type": "string"
    },
    {
        "name": "modificationDate",
        "baseName": "modificationDate",
        "type": "Date"
    },
    {
        "name": "number",
        "baseName": "number",
        "type": "string"
    },
    {
        "name": "owner",
        "baseName": "owner",
        "type": "OwnerEntity"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Document.TypeEnum"
    }
];
(function (Document) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BankStatement"] = "bankStatement";
        TypeEnum["DriversLicense"] = "driversLicense";
        TypeEnum["IdentityCard"] = "identityCard";
        TypeEnum["NationalIdNumber"] = "nationalIdNumber";
        TypeEnum["Passport"] = "passport";
        TypeEnum["ProofOfAddress"] = "proofOfAddress";
        TypeEnum["ProofOfNationalIdNumber"] = "proofOfNationalIdNumber";
        TypeEnum["ProofOfResidency"] = "proofOfResidency";
        TypeEnum["RegistrationDocument"] = "registrationDocument";
        TypeEnum["VatDocument"] = "vatDocument";
        TypeEnum["ProofOfOrganizationTaxInfo"] = "proofOfOrganizationTaxInfo";
        TypeEnum["ProofOfIndustry"] = "proofOfIndustry";
        TypeEnum["ConstitutionalDocument"] = "constitutionalDocument";
        TypeEnum["ProofOfFundingOrWealthSource"] = "proofOfFundingOrWealthSource";
    })(TypeEnum = Document.TypeEnum || (Document.TypeEnum = {}));
})(Document = exports.Document || (exports.Document = {}));
