// @adyen/api-library/lib/src/typings/legalEntityManagement/generatePciDescriptionRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeneratePciDescriptionRequest = void 0;
class GeneratePciDescriptionRequest {
    static getAttributeTypeMap() {
        return GeneratePciDescriptionRequest.attributeTypeMap;
    }
}
exports.GeneratePciDescriptionRequest = GeneratePciDescriptionRequest;
GeneratePciDescriptionRequest.discriminator = undefined;
GeneratePciDescriptionRequest.attributeTypeMap = [
    {
        "name": "additionalSalesChannels",
        "baseName": "additionalSalesChannels",
        "type": "Array<GeneratePciDescriptionRequest.AdditionalSalesChannelsEnum>"
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "string"
    }
];
(function (GeneratePciDescriptionRequest) {
    let AdditionalSalesChannelsEnum;
    (function (AdditionalSalesChannelsEnum) {
        AdditionalSalesChannelsEnum["ECommerce"] = "eCommerce";
        AdditionalSalesChannelsEnum["EcomMoto"] = "ecomMoto";
        AdditionalSalesChannelsEnum["Pos"] = "pos";
        AdditionalSalesChannelsEnum["PosMoto"] = "posMoto";
    })(AdditionalSalesChannelsEnum = GeneratePciDescriptionRequest.AdditionalSalesChannelsEnum || (GeneratePciDescriptionRequest.AdditionalSalesChannelsEnum = {}));
})(GeneratePciDescriptionRequest = exports.GeneratePciDescriptionRequest || (exports.GeneratePciDescriptionRequest = {}));
