// @adyen/api-library/lib/src/typings/legalEntityManagement/getTermsOfServiceDocumentResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTermsOfServiceDocumentResponse = void 0;
class GetTermsOfServiceDocumentResponse {
    static getAttributeTypeMap() {
        return GetTermsOfServiceDocumentResponse.attributeTypeMap;
    }
}
exports.GetTermsOfServiceDocumentResponse = GetTermsOfServiceDocumentResponse;
GetTermsOfServiceDocumentResponse.discriminator = undefined;
GetTermsOfServiceDocumentResponse.attributeTypeMap = [
    {
        "name": "document",
        "baseName": "document",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "string"
    },
    {
        "name": "termsOfServiceDocumentId",
        "baseName": "termsOfServiceDocumentId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "GetTermsOfServiceDocumentResponse.TypeEnum"
    }
];
(function (GetTermsOfServiceDocumentResponse) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AdyenAccount"] = "adyenAccount";
        TypeEnum["AdyenCapital"] = "adyenCapital";
        TypeEnum["AdyenCard"] = "adyenCard";
        TypeEnum["AdyenForPlatformsAdvanced"] = "adyenForPlatformsAdvanced";
        TypeEnum["AdyenForPlatformsManage"] = "adyenForPlatformsManage";
        TypeEnum["AdyenFranchisee"] = "adyenFranchisee";
        TypeEnum["AdyenIssuing"] = "adyenIssuing";
    })(TypeEnum = GetTermsOfServiceDocumentResponse.TypeEnum || (GetTermsOfServiceDocumentResponse.TypeEnum = {}));
})(GetTermsOfServiceDocumentResponse = exports.GetTermsOfServiceDocumentResponse || (exports.GetTermsOfServiceDocumentResponse = {}));
