// @adyen/api-library/lib/src/typings/legalEntityManagement/legalEntityCapability.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalEntityCapability = void 0;
class LegalEntityCapability {
    static getAttributeTypeMap() {
        return LegalEntityCapability.attributeTypeMap;
    }
}
exports.LegalEntityCapability = LegalEntityCapability;
LegalEntityCapability.discriminator = undefined;
LegalEntityCapability.attributeTypeMap = [
    {
        "name": "allowed",
        "baseName": "allowed",
        "type": "boolean"
    },
    {
        "name": "allowedLevel",
        "baseName": "allowedLevel",
        "type": "LegalEntityCapability.AllowedLevelEnum"
    },
    {
        "name": "allowedSettings",
        "baseName": "allowedSettings",
        "type": "CapabilitySettings"
    },
    {
        "name": "requested",
        "baseName": "requested",
        "type": "boolean"
    },
    {
        "name": "requestedLevel",
        "baseName": "requestedLevel",
        "type": "LegalEntityCapability.RequestedLevelEnum"
    },
    {
        "name": "requestedSettings",
        "baseName": "requestedSettings",
        "type": "CapabilitySettings"
    },
    {
        "name": "transferInstruments",
        "baseName": "transferInstruments",
        "type": "Array<SupportingEntityCapability>"
    },
    {
        "name": "verificationStatus",
        "baseName": "verificationStatus",
        "type": "string"
    }
];
(function (LegalEntityCapability) {
    let AllowedLevelEnum;
    (function (AllowedLevelEnum) {
        AllowedLevelEnum["High"] = "high";
        AllowedLevelEnum["Low"] = "low";
        AllowedLevelEnum["Medium"] = "medium";
        AllowedLevelEnum["NotApplicable"] = "notApplicable";
    })(AllowedLevelEnum = LegalEntityCapability.AllowedLevelEnum || (LegalEntityCapability.AllowedLevelEnum = {}));
    let RequestedLevelEnum;
    (function (RequestedLevelEnum) {
        RequestedLevelEnum["High"] = "high";
        RequestedLevelEnum["Low"] = "low";
        RequestedLevelEnum["Medium"] = "medium";
        RequestedLevelEnum["NotApplicable"] = "notApplicable";
    })(RequestedLevelEnum = LegalEntityCapability.RequestedLevelEnum || (LegalEntityCapability.RequestedLevelEnum = {}));
})(LegalEntityCapability = exports.LegalEntityCapability || (exports.LegalEntityCapability = {}));
