// @adyen/api-library/lib/src/typings/legalEntityManagement/termsOfServiceAcceptanceInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TermsOfServiceAcceptanceInfo = void 0;
class TermsOfServiceAcceptanceInfo {
    static getAttributeTypeMap() {
        return TermsOfServiceAcceptanceInfo.attributeTypeMap;
    }
}
exports.TermsOfServiceAcceptanceInfo = TermsOfServiceAcceptanceInfo;
TermsOfServiceAcceptanceInfo.discriminator = undefined;
TermsOfServiceAcceptanceInfo.attributeTypeMap = [
    {
        "name": "acceptedBy",
        "baseName": "acceptedBy",
        "type": "string"
    },
    {
        "name": "acceptedFor",
        "baseName": "acceptedFor",
        "type": "string"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TermsOfServiceAcceptanceInfo.TypeEnum"
    }
];
(function (TermsOfServiceAcceptanceInfo) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AdyenAccount"] = "adyenAccount";
        TypeEnum["AdyenCapital"] = "adyenCapital";
        TypeEnum["AdyenCard"] = "adyenCard";
        TypeEnum["AdyenForPlatformsAdvanced"] = "adyenForPlatformsAdvanced";
        TypeEnum["AdyenForPlatformsManage"] = "adyenForPlatformsManage";
        TypeEnum["AdyenFranchisee"] = "adyenFranchisee";
        TypeEnum["AdyenIssuing"] = "adyenIssuing";
    })(TypeEnum = TermsOfServiceAcceptanceInfo.TypeEnum || (TermsOfServiceAcceptanceInfo.TypeEnum = {}));
})(TermsOfServiceAcceptanceInfo = exports.TermsOfServiceAcceptanceInfo || (exports.TermsOfServiceAcceptanceInfo = {}));
