// @adyen/api-library/lib/src/typings/legalEntityManagement/transferInstrumentInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferInstrumentInfo = void 0;
class TransferInstrumentInfo {
    static getAttributeTypeMap() {
        return TransferInstrumentInfo.attributeTypeMap;
    }
}
exports.TransferInstrumentInfo = TransferInstrumentInfo;
TransferInstrumentInfo.discriminator = undefined;
TransferInstrumentInfo.attributeTypeMap = [
    {
        "name": "bankAccount",
        "baseName": "bankAccount",
        "type": "BankAccountInfo"
    },
    {
        "name": "legalEntityId",
        "baseName": "legalEntityId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransferInstrumentInfo.TypeEnum"
    }
];
(function (TransferInstrumentInfo) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BankAccount"] = "bankAccount";
        TypeEnum["RecurringDetail"] = "recurringDetail";
    })(TypeEnum = TransferInstrumentInfo.TypeEnum || (TransferInstrumentInfo.TypeEnum = {}));
})(TransferInstrumentInfo = exports.TransferInstrumentInfo || (exports.TransferInstrumentInfo = {}));
