// @adyen/api-library/lib/src/typings/legalEntityManagement/trust.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Trust = void 0;
class Trust {
    static getAttributeTypeMap() {
        return Trust.attributeTypeMap;
    }
}
exports.Trust = Trust;
Trust.discriminator = undefined;
Trust.attributeTypeMap = [
    {
        "name": "countryOfGoverningLaw",
        "baseName": "countryOfGoverningLaw",
        "type": "string"
    },
    {
        "name": "dateOfIncorporation",
        "baseName": "dateOfIncorporation",
        "type": "string"
    },
    {
        "name": "doingBusinessAs",
        "baseName": "doingBusinessAs",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "principalPlaceOfBusiness",
        "baseName": "principalPlaceOfBusiness",
        "type": "Address"
    },
    {
        "name": "registeredAddress",
        "baseName": "registeredAddress",
        "type": "Address"
    },
    {
        "name": "registrationNumber",
        "baseName": "registrationNumber",
        "type": "string"
    },
    {
        "name": "taxInformation",
        "baseName": "taxInformation",
        "type": "Array<TaxInformation>"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Trust.TypeEnum"
    },
    {
        "name": "undefinedBeneficiaryInfo",
        "baseName": "undefinedBeneficiaryInfo",
        "type": "Array<UndefinedBeneficiary>"
    },
    {
        "name": "vatAbsenceReason",
        "baseName": "vatAbsenceReason",
        "type": "Trust.VatAbsenceReasonEnum"
    },
    {
        "name": "vatNumber",
        "baseName": "vatNumber",
        "type": "string"
    }
];
(function (Trust) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["CashManagementTrust"] = "cashManagementTrust";
        TypeEnum["CorporateUnitTrust"] = "corporateUnitTrust";
        TypeEnum["DeceasedEstate"] = "deceasedEstate";
        TypeEnum["DiscretionaryInvestmentTrust"] = "discretionaryInvestmentTrust";
        TypeEnum["DiscretionaryServicesManagementTrust"] = "discretionaryServicesManagementTrust";
        TypeEnum["DiscretionaryTradingTrust"] = "discretionaryTradingTrust";
        TypeEnum["FirstHomeSaverAccountsTrust"] = "firstHomeSaverAccountsTrust";
        TypeEnum["FixedTrust"] = "fixedTrust";
        TypeEnum["FixedUnitTrust"] = "fixedUnitTrust";
        TypeEnum["HybridTrust"] = "hybridTrust";
        TypeEnum["ListedPublicUnitTrust"] = "listedPublicUnitTrust";
        TypeEnum["OtherTrust"] = "otherTrust";
        TypeEnum["PooledSuperannuationTrust"] = "pooledSuperannuationTrust";
        TypeEnum["PublicTradingTrust"] = "publicTradingTrust";
        TypeEnum["UnlistedPublicUnitTrust"] = "unlistedPublicUnitTrust";
    })(TypeEnum = Trust.TypeEnum || (Trust.TypeEnum = {}));
    let VatAbsenceReasonEnum;
    (function (VatAbsenceReasonEnum) {
        VatAbsenceReasonEnum["IndustryExemption"] = "industryExemption";
        VatAbsenceReasonEnum["BelowTaxThreshold"] = "belowTaxThreshold";
    })(VatAbsenceReasonEnum = Trust.VatAbsenceReasonEnum || (Trust.VatAbsenceReasonEnum = {}));
})(Trust = exports.Trust || (exports.Trust = {}));
