// @adyen/api-library/lib/src/typings/storedValue/models.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectSerializer = void 0;
__exportStar(require("./amount"), exports);
__exportStar(require("./serviceError"), exports);
__exportStar(require("./storedValueBalanceCheckRequest"), exports);
__exportStar(require("./storedValueBalanceCheckResponse"), exports);
__exportStar(require("./storedValueBalanceMergeRequest"), exports);
__exportStar(require("./storedValueBalanceMergeResponse"), exports);
__exportStar(require("./storedValueIssueRequest"), exports);
__exportStar(require("./storedValueIssueResponse"), exports);
__exportStar(require("./storedValueLoadRequest"), exports);
__exportStar(require("./storedValueLoadResponse"), exports);
__exportStar(require("./storedValueStatusChangeRequest"), exports);
__exportStar(require("./storedValueStatusChangeResponse"), exports);
__exportStar(require("./storedValueVoidRequest"), exports);
__exportStar(require("./storedValueVoidResponse"), exports);
const amount_1 = require("./amount");
const serviceError_1 = require("./serviceError");
const storedValueBalanceCheckRequest_1 = require("./storedValueBalanceCheckRequest");
const storedValueBalanceCheckResponse_1 = require("./storedValueBalanceCheckResponse");
const storedValueBalanceMergeRequest_1 = require("./storedValueBalanceMergeRequest");
const storedValueBalanceMergeResponse_1 = require("./storedValueBalanceMergeResponse");
const storedValueIssueRequest_1 = require("./storedValueIssueRequest");
const storedValueIssueResponse_1 = require("./storedValueIssueResponse");
const storedValueLoadRequest_1 = require("./storedValueLoadRequest");
const storedValueLoadResponse_1 = require("./storedValueLoadResponse");
const storedValueStatusChangeRequest_1 = require("./storedValueStatusChangeRequest");
const storedValueStatusChangeResponse_1 = require("./storedValueStatusChangeResponse");
const storedValueVoidRequest_1 = require("./storedValueVoidRequest");
const storedValueVoidResponse_1 = require("./storedValueVoidResponse");
/* tslint:disable:no-unused-variable */
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
let enumsMap = {
    "StoredValueBalanceCheckRequest.ShopperInteractionEnum": storedValueBalanceCheckRequest_1.StoredValueBalanceCheckRequest.ShopperInteractionEnum,
    "StoredValueBalanceCheckResponse.ResultCodeEnum": storedValueBalanceCheckResponse_1.StoredValueBalanceCheckResponse.ResultCodeEnum,
    "StoredValueBalanceMergeRequest.ShopperInteractionEnum": storedValueBalanceMergeRequest_1.StoredValueBalanceMergeRequest.ShopperInteractionEnum,
    "StoredValueBalanceMergeResponse.ResultCodeEnum": storedValueBalanceMergeResponse_1.StoredValueBalanceMergeResponse.ResultCodeEnum,
    "StoredValueIssueRequest.ShopperInteractionEnum": storedValueIssueRequest_1.StoredValueIssueRequest.ShopperInteractionEnum,
    "StoredValueIssueResponse.ResultCodeEnum": storedValueIssueResponse_1.StoredValueIssueResponse.ResultCodeEnum,
    "StoredValueLoadRequest.LoadTypeEnum": storedValueLoadRequest_1.StoredValueLoadRequest.LoadTypeEnum,
    "StoredValueLoadRequest.ShopperInteractionEnum": storedValueLoadRequest_1.StoredValueLoadRequest.ShopperInteractionEnum,
    "StoredValueLoadResponse.ResultCodeEnum": storedValueLoadResponse_1.StoredValueLoadResponse.ResultCodeEnum,
    "StoredValueStatusChangeRequest.ShopperInteractionEnum": storedValueStatusChangeRequest_1.StoredValueStatusChangeRequest.ShopperInteractionEnum,
    "StoredValueStatusChangeRequest.StatusEnum": storedValueStatusChangeRequest_1.StoredValueStatusChangeRequest.StatusEnum,
    "StoredValueStatusChangeResponse.ResultCodeEnum": storedValueStatusChangeResponse_1.StoredValueStatusChangeResponse.ResultCodeEnum,
    "StoredValueVoidResponse.ResultCodeEnum": storedValueVoidResponse_1.StoredValueVoidResponse.ResultCodeEnum,
};
let typeMap = {
    "Amount": amount_1.Amount,
    "ServiceError": serviceError_1.ServiceError,
    "StoredValueBalanceCheckRequest": storedValueBalanceCheckRequest_1.StoredValueBalanceCheckRequest,
    "StoredValueBalanceCheckResponse": storedValueBalanceCheckResponse_1.StoredValueBalanceCheckResponse,
    "StoredValueBalanceMergeRequest": storedValueBalanceMergeRequest_1.StoredValueBalanceMergeRequest,
    "StoredValueBalanceMergeResponse": storedValueBalanceMergeResponse_1.StoredValueBalanceMergeResponse,
    "StoredValueIssueRequest": storedValueIssueRequest_1.StoredValueIssueRequest,
    "StoredValueIssueResponse": storedValueIssueResponse_1.StoredValueIssueResponse,
    "StoredValueLoadRequest": storedValueLoadRequest_1.StoredValueLoadRequest,
    "StoredValueLoadResponse": storedValueLoadResponse_1.StoredValueLoadResponse,
    "StoredValueStatusChangeRequest": storedValueStatusChangeRequest_1.StoredValueStatusChangeRequest,
    "StoredValueStatusChangeResponse": storedValueStatusChangeResponse_1.StoredValueStatusChangeResponse,
    "StoredValueVoidRequest": storedValueVoidRequest_1.StoredValueVoidRequest,
    "StoredValueVoidResponse": storedValueVoidResponse_1.StoredValueVoidResponse,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }
            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType; // use the type given in the discriminator
                    }
                    else {
                        return expectedType; // discriminator did not map to a type
                    }
                }
                else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.serialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return data.toISOString();
        }
        else if (type === "SaleToAcquirerData") {
            const dataString = JSON.stringify(data);
            return Buffer.from(dataString).toString("base64");
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) { // in case we dont know the type
                return data;
            }
            // Get the actual type of this object
            type = this.findCorrectType(data, type);
            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.deserialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) { // is Enum
                return data;
            }
            if (!typeMap[type]) { // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
