// @adyen/api-library/lib/src/typings/storedValue/storedValueVoidResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueVoidResponse = void 0;
class StoredValueVoidResponse {
    static getAttributeTypeMap() {
        return StoredValueVoidResponse.attributeTypeMap;
    }
}
exports.StoredValueVoidResponse = StoredValueVoidResponse;
StoredValueVoidResponse.discriminator = undefined;
StoredValueVoidResponse.attributeTypeMap = [
    {
        "name": "currentBalance",
        "baseName": "currentBalance",
        "type": "Amount"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "StoredValueVoidResponse.ResultCodeEnum"
    },
    {
        "name": "thirdPartyRefusalReason",
        "baseName": "thirdPartyRefusalReason",
        "type": "string"
    }
];
(function (StoredValueVoidResponse) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["Success"] = "Success";
        ResultCodeEnum["Refused"] = "Refused";
        ResultCodeEnum["Error"] = "Error";
        ResultCodeEnum["NotEnoughBalance"] = "NotEnoughBalance";
    })(ResultCodeEnum = StoredValueVoidResponse.ResultCodeEnum || (StoredValueVoidResponse.ResultCodeEnum = {}));
})(StoredValueVoidResponse = exports.StoredValueVoidResponse || (exports.StoredValueVoidResponse = {}));
