// @adyen/api-library/lib/src/typings/storedValue/storedValueBalanceMergeRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueBalanceMergeRequest = void 0;
class StoredValueBalanceMergeRequest {
    static getAttributeTypeMap() {
        return StoredValueBalanceMergeRequest.attributeTypeMap;
    }
}
exports.StoredValueBalanceMergeRequest = StoredValueBalanceMergeRequest;
StoredValueBalanceMergeRequest.discriminator = undefined;
StoredValueBalanceMergeRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "StoredValueBalanceMergeRequest.ShopperInteractionEnum"
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string"
    },
    {
        "name": "sourcePaymentMethod",
        "baseName": "sourcePaymentMethod",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string"
    }
];
(function (StoredValueBalanceMergeRequest) {
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = StoredValueBalanceMergeRequest.ShopperInteractionEnum || (StoredValueBalanceMergeRequest.ShopperInteractionEnum = {}));
})(StoredValueBalanceMergeRequest = exports.StoredValueBalanceMergeRequest || (exports.StoredValueBalanceMergeRequest = {}));
