// @adyen/api-library/lib/src/typings/storedValue/storedValueLoadResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueLoadResponse = void 0;
class StoredValueLoadResponse {
    static getAttributeTypeMap() {
        return StoredValueLoadResponse.attributeTypeMap;
    }
}
exports.StoredValueLoadResponse = StoredValueLoadResponse;
StoredValueLoadResponse.discriminator = undefined;
StoredValueLoadResponse.attributeTypeMap = [
    {
        "name": "authCode",
        "baseName": "authCode",
        "type": "string"
    },
    {
        "name": "currentBalance",
        "baseName": "currentBalance",
        "type": "Amount"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "StoredValueLoadResponse.ResultCodeEnum"
    },
    {
        "name": "thirdPartyRefusalReason",
        "baseName": "thirdPartyRefusalReason",
        "type": "string"
    }
];
(function (StoredValueLoadResponse) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["Success"] = "Success";
        ResultCodeEnum["Refused"] = "Refused";
        ResultCodeEnum["Error"] = "Error";
        ResultCodeEnum["NotEnoughBalance"] = "NotEnoughBalance";
    })(ResultCodeEnum = StoredValueLoadResponse.ResultCodeEnum || (StoredValueLoadResponse.ResultCodeEnum = {}));
})(StoredValueLoadResponse = exports.StoredValueLoadResponse || (exports.StoredValueLoadResponse = {}));
