// @adyen/api-library/lib/src/typings/storedValue/storedValueIssueResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueIssueResponse = void 0;
class StoredValueIssueResponse {
    static getAttributeTypeMap() {
        return StoredValueIssueResponse.attributeTypeMap;
    }
}
exports.StoredValueIssueResponse = StoredValueIssueResponse;
StoredValueIssueResponse.discriminator = undefined;
StoredValueIssueResponse.attributeTypeMap = [
    {
        "name": "authCode",
        "baseName": "authCode",
        "type": "string"
    },
    {
        "name": "currentBalance",
        "baseName": "currentBalance",
        "type": "Amount"
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "StoredValueIssueResponse.ResultCodeEnum"
    },
    {
        "name": "thirdPartyRefusalReason",
        "baseName": "thirdPartyRefusalReason",
        "type": "string"
    }
];
(function (StoredValueIssueResponse) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["Success"] = "Success";
        ResultCodeEnum["Refused"] = "Refused";
        ResultCodeEnum["Error"] = "Error";
        ResultCodeEnum["NotEnoughBalance"] = "NotEnoughBalance";
    })(ResultCodeEnum = StoredValueIssueResponse.ResultCodeEnum || (StoredValueIssueResponse.ResultCodeEnum = {}));
})(StoredValueIssueResponse = exports.StoredValueIssueResponse || (exports.StoredValueIssueResponse = {}));
