// @adyen/api-library/lib/src/typings/platformsAccount/accountHolderStatus.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountHolderStatus = void 0;
class AccountHolderStatus {
    static getAttributeTypeMap() {
        return AccountHolderStatus.attributeTypeMap;
    }
}
exports.AccountHolderStatus = AccountHolderStatus;
AccountHolderStatus.discriminator = undefined;
AccountHolderStatus.attributeTypeMap = [
    {
        "name": "events",
        "baseName": "events",
        "type": "Array<AccountEvent>"
    },
    {
        "name": "payoutState",
        "baseName": "payoutState",
        "type": "AccountPayoutState"
    },
    {
        "name": "processingState",
        "baseName": "processingState",
        "type": "AccountProcessingState"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "AccountHolderStatus.StatusEnum"
    },
    {
        "name": "statusReason",
        "baseName": "statusReason",
        "type": "string"
    }
];
(function (AccountHolderStatus) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "Active";
        StatusEnum["Closed"] = "Closed";
        StatusEnum["Inactive"] = "Inactive";
        StatusEnum["Suspended"] = "Suspended";
    })(StatusEnum = AccountHolderStatus.StatusEnum || (AccountHolderStatus.StatusEnum = {}));
})(AccountHolderStatus = exports.AccountHolderStatus || (exports.AccountHolderStatus = {}));
