// @adyen/api-library/lib/src/typings/platformsAccount/closeAccountResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloseAccountResponse = void 0;
class CloseAccountResponse {
    static getAttributeTypeMap() {
        return CloseAccountResponse.attributeTypeMap;
    }
}
exports.CloseAccountResponse = CloseAccountResponse;
CloseAccountResponse.discriminator = undefined;
CloseAccountResponse.attributeTypeMap = [
    {
        "name": "accountCode",
        "baseName": "accountCode",
        "type": "string"
    },
    {
        "name": "invalidFields",
        "baseName": "invalidFields",
        "type": "Array<ErrorFieldType>"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "CloseAccountResponse.StatusEnum"
    }
];
(function (CloseAccountResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "Active";
        StatusEnum["Closed"] = "Closed";
        StatusEnum["Inactive"] = "Inactive";
        StatusEnum["Suspended"] = "Suspended";
    })(StatusEnum = CloseAccountResponse.StatusEnum || (CloseAccountResponse.StatusEnum = {}));
})(CloseAccountResponse = exports.CloseAccountResponse || (exports.CloseAccountResponse = {}));
