// @adyen/api-library/lib/src/typings/platformsAccount/payoutScheduleResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayoutScheduleResponse = void 0;
class PayoutScheduleResponse {
    static getAttributeTypeMap() {
        return PayoutScheduleResponse.attributeTypeMap;
    }
}
exports.PayoutScheduleResponse = PayoutScheduleResponse;
PayoutScheduleResponse.discriminator = undefined;
PayoutScheduleResponse.attributeTypeMap = [
    {
        "name": "nextScheduledPayout",
        "baseName": "nextScheduledPayout",
        "type": "Date"
    },
    {
        "name": "schedule",
        "baseName": "schedule",
        "type": "PayoutScheduleResponse.ScheduleEnum"
    }
];
(function (PayoutScheduleResponse) {
    let ScheduleEnum;
    (function (ScheduleEnum) {
        ScheduleEnum["BiweeklyOn1StAnd15ThAtMidnight"] = "BIWEEKLY_ON_1ST_AND_15TH_AT_MIDNIGHT";
        ScheduleEnum["Daily"] = "DAILY";
        ScheduleEnum["DailyAu"] = "DAILY_AU";
        ScheduleEnum["DailyEu"] = "DAILY_EU";
        ScheduleEnum["DailySg"] = "DAILY_SG";
        ScheduleEnum["DailyUs"] = "DAILY_US";
        ScheduleEnum["Hold"] = "HOLD";
        ScheduleEnum["Monthly"] = "MONTHLY";
        ScheduleEnum["Weekly"] = "WEEKLY";
        ScheduleEnum["WeeklyMonToFriAu"] = "WEEKLY_MON_TO_FRI_AU";
        ScheduleEnum["WeeklyMonToFriEu"] = "WEEKLY_MON_TO_FRI_EU";
        ScheduleEnum["WeeklyMonToFriUs"] = "WEEKLY_MON_TO_FRI_US";
        ScheduleEnum["WeeklyOnTueFriMidnight"] = "WEEKLY_ON_TUE_FRI_MIDNIGHT";
        ScheduleEnum["WeeklySunToThuAu"] = "WEEKLY_SUN_TO_THU_AU";
        ScheduleEnum["WeeklySunToThuUs"] = "WEEKLY_SUN_TO_THU_US";
    })(ScheduleEnum = PayoutScheduleResponse.ScheduleEnum || (PayoutScheduleResponse.ScheduleEnum = {}));
})(PayoutScheduleResponse = exports.PayoutScheduleResponse || (exports.PayoutScheduleResponse = {}));
