// @adyen/api-library/lib/src/typings/platformsAccount/updatePayoutScheduleRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatePayoutScheduleRequest = void 0;
class UpdatePayoutScheduleRequest {
    static getAttributeTypeMap() {
        return UpdatePayoutScheduleRequest.attributeTypeMap;
    }
}
exports.UpdatePayoutScheduleRequest = UpdatePayoutScheduleRequest;
UpdatePayoutScheduleRequest.discriminator = undefined;
UpdatePayoutScheduleRequest.attributeTypeMap = [
    {
        "name": "action",
        "baseName": "action",
        "type": "UpdatePayoutScheduleRequest.ActionEnum"
    },
    {
        "name": "reason",
        "baseName": "reason",
        "type": "string"
    },
    {
        "name": "schedule",
        "baseName": "schedule",
        "type": "UpdatePayoutScheduleRequest.ScheduleEnum"
    }
];
(function (UpdatePayoutScheduleRequest) {
    let ActionEnum;
    (function (ActionEnum) {
        ActionEnum["Close"] = "CLOSE";
        ActionEnum["Nothing"] = "NOTHING";
        ActionEnum["Update"] = "UPDATE";
    })(ActionEnum = UpdatePayoutScheduleRequest.ActionEnum || (UpdatePayoutScheduleRequest.ActionEnum = {}));
    let ScheduleEnum;
    (function (ScheduleEnum) {
        ScheduleEnum["BiweeklyOn1StAnd15ThAtMidnight"] = "BIWEEKLY_ON_1ST_AND_15TH_AT_MIDNIGHT";
        ScheduleEnum["Daily"] = "DAILY";
        ScheduleEnum["DailyAu"] = "DAILY_AU";
        ScheduleEnum["DailyEu"] = "DAILY_EU";
        ScheduleEnum["DailySg"] = "DAILY_SG";
        ScheduleEnum["DailyUs"] = "DAILY_US";
        ScheduleEnum["Hold"] = "HOLD";
        ScheduleEnum["Monthly"] = "MONTHLY";
        ScheduleEnum["Weekly"] = "WEEKLY";
        ScheduleEnum["WeeklyMonToFriAu"] = "WEEKLY_MON_TO_FRI_AU";
        ScheduleEnum["WeeklyMonToFriEu"] = "WEEKLY_MON_TO_FRI_EU";
        ScheduleEnum["WeeklyMonToFriUs"] = "WEEKLY_MON_TO_FRI_US";
        ScheduleEnum["WeeklyOnTueFriMidnight"] = "WEEKLY_ON_TUE_FRI_MIDNIGHT";
        ScheduleEnum["WeeklySunToThuAu"] = "WEEKLY_SUN_TO_THU_AU";
        ScheduleEnum["WeeklySunToThuUs"] = "WEEKLY_SUN_TO_THU_US";
    })(ScheduleEnum = UpdatePayoutScheduleRequest.ScheduleEnum || (UpdatePayoutScheduleRequest.ScheduleEnum = {}));
})(UpdatePayoutScheduleRequest = exports.UpdatePayoutScheduleRequest || (exports.UpdatePayoutScheduleRequest = {}));
