// @adyen/api-library/lib/src/typings/platformsAccount/updateAccountRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAccountRequest = void 0;
class UpdateAccountRequest {
    static getAttributeTypeMap() {
        return UpdateAccountRequest.attributeTypeMap;
    }
}
exports.UpdateAccountRequest = UpdateAccountRequest;
UpdateAccountRequest.discriminator = undefined;
UpdateAccountRequest.attributeTypeMap = [
    {
        "name": "accountCode",
        "baseName": "accountCode",
        "type": "string"
    },
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "payoutMethodCode",
        "baseName": "payoutMethodCode",
        "type": "string"
    },
    {
        "name": "payoutSchedule",
        "baseName": "payoutSchedule",
        "type": "UpdatePayoutScheduleRequest"
    },
    {
        "name": "payoutSpeed",
        "baseName": "payoutSpeed",
        "type": "UpdateAccountRequest.PayoutSpeedEnum"
    }
];
(function (UpdateAccountRequest) {
    let PayoutSpeedEnum;
    (function (PayoutSpeedEnum) {
        PayoutSpeedEnum["Instant"] = "INSTANT";
        PayoutSpeedEnum["SameDay"] = "SAME_DAY";
        PayoutSpeedEnum["Standard"] = "STANDARD";
    })(PayoutSpeedEnum = UpdateAccountRequest.PayoutSpeedEnum || (UpdateAccountRequest.PayoutSpeedEnum = {}));
})(UpdateAccountRequest = exports.UpdateAccountRequest || (exports.UpdateAccountRequest = {}));
