// @adyen/api-library/lib/src/typings/platformsAccount/viasPhoneNumber.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViasPhoneNumber = void 0;
class ViasPhoneNumber {
    static getAttributeTypeMap() {
        return ViasPhoneNumber.attributeTypeMap;
    }
}
exports.ViasPhoneNumber = ViasPhoneNumber;
ViasPhoneNumber.discriminator = undefined;
ViasPhoneNumber.attributeTypeMap = [
    {
        "name": "phoneCountryCode",
        "baseName": "phoneCountryCode",
        "type": "string"
    },
    {
        "name": "phoneNumber",
        "baseName": "phoneNumber",
        "type": "string"
    },
    {
        "name": "phoneType",
        "baseName": "phoneType",
        "type": "ViasPhoneNumber.PhoneTypeEnum"
    }
];
(function (ViasPhoneNumber) {
    let PhoneTypeEnum;
    (function (PhoneTypeEnum) {
        PhoneTypeEnum["Fax"] = "Fax";
        PhoneTypeEnum["Landline"] = "Landline";
        PhoneTypeEnum["Mobile"] = "Mobile";
        PhoneTypeEnum["Sip"] = "SIP";
    })(PhoneTypeEnum = ViasPhoneNumber.PhoneTypeEnum || (ViasPhoneNumber.PhoneTypeEnum = {}));
})(ViasPhoneNumber = exports.ViasPhoneNumber || (exports.ViasPhoneNumber = {}));
