// @adyen/api-library/lib/src/typings/checkout/achDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchDetails = void 0;
class AchDetails {
    static getAttributeTypeMap() {
        return AchDetails.attributeTypeMap;
    }
}
exports.AchDetails = AchDetails;
AchDetails.discriminator = undefined;
AchDetails.attributeTypeMap = [
    {
        "name": "bankAccountNumber",
        "baseName": "bankAccountNumber",
        "type": "string"
    },
    {
        "name": "bankAccountType",
        "baseName": "bankAccountType",
        "type": "AchDetails.BankAccountTypeEnum"
    },
    {
        "name": "bankLocationId",
        "baseName": "bankLocationId",
        "type": "string"
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "encryptedBankAccountNumber",
        "baseName": "encryptedBankAccountNumber",
        "type": "string"
    },
    {
        "name": "encryptedBankLocationId",
        "baseName": "encryptedBankLocationId",
        "type": "string"
    },
    {
        "name": "ownerName",
        "baseName": "ownerName",
        "type": "string"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "AchDetails.TypeEnum"
    }
];
(function (AchDetails) {
    let BankAccountTypeEnum;
    (function (BankAccountTypeEnum) {
        BankAccountTypeEnum["Balance"] = "balance";
        BankAccountTypeEnum["Checking"] = "checking";
        BankAccountTypeEnum["Deposit"] = "deposit";
        BankAccountTypeEnum["General"] = "general";
        BankAccountTypeEnum["Other"] = "other";
        BankAccountTypeEnum["Payment"] = "payment";
        BankAccountTypeEnum["Savings"] = "savings";
    })(BankAccountTypeEnum = AchDetails.BankAccountTypeEnum || (AchDetails.BankAccountTypeEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Ach"] = "ach";
        TypeEnum["AchPlaid"] = "ach_plaid";
    })(TypeEnum = AchDetails.TypeEnum || (AchDetails.TypeEnum = {}));
})(AchDetails = exports.AchDetails || (exports.AchDetails = {}));
