// @adyen/api-library/lib/src/typings/checkout/applePayDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplePayDetails = void 0;
class ApplePayDetails {
    static getAttributeTypeMap() {
        return ApplePayDetails.attributeTypeMap;
    }
}
exports.ApplePayDetails = ApplePayDetails;
ApplePayDetails.discriminator = undefined;
ApplePayDetails.attributeTypeMap = [
    {
        "name": "applePayToken",
        "baseName": "applePayToken",
        "type": "string"
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "ApplePayDetails.FundingSourceEnum"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "ApplePayDetails.TypeEnum"
    }
];
(function (ApplePayDetails) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = ApplePayDetails.FundingSourceEnum || (ApplePayDetails.FundingSourceEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Applepay"] = "applepay";
    })(TypeEnum = ApplePayDetails.TypeEnum || (ApplePayDetails.TypeEnum = {}));
})(ApplePayDetails = exports.ApplePayDetails || (exports.ApplePayDetails = {}));
