// @adyen/api-library/lib/src/typings/checkout/avs.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Avs = void 0;
class Avs {
    static getAttributeTypeMap() {
        return Avs.attributeTypeMap;
    }
}
exports.Avs = Avs;
Avs.discriminator = undefined;
Avs.attributeTypeMap = [
    {
        "name": "addressEditable",
        "baseName": "addressEditable",
        "type": "boolean"
    },
    {
        "name": "enabled",
        "baseName": "enabled",
        "type": "Avs.EnabledEnum"
    }
];
(function (Avs) {
    let EnabledEnum;
    (function (EnabledEnum) {
        EnabledEnum["Yes"] = "yes";
        EnabledEnum["No"] = "no";
        EnabledEnum["Automatic"] = "automatic";
    })(EnabledEnum = Avs.EnabledEnum || (Avs.EnabledEnum = {}));
})(Avs = exports.Avs || (exports.Avs = {}));
