// @adyen/api-library/lib/src/typings/checkout/billDeskDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillDeskDetails = void 0;
class BillDeskDetails {
    static getAttributeTypeMap() {
        return BillDeskDetails.attributeTypeMap;
    }
}
exports.BillDeskDetails = BillDeskDetails;
BillDeskDetails.discriminator = undefined;
BillDeskDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "issuer",
        "baseName": "issuer",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "BillDeskDetails.TypeEnum"
    }
];
(function (BillDeskDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BilldeskOnline"] = "billdesk_online";
        TypeEnum["BilldeskWallet"] = "billdesk_wallet";
        TypeEnum["OnlinebankingIn"] = "onlinebanking_IN";
        TypeEnum["WalletIn"] = "wallet_IN";
    })(TypeEnum = BillDeskDetails.TypeEnum || (BillDeskDetails.TypeEnum = {}));
})(BillDeskDetails = exports.BillDeskDetails || (exports.BillDeskDetails = {}));
