// @adyen/api-library/lib/src/typings/checkout/checkoutSessionInstallmentOption.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutSessionInstallmentOption = void 0;
class CheckoutSessionInstallmentOption {
    static getAttributeTypeMap() {
        return CheckoutSessionInstallmentOption.attributeTypeMap;
    }
}
exports.CheckoutSessionInstallmentOption = CheckoutSessionInstallmentOption;
CheckoutSessionInstallmentOption.discriminator = undefined;
CheckoutSessionInstallmentOption.attributeTypeMap = [
    {
        "name": "plans",
        "baseName": "plans",
        "type": "Array<CheckoutSessionInstallmentOption.PlansEnum>"
    },
    {
        "name": "preselectedValue",
        "baseName": "preselectedValue",
        "type": "number"
    },
    {
        "name": "values",
        "baseName": "values",
        "type": "Array<number>"
    }
];
(function (CheckoutSessionInstallmentOption) {
    let PlansEnum;
    (function (PlansEnum) {
        PlansEnum["Regular"] = "regular";
        PlansEnum["Revolving"] = "revolving";
    })(PlansEnum = CheckoutSessionInstallmentOption.PlansEnum || (CheckoutSessionInstallmentOption.PlansEnum = {}));
})(CheckoutSessionInstallmentOption = exports.CheckoutSessionInstallmentOption || (exports.CheckoutSessionInstallmentOption = {}));
