// @adyen/api-library/lib/src/typings/checkout/createCheckoutSessionRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCheckoutSessionRequest = void 0;
class CreateCheckoutSessionRequest {
    static getAttributeTypeMap() {
        return CreateCheckoutSessionRequest.attributeTypeMap;
    }
}
exports.CreateCheckoutSessionRequest = CreateCheckoutSessionRequest;
CreateCheckoutSessionRequest.discriminator = undefined;
CreateCheckoutSessionRequest.attributeTypeMap = [
    {
        "name": "accountInfo",
        "baseName": "accountInfo",
        "type": "AccountInfo"
    },
    {
        "name": "additionalAmount",
        "baseName": "additionalAmount",
        "type": "Amount"
    },
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "allowedPaymentMethods",
        "baseName": "allowedPaymentMethods",
        "type": "Array<string>"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "applicationInfo",
        "baseName": "applicationInfo",
        "type": "ApplicationInfo"
    },
    {
        "name": "authenticationData",
        "baseName": "authenticationData",
        "type": "AuthenticationData"
    },
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "BillingAddress"
    },
    {
        "name": "blockedPaymentMethods",
        "baseName": "blockedPaymentMethods",
        "type": "Array<string>"
    },
    {
        "name": "captureDelayHours",
        "baseName": "captureDelayHours",
        "type": "number"
    },
    {
        "name": "channel",
        "baseName": "channel",
        "type": "CreateCheckoutSessionRequest.ChannelEnum"
    },
    {
        "name": "company",
        "baseName": "company",
        "type": "Company"
    },
    {
        "name": "countryCode",
        "baseName": "countryCode",
        "type": "string"
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "string"
    },
    {
        "name": "deliverAt",
        "baseName": "deliverAt",
        "type": "Date"
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "DeliveryAddress"
    },
    {
        "name": "enableOneClick",
        "baseName": "enableOneClick",
        "type": "boolean"
    },
    {
        "name": "enablePayOut",
        "baseName": "enablePayOut",
        "type": "boolean"
    },
    {
        "name": "enableRecurring",
        "baseName": "enableRecurring",
        "type": "boolean"
    },
    {
        "name": "expiresAt",
        "baseName": "expiresAt",
        "type": "Date"
    },
    {
        "name": "fundOrigin",
        "baseName": "fundOrigin",
        "type": "FundOrigin"
    },
    {
        "name": "fundRecipient",
        "baseName": "fundRecipient",
        "type": "FundRecipient"
    },
    {
        "name": "installmentOptions",
        "baseName": "installmentOptions",
        "type": "{ [key: string]: CheckoutSessionInstallmentOption; }"
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>"
    },
    {
        "name": "mandate",
        "baseName": "mandate",
        "type": "Mandate"
    },
    {
        "name": "mcc",
        "baseName": "mcc",
        "type": "string"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "merchantOrderReference",
        "baseName": "merchantOrderReference",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "mpiData",
        "baseName": "mpiData",
        "type": "ThreeDSecureData"
    },
    {
        "name": "platformChargebackLogic",
        "baseName": "platformChargebackLogic",
        "type": "PlatformChargebackLogic"
    },
    {
        "name": "recurringExpiry",
        "baseName": "recurringExpiry",
        "type": "string"
    },
    {
        "name": "recurringFrequency",
        "baseName": "recurringFrequency",
        "type": "string"
    },
    {
        "name": "recurringProcessingModel",
        "baseName": "recurringProcessingModel",
        "type": "CreateCheckoutSessionRequest.RecurringProcessingModelEnum"
    },
    {
        "name": "redirectFromIssuerMethod",
        "baseName": "redirectFromIssuerMethod",
        "type": "string"
    },
    {
        "name": "redirectToIssuerMethod",
        "baseName": "redirectToIssuerMethod",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "returnUrl",
        "baseName": "returnUrl",
        "type": "string"
    },
    {
        "name": "riskData",
        "baseName": "riskData",
        "type": "RiskData"
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string"
    },
    {
        "name": "shopperIP",
        "baseName": "shopperIP",
        "type": "string"
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "CreateCheckoutSessionRequest.ShopperInteractionEnum"
    },
    {
        "name": "shopperLocale",
        "baseName": "shopperLocale",
        "type": "string"
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name"
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string"
    },
    {
        "name": "shopperStatement",
        "baseName": "shopperStatement",
        "type": "string"
    },
    {
        "name": "socialSecurityNumber",
        "baseName": "socialSecurityNumber",
        "type": "string"
    },
    {
        "name": "splitCardFundingSources",
        "baseName": "splitCardFundingSources",
        "type": "boolean"
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>"
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string"
    },
    {
        "name": "storePaymentMethod",
        "baseName": "storePaymentMethod",
        "type": "boolean"
    },
    {
        "name": "storePaymentMethodMode",
        "baseName": "storePaymentMethodMode",
        "type": "CreateCheckoutSessionRequest.StorePaymentMethodModeEnum"
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string"
    },
    {
        "name": "threeDSAuthenticationOnly",
        "baseName": "threeDSAuthenticationOnly",
        "type": "boolean"
    },
    {
        "name": "trustedShopper",
        "baseName": "trustedShopper",
        "type": "boolean"
    }
];
(function (CreateCheckoutSessionRequest) {
    let ChannelEnum;
    (function (ChannelEnum) {
        ChannelEnum["IOs"] = "iOS";
        ChannelEnum["Android"] = "Android";
        ChannelEnum["Web"] = "Web";
    })(ChannelEnum = CreateCheckoutSessionRequest.ChannelEnum || (CreateCheckoutSessionRequest.ChannelEnum = {}));
    let RecurringProcessingModelEnum;
    (function (RecurringProcessingModelEnum) {
        RecurringProcessingModelEnum["CardOnFile"] = "CardOnFile";
        RecurringProcessingModelEnum["Subscription"] = "Subscription";
        RecurringProcessingModelEnum["UnscheduledCardOnFile"] = "UnscheduledCardOnFile";
    })(RecurringProcessingModelEnum = CreateCheckoutSessionRequest.RecurringProcessingModelEnum || (CreateCheckoutSessionRequest.RecurringProcessingModelEnum = {}));
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = CreateCheckoutSessionRequest.ShopperInteractionEnum || (CreateCheckoutSessionRequest.ShopperInteractionEnum = {}));
    let StorePaymentMethodModeEnum;
    (function (StorePaymentMethodModeEnum) {
        StorePaymentMethodModeEnum["AskForConsent"] = "askForConsent";
        StorePaymentMethodModeEnum["Disabled"] = "disabled";
        StorePaymentMethodModeEnum["Enabled"] = "enabled";
    })(StorePaymentMethodModeEnum = CreateCheckoutSessionRequest.StorePaymentMethodModeEnum || (CreateCheckoutSessionRequest.StorePaymentMethodModeEnum = {}));
})(CreateCheckoutSessionRequest = exports.CreateCheckoutSessionRequest || (exports.CreateCheckoutSessionRequest = {}));
