// @adyen/api-library/lib/src/typings/checkout/econtextVoucherDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EcontextVoucherDetails = void 0;
class EcontextVoucherDetails {
    static getAttributeTypeMap() {
        return EcontextVoucherDetails.attributeTypeMap;
    }
}
exports.EcontextVoucherDetails = EcontextVoucherDetails;
EcontextVoucherDetails.discriminator = undefined;
EcontextVoucherDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "firstName",
        "baseName": "firstName",
        "type": "string"
    },
    {
        "name": "lastName",
        "baseName": "lastName",
        "type": "string"
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string"
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "EcontextVoucherDetails.TypeEnum"
    }
];
(function (EcontextVoucherDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Seveneleven"] = "econtext_seveneleven";
        TypeEnum["Stores"] = "econtext_stores";
    })(TypeEnum = EcontextVoucherDetails.TypeEnum || (EcontextVoucherDetails.TypeEnum = {}));
})(EcontextVoucherDetails = exports.EcontextVoucherDetails || (exports.EcontextVoucherDetails = {}));
