// @adyen/api-library/lib/src/typings/checkout/installmentOption.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstallmentOption = void 0;
class InstallmentOption {
    static getAttributeTypeMap() {
        return InstallmentOption.attributeTypeMap;
    }
}
exports.InstallmentOption = InstallmentOption;
InstallmentOption.discriminator = undefined;
InstallmentOption.attributeTypeMap = [
    {
        "name": "maxValue",
        "baseName": "maxValue",
        "type": "number"
    },
    {
        "name": "plans",
        "baseName": "plans",
        "type": "Array<InstallmentOption.PlansEnum>"
    },
    {
        "name": "preselectedValue",
        "baseName": "preselectedValue",
        "type": "number"
    },
    {
        "name": "values",
        "baseName": "values",
        "type": "Array<number>"
    }
];
(function (InstallmentOption) {
    let PlansEnum;
    (function (PlansEnum) {
        PlansEnum["Regular"] = "regular";
        PlansEnum["Revolving"] = "revolving";
    })(PlansEnum = InstallmentOption.PlansEnum || (InstallmentOption.PlansEnum = {}));
})(InstallmentOption = exports.InstallmentOption || (exports.InstallmentOption = {}));
