// @adyen/api-library/lib/src/typings/checkout/installments.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Installments = void 0;
class Installments {
    static getAttributeTypeMap() {
        return Installments.attributeTypeMap;
    }
}
exports.Installments = Installments;
Installments.discriminator = undefined;
Installments.attributeTypeMap = [
    {
        "name": "plan",
        "baseName": "plan",
        "type": "Installments.PlanEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "number"
    }
];
(function (Installments) {
    let PlanEnum;
    (function (PlanEnum) {
        PlanEnum["Regular"] = "regular";
        PlanEnum["Revolving"] = "revolving";
    })(PlanEnum = Installments.PlanEnum || (Installments.PlanEnum = {}));
})(Installments = exports.Installments || (exports.Installments = {}));
