// @adyen/api-library/lib/src/typings/checkout/mandate.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Mandate = void 0;
class Mandate {
    static getAttributeTypeMap() {
        return Mandate.attributeTypeMap;
    }
}
exports.Mandate = Mandate;
Mandate.discriminator = undefined;
Mandate.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "string"
    },
    {
        "name": "amountRule",
        "baseName": "amountRule",
        "type": "Mandate.AmountRuleEnum"
    },
    {
        "name": "billingAttemptsRule",
        "baseName": "billingAttemptsRule",
        "type": "Mandate.BillingAttemptsRuleEnum"
    },
    {
        "name": "billingDay",
        "baseName": "billingDay",
        "type": "string"
    },
    {
        "name": "endsAt",
        "baseName": "endsAt",
        "type": "string"
    },
    {
        "name": "frequency",
        "baseName": "frequency",
        "type": "Mandate.FrequencyEnum"
    },
    {
        "name": "remarks",
        "baseName": "remarks",
        "type": "string"
    },
    {
        "name": "startsAt",
        "baseName": "startsAt",
        "type": "string"
    }
];
(function (Mandate) {
    let AmountRuleEnum;
    (function (AmountRuleEnum) {
        AmountRuleEnum["Max"] = "max";
        AmountRuleEnum["Exact"] = "exact";
    })(AmountRuleEnum = Mandate.AmountRuleEnum || (Mandate.AmountRuleEnum = {}));
    let BillingAttemptsRuleEnum;
    (function (BillingAttemptsRuleEnum) {
        BillingAttemptsRuleEnum["On"] = "on";
        BillingAttemptsRuleEnum["Before"] = "before";
        BillingAttemptsRuleEnum["After"] = "after";
    })(BillingAttemptsRuleEnum = Mandate.BillingAttemptsRuleEnum || (Mandate.BillingAttemptsRuleEnum = {}));
    let FrequencyEnum;
    (function (FrequencyEnum) {
        FrequencyEnum["Adhoc"] = "adhoc";
        FrequencyEnum["Daily"] = "daily";
        FrequencyEnum["Weekly"] = "weekly";
        FrequencyEnum["BiWeekly"] = "biWeekly";
        FrequencyEnum["Monthly"] = "monthly";
        FrequencyEnum["Quarterly"] = "quarterly";
        FrequencyEnum["HalfYearly"] = "halfYearly";
        FrequencyEnum["Yearly"] = "yearly";
    })(FrequencyEnum = Mandate.FrequencyEnum || (Mandate.FrequencyEnum = {}));
})(Mandate = exports.Mandate || (exports.Mandate = {}));
