// @adyen/api-library/lib/src/typings/checkout/models.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectSerializer = void 0;
__exportStar(require("./accountInfo"), exports);
__exportStar(require("./acctInfo"), exports);
__exportStar(require("./achDetails"), exports);
__exportStar(require("./additionalData3DSecure"), exports);
__exportStar(require("./additionalDataAirline"), exports);
__exportStar(require("./additionalDataCarRental"), exports);
__exportStar(require("./additionalDataCommon"), exports);
__exportStar(require("./additionalDataLevel23"), exports);
__exportStar(require("./additionalDataLodging"), exports);
__exportStar(require("./additionalDataOpenInvoice"), exports);
__exportStar(require("./additionalDataOpi"), exports);
__exportStar(require("./additionalDataRatepay"), exports);
__exportStar(require("./additionalDataRetry"), exports);
__exportStar(require("./additionalDataRisk"), exports);
__exportStar(require("./additionalDataRiskStandalone"), exports);
__exportStar(require("./additionalDataSubMerchant"), exports);
__exportStar(require("./additionalDataTemporaryServices"), exports);
__exportStar(require("./additionalDataWallets"), exports);
__exportStar(require("./address"), exports);
__exportStar(require("./afterpayDetails"), exports);
__exportStar(require("./amazonPayDetails"), exports);
__exportStar(require("./amount"), exports);
__exportStar(require("./androidPayDetails"), exports);
__exportStar(require("./applePayDetails"), exports);
__exportStar(require("./applePaySessionRequest"), exports);
__exportStar(require("./applePaySessionResponse"), exports);
__exportStar(require("./applicationInfo"), exports);
__exportStar(require("./authenticationData"), exports);
__exportStar(require("./avs"), exports);
__exportStar(require("./bacsDirectDebitDetails"), exports);
__exportStar(require("./balanceCheckRequest"), exports);
__exportStar(require("./balanceCheckResponse"), exports);
__exportStar(require("./bankAccount"), exports);
__exportStar(require("./billDeskDetails"), exports);
__exportStar(require("./billingAddress"), exports);
__exportStar(require("./blikDetails"), exports);
__exportStar(require("./browserInfo"), exports);
__exportStar(require("./cancelOrderRequest"), exports);
__exportStar(require("./cancelOrderResponse"), exports);
__exportStar(require("./card"), exports);
__exportStar(require("./cardBrandDetails"), exports);
__exportStar(require("./cardDetails"), exports);
__exportStar(require("./cardDetailsRequest"), exports);
__exportStar(require("./cardDetailsResponse"), exports);
__exportStar(require("./cellulantDetails"), exports);
__exportStar(require("./checkoutAwaitAction"), exports);
__exportStar(require("./checkoutDelegatedAuthenticationAction"), exports);
__exportStar(require("./checkoutNativeRedirectAction"), exports);
__exportStar(require("./checkoutOrderResponse"), exports);
__exportStar(require("./checkoutQrCodeAction"), exports);
__exportStar(require("./checkoutRedirectAction"), exports);
__exportStar(require("./checkoutSDKAction"), exports);
__exportStar(require("./checkoutSessionInstallmentOption"), exports);
__exportStar(require("./checkoutThreeDS2Action"), exports);
__exportStar(require("./checkoutVoucherAction"), exports);
__exportStar(require("./commonField"), exports);
__exportStar(require("./company"), exports);
__exportStar(require("./configuration"), exports);
__exportStar(require("./createCheckoutSessionRequest"), exports);
__exportStar(require("./createCheckoutSessionResponse"), exports);
__exportStar(require("./createOrderRequest"), exports);
__exportStar(require("./createOrderResponse"), exports);
__exportStar(require("./deliveryAddress"), exports);
__exportStar(require("./detailsRequestAuthenticationData"), exports);
__exportStar(require("./deviceRenderOptions"), exports);
__exportStar(require("./dokuDetails"), exports);
__exportStar(require("./donationPaymentRequest"), exports);
__exportStar(require("./donationPaymentResponse"), exports);
__exportStar(require("./dotpayDetails"), exports);
__exportStar(require("./dragonpayDetails"), exports);
__exportStar(require("./econtextVoucherDetails"), exports);
__exportStar(require("./encryptedOrderData"), exports);
__exportStar(require("./externalPlatform"), exports);
__exportStar(require("./forexQuote"), exports);
__exportStar(require("./fraudCheckResult"), exports);
__exportStar(require("./fraudResult"), exports);
__exportStar(require("./fundOrigin"), exports);
__exportStar(require("./fundRecipient"), exports);
__exportStar(require("./genericIssuerPaymentMethodDetails"), exports);
__exportStar(require("./giropayDetails"), exports);
__exportStar(require("./googlePayDetails"), exports);
__exportStar(require("./idealDetails"), exports);
__exportStar(require("./inputDetail"), exports);
__exportStar(require("./installmentOption"), exports);
__exportStar(require("./installments"), exports);
__exportStar(require("./installmentsNumber"), exports);
__exportStar(require("./item"), exports);
__exportStar(require("./klarnaDetails"), exports);
__exportStar(require("./lineItem"), exports);
__exportStar(require("./listStoredPaymentMethodsResponse"), exports);
__exportStar(require("./mandate"), exports);
__exportStar(require("./masterpassDetails"), exports);
__exportStar(require("./mbwayDetails"), exports);
__exportStar(require("./merchantDevice"), exports);
__exportStar(require("./merchantRiskIndicator"), exports);
__exportStar(require("./mobilePayDetails"), exports);
__exportStar(require("./molPayDetails"), exports);
__exportStar(require("./name"), exports);
__exportStar(require("./openInvoiceDetails"), exports);
__exportStar(require("./payPalDetails"), exports);
__exportStar(require("./payUUpiDetails"), exports);
__exportStar(require("./payWithGoogleDetails"), exports);
__exportStar(require("./paymentAmountUpdateRequest"), exports);
__exportStar(require("./paymentAmountUpdateResponse"), exports);
__exportStar(require("./paymentCancelRequest"), exports);
__exportStar(require("./paymentCancelResponse"), exports);
__exportStar(require("./paymentCaptureRequest"), exports);
__exportStar(require("./paymentCaptureResponse"), exports);
__exportStar(require("./paymentCompletionDetails"), exports);
__exportStar(require("./paymentDetails"), exports);
__exportStar(require("./paymentDetailsRequest"), exports);
__exportStar(require("./paymentDetailsResponse"), exports);
__exportStar(require("./paymentLinkRequest"), exports);
__exportStar(require("./paymentLinkResponse"), exports);
__exportStar(require("./paymentMethod"), exports);
__exportStar(require("./paymentMethodGroup"), exports);
__exportStar(require("./paymentMethodIssuer"), exports);
__exportStar(require("./paymentMethodsRequest"), exports);
__exportStar(require("./paymentMethodsResponse"), exports);
__exportStar(require("./paymentRefundRequest"), exports);
__exportStar(require("./paymentRefundResponse"), exports);
__exportStar(require("./paymentRequest"), exports);
__exportStar(require("./paymentResponse"), exports);
__exportStar(require("./paymentReversalRequest"), exports);
__exportStar(require("./paymentReversalResponse"), exports);
__exportStar(require("./paymentSetupRequest"), exports);
__exportStar(require("./paymentSetupResponse"), exports);
__exportStar(require("./paymentVerificationRequest"), exports);
__exportStar(require("./paymentVerificationResponse"), exports);
__exportStar(require("./phone"), exports);
__exportStar(require("./platformChargebackLogic"), exports);
__exportStar(require("./ratepayDetails"), exports);
__exportStar(require("./recurring"), exports);
__exportStar(require("./recurringDetail"), exports);
__exportStar(require("./responseAdditionalData3DSecure"), exports);
__exportStar(require("./responseAdditionalDataBillingAddress"), exports);
__exportStar(require("./responseAdditionalDataCard"), exports);
__exportStar(require("./responseAdditionalDataCommon"), exports);
__exportStar(require("./responseAdditionalDataInstallments"), exports);
__exportStar(require("./responseAdditionalDataNetworkTokens"), exports);
__exportStar(require("./responseAdditionalDataOpi"), exports);
__exportStar(require("./responseAdditionalDataSepa"), exports);
__exportStar(require("./responsePaymentMethod"), exports);
__exportStar(require("./riskData"), exports);
__exportStar(require("./sDKEphemPubKey"), exports);
__exportStar(require("./samsungPayDetails"), exports);
__exportStar(require("./sepaDirectDebitDetails"), exports);
__exportStar(require("./serviceError"), exports);
__exportStar(require("./serviceError2"), exports);
__exportStar(require("./sessionResultResponse"), exports);
__exportStar(require("./shopperInput"), exports);
__exportStar(require("./shopperInteractionDevice"), exports);
__exportStar(require("./split"), exports);
__exportStar(require("./splitAmount"), exports);
__exportStar(require("./standalonePaymentCancelRequest"), exports);
__exportStar(require("./standalonePaymentCancelResponse"), exports);
__exportStar(require("./storedDetails"), exports);
__exportStar(require("./storedPaymentMethod"), exports);
__exportStar(require("./storedPaymentMethodDetails"), exports);
__exportStar(require("./storedPaymentMethodResource"), exports);
__exportStar(require("./subInputDetail"), exports);
__exportStar(require("./subMerchant"), exports);
__exportStar(require("./subMerchantInfo"), exports);
__exportStar(require("./threeDS2RequestData"), exports);
__exportStar(require("./threeDS2RequestData2"), exports);
__exportStar(require("./threeDS2ResponseData"), exports);
__exportStar(require("./threeDS2Result"), exports);
__exportStar(require("./threeDSRequestData"), exports);
__exportStar(require("./threeDSRequestorAuthenticationInfo"), exports);
__exportStar(require("./threeDSRequestorPriorAuthenticationInfo"), exports);
__exportStar(require("./threeDSecureData"), exports);
__exportStar(require("./updatePaymentLinkRequest"), exports);
__exportStar(require("./upiCollectDetails"), exports);
__exportStar(require("./upiIntentDetails"), exports);
__exportStar(require("./utilityRequest"), exports);
__exportStar(require("./utilityResponse"), exports);
__exportStar(require("./vippsDetails"), exports);
__exportStar(require("./visaCheckoutDetails"), exports);
__exportStar(require("./weChatPayDetails"), exports);
__exportStar(require("./weChatPayMiniProgramDetails"), exports);
__exportStar(require("./zipDetails"), exports);
const accountInfo_1 = require("./accountInfo");
const acctInfo_1 = require("./acctInfo");
const achDetails_1 = require("./achDetails");
const additionalData3DSecure_1 = require("./additionalData3DSecure");
const additionalDataAirline_1 = require("./additionalDataAirline");
const additionalDataCarRental_1 = require("./additionalDataCarRental");
const additionalDataCommon_1 = require("./additionalDataCommon");
const additionalDataLevel23_1 = require("./additionalDataLevel23");
const additionalDataLodging_1 = require("./additionalDataLodging");
const additionalDataOpenInvoice_1 = require("./additionalDataOpenInvoice");
const additionalDataOpi_1 = require("./additionalDataOpi");
const additionalDataRatepay_1 = require("./additionalDataRatepay");
const additionalDataRetry_1 = require("./additionalDataRetry");
const additionalDataRisk_1 = require("./additionalDataRisk");
const additionalDataRiskStandalone_1 = require("./additionalDataRiskStandalone");
const additionalDataSubMerchant_1 = require("./additionalDataSubMerchant");
const additionalDataTemporaryServices_1 = require("./additionalDataTemporaryServices");
const additionalDataWallets_1 = require("./additionalDataWallets");
const address_1 = require("./address");
const afterpayDetails_1 = require("./afterpayDetails");
const amazonPayDetails_1 = require("./amazonPayDetails");
const amount_1 = require("./amount");
const androidPayDetails_1 = require("./androidPayDetails");
const applePayDetails_1 = require("./applePayDetails");
const applePaySessionRequest_1 = require("./applePaySessionRequest");
const applePaySessionResponse_1 = require("./applePaySessionResponse");
const applicationInfo_1 = require("./applicationInfo");
const authenticationData_1 = require("./authenticationData");
const avs_1 = require("./avs");
const bacsDirectDebitDetails_1 = require("./bacsDirectDebitDetails");
const balanceCheckRequest_1 = require("./balanceCheckRequest");
const balanceCheckResponse_1 = require("./balanceCheckResponse");
const bankAccount_1 = require("./bankAccount");
const billDeskDetails_1 = require("./billDeskDetails");
const billingAddress_1 = require("./billingAddress");
const blikDetails_1 = require("./blikDetails");
const browserInfo_1 = require("./browserInfo");
const cancelOrderRequest_1 = require("./cancelOrderRequest");
const cancelOrderResponse_1 = require("./cancelOrderResponse");
const card_1 = require("./card");
const cardBrandDetails_1 = require("./cardBrandDetails");
const cardDetails_1 = require("./cardDetails");
const cardDetailsRequest_1 = require("./cardDetailsRequest");
const cardDetailsResponse_1 = require("./cardDetailsResponse");
const cellulantDetails_1 = require("./cellulantDetails");
const checkoutAwaitAction_1 = require("./checkoutAwaitAction");
const checkoutDelegatedAuthenticationAction_1 = require("./checkoutDelegatedAuthenticationAction");
const checkoutNativeRedirectAction_1 = require("./checkoutNativeRedirectAction");
const checkoutOrderResponse_1 = require("./checkoutOrderResponse");
const checkoutQrCodeAction_1 = require("./checkoutQrCodeAction");
const checkoutRedirectAction_1 = require("./checkoutRedirectAction");
const checkoutSDKAction_1 = require("./checkoutSDKAction");
const checkoutSessionInstallmentOption_1 = require("./checkoutSessionInstallmentOption");
const checkoutThreeDS2Action_1 = require("./checkoutThreeDS2Action");
const checkoutVoucherAction_1 = require("./checkoutVoucherAction");
const commonField_1 = require("./commonField");
const company_1 = require("./company");
const configuration_1 = require("./configuration");
const createCheckoutSessionRequest_1 = require("./createCheckoutSessionRequest");
const createCheckoutSessionResponse_1 = require("./createCheckoutSessionResponse");
const createOrderRequest_1 = require("./createOrderRequest");
const createOrderResponse_1 = require("./createOrderResponse");
const deliveryAddress_1 = require("./deliveryAddress");
const detailsRequestAuthenticationData_1 = require("./detailsRequestAuthenticationData");
const deviceRenderOptions_1 = require("./deviceRenderOptions");
const dokuDetails_1 = require("./dokuDetails");
const donationPaymentRequest_1 = require("./donationPaymentRequest");
const donationPaymentResponse_1 = require("./donationPaymentResponse");
const dotpayDetails_1 = require("./dotpayDetails");
const dragonpayDetails_1 = require("./dragonpayDetails");
const econtextVoucherDetails_1 = require("./econtextVoucherDetails");
const encryptedOrderData_1 = require("./encryptedOrderData");
const externalPlatform_1 = require("./externalPlatform");
const forexQuote_1 = require("./forexQuote");
const fraudCheckResult_1 = require("./fraudCheckResult");
const fraudResult_1 = require("./fraudResult");
const fundOrigin_1 = require("./fundOrigin");
const fundRecipient_1 = require("./fundRecipient");
const genericIssuerPaymentMethodDetails_1 = require("./genericIssuerPaymentMethodDetails");
const giropayDetails_1 = require("./giropayDetails");
const googlePayDetails_1 = require("./googlePayDetails");
const idealDetails_1 = require("./idealDetails");
const inputDetail_1 = require("./inputDetail");
const installmentOption_1 = require("./installmentOption");
const installments_1 = require("./installments");
const installmentsNumber_1 = require("./installmentsNumber");
const item_1 = require("./item");
const klarnaDetails_1 = require("./klarnaDetails");
const lineItem_1 = require("./lineItem");
const listStoredPaymentMethodsResponse_1 = require("./listStoredPaymentMethodsResponse");
const mandate_1 = require("./mandate");
const masterpassDetails_1 = require("./masterpassDetails");
const mbwayDetails_1 = require("./mbwayDetails");
const merchantDevice_1 = require("./merchantDevice");
const merchantRiskIndicator_1 = require("./merchantRiskIndicator");
const mobilePayDetails_1 = require("./mobilePayDetails");
const molPayDetails_1 = require("./molPayDetails");
const name_1 = require("./name");
const openInvoiceDetails_1 = require("./openInvoiceDetails");
const payPalDetails_1 = require("./payPalDetails");
const payUUpiDetails_1 = require("./payUUpiDetails");
const payWithGoogleDetails_1 = require("./payWithGoogleDetails");
const paymentAmountUpdateRequest_1 = require("./paymentAmountUpdateRequest");
const paymentAmountUpdateResponse_1 = require("./paymentAmountUpdateResponse");
const paymentCancelRequest_1 = require("./paymentCancelRequest");
const paymentCancelResponse_1 = require("./paymentCancelResponse");
const paymentCaptureRequest_1 = require("./paymentCaptureRequest");
const paymentCaptureResponse_1 = require("./paymentCaptureResponse");
const paymentCompletionDetails_1 = require("./paymentCompletionDetails");
const paymentDetails_1 = require("./paymentDetails");
const paymentDetailsRequest_1 = require("./paymentDetailsRequest");
const paymentDetailsResponse_1 = require("./paymentDetailsResponse");
const paymentLinkRequest_1 = require("./paymentLinkRequest");
const paymentLinkResponse_1 = require("./paymentLinkResponse");
const paymentMethod_1 = require("./paymentMethod");
const paymentMethodGroup_1 = require("./paymentMethodGroup");
const paymentMethodIssuer_1 = require("./paymentMethodIssuer");
const paymentMethodsRequest_1 = require("./paymentMethodsRequest");
const paymentMethodsResponse_1 = require("./paymentMethodsResponse");
const paymentRefundRequest_1 = require("./paymentRefundRequest");
const paymentRefundResponse_1 = require("./paymentRefundResponse");
const paymentRequest_1 = require("./paymentRequest");
const paymentResponse_1 = require("./paymentResponse");
const paymentReversalRequest_1 = require("./paymentReversalRequest");
const paymentReversalResponse_1 = require("./paymentReversalResponse");
const paymentSetupRequest_1 = require("./paymentSetupRequest");
const paymentSetupResponse_1 = require("./paymentSetupResponse");
const paymentVerificationRequest_1 = require("./paymentVerificationRequest");
const paymentVerificationResponse_1 = require("./paymentVerificationResponse");
const phone_1 = require("./phone");
const platformChargebackLogic_1 = require("./platformChargebackLogic");
const ratepayDetails_1 = require("./ratepayDetails");
const recurring_1 = require("./recurring");
const recurringDetail_1 = require("./recurringDetail");
const responseAdditionalData3DSecure_1 = require("./responseAdditionalData3DSecure");
const responseAdditionalDataBillingAddress_1 = require("./responseAdditionalDataBillingAddress");
const responseAdditionalDataCard_1 = require("./responseAdditionalDataCard");
const responseAdditionalDataCommon_1 = require("./responseAdditionalDataCommon");
const responseAdditionalDataInstallments_1 = require("./responseAdditionalDataInstallments");
const responseAdditionalDataNetworkTokens_1 = require("./responseAdditionalDataNetworkTokens");
const responseAdditionalDataOpi_1 = require("./responseAdditionalDataOpi");
const responseAdditionalDataSepa_1 = require("./responseAdditionalDataSepa");
const responsePaymentMethod_1 = require("./responsePaymentMethod");
const riskData_1 = require("./riskData");
const sDKEphemPubKey_1 = require("./sDKEphemPubKey");
const samsungPayDetails_1 = require("./samsungPayDetails");
const sepaDirectDebitDetails_1 = require("./sepaDirectDebitDetails");
const serviceError_1 = require("./serviceError");
const serviceError2_1 = require("./serviceError2");
const sessionResultResponse_1 = require("./sessionResultResponse");
const shopperInput_1 = require("./shopperInput");
const shopperInteractionDevice_1 = require("./shopperInteractionDevice");
const split_1 = require("./split");
const splitAmount_1 = require("./splitAmount");
const standalonePaymentCancelRequest_1 = require("./standalonePaymentCancelRequest");
const standalonePaymentCancelResponse_1 = require("./standalonePaymentCancelResponse");
const storedDetails_1 = require("./storedDetails");
const storedPaymentMethod_1 = require("./storedPaymentMethod");
const storedPaymentMethodDetails_1 = require("./storedPaymentMethodDetails");
const storedPaymentMethodResource_1 = require("./storedPaymentMethodResource");
const subInputDetail_1 = require("./subInputDetail");
const subMerchant_1 = require("./subMerchant");
const subMerchantInfo_1 = require("./subMerchantInfo");
const threeDS2RequestData_1 = require("./threeDS2RequestData");
const threeDS2RequestData2_1 = require("./threeDS2RequestData2");
const threeDS2ResponseData_1 = require("./threeDS2ResponseData");
const threeDS2Result_1 = require("./threeDS2Result");
const threeDSRequestData_1 = require("./threeDSRequestData");
const threeDSRequestorAuthenticationInfo_1 = require("./threeDSRequestorAuthenticationInfo");
const threeDSRequestorPriorAuthenticationInfo_1 = require("./threeDSRequestorPriorAuthenticationInfo");
const threeDSecureData_1 = require("./threeDSecureData");
const updatePaymentLinkRequest_1 = require("./updatePaymentLinkRequest");
const upiCollectDetails_1 = require("./upiCollectDetails");
const upiIntentDetails_1 = require("./upiIntentDetails");
const utilityRequest_1 = require("./utilityRequest");
const utilityResponse_1 = require("./utilityResponse");
const vippsDetails_1 = require("./vippsDetails");
const visaCheckoutDetails_1 = require("./visaCheckoutDetails");
const weChatPayDetails_1 = require("./weChatPayDetails");
const weChatPayMiniProgramDetails_1 = require("./weChatPayMiniProgramDetails");
const zipDetails_1 = require("./zipDetails");
/* tslint:disable:no-unused-variable */
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
let enumsMap = {
    "AccountInfo.AccountAgeIndicatorEnum": accountInfo_1.AccountInfo.AccountAgeIndicatorEnum,
    "AccountInfo.AccountChangeIndicatorEnum": accountInfo_1.AccountInfo.AccountChangeIndicatorEnum,
    "AccountInfo.AccountTypeEnum": accountInfo_1.AccountInfo.AccountTypeEnum,
    "AccountInfo.DeliveryAddressUsageIndicatorEnum": accountInfo_1.AccountInfo.DeliveryAddressUsageIndicatorEnum,
    "AccountInfo.PasswordChangeIndicatorEnum": accountInfo_1.AccountInfo.PasswordChangeIndicatorEnum,
    "AccountInfo.PaymentAccountIndicatorEnum": accountInfo_1.AccountInfo.PaymentAccountIndicatorEnum,
    "AcctInfo.ChAccAgeIndEnum": acctInfo_1.AcctInfo.ChAccAgeIndEnum,
    "AcctInfo.ChAccChangeIndEnum": acctInfo_1.AcctInfo.ChAccChangeIndEnum,
    "AcctInfo.ChAccPwChangeIndEnum": acctInfo_1.AcctInfo.ChAccPwChangeIndEnum,
    "AcctInfo.PaymentAccIndEnum": acctInfo_1.AcctInfo.PaymentAccIndEnum,
    "AcctInfo.ShipAddressUsageIndEnum": acctInfo_1.AcctInfo.ShipAddressUsageIndEnum,
    "AcctInfo.ShipNameIndicatorEnum": acctInfo_1.AcctInfo.ShipNameIndicatorEnum,
    "AcctInfo.SuspiciousAccActivityEnum": acctInfo_1.AcctInfo.SuspiciousAccActivityEnum,
    "AchDetails.BankAccountTypeEnum": achDetails_1.AchDetails.BankAccountTypeEnum,
    "AchDetails.TypeEnum": achDetails_1.AchDetails.TypeEnum,
    "AdditionalData3DSecure.ChallengeWindowSizeEnum": additionalData3DSecure_1.AdditionalData3DSecure.ChallengeWindowSizeEnum,
    "AdditionalDataCommon.IndustryUsageEnum": additionalDataCommon_1.AdditionalDataCommon.IndustryUsageEnum,
    "AfterpayDetails.TypeEnum": afterpayDetails_1.AfterpayDetails.TypeEnum,
    "AmazonPayDetails.TypeEnum": amazonPayDetails_1.AmazonPayDetails.TypeEnum,
    "AndroidPayDetails.TypeEnum": androidPayDetails_1.AndroidPayDetails.TypeEnum,
    "ApplePayDetails.FundingSourceEnum": applePayDetails_1.ApplePayDetails.FundingSourceEnum,
    "ApplePayDetails.TypeEnum": applePayDetails_1.ApplePayDetails.TypeEnum,
    "AuthenticationData.AttemptAuthenticationEnum": authenticationData_1.AuthenticationData.AttemptAuthenticationEnum,
    "Avs.EnabledEnum": avs_1.Avs.EnabledEnum,
    "BacsDirectDebitDetails.TypeEnum": bacsDirectDebitDetails_1.BacsDirectDebitDetails.TypeEnum,
    "BalanceCheckRequest.RecurringProcessingModelEnum": balanceCheckRequest_1.BalanceCheckRequest.RecurringProcessingModelEnum,
    "BalanceCheckRequest.ShopperInteractionEnum": balanceCheckRequest_1.BalanceCheckRequest.ShopperInteractionEnum,
    "BalanceCheckResponse.ResultCodeEnum": balanceCheckResponse_1.BalanceCheckResponse.ResultCodeEnum,
    "BillDeskDetails.TypeEnum": billDeskDetails_1.BillDeskDetails.TypeEnum,
    "BlikDetails.TypeEnum": blikDetails_1.BlikDetails.TypeEnum,
    "CancelOrderResponse.ResultCodeEnum": cancelOrderResponse_1.CancelOrderResponse.ResultCodeEnum,
    "CardDetails.FundingSourceEnum": cardDetails_1.CardDetails.FundingSourceEnum,
    "CardDetails.TypeEnum": cardDetails_1.CardDetails.TypeEnum,
    "CellulantDetails.TypeEnum": cellulantDetails_1.CellulantDetails.TypeEnum,
    "CheckoutAwaitAction.TypeEnum": checkoutAwaitAction_1.CheckoutAwaitAction.TypeEnum,
    "CheckoutDelegatedAuthenticationAction.TypeEnum": checkoutDelegatedAuthenticationAction_1.CheckoutDelegatedAuthenticationAction.TypeEnum,
    "CheckoutNativeRedirectAction.TypeEnum": checkoutNativeRedirectAction_1.CheckoutNativeRedirectAction.TypeEnum,
    "CheckoutQrCodeAction.TypeEnum": checkoutQrCodeAction_1.CheckoutQrCodeAction.TypeEnum,
    "CheckoutRedirectAction.TypeEnum": checkoutRedirectAction_1.CheckoutRedirectAction.TypeEnum,
    "CheckoutSDKAction.TypeEnum": checkoutSDKAction_1.CheckoutSDKAction.TypeEnum,
    "CheckoutSessionInstallmentOption.PlansEnum": checkoutSessionInstallmentOption_1.CheckoutSessionInstallmentOption.PlansEnum,
    "CheckoutThreeDS2Action.TypeEnum": checkoutThreeDS2Action_1.CheckoutThreeDS2Action.TypeEnum,
    "CheckoutVoucherAction.TypeEnum": checkoutVoucherAction_1.CheckoutVoucherAction.TypeEnum,
    "Configuration.CardHolderNameEnum": configuration_1.Configuration.CardHolderNameEnum,
    "CreateCheckoutSessionRequest.ChannelEnum": createCheckoutSessionRequest_1.CreateCheckoutSessionRequest.ChannelEnum,
    "CreateCheckoutSessionRequest.RecurringProcessingModelEnum": createCheckoutSessionRequest_1.CreateCheckoutSessionRequest.RecurringProcessingModelEnum,
    "CreateCheckoutSessionRequest.ShopperInteractionEnum": createCheckoutSessionRequest_1.CreateCheckoutSessionRequest.ShopperInteractionEnum,
    "CreateCheckoutSessionRequest.StorePaymentMethodModeEnum": createCheckoutSessionRequest_1.CreateCheckoutSessionRequest.StorePaymentMethodModeEnum,
    "CreateCheckoutSessionResponse.ChannelEnum": createCheckoutSessionResponse_1.CreateCheckoutSessionResponse.ChannelEnum,
    "CreateCheckoutSessionResponse.ModeEnum": createCheckoutSessionResponse_1.CreateCheckoutSessionResponse.ModeEnum,
    "CreateCheckoutSessionResponse.RecurringProcessingModelEnum": createCheckoutSessionResponse_1.CreateCheckoutSessionResponse.RecurringProcessingModelEnum,
    "CreateCheckoutSessionResponse.ShopperInteractionEnum": createCheckoutSessionResponse_1.CreateCheckoutSessionResponse.ShopperInteractionEnum,
    "CreateCheckoutSessionResponse.StorePaymentMethodModeEnum": createCheckoutSessionResponse_1.CreateCheckoutSessionResponse.StorePaymentMethodModeEnum,
    "CreateOrderResponse.ResultCodeEnum": createOrderResponse_1.CreateOrderResponse.ResultCodeEnum,
    "DeviceRenderOptions.SdkInterfaceEnum": deviceRenderOptions_1.DeviceRenderOptions.SdkInterfaceEnum,
    "DeviceRenderOptions.SdkUiTypeEnum": deviceRenderOptions_1.DeviceRenderOptions.SdkUiTypeEnum,
    "DokuDetails.TypeEnum": dokuDetails_1.DokuDetails.TypeEnum,
    "DonationPaymentRequest.ChannelEnum": donationPaymentRequest_1.DonationPaymentRequest.ChannelEnum,
    "DonationPaymentRequest.EntityTypeEnum": donationPaymentRequest_1.DonationPaymentRequest.EntityTypeEnum,
    "DonationPaymentRequest.FundingSourceEnum": donationPaymentRequest_1.DonationPaymentRequest.FundingSourceEnum,
    "DonationPaymentRequest.IndustryUsageEnum": donationPaymentRequest_1.DonationPaymentRequest.IndustryUsageEnum,
    "DonationPaymentRequest.RecurringProcessingModelEnum": donationPaymentRequest_1.DonationPaymentRequest.RecurringProcessingModelEnum,
    "DonationPaymentRequest.ShopperInteractionEnum": donationPaymentRequest_1.DonationPaymentRequest.ShopperInteractionEnum,
    "DonationPaymentResponse.StatusEnum": donationPaymentResponse_1.DonationPaymentResponse.StatusEnum,
    "DotpayDetails.TypeEnum": dotpayDetails_1.DotpayDetails.TypeEnum,
    "DragonpayDetails.TypeEnum": dragonpayDetails_1.DragonpayDetails.TypeEnum,
    "EcontextVoucherDetails.TypeEnum": econtextVoucherDetails_1.EcontextVoucherDetails.TypeEnum,
    "GenericIssuerPaymentMethodDetails.TypeEnum": genericIssuerPaymentMethodDetails_1.GenericIssuerPaymentMethodDetails.TypeEnum,
    "GiropayDetails.TypeEnum": giropayDetails_1.GiropayDetails.TypeEnum,
    "GooglePayDetails.FundingSourceEnum": googlePayDetails_1.GooglePayDetails.FundingSourceEnum,
    "GooglePayDetails.TypeEnum": googlePayDetails_1.GooglePayDetails.TypeEnum,
    "IdealDetails.TypeEnum": idealDetails_1.IdealDetails.TypeEnum,
    "InstallmentOption.PlansEnum": installmentOption_1.InstallmentOption.PlansEnum,
    "Installments.PlanEnum": installments_1.Installments.PlanEnum,
    "KlarnaDetails.TypeEnum": klarnaDetails_1.KlarnaDetails.TypeEnum,
    "Mandate.AmountRuleEnum": mandate_1.Mandate.AmountRuleEnum,
    "Mandate.BillingAttemptsRuleEnum": mandate_1.Mandate.BillingAttemptsRuleEnum,
    "Mandate.FrequencyEnum": mandate_1.Mandate.FrequencyEnum,
    "MasterpassDetails.FundingSourceEnum": masterpassDetails_1.MasterpassDetails.FundingSourceEnum,
    "MasterpassDetails.TypeEnum": masterpassDetails_1.MasterpassDetails.TypeEnum,
    "MbwayDetails.TypeEnum": mbwayDetails_1.MbwayDetails.TypeEnum,
    "MerchantRiskIndicator.DeliveryAddressIndicatorEnum": merchantRiskIndicator_1.MerchantRiskIndicator.DeliveryAddressIndicatorEnum,
    "MerchantRiskIndicator.DeliveryTimeframeEnum": merchantRiskIndicator_1.MerchantRiskIndicator.DeliveryTimeframeEnum,
    "MobilePayDetails.TypeEnum": mobilePayDetails_1.MobilePayDetails.TypeEnum,
    "MolPayDetails.TypeEnum": molPayDetails_1.MolPayDetails.TypeEnum,
    "OpenInvoiceDetails.TypeEnum": openInvoiceDetails_1.OpenInvoiceDetails.TypeEnum,
    "PayPalDetails.SubtypeEnum": payPalDetails_1.PayPalDetails.SubtypeEnum,
    "PayPalDetails.TypeEnum": payPalDetails_1.PayPalDetails.TypeEnum,
    "PayUUpiDetails.TypeEnum": payUUpiDetails_1.PayUUpiDetails.TypeEnum,
    "PayWithGoogleDetails.FundingSourceEnum": payWithGoogleDetails_1.PayWithGoogleDetails.FundingSourceEnum,
    "PayWithGoogleDetails.TypeEnum": payWithGoogleDetails_1.PayWithGoogleDetails.TypeEnum,
    "PaymentAmountUpdateRequest.IndustryUsageEnum": paymentAmountUpdateRequest_1.PaymentAmountUpdateRequest.IndustryUsageEnum,
    "PaymentAmountUpdateResponse.IndustryUsageEnum": paymentAmountUpdateResponse_1.PaymentAmountUpdateResponse.IndustryUsageEnum,
    "PaymentAmountUpdateResponse.StatusEnum": paymentAmountUpdateResponse_1.PaymentAmountUpdateResponse.StatusEnum,
    "PaymentCancelResponse.StatusEnum": paymentCancelResponse_1.PaymentCancelResponse.StatusEnum,
    "PaymentCaptureResponse.StatusEnum": paymentCaptureResponse_1.PaymentCaptureResponse.StatusEnum,
    "PaymentDetails.TypeEnum": paymentDetails_1.PaymentDetails.TypeEnum,
    "PaymentDetailsResponse.ResultCodeEnum": paymentDetailsResponse_1.PaymentDetailsResponse.ResultCodeEnum,
    "PaymentLinkRequest.RecurringProcessingModelEnum": paymentLinkRequest_1.PaymentLinkRequest.RecurringProcessingModelEnum,
    "PaymentLinkRequest.RequiredShopperFieldsEnum": paymentLinkRequest_1.PaymentLinkRequest.RequiredShopperFieldsEnum,
    "PaymentLinkRequest.StorePaymentMethodModeEnum": paymentLinkRequest_1.PaymentLinkRequest.StorePaymentMethodModeEnum,
    "PaymentLinkResponse.RecurringProcessingModelEnum": paymentLinkResponse_1.PaymentLinkResponse.RecurringProcessingModelEnum,
    "PaymentLinkResponse.RequiredShopperFieldsEnum": paymentLinkResponse_1.PaymentLinkResponse.RequiredShopperFieldsEnum,
    "PaymentLinkResponse.StatusEnum": paymentLinkResponse_1.PaymentLinkResponse.StatusEnum,
    "PaymentLinkResponse.StorePaymentMethodModeEnum": paymentLinkResponse_1.PaymentLinkResponse.StorePaymentMethodModeEnum,
    "PaymentMethod.FundingSourceEnum": paymentMethod_1.PaymentMethod.FundingSourceEnum,
    "PaymentMethodsRequest.ChannelEnum": paymentMethodsRequest_1.PaymentMethodsRequest.ChannelEnum,
    "PaymentRefundRequest.MerchantRefundReasonEnum": paymentRefundRequest_1.PaymentRefundRequest.MerchantRefundReasonEnum,
    "PaymentRefundResponse.MerchantRefundReasonEnum": paymentRefundResponse_1.PaymentRefundResponse.MerchantRefundReasonEnum,
    "PaymentRefundResponse.StatusEnum": paymentRefundResponse_1.PaymentRefundResponse.StatusEnum,
    "PaymentRequest.ChannelEnum": paymentRequest_1.PaymentRequest.ChannelEnum,
    "PaymentRequest.EntityTypeEnum": paymentRequest_1.PaymentRequest.EntityTypeEnum,
    "PaymentRequest.IndustryUsageEnum": paymentRequest_1.PaymentRequest.IndustryUsageEnum,
    "PaymentRequest.RecurringProcessingModelEnum": paymentRequest_1.PaymentRequest.RecurringProcessingModelEnum,
    "PaymentRequest.ShopperInteractionEnum": paymentRequest_1.PaymentRequest.ShopperInteractionEnum,
    "PaymentResponse.ResultCodeEnum": paymentResponse_1.PaymentResponse.ResultCodeEnum,
    "PaymentReversalResponse.StatusEnum": paymentReversalResponse_1.PaymentReversalResponse.StatusEnum,
    "PaymentSetupRequest.ChannelEnum": paymentSetupRequest_1.PaymentSetupRequest.ChannelEnum,
    "PaymentSetupRequest.EntityTypeEnum": paymentSetupRequest_1.PaymentSetupRequest.EntityTypeEnum,
    "PaymentSetupRequest.ShopperInteractionEnum": paymentSetupRequest_1.PaymentSetupRequest.ShopperInteractionEnum,
    "PaymentVerificationResponse.ResultCodeEnum": paymentVerificationResponse_1.PaymentVerificationResponse.ResultCodeEnum,
    "PlatformChargebackLogic.BehaviorEnum": platformChargebackLogic_1.PlatformChargebackLogic.BehaviorEnum,
    "RatepayDetails.TypeEnum": ratepayDetails_1.RatepayDetails.TypeEnum,
    "Recurring.ContractEnum": recurring_1.Recurring.ContractEnum,
    "Recurring.TokenServiceEnum": recurring_1.Recurring.TokenServiceEnum,
    "RecurringDetail.FundingSourceEnum": recurringDetail_1.RecurringDetail.FundingSourceEnum,
    "ResponseAdditionalDataCommon.FraudResultTypeEnum": responseAdditionalDataCommon_1.ResponseAdditionalDataCommon.FraudResultTypeEnum,
    "ResponseAdditionalDataCommon.RecurringProcessingModelEnum": responseAdditionalDataCommon_1.ResponseAdditionalDataCommon.RecurringProcessingModelEnum,
    "SamsungPayDetails.FundingSourceEnum": samsungPayDetails_1.SamsungPayDetails.FundingSourceEnum,
    "SamsungPayDetails.TypeEnum": samsungPayDetails_1.SamsungPayDetails.TypeEnum,
    "SepaDirectDebitDetails.TypeEnum": sepaDirectDebitDetails_1.SepaDirectDebitDetails.TypeEnum,
    "SessionResultResponse.StatusEnum": sessionResultResponse_1.SessionResultResponse.StatusEnum,
    "ShopperInput.BillingAddressEnum": shopperInput_1.ShopperInput.BillingAddressEnum,
    "ShopperInput.DeliveryAddressEnum": shopperInput_1.ShopperInput.DeliveryAddressEnum,
    "ShopperInput.PersonalDetailsEnum": shopperInput_1.ShopperInput.PersonalDetailsEnum,
    "Split.TypeEnum": split_1.Split.TypeEnum,
    "StandalonePaymentCancelResponse.StatusEnum": standalonePaymentCancelResponse_1.StandalonePaymentCancelResponse.StatusEnum,
    "StoredPaymentMethodDetails.TypeEnum": storedPaymentMethodDetails_1.StoredPaymentMethodDetails.TypeEnum,
    "ThreeDS2RequestData.AcctTypeEnum": threeDS2RequestData_1.ThreeDS2RequestData.AcctTypeEnum,
    "ThreeDS2RequestData.AddrMatchEnum": threeDS2RequestData_1.ThreeDS2RequestData.AddrMatchEnum,
    "ThreeDS2RequestData.ChallengeIndicatorEnum": threeDS2RequestData_1.ThreeDS2RequestData.ChallengeIndicatorEnum,
    "ThreeDS2RequestData.ThreeDSRequestorChallengeIndEnum": threeDS2RequestData_1.ThreeDS2RequestData.ThreeDSRequestorChallengeIndEnum,
    "ThreeDS2RequestData.TransTypeEnum": threeDS2RequestData_1.ThreeDS2RequestData.TransTypeEnum,
    "ThreeDS2RequestData.TransactionTypeEnum": threeDS2RequestData_1.ThreeDS2RequestData.TransactionTypeEnum,
    "ThreeDS2RequestData2.AcctTypeEnum": threeDS2RequestData2_1.ThreeDS2RequestData2.AcctTypeEnum,
    "ThreeDS2RequestData2.AddrMatchEnum": threeDS2RequestData2_1.ThreeDS2RequestData2.AddrMatchEnum,
    "ThreeDS2RequestData2.ChallengeIndicatorEnum": threeDS2RequestData2_1.ThreeDS2RequestData2.ChallengeIndicatorEnum,
    "ThreeDS2RequestData2.ThreeDSRequestorChallengeIndEnum": threeDS2RequestData2_1.ThreeDS2RequestData2.ThreeDSRequestorChallengeIndEnum,
    "ThreeDS2RequestData2.TransTypeEnum": threeDS2RequestData2_1.ThreeDS2RequestData2.TransTypeEnum,
    "ThreeDS2RequestData2.TransactionTypeEnum": threeDS2RequestData2_1.ThreeDS2RequestData2.TransactionTypeEnum,
    "ThreeDS2Result.ChallengeCancelEnum": threeDS2Result_1.ThreeDS2Result.ChallengeCancelEnum,
    "ThreeDS2Result.ExemptionIndicatorEnum": threeDS2Result_1.ThreeDS2Result.ExemptionIndicatorEnum,
    "ThreeDS2Result.ThreeDSRequestorChallengeIndEnum": threeDS2Result_1.ThreeDS2Result.ThreeDSRequestorChallengeIndEnum,
    "ThreeDSRequestData.ChallengeWindowSizeEnum": threeDSRequestData_1.ThreeDSRequestData.ChallengeWindowSizeEnum,
    "ThreeDSRequestData.DataOnlyEnum": threeDSRequestData_1.ThreeDSRequestData.DataOnlyEnum,
    "ThreeDSRequestData.NativeThreeDSEnum": threeDSRequestData_1.ThreeDSRequestData.NativeThreeDSEnum,
    "ThreeDSRequestData.ThreeDSVersionEnum": threeDSRequestData_1.ThreeDSRequestData.ThreeDSVersionEnum,
    "ThreeDSRequestorAuthenticationInfo.ThreeDSReqAuthMethodEnum": threeDSRequestorAuthenticationInfo_1.ThreeDSRequestorAuthenticationInfo.ThreeDSReqAuthMethodEnum,
    "ThreeDSRequestorPriorAuthenticationInfo.ThreeDSReqPriorAuthMethodEnum": threeDSRequestorPriorAuthenticationInfo_1.ThreeDSRequestorPriorAuthenticationInfo.ThreeDSReqPriorAuthMethodEnum,
    "ThreeDSecureData.AuthenticationResponseEnum": threeDSecureData_1.ThreeDSecureData.AuthenticationResponseEnum,
    "ThreeDSecureData.ChallengeCancelEnum": threeDSecureData_1.ThreeDSecureData.ChallengeCancelEnum,
    "ThreeDSecureData.DirectoryResponseEnum": threeDSecureData_1.ThreeDSecureData.DirectoryResponseEnum,
    "UpdatePaymentLinkRequest.StatusEnum": updatePaymentLinkRequest_1.UpdatePaymentLinkRequest.StatusEnum,
    "UpiCollectDetails.TypeEnum": upiCollectDetails_1.UpiCollectDetails.TypeEnum,
    "UpiIntentDetails.TypeEnum": upiIntentDetails_1.UpiIntentDetails.TypeEnum,
    "VippsDetails.TypeEnum": vippsDetails_1.VippsDetails.TypeEnum,
    "VisaCheckoutDetails.FundingSourceEnum": visaCheckoutDetails_1.VisaCheckoutDetails.FundingSourceEnum,
    "VisaCheckoutDetails.TypeEnum": visaCheckoutDetails_1.VisaCheckoutDetails.TypeEnum,
    "WeChatPayDetails.TypeEnum": weChatPayDetails_1.WeChatPayDetails.TypeEnum,
    "WeChatPayMiniProgramDetails.TypeEnum": weChatPayMiniProgramDetails_1.WeChatPayMiniProgramDetails.TypeEnum,
    "ZipDetails.TypeEnum": zipDetails_1.ZipDetails.TypeEnum,
};
let typeMap = {
    "AccountInfo": accountInfo_1.AccountInfo,
    "AcctInfo": acctInfo_1.AcctInfo,
    "AchDetails": achDetails_1.AchDetails,
    "AdditionalData3DSecure": additionalData3DSecure_1.AdditionalData3DSecure,
    "AdditionalDataAirline": additionalDataAirline_1.AdditionalDataAirline,
    "AdditionalDataCarRental": additionalDataCarRental_1.AdditionalDataCarRental,
    "AdditionalDataCommon": additionalDataCommon_1.AdditionalDataCommon,
    "AdditionalDataLevel23": additionalDataLevel23_1.AdditionalDataLevel23,
    "AdditionalDataLodging": additionalDataLodging_1.AdditionalDataLodging,
    "AdditionalDataOpenInvoice": additionalDataOpenInvoice_1.AdditionalDataOpenInvoice,
    "AdditionalDataOpi": additionalDataOpi_1.AdditionalDataOpi,
    "AdditionalDataRatepay": additionalDataRatepay_1.AdditionalDataRatepay,
    "AdditionalDataRetry": additionalDataRetry_1.AdditionalDataRetry,
    "AdditionalDataRisk": additionalDataRisk_1.AdditionalDataRisk,
    "AdditionalDataRiskStandalone": additionalDataRiskStandalone_1.AdditionalDataRiskStandalone,
    "AdditionalDataSubMerchant": additionalDataSubMerchant_1.AdditionalDataSubMerchant,
    "AdditionalDataTemporaryServices": additionalDataTemporaryServices_1.AdditionalDataTemporaryServices,
    "AdditionalDataWallets": additionalDataWallets_1.AdditionalDataWallets,
    "Address": address_1.Address,
    "AfterpayDetails": afterpayDetails_1.AfterpayDetails,
    "AmazonPayDetails": amazonPayDetails_1.AmazonPayDetails,
    "Amount": amount_1.Amount,
    "AndroidPayDetails": androidPayDetails_1.AndroidPayDetails,
    "ApplePayDetails": applePayDetails_1.ApplePayDetails,
    "ApplePaySessionRequest": applePaySessionRequest_1.ApplePaySessionRequest,
    "ApplePaySessionResponse": applePaySessionResponse_1.ApplePaySessionResponse,
    "ApplicationInfo": applicationInfo_1.ApplicationInfo,
    "AuthenticationData": authenticationData_1.AuthenticationData,
    "Avs": avs_1.Avs,
    "BacsDirectDebitDetails": bacsDirectDebitDetails_1.BacsDirectDebitDetails,
    "BalanceCheckRequest": balanceCheckRequest_1.BalanceCheckRequest,
    "BalanceCheckResponse": balanceCheckResponse_1.BalanceCheckResponse,
    "BankAccount": bankAccount_1.BankAccount,
    "BillDeskDetails": billDeskDetails_1.BillDeskDetails,
    "BillingAddress": billingAddress_1.BillingAddress,
    "BlikDetails": blikDetails_1.BlikDetails,
    "BrowserInfo": browserInfo_1.BrowserInfo,
    "CancelOrderRequest": cancelOrderRequest_1.CancelOrderRequest,
    "CancelOrderResponse": cancelOrderResponse_1.CancelOrderResponse,
    "Card": card_1.Card,
    "CardBrandDetails": cardBrandDetails_1.CardBrandDetails,
    "CardDetails": cardDetails_1.CardDetails,
    "CardDetailsRequest": cardDetailsRequest_1.CardDetailsRequest,
    "CardDetailsResponse": cardDetailsResponse_1.CardDetailsResponse,
    "CellulantDetails": cellulantDetails_1.CellulantDetails,
    "CheckoutAwaitAction": checkoutAwaitAction_1.CheckoutAwaitAction,
    "CheckoutDelegatedAuthenticationAction": checkoutDelegatedAuthenticationAction_1.CheckoutDelegatedAuthenticationAction,
    "CheckoutNativeRedirectAction": checkoutNativeRedirectAction_1.CheckoutNativeRedirectAction,
    "CheckoutOrderResponse": checkoutOrderResponse_1.CheckoutOrderResponse,
    "CheckoutQrCodeAction": checkoutQrCodeAction_1.CheckoutQrCodeAction,
    "CheckoutRedirectAction": checkoutRedirectAction_1.CheckoutRedirectAction,
    "CheckoutSDKAction": checkoutSDKAction_1.CheckoutSDKAction,
    "CheckoutSessionInstallmentOption": checkoutSessionInstallmentOption_1.CheckoutSessionInstallmentOption,
    "CheckoutThreeDS2Action": checkoutThreeDS2Action_1.CheckoutThreeDS2Action,
    "CheckoutVoucherAction": checkoutVoucherAction_1.CheckoutVoucherAction,
    "CommonField": commonField_1.CommonField,
    "Company": company_1.Company,
    "Configuration": configuration_1.Configuration,
    "CreateCheckoutSessionRequest": createCheckoutSessionRequest_1.CreateCheckoutSessionRequest,
    "CreateCheckoutSessionResponse": createCheckoutSessionResponse_1.CreateCheckoutSessionResponse,
    "CreateOrderRequest": createOrderRequest_1.CreateOrderRequest,
    "CreateOrderResponse": createOrderResponse_1.CreateOrderResponse,
    "DeliveryAddress": deliveryAddress_1.DeliveryAddress,
    "DetailsRequestAuthenticationData": detailsRequestAuthenticationData_1.DetailsRequestAuthenticationData,
    "DeviceRenderOptions": deviceRenderOptions_1.DeviceRenderOptions,
    "DokuDetails": dokuDetails_1.DokuDetails,
    "DonationPaymentRequest": donationPaymentRequest_1.DonationPaymentRequest,
    "DonationPaymentResponse": donationPaymentResponse_1.DonationPaymentResponse,
    "DotpayDetails": dotpayDetails_1.DotpayDetails,
    "DragonpayDetails": dragonpayDetails_1.DragonpayDetails,
    "EcontextVoucherDetails": econtextVoucherDetails_1.EcontextVoucherDetails,
    "EncryptedOrderData": encryptedOrderData_1.EncryptedOrderData,
    "ExternalPlatform": externalPlatform_1.ExternalPlatform,
    "ForexQuote": forexQuote_1.ForexQuote,
    "FraudCheckResult": fraudCheckResult_1.FraudCheckResult,
    "FraudResult": fraudResult_1.FraudResult,
    "FundOrigin": fundOrigin_1.FundOrigin,
    "FundRecipient": fundRecipient_1.FundRecipient,
    "GenericIssuerPaymentMethodDetails": genericIssuerPaymentMethodDetails_1.GenericIssuerPaymentMethodDetails,
    "GiropayDetails": giropayDetails_1.GiropayDetails,
    "GooglePayDetails": googlePayDetails_1.GooglePayDetails,
    "IdealDetails": idealDetails_1.IdealDetails,
    "InputDetail": inputDetail_1.InputDetail,
    "InstallmentOption": installmentOption_1.InstallmentOption,
    "Installments": installments_1.Installments,
    "InstallmentsNumber": installmentsNumber_1.InstallmentsNumber,
    "Item": item_1.Item,
    "KlarnaDetails": klarnaDetails_1.KlarnaDetails,
    "LineItem": lineItem_1.LineItem,
    "ListStoredPaymentMethodsResponse": listStoredPaymentMethodsResponse_1.ListStoredPaymentMethodsResponse,
    "Mandate": mandate_1.Mandate,
    "MasterpassDetails": masterpassDetails_1.MasterpassDetails,
    "MbwayDetails": mbwayDetails_1.MbwayDetails,
    "MerchantDevice": merchantDevice_1.MerchantDevice,
    "MerchantRiskIndicator": merchantRiskIndicator_1.MerchantRiskIndicator,
    "MobilePayDetails": mobilePayDetails_1.MobilePayDetails,
    "MolPayDetails": molPayDetails_1.MolPayDetails,
    "Name": name_1.Name,
    "OpenInvoiceDetails": openInvoiceDetails_1.OpenInvoiceDetails,
    "PayPalDetails": payPalDetails_1.PayPalDetails,
    "PayUUpiDetails": payUUpiDetails_1.PayUUpiDetails,
    "PayWithGoogleDetails": payWithGoogleDetails_1.PayWithGoogleDetails,
    "PaymentAmountUpdateRequest": paymentAmountUpdateRequest_1.PaymentAmountUpdateRequest,
    "PaymentAmountUpdateResponse": paymentAmountUpdateResponse_1.PaymentAmountUpdateResponse,
    "PaymentCancelRequest": paymentCancelRequest_1.PaymentCancelRequest,
    "PaymentCancelResponse": paymentCancelResponse_1.PaymentCancelResponse,
    "PaymentCaptureRequest": paymentCaptureRequest_1.PaymentCaptureRequest,
    "PaymentCaptureResponse": paymentCaptureResponse_1.PaymentCaptureResponse,
    "PaymentCompletionDetails": paymentCompletionDetails_1.PaymentCompletionDetails,
    "PaymentDetails": paymentDetails_1.PaymentDetails,
    "PaymentDetailsRequest": paymentDetailsRequest_1.PaymentDetailsRequest,
    "PaymentDetailsResponse": paymentDetailsResponse_1.PaymentDetailsResponse,
    "PaymentLinkRequest": paymentLinkRequest_1.PaymentLinkRequest,
    "PaymentLinkResponse": paymentLinkResponse_1.PaymentLinkResponse,
    "PaymentMethod": paymentMethod_1.PaymentMethod,
    "PaymentMethodGroup": paymentMethodGroup_1.PaymentMethodGroup,
    "PaymentMethodIssuer": paymentMethodIssuer_1.PaymentMethodIssuer,
    "PaymentMethodsRequest": paymentMethodsRequest_1.PaymentMethodsRequest,
    "PaymentMethodsResponse": paymentMethodsResponse_1.PaymentMethodsResponse,
    "PaymentRefundRequest": paymentRefundRequest_1.PaymentRefundRequest,
    "PaymentRefundResponse": paymentRefundResponse_1.PaymentRefundResponse,
    "PaymentRequest": paymentRequest_1.PaymentRequest,
    "PaymentResponse": paymentResponse_1.PaymentResponse,
    "PaymentReversalRequest": paymentReversalRequest_1.PaymentReversalRequest,
    "PaymentReversalResponse": paymentReversalResponse_1.PaymentReversalResponse,
    "PaymentSetupRequest": paymentSetupRequest_1.PaymentSetupRequest,
    "PaymentSetupResponse": paymentSetupResponse_1.PaymentSetupResponse,
    "PaymentVerificationRequest": paymentVerificationRequest_1.PaymentVerificationRequest,
    "PaymentVerificationResponse": paymentVerificationResponse_1.PaymentVerificationResponse,
    "Phone": phone_1.Phone,
    "PlatformChargebackLogic": platformChargebackLogic_1.PlatformChargebackLogic,
    "RatepayDetails": ratepayDetails_1.RatepayDetails,
    "Recurring": recurring_1.Recurring,
    "RecurringDetail": recurringDetail_1.RecurringDetail,
    "ResponseAdditionalData3DSecure": responseAdditionalData3DSecure_1.ResponseAdditionalData3DSecure,
    "ResponseAdditionalDataBillingAddress": responseAdditionalDataBillingAddress_1.ResponseAdditionalDataBillingAddress,
    "ResponseAdditionalDataCard": responseAdditionalDataCard_1.ResponseAdditionalDataCard,
    "ResponseAdditionalDataCommon": responseAdditionalDataCommon_1.ResponseAdditionalDataCommon,
    "ResponseAdditionalDataInstallments": responseAdditionalDataInstallments_1.ResponseAdditionalDataInstallments,
    "ResponseAdditionalDataNetworkTokens": responseAdditionalDataNetworkTokens_1.ResponseAdditionalDataNetworkTokens,
    "ResponseAdditionalDataOpi": responseAdditionalDataOpi_1.ResponseAdditionalDataOpi,
    "ResponseAdditionalDataSepa": responseAdditionalDataSepa_1.ResponseAdditionalDataSepa,
    "ResponsePaymentMethod": responsePaymentMethod_1.ResponsePaymentMethod,
    "RiskData": riskData_1.RiskData,
    "SDKEphemPubKey": sDKEphemPubKey_1.SDKEphemPubKey,
    "SamsungPayDetails": samsungPayDetails_1.SamsungPayDetails,
    "SepaDirectDebitDetails": sepaDirectDebitDetails_1.SepaDirectDebitDetails,
    "ServiceError": serviceError_1.ServiceError,
    "ServiceError2": serviceError2_1.ServiceError2,
    "SessionResultResponse": sessionResultResponse_1.SessionResultResponse,
    "ShopperInput": shopperInput_1.ShopperInput,
    "ShopperInteractionDevice": shopperInteractionDevice_1.ShopperInteractionDevice,
    "Split": split_1.Split,
    "SplitAmount": splitAmount_1.SplitAmount,
    "StandalonePaymentCancelRequest": standalonePaymentCancelRequest_1.StandalonePaymentCancelRequest,
    "StandalonePaymentCancelResponse": standalonePaymentCancelResponse_1.StandalonePaymentCancelResponse,
    "StoredDetails": storedDetails_1.StoredDetails,
    "StoredPaymentMethod": storedPaymentMethod_1.StoredPaymentMethod,
    "StoredPaymentMethodDetails": storedPaymentMethodDetails_1.StoredPaymentMethodDetails,
    "StoredPaymentMethodResource": storedPaymentMethodResource_1.StoredPaymentMethodResource,
    "SubInputDetail": subInputDetail_1.SubInputDetail,
    "SubMerchant": subMerchant_1.SubMerchant,
    "SubMerchantInfo": subMerchantInfo_1.SubMerchantInfo,
    "ThreeDS2RequestData": threeDS2RequestData_1.ThreeDS2RequestData,
    "ThreeDS2RequestData2": threeDS2RequestData2_1.ThreeDS2RequestData2,
    "ThreeDS2ResponseData": threeDS2ResponseData_1.ThreeDS2ResponseData,
    "ThreeDS2Result": threeDS2Result_1.ThreeDS2Result,
    "ThreeDSRequestData": threeDSRequestData_1.ThreeDSRequestData,
    "ThreeDSRequestorAuthenticationInfo": threeDSRequestorAuthenticationInfo_1.ThreeDSRequestorAuthenticationInfo,
    "ThreeDSRequestorPriorAuthenticationInfo": threeDSRequestorPriorAuthenticationInfo_1.ThreeDSRequestorPriorAuthenticationInfo,
    "ThreeDSecureData": threeDSecureData_1.ThreeDSecureData,
    "UpdatePaymentLinkRequest": updatePaymentLinkRequest_1.UpdatePaymentLinkRequest,
    "UpiCollectDetails": upiCollectDetails_1.UpiCollectDetails,
    "UpiIntentDetails": upiIntentDetails_1.UpiIntentDetails,
    "UtilityRequest": utilityRequest_1.UtilityRequest,
    "UtilityResponse": utilityResponse_1.UtilityResponse,
    "VippsDetails": vippsDetails_1.VippsDetails,
    "VisaCheckoutDetails": visaCheckoutDetails_1.VisaCheckoutDetails,
    "WeChatPayDetails": weChatPayDetails_1.WeChatPayDetails,
    "WeChatPayMiniProgramDetails": weChatPayMiniProgramDetails_1.WeChatPayMiniProgramDetails,
    "ZipDetails": zipDetails_1.ZipDetails,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }
            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType; // use the type given in the discriminator
                    }
                    else {
                        return expectedType; // discriminator did not map to a type
                    }
                }
                else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.serialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return data.toISOString();
        }
        else if (type === "SaleToAcquirerData") {
            const dataString = JSON.stringify(data);
            return Buffer.from(dataString).toString("base64");
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) { // in case we dont know the type
                return data;
            }
            // Get the actual type of this object
            type = this.findCorrectType(data, type);
            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.deserialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) { // is Enum
                return data;
            }
            if (!typeMap[type]) { // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
