// @adyen/api-library/lib/src/typings/checkout/paymentMethodsRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodsRequest = void 0;
class PaymentMethodsRequest {
    static getAttributeTypeMap() {
        return PaymentMethodsRequest.attributeTypeMap;
    }
}
exports.PaymentMethodsRequest = PaymentMethodsRequest;
PaymentMethodsRequest.discriminator = undefined;
PaymentMethodsRequest.attributeTypeMap = [
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "allowedPaymentMethods",
        "baseName": "allowedPaymentMethods",
        "type": "Array<string>"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "blockedPaymentMethods",
        "baseName": "blockedPaymentMethods",
        "type": "Array<string>"
    },
    {
        "name": "channel",
        "baseName": "channel",
        "type": "PaymentMethodsRequest.ChannelEnum"
    },
    {
        "name": "countryCode",
        "baseName": "countryCode",
        "type": "string"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "order",
        "baseName": "order",
        "type": "EncryptedOrderData"
    },
    {
        "name": "shopperLocale",
        "baseName": "shopperLocale",
        "type": "string"
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string"
    },
    {
        "name": "splitCardFundingSources",
        "baseName": "splitCardFundingSources",
        "type": "boolean"
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string"
    }
];
(function (PaymentMethodsRequest) {
    let ChannelEnum;
    (function (ChannelEnum) {
        ChannelEnum["IOs"] = "iOS";
        ChannelEnum["Android"] = "Android";
        ChannelEnum["Web"] = "Web";
    })(ChannelEnum = PaymentMethodsRequest.ChannelEnum || (PaymentMethodsRequest.ChannelEnum = {}));
})(PaymentMethodsRequest = exports.PaymentMethodsRequest || (exports.PaymentMethodsRequest = {}));
