// @adyen/api-library/lib/src/typings/checkout/paymentSetupRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentSetupRequest = void 0;
class PaymentSetupRequest {
    static getAttributeTypeMap() {
        return PaymentSetupRequest.attributeTypeMap;
    }
}
exports.PaymentSetupRequest = PaymentSetupRequest;
PaymentSetupRequest.discriminator = undefined;
PaymentSetupRequest.attributeTypeMap = [
    {
        "name": "additionalAmount",
        "baseName": "additionalAmount",
        "type": "Amount"
    },
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "allowedPaymentMethods",
        "baseName": "allowedPaymentMethods",
        "type": "Array<string>"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "applicationInfo",
        "baseName": "applicationInfo",
        "type": "ApplicationInfo"
    },
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "Address"
    },
    {
        "name": "blockedPaymentMethods",
        "baseName": "blockedPaymentMethods",
        "type": "Array<string>"
    },
    {
        "name": "captureDelayHours",
        "baseName": "captureDelayHours",
        "type": "number"
    },
    {
        "name": "channel",
        "baseName": "channel",
        "type": "PaymentSetupRequest.ChannelEnum"
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "company",
        "baseName": "company",
        "type": "Company"
    },
    {
        "name": "configuration",
        "baseName": "configuration",
        "type": "Configuration"
    },
    {
        "name": "conversionId",
        "baseName": "conversionId",
        "type": "string"
    },
    {
        "name": "countryCode",
        "baseName": "countryCode",
        "type": "string"
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "string"
    },
    {
        "name": "dccQuote",
        "baseName": "dccQuote",
        "type": "ForexQuote"
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "Address"
    },
    {
        "name": "deliveryDate",
        "baseName": "deliveryDate",
        "type": "Date"
    },
    {
        "name": "enableOneClick",
        "baseName": "enableOneClick",
        "type": "boolean"
    },
    {
        "name": "enablePayOut",
        "baseName": "enablePayOut",
        "type": "boolean"
    },
    {
        "name": "enableRecurring",
        "baseName": "enableRecurring",
        "type": "boolean"
    },
    {
        "name": "entityType",
        "baseName": "entityType",
        "type": "PaymentSetupRequest.EntityTypeEnum"
    },
    {
        "name": "fraudOffset",
        "baseName": "fraudOffset",
        "type": "number"
    },
    {
        "name": "installments",
        "baseName": "installments",
        "type": "Installments"
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>"
    },
    {
        "name": "localizedShopperStatement",
        "baseName": "localizedShopperStatement",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "mandate",
        "baseName": "mandate",
        "type": "Mandate"
    },
    {
        "name": "mcc",
        "baseName": "mcc",
        "type": "string"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "merchantOrderReference",
        "baseName": "merchantOrderReference",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "orderReference",
        "baseName": "orderReference",
        "type": "string"
    },
    {
        "name": "origin",
        "baseName": "origin",
        "type": "string"
    },
    {
        "name": "platformChargebackLogic",
        "baseName": "platformChargebackLogic",
        "type": "PlatformChargebackLogic"
    },
    {
        "name": "recurringExpiry",
        "baseName": "recurringExpiry",
        "type": "string"
    },
    {
        "name": "recurringFrequency",
        "baseName": "recurringFrequency",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "returnUrl",
        "baseName": "returnUrl",
        "type": "string"
    },
    {
        "name": "riskData",
        "baseName": "riskData",
        "type": "RiskData"
    },
    {
        "name": "sdkVersion",
        "baseName": "sdkVersion",
        "type": "string"
    },
    {
        "name": "sessionValidity",
        "baseName": "sessionValidity",
        "type": "string"
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string"
    },
    {
        "name": "shopperIP",
        "baseName": "shopperIP",
        "type": "string"
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "PaymentSetupRequest.ShopperInteractionEnum"
    },
    {
        "name": "shopperLocale",
        "baseName": "shopperLocale",
        "type": "string"
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name"
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string"
    },
    {
        "name": "shopperStatement",
        "baseName": "shopperStatement",
        "type": "string"
    },
    {
        "name": "socialSecurityNumber",
        "baseName": "socialSecurityNumber",
        "type": "string"
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>"
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string"
    },
    {
        "name": "storePaymentMethod",
        "baseName": "storePaymentMethod",
        "type": "boolean"
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string"
    },
    {
        "name": "threeDSAuthenticationOnly",
        "baseName": "threeDSAuthenticationOnly",
        "type": "boolean"
    },
    {
        "name": "token",
        "baseName": "token",
        "type": "string"
    },
    {
        "name": "trustedShopper",
        "baseName": "trustedShopper",
        "type": "boolean"
    }
];
(function (PaymentSetupRequest) {
    let ChannelEnum;
    (function (ChannelEnum) {
        ChannelEnum["IOs"] = "iOS";
        ChannelEnum["Android"] = "Android";
        ChannelEnum["Web"] = "Web";
    })(ChannelEnum = PaymentSetupRequest.ChannelEnum || (PaymentSetupRequest.ChannelEnum = {}));
    let EntityTypeEnum;
    (function (EntityTypeEnum) {
        EntityTypeEnum["NaturalPerson"] = "NaturalPerson";
        EntityTypeEnum["CompanyName"] = "CompanyName";
    })(EntityTypeEnum = PaymentSetupRequest.EntityTypeEnum || (PaymentSetupRequest.EntityTypeEnum = {}));
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = PaymentSetupRequest.ShopperInteractionEnum || (PaymentSetupRequest.ShopperInteractionEnum = {}));
})(PaymentSetupRequest = exports.PaymentSetupRequest || (exports.PaymentSetupRequest = {}));
