// @adyen/api-library/lib/src/typings/checkout/paymentDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentDetails = void 0;
class PaymentDetails {
    static getAttributeTypeMap() {
        return PaymentDetails.attributeTypeMap;
    }
}
exports.PaymentDetails = PaymentDetails;
PaymentDetails.discriminator = undefined;
PaymentDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PaymentDetails.TypeEnum"
    }
];
(function (PaymentDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Alipay"] = "alipay";
        TypeEnum["Multibanco"] = "multibanco";
        TypeEnum["BankTransferIban"] = "bankTransfer_IBAN";
        TypeEnum["Paybright"] = "paybright";
        TypeEnum["Paynow"] = "paynow";
        TypeEnum["Affirm"] = "affirm";
        TypeEnum["AffirmPos"] = "affirm_pos";
        TypeEnum["Trustly"] = "trustly";
        TypeEnum["Trustlyvector"] = "trustlyvector";
        TypeEnum["Oney"] = "oney";
        TypeEnum["Facilypay"] = "facilypay";
        TypeEnum["Facilypay3x"] = "facilypay_3x";
        TypeEnum["Facilypay4x"] = "facilypay_4x";
        TypeEnum["Facilypay6x"] = "facilypay_6x";
        TypeEnum["Facilypay10x"] = "facilypay_10x";
        TypeEnum["Facilypay12x"] = "facilypay_12x";
        TypeEnum["Unionpay"] = "unionpay";
        TypeEnum["KcpBanktransfer"] = "kcp_banktransfer";
        TypeEnum["KcpPayco"] = "kcp_payco";
        TypeEnum["KcpCreditcard"] = "kcp_creditcard";
        TypeEnum["WechatpaySdk"] = "wechatpaySDK";
        TypeEnum["WechatpayQr"] = "wechatpayQR";
        TypeEnum["WechatpayWeb"] = "wechatpayWeb";
        TypeEnum["MolpayBoost"] = "molpay_boost";
        TypeEnum["WalletIn"] = "wallet_IN";
        TypeEnum["PayuInCashcard"] = "payu_IN_cashcard";
        TypeEnum["PayuInNb"] = "payu_IN_nb";
        TypeEnum["UpiQr"] = "upi_qr";
        TypeEnum["Paytm"] = "paytm";
        TypeEnum["MolpayEbankingVn"] = "molpay_ebanking_VN";
        TypeEnum["Paybybank"] = "paybybank";
        TypeEnum["EbankingFi"] = "ebanking_FI";
        TypeEnum["MolpayEbankingMy"] = "molpay_ebanking_MY";
        TypeEnum["MolpayEbankingDirectMy"] = "molpay_ebanking_direct_MY";
        TypeEnum["Swish"] = "swish";
        TypeEnum["Pix"] = "pix";
        TypeEnum["Walley"] = "walley";
        TypeEnum["WalleyB2b"] = "walley_b2b";
        TypeEnum["Alma"] = "alma";
        TypeEnum["MolpayFpx"] = "molpay_fpx";
        TypeEnum["Konbini"] = "konbini";
        TypeEnum["DirectEbanking"] = "directEbanking";
        TypeEnum["Boletobancario"] = "boletobancario";
        TypeEnum["Neteller"] = "neteller";
        TypeEnum["Paysafecard"] = "paysafecard";
        TypeEnum["Cashticket"] = "cashticket";
        TypeEnum["Ikano"] = "ikano";
        TypeEnum["Karenmillen"] = "karenmillen";
        TypeEnum["Oasis"] = "oasis";
        TypeEnum["Warehouse"] = "warehouse";
        TypeEnum["PrimeiropayBoleto"] = "primeiropay_boleto";
        TypeEnum["Mada"] = "mada";
        TypeEnum["Benefit"] = "benefit";
        TypeEnum["Knet"] = "knet";
        TypeEnum["Omannet"] = "omannet";
        TypeEnum["GopayWallet"] = "gopay_wallet";
        TypeEnum["KcpNaverpay"] = "kcp_naverpay";
        TypeEnum["OnlinebankingIn"] = "onlinebanking_IN";
        TypeEnum["Fawry"] = "fawry";
        TypeEnum["Atome"] = "atome";
        TypeEnum["Moneybookers"] = "moneybookers";
        TypeEnum["Naps"] = "naps";
        TypeEnum["Nordea"] = "nordea";
        TypeEnum["BoletobancarioBradesco"] = "boletobancario_bradesco";
        TypeEnum["BoletobancarioItau"] = "boletobancario_itau";
        TypeEnum["BoletobancarioSantander"] = "boletobancario_santander";
        TypeEnum["BoletobancarioBancodobrasil"] = "boletobancario_bancodobrasil";
        TypeEnum["BoletobancarioHsbc"] = "boletobancario_hsbc";
        TypeEnum["MolpayMaybank2u"] = "molpay_maybank2u";
        TypeEnum["MolpayCimb"] = "molpay_cimb";
        TypeEnum["MolpayRhb"] = "molpay_rhb";
        TypeEnum["MolpayAmb"] = "molpay_amb";
        TypeEnum["MolpayHlb"] = "molpay_hlb";
        TypeEnum["MolpayAffinEpg"] = "molpay_affin_epg";
        TypeEnum["MolpayBankislam"] = "molpay_bankislam";
        TypeEnum["MolpayPublicbank"] = "molpay_publicbank";
        TypeEnum["FpxAgrobank"] = "fpx_agrobank";
        TypeEnum["Touchngo"] = "touchngo";
        TypeEnum["Maybank2uMae"] = "maybank2u_mae";
        TypeEnum["Duitnow"] = "duitnow";
        TypeEnum["Promptpay"] = "promptpay";
        TypeEnum["TwintPos"] = "twint_pos";
        TypeEnum["AlipayHk"] = "alipay_hk";
        TypeEnum["AlipayHkWeb"] = "alipay_hk_web";
        TypeEnum["AlipayHkWap"] = "alipay_hk_wap";
        TypeEnum["AlipayWap"] = "alipay_wap";
        TypeEnum["Balanceplatform"] = "balanceplatform";
    })(TypeEnum = PaymentDetails.TypeEnum || (PaymentDetails.TypeEnum = {}));
})(PaymentDetails = exports.PaymentDetails || (exports.PaymentDetails = {}));
