// @adyen/api-library/lib/src/typings/checkout/paymentResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentResponse = void 0;
class PaymentResponse {
    static getAttributeTypeMap() {
        return PaymentResponse.attributeTypeMap;
    }
}
exports.PaymentResponse = PaymentResponse;
PaymentResponse.discriminator = undefined;
PaymentResponse.attributeTypeMap = [
    {
        "name": "action",
        "baseName": "action",
        "type": "CheckoutAwaitAction | CheckoutDelegatedAuthenticationAction | CheckoutNativeRedirectAction | CheckoutQrCodeAction | CheckoutRedirectAction | CheckoutSDKAction | CheckoutThreeDS2Action | CheckoutVoucherAction"
    },
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "donationToken",
        "baseName": "donationToken",
        "type": "string"
    },
    {
        "name": "fraudResult",
        "baseName": "fraudResult",
        "type": "FraudResult"
    },
    {
        "name": "merchantReference",
        "baseName": "merchantReference",
        "type": "string"
    },
    {
        "name": "order",
        "baseName": "order",
        "type": "CheckoutOrderResponse"
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "ResponsePaymentMethod"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string"
    },
    {
        "name": "refusalReasonCode",
        "baseName": "refusalReasonCode",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "PaymentResponse.ResultCodeEnum"
    },
    {
        "name": "threeDS2ResponseData",
        "baseName": "threeDS2ResponseData",
        "type": "ThreeDS2ResponseData"
    },
    {
        "name": "threeDS2Result",
        "baseName": "threeDS2Result",
        "type": "ThreeDS2Result"
    },
    {
        "name": "threeDSPaymentData",
        "baseName": "threeDSPaymentData",
        "type": "string"
    }
];
(function (PaymentResponse) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["AuthenticationFinished"] = "AuthenticationFinished";
        ResultCodeEnum["AuthenticationNotRequired"] = "AuthenticationNotRequired";
        ResultCodeEnum["Authorised"] = "Authorised";
        ResultCodeEnum["Cancelled"] = "Cancelled";
        ResultCodeEnum["ChallengeShopper"] = "ChallengeShopper";
        ResultCodeEnum["Error"] = "Error";
        ResultCodeEnum["IdentifyShopper"] = "IdentifyShopper";
        ResultCodeEnum["PartiallyAuthorised"] = "PartiallyAuthorised";
        ResultCodeEnum["Pending"] = "Pending";
        ResultCodeEnum["PresentToShopper"] = "PresentToShopper";
        ResultCodeEnum["Received"] = "Received";
        ResultCodeEnum["RedirectShopper"] = "RedirectShopper";
        ResultCodeEnum["Refused"] = "Refused";
        ResultCodeEnum["Success"] = "Success";
    })(ResultCodeEnum = PaymentResponse.ResultCodeEnum || (PaymentResponse.ResultCodeEnum = {}));
})(PaymentResponse = exports.PaymentResponse || (exports.PaymentResponse = {}));
