// @adyen/api-library/lib/src/typings/checkout/paymentRefundRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRefundRequest = void 0;
class PaymentRefundRequest {
    static getAttributeTypeMap() {
        return PaymentRefundRequest.attributeTypeMap;
    }
}
exports.PaymentRefundRequest = PaymentRefundRequest;
PaymentRefundRequest.discriminator = undefined;
PaymentRefundRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "merchantRefundReason",
        "baseName": "merchantRefundReason",
        "type": "PaymentRefundRequest.MerchantRefundReasonEnum"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>"
    }
];
(function (PaymentRefundRequest) {
    let MerchantRefundReasonEnum;
    (function (MerchantRefundReasonEnum) {
        MerchantRefundReasonEnum["Fraud"] = "FRAUD";
        MerchantRefundReasonEnum["CustomerRequest"] = "CUSTOMER REQUEST";
        MerchantRefundReasonEnum["Return"] = "RETURN";
        MerchantRefundReasonEnum["Duplicate"] = "DUPLICATE";
        MerchantRefundReasonEnum["Other"] = "OTHER";
    })(MerchantRefundReasonEnum = PaymentRefundRequest.MerchantRefundReasonEnum || (PaymentRefundRequest.MerchantRefundReasonEnum = {}));
})(PaymentRefundRequest = exports.PaymentRefundRequest || (exports.PaymentRefundRequest = {}));
