// @adyen/api-library/lib/src/typings/checkout/payPalDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayPalDetails = void 0;
class PayPalDetails {
    static getAttributeTypeMap() {
        return PayPalDetails.attributeTypeMap;
    }
}
exports.PayPalDetails = PayPalDetails;
PayPalDetails.discriminator = undefined;
PayPalDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "orderID",
        "baseName": "orderID",
        "type": "string"
    },
    {
        "name": "payeePreferred",
        "baseName": "payeePreferred",
        "type": "string"
    },
    {
        "name": "payerID",
        "baseName": "payerID",
        "type": "string"
    },
    {
        "name": "payerSelected",
        "baseName": "payerSelected",
        "type": "string"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "subtype",
        "baseName": "subtype",
        "type": "PayPalDetails.SubtypeEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PayPalDetails.TypeEnum"
    }
];
(function (PayPalDetails) {
    let SubtypeEnum;
    (function (SubtypeEnum) {
        SubtypeEnum["Redirect"] = "redirect";
        SubtypeEnum["Sdk"] = "sdk";
    })(SubtypeEnum = PayPalDetails.SubtypeEnum || (PayPalDetails.SubtypeEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Paypal"] = "paypal";
    })(TypeEnum = PayPalDetails.TypeEnum || (PayPalDetails.TypeEnum = {}));
})(PayPalDetails = exports.PayPalDetails || (exports.PayPalDetails = {}));
