// @adyen/api-library/lib/src/typings/checkout/payWithGoogleDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayWithGoogleDetails = void 0;
class PayWithGoogleDetails {
    static getAttributeTypeMap() {
        return PayWithGoogleDetails.attributeTypeMap;
    }
}
exports.PayWithGoogleDetails = PayWithGoogleDetails;
PayWithGoogleDetails.discriminator = undefined;
PayWithGoogleDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "PayWithGoogleDetails.FundingSourceEnum"
    },
    {
        "name": "googlePayToken",
        "baseName": "googlePayToken",
        "type": "string"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PayWithGoogleDetails.TypeEnum"
    }
];
(function (PayWithGoogleDetails) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = PayWithGoogleDetails.FundingSourceEnum || (PayWithGoogleDetails.FundingSourceEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Paywithgoogle"] = "paywithgoogle";
    })(TypeEnum = PayWithGoogleDetails.TypeEnum || (PayWithGoogleDetails.TypeEnum = {}));
})(PayWithGoogleDetails = exports.PayWithGoogleDetails || (exports.PayWithGoogleDetails = {}));
