// @adyen/api-library/lib/src/typings/checkout/threeDS2RequestData2.js
"use strict";
/*
 * The version of the OpenAPI document: v70
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreeDS2RequestData2 = void 0;
class ThreeDS2RequestData2 {
    static getAttributeTypeMap() {
        return ThreeDS2RequestData2.attributeTypeMap;
    }
}
exports.ThreeDS2RequestData2 = ThreeDS2RequestData2;
ThreeDS2RequestData2.discriminator = undefined;
ThreeDS2RequestData2.attributeTypeMap = [
    {
        "name": "acctInfo",
        "baseName": "acctInfo",
        "type": "AcctInfo"
    },
    {
        "name": "acctType",
        "baseName": "acctType",
        "type": "ThreeDS2RequestData2.AcctTypeEnum"
    },
    {
        "name": "acquirerBIN",
        "baseName": "acquirerBIN",
        "type": "string"
    },
    {
        "name": "acquirerMerchantID",
        "baseName": "acquirerMerchantID",
        "type": "string"
    },
    {
        "name": "addrMatch",
        "baseName": "addrMatch",
        "type": "ThreeDS2RequestData2.AddrMatchEnum"
    },
    {
        "name": "authenticationOnly",
        "baseName": "authenticationOnly",
        "type": "boolean"
    },
    {
        "name": "challengeIndicator",
        "baseName": "challengeIndicator",
        "type": "ThreeDS2RequestData2.ChallengeIndicatorEnum"
    },
    {
        "name": "deviceRenderOptions",
        "baseName": "deviceRenderOptions",
        "type": "DeviceRenderOptions"
    },
    {
        "name": "homePhone",
        "baseName": "homePhone",
        "type": "Phone"
    },
    {
        "name": "mcc",
        "baseName": "mcc",
        "type": "string"
    },
    {
        "name": "merchantName",
        "baseName": "merchantName",
        "type": "string"
    },
    {
        "name": "messageVersion",
        "baseName": "messageVersion",
        "type": "string"
    },
    {
        "name": "mobilePhone",
        "baseName": "mobilePhone",
        "type": "Phone"
    },
    {
        "name": "notificationURL",
        "baseName": "notificationURL",
        "type": "string"
    },
    {
        "name": "payTokenInd",
        "baseName": "payTokenInd",
        "type": "boolean"
    },
    {
        "name": "paymentAuthenticationUseCase",
        "baseName": "paymentAuthenticationUseCase",
        "type": "string"
    },
    {
        "name": "purchaseInstalData",
        "baseName": "purchaseInstalData",
        "type": "string"
    },
    {
        "name": "recurringExpiry",
        "baseName": "recurringExpiry",
        "type": "string"
    },
    {
        "name": "recurringFrequency",
        "baseName": "recurringFrequency",
        "type": "string"
    },
    {
        "name": "sdkAppID",
        "baseName": "sdkAppID",
        "type": "string"
    },
    {
        "name": "sdkEphemPubKey",
        "baseName": "sdkEphemPubKey",
        "type": "SDKEphemPubKey"
    },
    {
        "name": "sdkMaxTimeout",
        "baseName": "sdkMaxTimeout",
        "type": "number"
    },
    {
        "name": "sdkReferenceNumber",
        "baseName": "sdkReferenceNumber",
        "type": "string"
    },
    {
        "name": "sdkTransID",
        "baseName": "sdkTransID",
        "type": "string"
    },
    {
        "name": "threeDSCompInd",
        "baseName": "threeDSCompInd",
        "type": "string"
    },
    {
        "name": "threeDSRequestorAuthenticationInd",
        "baseName": "threeDSRequestorAuthenticationInd",
        "type": "string"
    },
    {
        "name": "threeDSRequestorAuthenticationInfo",
        "baseName": "threeDSRequestorAuthenticationInfo",
        "type": "ThreeDSRequestorAuthenticationInfo"
    },
    {
        "name": "threeDSRequestorChallengeInd",
        "baseName": "threeDSRequestorChallengeInd",
        "type": "ThreeDS2RequestData2.ThreeDSRequestorChallengeIndEnum"
    },
    {
        "name": "threeDSRequestorID",
        "baseName": "threeDSRequestorID",
        "type": "string"
    },
    {
        "name": "threeDSRequestorName",
        "baseName": "threeDSRequestorName",
        "type": "string"
    },
    {
        "name": "threeDSRequestorPriorAuthenticationInfo",
        "baseName": "threeDSRequestorPriorAuthenticationInfo",
        "type": "ThreeDSRequestorPriorAuthenticationInfo"
    },
    {
        "name": "threeDSRequestorURL",
        "baseName": "threeDSRequestorURL",
        "type": "string"
    },
    {
        "name": "transType",
        "baseName": "transType",
        "type": "ThreeDS2RequestData2.TransTypeEnum"
    },
    {
        "name": "transactionType",
        "baseName": "transactionType",
        "type": "ThreeDS2RequestData2.TransactionTypeEnum"
    },
    {
        "name": "whiteListStatus",
        "baseName": "whiteListStatus",
        "type": "string"
    },
    {
        "name": "workPhone",
        "baseName": "workPhone",
        "type": "Phone"
    }
];
(function (ThreeDS2RequestData2) {
    let AcctTypeEnum;
    (function (AcctTypeEnum) {
        AcctTypeEnum["_01"] = "01";
        AcctTypeEnum["_02"] = "02";
        AcctTypeEnum["_03"] = "03";
    })(AcctTypeEnum = ThreeDS2RequestData2.AcctTypeEnum || (ThreeDS2RequestData2.AcctTypeEnum = {}));
    let AddrMatchEnum;
    (function (AddrMatchEnum) {
        AddrMatchEnum["Y"] = "Y";
        AddrMatchEnum["N"] = "N";
    })(AddrMatchEnum = ThreeDS2RequestData2.AddrMatchEnum || (ThreeDS2RequestData2.AddrMatchEnum = {}));
    let ChallengeIndicatorEnum;
    (function (ChallengeIndicatorEnum) {
        ChallengeIndicatorEnum["NoPreference"] = "noPreference";
        ChallengeIndicatorEnum["RequestNoChallenge"] = "requestNoChallenge";
        ChallengeIndicatorEnum["RequestChallenge"] = "requestChallenge";
        ChallengeIndicatorEnum["RequestChallengeAsMandate"] = "requestChallengeAsMandate";
    })(ChallengeIndicatorEnum = ThreeDS2RequestData2.ChallengeIndicatorEnum || (ThreeDS2RequestData2.ChallengeIndicatorEnum = {}));
    let ThreeDSRequestorChallengeIndEnum;
    (function (ThreeDSRequestorChallengeIndEnum) {
        ThreeDSRequestorChallengeIndEnum["_01"] = "01";
        ThreeDSRequestorChallengeIndEnum["_02"] = "02";
        ThreeDSRequestorChallengeIndEnum["_03"] = "03";
        ThreeDSRequestorChallengeIndEnum["_04"] = "04";
        ThreeDSRequestorChallengeIndEnum["_05"] = "05";
        ThreeDSRequestorChallengeIndEnum["_06"] = "06";
    })(ThreeDSRequestorChallengeIndEnum = ThreeDS2RequestData2.ThreeDSRequestorChallengeIndEnum || (ThreeDS2RequestData2.ThreeDSRequestorChallengeIndEnum = {}));
    let TransTypeEnum;
    (function (TransTypeEnum) {
        TransTypeEnum["_01"] = "01";
        TransTypeEnum["_03"] = "03";
        TransTypeEnum["_10"] = "10";
        TransTypeEnum["_11"] = "11";
        TransTypeEnum["_28"] = "28";
    })(TransTypeEnum = ThreeDS2RequestData2.TransTypeEnum || (ThreeDS2RequestData2.TransTypeEnum = {}));
    let TransactionTypeEnum;
    (function (TransactionTypeEnum) {
        TransactionTypeEnum["GoodsOrServicePurchase"] = "goodsOrServicePurchase";
        TransactionTypeEnum["CheckAcceptance"] = "checkAcceptance";
        TransactionTypeEnum["AccountFunding"] = "accountFunding";
        TransactionTypeEnum["QuasiCashTransaction"] = "quasiCashTransaction";
        TransactionTypeEnum["PrepaidActivationAndLoad"] = "prepaidActivationAndLoad";
    })(TransactionTypeEnum = ThreeDS2RequestData2.TransactionTypeEnum || (ThreeDS2RequestData2.TransactionTypeEnum = {}));
})(ThreeDS2RequestData2 = exports.ThreeDS2RequestData2 || (exports.ThreeDS2RequestData2 = {}));
