// @adyen/api-library/lib/src/typings/checkout/visaCheckoutDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VisaCheckoutDetails = void 0;
class VisaCheckoutDetails {
    static getAttributeTypeMap() {
        return VisaCheckoutDetails.attributeTypeMap;
    }
}
exports.VisaCheckoutDetails = VisaCheckoutDetails;
VisaCheckoutDetails.discriminator = undefined;
VisaCheckoutDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "VisaCheckoutDetails.FundingSourceEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "VisaCheckoutDetails.TypeEnum"
    },
    {
        "name": "visaCheckoutCallId",
        "baseName": "visaCheckoutCallId",
        "type": "string"
    }
];
(function (VisaCheckoutDetails) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = VisaCheckoutDetails.FundingSourceEnum || (VisaCheckoutDetails.FundingSourceEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Visacheckout"] = "visacheckout";
    })(TypeEnum = VisaCheckoutDetails.TypeEnum || (VisaCheckoutDetails.TypeEnum = {}));
})(VisaCheckoutDetails = exports.VisaCheckoutDetails || (exports.VisaCheckoutDetails = {}));
