// @adyen/api-library/lib/src/typings/checkout/weChatPayDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeChatPayDetails = void 0;
class WeChatPayDetails {
    static getAttributeTypeMap() {
        return WeChatPayDetails.attributeTypeMap;
    }
}
exports.WeChatPayDetails = WeChatPayDetails;
WeChatPayDetails.discriminator = undefined;
WeChatPayDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "WeChatPayDetails.TypeEnum"
    }
];
(function (WeChatPayDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Wechatpay"] = "wechatpay";
        TypeEnum["WechatpayPos"] = "wechatpay_pos";
    })(TypeEnum = WeChatPayDetails.TypeEnum || (WeChatPayDetails.TypeEnum = {}));
})(WeChatPayDetails = exports.WeChatPayDetails || (exports.WeChatPayDetails = {}));
