// @adyen/api-library/lib/src/typings/checkout/zipDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZipDetails = void 0;
class ZipDetails {
    static getAttributeTypeMap() {
        return ZipDetails.attributeTypeMap;
    }
}
exports.ZipDetails = ZipDetails;
ZipDetails.discriminator = undefined;
ZipDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "clickAndCollect",
        "baseName": "clickAndCollect",
        "type": "string"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "ZipDetails.TypeEnum"
    }
];
(function (ZipDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Zip"] = "zip";
        TypeEnum["ZipPos"] = "zip_pos";
    })(TypeEnum = ZipDetails.TypeEnum || (ZipDetails.TypeEnum = {}));
})(ZipDetails = exports.ZipDetails || (exports.ZipDetails = {}));
