// @adyen/api-library/lib/src/typings/terminal/algorithmType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlgorithmType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var AlgorithmType;
(function (AlgorithmType) {
    AlgorithmType[AlgorithmType["DesEde3Cbc"] = 'Des-ede3-cbc'] = "DesEde3Cbc";
    AlgorithmType[AlgorithmType["DesEde3Ecb"] = 'Des-ede3-ecb'] = "DesEde3Ecb";
    AlgorithmType[AlgorithmType["IdDukptWrap"] = 'Id-dukpt-wrap'] = "IdDukptWrap";
    AlgorithmType[AlgorithmType["IdRetailCbcMac"] = 'Id-retail-cbc-MAC'] = "IdRetailCbcMac";
    AlgorithmType[AlgorithmType["IdRetailCbcMacSha256"] = 'Id-retail-cbc-MAC-sha-256'] = "IdRetailCbcMacSha256";
    AlgorithmType[AlgorithmType["IdSha256"] = 'Id-sha256'] = "IdSha256";
    AlgorithmType[AlgorithmType["IdUkptWrap"] = 'id-ukpt-wrap '] = "IdUkptWrap";
    AlgorithmType[AlgorithmType["RsaEncryption"] = 'RsaEncryption'] = "RsaEncryption";
    AlgorithmType[AlgorithmType["Sha256WithRsaEncryption"] = 'Sha256WithRSAEncryption'] = "Sha256WithRsaEncryption";
})(AlgorithmType = exports.AlgorithmType || (exports.AlgorithmType = {}));
