// @adyen/api-library/lib/src/typings/terminal/entryModeType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntryModeType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var EntryModeType;
(function (EntryModeType) {
    EntryModeType[EntryModeType["Contactless"] = 'Contactless'] = "Contactless";
    EntryModeType[EntryModeType["File"] = 'File'] = "File";
    EntryModeType[EntryModeType["Icc"] = 'ICC'] = "Icc";
    EntryModeType[EntryModeType["Keyed"] = 'Keyed'] = "Keyed";
    EntryModeType[EntryModeType["MagStripe"] = 'MagStripe'] = "MagStripe";
    EntryModeType[EntryModeType["Manual"] = 'Manual'] = "Manual";
    EntryModeType[EntryModeType["Mobile"] = 'Mobile'] = "Mobile";
    EntryModeType[EntryModeType["Rfid"] = 'RFID'] = "Rfid";
    EntryModeType[EntryModeType["Scanned"] = 'Scanned'] = "Scanned";
    EntryModeType[EntryModeType["SynchronousIcc"] = 'SynchronousICC'] = "SynchronousIcc";
    EntryModeType[EntryModeType["Tapped"] = 'Tapped'] = "Tapped";
})(EntryModeType = exports.EntryModeType || (exports.EntryModeType = {}));
