// @adyen/api-library/lib/src/typings/terminal/errorConditionType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorConditionType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var ErrorConditionType;
(function (ErrorConditionType) {
    ErrorConditionType[ErrorConditionType["Aborted"] = 'Aborted'] = "Aborted";
    ErrorConditionType[ErrorConditionType["Busy"] = 'Busy'] = "Busy";
    ErrorConditionType[ErrorConditionType["Cancel"] = 'Cancel'] = "Cancel";
    ErrorConditionType[ErrorConditionType["DeviceOut"] = 'DeviceOut'] = "DeviceOut";
    ErrorConditionType[ErrorConditionType["InProgress"] = 'InProgress'] = "InProgress";
    ErrorConditionType[ErrorConditionType["InsertedCard"] = 'InsertedCard'] = "InsertedCard";
    ErrorConditionType[ErrorConditionType["InvalidCard"] = 'InvalidCard'] = "InvalidCard";
    ErrorConditionType[ErrorConditionType["LoggedOut"] = 'LoggedOut'] = "LoggedOut";
    ErrorConditionType[ErrorConditionType["MessageFormat"] = 'MessageFormat'] = "MessageFormat";
    ErrorConditionType[ErrorConditionType["NotAllowed"] = 'NotAllowed'] = "NotAllowed";
    ErrorConditionType[ErrorConditionType["NotFound"] = 'NotFound'] = "NotFound";
    ErrorConditionType[ErrorConditionType["PaymentRestriction"] = 'PaymentRestriction'] = "PaymentRestriction";
    ErrorConditionType[ErrorConditionType["Refusal"] = 'Refusal'] = "Refusal";
    ErrorConditionType[ErrorConditionType["UnavailableDevice"] = 'UnavailableDevice'] = "UnavailableDevice";
    ErrorConditionType[ErrorConditionType["UnavailableService"] = 'UnavailableService'] = "UnavailableService";
    ErrorConditionType[ErrorConditionType["UnreachableHost"] = 'UnreachableHost'] = "UnreachableHost";
    ErrorConditionType[ErrorConditionType["WrongPin"] = 'WrongPIN'] = "WrongPin";
})(ErrorConditionType = exports.ErrorConditionType || (exports.ErrorConditionType = {}));
