// @adyen/api-library/lib/src/typings/terminal/loyaltyHandlingType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoyaltyHandlingType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var LoyaltyHandlingType;
(function (LoyaltyHandlingType) {
    LoyaltyHandlingType[LoyaltyHandlingType["Allowed"] = 'Allowed'] = "Allowed";
    LoyaltyHandlingType[LoyaltyHandlingType["Forbidden"] = 'Forbidden'] = "Forbidden";
    LoyaltyHandlingType[LoyaltyHandlingType["Processed"] = 'Processed'] = "Processed";
    LoyaltyHandlingType[LoyaltyHandlingType["Proposed"] = 'Proposed'] = "Proposed";
    LoyaltyHandlingType[LoyaltyHandlingType["Required"] = 'Required'] = "Required";
})(LoyaltyHandlingType = exports.LoyaltyHandlingType || (exports.LoyaltyHandlingType = {}));
