// @adyen/api-library/lib/src/typings/terminal/paymentType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var PaymentType;
(function (PaymentType) {
    PaymentType[PaymentType["CashAdvance"] = 'CashAdvance'] = "CashAdvance";
    PaymentType[PaymentType["CashDeposit"] = 'CashDeposit'] = "CashDeposit";
    PaymentType[PaymentType["Completion"] = 'Completion'] = "Completion";
    PaymentType[PaymentType["FirstReservation"] = 'FirstReservation'] = "FirstReservation";
    PaymentType[PaymentType["Instalment"] = 'Instalment'] = "Instalment";
    PaymentType[PaymentType["IssuerInstalment"] = 'IssuerInstalment'] = "IssuerInstalment";
    PaymentType[PaymentType["Normal"] = 'Normal'] = "Normal";
    PaymentType[PaymentType["OneTimeReservation"] = 'OneTimeReservation'] = "OneTimeReservation";
    PaymentType[PaymentType["PaidOut"] = 'PaidOut'] = "PaidOut";
    PaymentType[PaymentType["Recurring"] = 'Recurring'] = "Recurring";
    PaymentType[PaymentType["Refund"] = 'Refund'] = "Refund";
    PaymentType[PaymentType["UpdateReservation"] = 'UpdateReservation'] = "UpdateReservation";
})(PaymentType = exports.PaymentType || (exports.PaymentType = {}));
