// @adyen/api-library/lib/src/typings/terminalManagement/getTerminalDetailsResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTerminalDetailsResponse = void 0;
class GetTerminalDetailsResponse {
    static getAttributeTypeMap() {
        return GetTerminalDetailsResponse.attributeTypeMap;
    }
}
exports.GetTerminalDetailsResponse = GetTerminalDetailsResponse;
GetTerminalDetailsResponse.discriminator = undefined;
GetTerminalDetailsResponse.attributeTypeMap = [
    {
        "name": "bluetoothIp",
        "baseName": "bluetoothIp",
        "type": "string"
    },
    {
        "name": "bluetoothMac",
        "baseName": "bluetoothMac",
        "type": "string"
    },
    {
        "name": "companyAccount",
        "baseName": "companyAccount",
        "type": "string"
    },
    {
        "name": "country",
        "baseName": "country",
        "type": "string"
    },
    {
        "name": "deviceModel",
        "baseName": "deviceModel",
        "type": "string"
    },
    {
        "name": "dhcpEnabled",
        "baseName": "dhcpEnabled",
        "type": "boolean"
    },
    {
        "name": "displayLabel",
        "baseName": "displayLabel",
        "type": "string"
    },
    {
        "name": "ethernetIp",
        "baseName": "ethernetIp",
        "type": "string"
    },
    {
        "name": "ethernetMac",
        "baseName": "ethernetMac",
        "type": "string"
    },
    {
        "name": "firmwareVersion",
        "baseName": "firmwareVersion",
        "type": "string"
    },
    {
        "name": "iccid",
        "baseName": "iccid",
        "type": "string"
    },
    {
        "name": "lastActivityDateTime",
        "baseName": "lastActivityDateTime",
        "type": "Date"
    },
    {
        "name": "lastTransactionDateTime",
        "baseName": "lastTransactionDateTime",
        "type": "Date"
    },
    {
        "name": "linkNegotiation",
        "baseName": "linkNegotiation",
        "type": "string"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "merchantInventory",
        "baseName": "merchantInventory",
        "type": "boolean"
    },
    {
        "name": "permanentTerminalId",
        "baseName": "permanentTerminalId",
        "type": "string"
    },
    {
        "name": "serialNumber",
        "baseName": "serialNumber",
        "type": "string"
    },
    {
        "name": "simStatus",
        "baseName": "simStatus",
        "type": "string"
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string"
    },
    {
        "name": "storeDetails",
        "baseName": "storeDetails",
        "type": "Store"
    },
    {
        "name": "terminal",
        "baseName": "terminal",
        "type": "string"
    },
    {
        "name": "terminalStatus",
        "baseName": "terminalStatus",
        "type": "GetTerminalDetailsResponse.TerminalStatusEnum"
    },
    {
        "name": "wifiIp",
        "baseName": "wifiIp",
        "type": "string"
    },
    {
        "name": "wifiMac",
        "baseName": "wifiMac",
        "type": "string"
    }
];
(function (GetTerminalDetailsResponse) {
    let TerminalStatusEnum;
    (function (TerminalStatusEnum) {
        TerminalStatusEnum["OnlineLast1Day"] = "OnlineLast1Day";
        TerminalStatusEnum["OnlineLast2Days"] = "OnlineLast2Days";
        TerminalStatusEnum["OnlineLast3Days"] = "OnlineLast3Days";
        TerminalStatusEnum["OnlineLast4Days"] = "OnlineLast4Days";
        TerminalStatusEnum["OnlineLast5Days"] = "OnlineLast5Days";
        TerminalStatusEnum["OnlineLast6Days"] = "OnlineLast6Days";
        TerminalStatusEnum["OnlineLast7Days"] = "OnlineLast7Days";
        TerminalStatusEnum["OnlineToday"] = "OnlineToday";
        TerminalStatusEnum["ReAssignToInventoryPending"] = "ReAssignToInventoryPending";
        TerminalStatusEnum["ReAssignToMerchantInventoryPending"] = "ReAssignToMerchantInventoryPending";
        TerminalStatusEnum["ReAssignToStorePending"] = "ReAssignToStorePending";
        TerminalStatusEnum["SwitchedOff"] = "SwitchedOff";
    })(TerminalStatusEnum = GetTerminalDetailsResponse.TerminalStatusEnum || (GetTerminalDetailsResponse.TerminalStatusEnum = {}));
})(GetTerminalDetailsResponse = exports.GetTerminalDetailsResponse || (exports.GetTerminalDetailsResponse = {}));
