// @adyen/api-library/lib/src/typings/balancePlatform/accountHolderUpdateRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountHolderUpdateRequest = void 0;
class AccountHolderUpdateRequest {
    static getAttributeTypeMap() {
        return AccountHolderUpdateRequest.attributeTypeMap;
    }
}
exports.AccountHolderUpdateRequest = AccountHolderUpdateRequest;
AccountHolderUpdateRequest.discriminator = undefined;
AccountHolderUpdateRequest.attributeTypeMap = [
    {
        "name": "balancePlatform",
        "baseName": "balancePlatform",
        "type": "string"
    },
    {
        "name": "capabilities",
        "baseName": "capabilities",
        "type": "{ [key: string]: AccountHolderCapability; }"
    },
    {
        "name": "contactDetails",
        "baseName": "contactDetails",
        "type": "ContactDetails"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "migratedAccountHolderCode",
        "baseName": "migratedAccountHolderCode",
        "type": "string"
    },
    {
        "name": "primaryBalanceAccount",
        "baseName": "primaryBalanceAccount",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "AccountHolderUpdateRequest.StatusEnum"
    },
    {
        "name": "timeZone",
        "baseName": "timeZone",
        "type": "string"
    },
    {
        "name": "verificationDeadlines",
        "baseName": "verificationDeadlines",
        "type": "Array<VerificationDeadline>"
    }
];
(function (AccountHolderUpdateRequest) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Closed"] = "closed";
        StatusEnum["Inactive"] = "inactive";
        StatusEnum["Suspended"] = "suspended";
    })(StatusEnum = AccountHolderUpdateRequest.StatusEnum || (AccountHolderUpdateRequest.StatusEnum = {}));
})(AccountHolderUpdateRequest = exports.AccountHolderUpdateRequest || (exports.AccountHolderUpdateRequest = {}));
