// @adyen/api-library/lib/src/typings/balancePlatform/balanceAccount.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceAccount = void 0;
class BalanceAccount {
    static getAttributeTypeMap() {
        return BalanceAccount.attributeTypeMap;
    }
}
exports.BalanceAccount = BalanceAccount;
BalanceAccount.discriminator = undefined;
BalanceAccount.attributeTypeMap = [
    {
        "name": "accountHolderId",
        "baseName": "accountHolderId",
        "type": "string"
    },
    {
        "name": "balances",
        "baseName": "balances",
        "type": "Array<Balance>"
    },
    {
        "name": "defaultCurrencyCode",
        "baseName": "defaultCurrencyCode",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "migratedAccountCode",
        "baseName": "migratedAccountCode",
        "type": "string"
    },
    {
        "name": "platformPaymentConfiguration",
        "baseName": "platformPaymentConfiguration",
        "type": "PlatformPaymentConfiguration"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BalanceAccount.StatusEnum"
    },
    {
        "name": "timeZone",
        "baseName": "timeZone",
        "type": "string"
    }
];
(function (BalanceAccount) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Closed"] = "closed";
        StatusEnum["Inactive"] = "inactive";
        StatusEnum["Suspended"] = "suspended";
    })(StatusEnum = BalanceAccount.StatusEnum || (BalanceAccount.StatusEnum = {}));
})(BalanceAccount = exports.BalanceAccount || (exports.BalanceAccount = {}));
