// @adyen/api-library/lib/src/typings/balancePlatform/cALocalAccountIdentification.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CALocalAccountIdentification = void 0;
class CALocalAccountIdentification {
    static getAttributeTypeMap() {
        return CALocalAccountIdentification.attributeTypeMap;
    }
}
exports.CALocalAccountIdentification = CALocalAccountIdentification;
CALocalAccountIdentification.discriminator = undefined;
CALocalAccountIdentification.attributeTypeMap = [
    {
        "name": "accountNumber",
        "baseName": "accountNumber",
        "type": "string"
    },
    {
        "name": "accountType",
        "baseName": "accountType",
        "type": "CALocalAccountIdentification.AccountTypeEnum"
    },
    {
        "name": "institutionNumber",
        "baseName": "institutionNumber",
        "type": "string"
    },
    {
        "name": "transitNumber",
        "baseName": "transitNumber",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "CALocalAccountIdentification.TypeEnum"
    }
];
(function (CALocalAccountIdentification) {
    let AccountTypeEnum;
    (function (AccountTypeEnum) {
        AccountTypeEnum["Checking"] = "checking";
        AccountTypeEnum["Savings"] = "savings";
    })(AccountTypeEnum = CALocalAccountIdentification.AccountTypeEnum || (CALocalAccountIdentification.AccountTypeEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["CaLocal"] = "caLocal";
    })(TypeEnum = CALocalAccountIdentification.TypeEnum || (CALocalAccountIdentification.TypeEnum = {}));
})(CALocalAccountIdentification = exports.CALocalAccountIdentification || (exports.CALocalAccountIdentification = {}));
