// @adyen/api-library/lib/src/typings/balancePlatform/entryModesRestriction.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntryModesRestriction = void 0;
class EntryModesRestriction {
    static getAttributeTypeMap() {
        return EntryModesRestriction.attributeTypeMap;
    }
}
exports.EntryModesRestriction = EntryModesRestriction;
EntryModesRestriction.discriminator = undefined;
EntryModesRestriction.attributeTypeMap = [
    {
        "name": "operation",
        "baseName": "operation",
        "type": "string"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "Array<EntryModesRestriction.ValueEnum>"
    }
];
(function (EntryModesRestriction) {
    let ValueEnum;
    (function (ValueEnum) {
        ValueEnum["Barcode"] = "barcode";
        ValueEnum["Chip"] = "chip";
        ValueEnum["Cof"] = "cof";
        ValueEnum["Contactless"] = "contactless";
        ValueEnum["Magstripe"] = "magstripe";
        ValueEnum["Manual"] = "manual";
        ValueEnum["Ocr"] = "ocr";
        ValueEnum["Server"] = "server";
        ValueEnum["Unknown"] = "unknown";
    })(ValueEnum = EntryModesRestriction.ValueEnum || (EntryModesRestriction.ValueEnum = {}));
})(EntryModesRestriction = exports.EntryModesRestriction || (exports.EntryModesRestriction = {}));
