// @adyen/api-library/lib/src/typings/balancePlatform/models.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectSerializer = void 0;
__exportStar(require("./aULocalAccountIdentification"), exports);
__exportStar(require("./accountHolder"), exports);
__exportStar(require("./accountHolderCapability"), exports);
__exportStar(require("./accountHolderInfo"), exports);
__exportStar(require("./accountHolderUpdateRequest"), exports);
__exportStar(require("./accountSupportingEntityCapability"), exports);
__exportStar(require("./activeNetworkTokensRestriction"), exports);
__exportStar(require("./additionalBankIdentification"), exports);
__exportStar(require("./address"), exports);
__exportStar(require("./addressRequirement"), exports);
__exportStar(require("./amount"), exports);
__exportStar(require("./amountMinMaxRequirement"), exports);
__exportStar(require("./authentication"), exports);
__exportStar(require("./bRLocalAccountIdentification"), exports);
__exportStar(require("./balance"), exports);
__exportStar(require("./balanceAccount"), exports);
__exportStar(require("./balanceAccountBase"), exports);
__exportStar(require("./balanceAccountInfo"), exports);
__exportStar(require("./balanceAccountUpdateRequest"), exports);
__exportStar(require("./balancePlatform"), exports);
__exportStar(require("./balanceSweepConfigurationsResponse"), exports);
__exportStar(require("./bankAccount"), exports);
__exportStar(require("./bankAccountIdentificationTypeRequirement"), exports);
__exportStar(require("./bankAccountIdentificationValidationRequest"), exports);
__exportStar(require("./bankIdentification"), exports);
__exportStar(require("./brandVariantsRestriction"), exports);
__exportStar(require("./bulkAddress"), exports);
__exportStar(require("./cALocalAccountIdentification"), exports);
__exportStar(require("./cZLocalAccountIdentification"), exports);
__exportStar(require("./capabilityProblem"), exports);
__exportStar(require("./capabilityProblemEntity"), exports);
__exportStar(require("./capabilityProblemEntityRecursive"), exports);
__exportStar(require("./capabilitySettings"), exports);
__exportStar(require("./capitalBalance"), exports);
__exportStar(require("./capitalGrantAccount"), exports);
__exportStar(require("./card"), exports);
__exportStar(require("./cardConfiguration"), exports);
__exportStar(require("./cardInfo"), exports);
__exportStar(require("./contactDetails"), exports);
__exportStar(require("./counterparty"), exports);
__exportStar(require("./counterpartyBankRestriction"), exports);
__exportStar(require("./countriesRestriction"), exports);
__exportStar(require("./createSweepConfigurationV2"), exports);
__exportStar(require("./dKLocalAccountIdentification"), exports);
__exportStar(require("./dayOfWeekRestriction"), exports);
__exportStar(require("./deliveryAddress"), exports);
__exportStar(require("./deliveryContact"), exports);
__exportStar(require("./deviceInfo"), exports);
__exportStar(require("./differentCurrenciesRestriction"), exports);
__exportStar(require("./duration"), exports);
__exportStar(require("./entryModesRestriction"), exports);
__exportStar(require("./expiry"), exports);
__exportStar(require("./fee"), exports);
__exportStar(require("./getNetworkTokenResponse"), exports);
__exportStar(require("./getTaxFormResponse"), exports);
__exportStar(require("./grantLimit"), exports);
__exportStar(require("./grantOffer"), exports);
__exportStar(require("./grantOffers"), exports);
__exportStar(require("./hKLocalAccountIdentification"), exports);
__exportStar(require("./hULocalAccountIdentification"), exports);
__exportStar(require("./ibanAccountIdentification"), exports);
__exportStar(require("./internationalTransactionRestriction"), exports);
__exportStar(require("./invalidField"), exports);
__exportStar(require("./jSONObject"), exports);
__exportStar(require("./jSONPath"), exports);
__exportStar(require("./listNetworkTokensResponse"), exports);
__exportStar(require("./matchingTransactionsRestriction"), exports);
__exportStar(require("./mccsRestriction"), exports);
__exportStar(require("./merchantAcquirerPair"), exports);
__exportStar(require("./merchantNamesRestriction"), exports);
__exportStar(require("./merchantsRestriction"), exports);
__exportStar(require("./nOLocalAccountIdentification"), exports);
__exportStar(require("./nZLocalAccountIdentification"), exports);
__exportStar(require("./name"), exports);
__exportStar(require("./networkToken"), exports);
__exportStar(require("./numberAndBicAccountIdentification"), exports);
__exportStar(require("./pLLocalAccountIdentification"), exports);
__exportStar(require("./paginatedAccountHoldersResponse"), exports);
__exportStar(require("./paginatedBalanceAccountsResponse"), exports);
__exportStar(require("./paginatedPaymentInstrumentsResponse"), exports);
__exportStar(require("./paymentInstrument"), exports);
__exportStar(require("./paymentInstrumentGroup"), exports);
__exportStar(require("./paymentInstrumentGroupInfo"), exports);
__exportStar(require("./paymentInstrumentInfo"), exports);
__exportStar(require("./paymentInstrumentRequirement"), exports);
__exportStar(require("./paymentInstrumentRevealInfo"), exports);
__exportStar(require("./paymentInstrumentUpdateRequest"), exports);
__exportStar(require("./phone"), exports);
__exportStar(require("./phoneNumber"), exports);
__exportStar(require("./platformPaymentConfiguration"), exports);
__exportStar(require("./processingTypesRestriction"), exports);
__exportStar(require("./remediatingAction"), exports);
__exportStar(require("./repayment"), exports);
__exportStar(require("./repaymentTerm"), exports);
__exportStar(require("./restServiceError"), exports);
__exportStar(require("./sELocalAccountIdentification"), exports);
__exportStar(require("./sGLocalAccountIdentification"), exports);
__exportStar(require("./stringMatch"), exports);
__exportStar(require("./sweepConfigurationV2"), exports);
__exportStar(require("./sweepCounterparty"), exports);
__exportStar(require("./sweepSchedule"), exports);
__exportStar(require("./thresholdRepayment"), exports);
__exportStar(require("./timeOfDay"), exports);
__exportStar(require("./timeOfDayRestriction"), exports);
__exportStar(require("./totalAmountRestriction"), exports);
__exportStar(require("./transactionRule"), exports);
__exportStar(require("./transactionRuleEntityKey"), exports);
__exportStar(require("./transactionRuleInfo"), exports);
__exportStar(require("./transactionRuleInterval"), exports);
__exportStar(require("./transactionRuleResponse"), exports);
__exportStar(require("./transactionRuleRestrictions"), exports);
__exportStar(require("./transactionRulesResponse"), exports);
__exportStar(require("./transferRoute"), exports);
__exportStar(require("./transferRouteRequest"), exports);
__exportStar(require("./transferRouteResponse"), exports);
__exportStar(require("./uKLocalAccountIdentification"), exports);
__exportStar(require("./uSLocalAccountIdentification"), exports);
__exportStar(require("./updateNetworkTokenRequest"), exports);
__exportStar(require("./updatePaymentInstrument"), exports);
__exportStar(require("./updateSweepConfigurationV2"), exports);
__exportStar(require("./verificationDeadline"), exports);
__exportStar(require("./verificationError"), exports);
__exportStar(require("./verificationErrorRecursive"), exports);
const aULocalAccountIdentification_1 = require("./aULocalAccountIdentification");
const accountHolder_1 = require("./accountHolder");
const accountHolderCapability_1 = require("./accountHolderCapability");
const accountHolderInfo_1 = require("./accountHolderInfo");
const accountHolderUpdateRequest_1 = require("./accountHolderUpdateRequest");
const accountSupportingEntityCapability_1 = require("./accountSupportingEntityCapability");
const activeNetworkTokensRestriction_1 = require("./activeNetworkTokensRestriction");
const additionalBankIdentification_1 = require("./additionalBankIdentification");
const address_1 = require("./address");
const addressRequirement_1 = require("./addressRequirement");
const amount_1 = require("./amount");
const amountMinMaxRequirement_1 = require("./amountMinMaxRequirement");
const authentication_1 = require("./authentication");
const bRLocalAccountIdentification_1 = require("./bRLocalAccountIdentification");
const balance_1 = require("./balance");
const balanceAccount_1 = require("./balanceAccount");
const balanceAccountBase_1 = require("./balanceAccountBase");
const balanceAccountInfo_1 = require("./balanceAccountInfo");
const balanceAccountUpdateRequest_1 = require("./balanceAccountUpdateRequest");
const balancePlatform_1 = require("./balancePlatform");
const balanceSweepConfigurationsResponse_1 = require("./balanceSweepConfigurationsResponse");
const bankAccount_1 = require("./bankAccount");
const bankAccountIdentificationTypeRequirement_1 = require("./bankAccountIdentificationTypeRequirement");
const bankAccountIdentificationValidationRequest_1 = require("./bankAccountIdentificationValidationRequest");
const bankIdentification_1 = require("./bankIdentification");
const brandVariantsRestriction_1 = require("./brandVariantsRestriction");
const bulkAddress_1 = require("./bulkAddress");
const cALocalAccountIdentification_1 = require("./cALocalAccountIdentification");
const cZLocalAccountIdentification_1 = require("./cZLocalAccountIdentification");
const capabilityProblem_1 = require("./capabilityProblem");
const capabilityProblemEntity_1 = require("./capabilityProblemEntity");
const capabilityProblemEntityRecursive_1 = require("./capabilityProblemEntityRecursive");
const capabilitySettings_1 = require("./capabilitySettings");
const capitalBalance_1 = require("./capitalBalance");
const capitalGrantAccount_1 = require("./capitalGrantAccount");
const card_1 = require("./card");
const cardConfiguration_1 = require("./cardConfiguration");
const cardInfo_1 = require("./cardInfo");
const contactDetails_1 = require("./contactDetails");
const counterparty_1 = require("./counterparty");
const counterpartyBankRestriction_1 = require("./counterpartyBankRestriction");
const countriesRestriction_1 = require("./countriesRestriction");
const createSweepConfigurationV2_1 = require("./createSweepConfigurationV2");
const dKLocalAccountIdentification_1 = require("./dKLocalAccountIdentification");
const dayOfWeekRestriction_1 = require("./dayOfWeekRestriction");
const deliveryAddress_1 = require("./deliveryAddress");
const deliveryContact_1 = require("./deliveryContact");
const deviceInfo_1 = require("./deviceInfo");
const differentCurrenciesRestriction_1 = require("./differentCurrenciesRestriction");
const duration_1 = require("./duration");
const entryModesRestriction_1 = require("./entryModesRestriction");
const expiry_1 = require("./expiry");
const fee_1 = require("./fee");
const getNetworkTokenResponse_1 = require("./getNetworkTokenResponse");
const getTaxFormResponse_1 = require("./getTaxFormResponse");
const grantLimit_1 = require("./grantLimit");
const grantOffer_1 = require("./grantOffer");
const grantOffers_1 = require("./grantOffers");
const hKLocalAccountIdentification_1 = require("./hKLocalAccountIdentification");
const hULocalAccountIdentification_1 = require("./hULocalAccountIdentification");
const ibanAccountIdentification_1 = require("./ibanAccountIdentification");
const internationalTransactionRestriction_1 = require("./internationalTransactionRestriction");
const invalidField_1 = require("./invalidField");
const jSONObject_1 = require("./jSONObject");
const jSONPath_1 = require("./jSONPath");
const listNetworkTokensResponse_1 = require("./listNetworkTokensResponse");
const matchingTransactionsRestriction_1 = require("./matchingTransactionsRestriction");
const mccsRestriction_1 = require("./mccsRestriction");
const merchantAcquirerPair_1 = require("./merchantAcquirerPair");
const merchantNamesRestriction_1 = require("./merchantNamesRestriction");
const merchantsRestriction_1 = require("./merchantsRestriction");
const nOLocalAccountIdentification_1 = require("./nOLocalAccountIdentification");
const nZLocalAccountIdentification_1 = require("./nZLocalAccountIdentification");
const name_1 = require("./name");
const networkToken_1 = require("./networkToken");
const numberAndBicAccountIdentification_1 = require("./numberAndBicAccountIdentification");
const pLLocalAccountIdentification_1 = require("./pLLocalAccountIdentification");
const paginatedAccountHoldersResponse_1 = require("./paginatedAccountHoldersResponse");
const paginatedBalanceAccountsResponse_1 = require("./paginatedBalanceAccountsResponse");
const paginatedPaymentInstrumentsResponse_1 = require("./paginatedPaymentInstrumentsResponse");
const paymentInstrument_1 = require("./paymentInstrument");
const paymentInstrumentGroup_1 = require("./paymentInstrumentGroup");
const paymentInstrumentGroupInfo_1 = require("./paymentInstrumentGroupInfo");
const paymentInstrumentInfo_1 = require("./paymentInstrumentInfo");
const paymentInstrumentRequirement_1 = require("./paymentInstrumentRequirement");
const paymentInstrumentRevealInfo_1 = require("./paymentInstrumentRevealInfo");
const paymentInstrumentUpdateRequest_1 = require("./paymentInstrumentUpdateRequest");
const phone_1 = require("./phone");
const phoneNumber_1 = require("./phoneNumber");
const platformPaymentConfiguration_1 = require("./platformPaymentConfiguration");
const processingTypesRestriction_1 = require("./processingTypesRestriction");
const remediatingAction_1 = require("./remediatingAction");
const repayment_1 = require("./repayment");
const repaymentTerm_1 = require("./repaymentTerm");
const restServiceError_1 = require("./restServiceError");
const sELocalAccountIdentification_1 = require("./sELocalAccountIdentification");
const sGLocalAccountIdentification_1 = require("./sGLocalAccountIdentification");
const stringMatch_1 = require("./stringMatch");
const sweepConfigurationV2_1 = require("./sweepConfigurationV2");
const sweepCounterparty_1 = require("./sweepCounterparty");
const sweepSchedule_1 = require("./sweepSchedule");
const thresholdRepayment_1 = require("./thresholdRepayment");
const timeOfDay_1 = require("./timeOfDay");
const timeOfDayRestriction_1 = require("./timeOfDayRestriction");
const totalAmountRestriction_1 = require("./totalAmountRestriction");
const transactionRule_1 = require("./transactionRule");
const transactionRuleEntityKey_1 = require("./transactionRuleEntityKey");
const transactionRuleInfo_1 = require("./transactionRuleInfo");
const transactionRuleInterval_1 = require("./transactionRuleInterval");
const transactionRuleResponse_1 = require("./transactionRuleResponse");
const transactionRuleRestrictions_1 = require("./transactionRuleRestrictions");
const transactionRulesResponse_1 = require("./transactionRulesResponse");
const transferRoute_1 = require("./transferRoute");
const transferRouteRequest_1 = require("./transferRouteRequest");
const transferRouteResponse_1 = require("./transferRouteResponse");
const uKLocalAccountIdentification_1 = require("./uKLocalAccountIdentification");
const uSLocalAccountIdentification_1 = require("./uSLocalAccountIdentification");
const updateNetworkTokenRequest_1 = require("./updateNetworkTokenRequest");
const updatePaymentInstrument_1 = require("./updatePaymentInstrument");
const updateSweepConfigurationV2_1 = require("./updateSweepConfigurationV2");
const verificationDeadline_1 = require("./verificationDeadline");
const verificationError_1 = require("./verificationError");
const verificationErrorRecursive_1 = require("./verificationErrorRecursive");
/* tslint:disable:no-unused-variable */
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
let enumsMap = {
    "AULocalAccountIdentification.TypeEnum": aULocalAccountIdentification_1.AULocalAccountIdentification.TypeEnum,
    "AccountHolder.StatusEnum": accountHolder_1.AccountHolder.StatusEnum,
    "AccountHolderCapability.AllowedLevelEnum": accountHolderCapability_1.AccountHolderCapability.AllowedLevelEnum,
    "AccountHolderCapability.RequestedLevelEnum": accountHolderCapability_1.AccountHolderCapability.RequestedLevelEnum,
    "AccountHolderCapability.VerificationStatusEnum": accountHolderCapability_1.AccountHolderCapability.VerificationStatusEnum,
    "AccountHolderUpdateRequest.StatusEnum": accountHolderUpdateRequest_1.AccountHolderUpdateRequest.StatusEnum,
    "AccountSupportingEntityCapability.AllowedLevelEnum": accountSupportingEntityCapability_1.AccountSupportingEntityCapability.AllowedLevelEnum,
    "AccountSupportingEntityCapability.RequestedLevelEnum": accountSupportingEntityCapability_1.AccountSupportingEntityCapability.RequestedLevelEnum,
    "AccountSupportingEntityCapability.VerificationStatusEnum": accountSupportingEntityCapability_1.AccountSupportingEntityCapability.VerificationStatusEnum,
    "AdditionalBankIdentification.TypeEnum": additionalBankIdentification_1.AdditionalBankIdentification.TypeEnum,
    "AddressRequirement.TypeEnum": addressRequirement_1.AddressRequirement.TypeEnum,
    "AmountMinMaxRequirement.TypeEnum": amountMinMaxRequirement_1.AmountMinMaxRequirement.TypeEnum,
    "BRLocalAccountIdentification.TypeEnum": bRLocalAccountIdentification_1.BRLocalAccountIdentification.TypeEnum,
    "BalanceAccount.StatusEnum": balanceAccount_1.BalanceAccount.StatusEnum,
    "BalanceAccountBase.StatusEnum": balanceAccountBase_1.BalanceAccountBase.StatusEnum,
    "BalanceAccountUpdateRequest.StatusEnum": balanceAccountUpdateRequest_1.BalanceAccountUpdateRequest.StatusEnum,
    "BankAccountIdentificationTypeRequirement.BankAccountIdentificationTypesEnum": bankAccountIdentificationTypeRequirement_1.BankAccountIdentificationTypeRequirement.BankAccountIdentificationTypesEnum,
    "BankAccountIdentificationTypeRequirement.TypeEnum": bankAccountIdentificationTypeRequirement_1.BankAccountIdentificationTypeRequirement.TypeEnum,
    "BankIdentification.IdentificationTypeEnum": bankIdentification_1.BankIdentification.IdentificationTypeEnum,
    "CALocalAccountIdentification.AccountTypeEnum": cALocalAccountIdentification_1.CALocalAccountIdentification.AccountTypeEnum,
    "CALocalAccountIdentification.TypeEnum": cALocalAccountIdentification_1.CALocalAccountIdentification.TypeEnum,
    "CZLocalAccountIdentification.TypeEnum": cZLocalAccountIdentification_1.CZLocalAccountIdentification.TypeEnum,
    "CapabilityProblemEntity.TypeEnum": capabilityProblemEntity_1.CapabilityProblemEntity.TypeEnum,
    "CapabilityProblemEntityRecursive.TypeEnum": capabilityProblemEntityRecursive_1.CapabilityProblemEntityRecursive.TypeEnum,
    "CapabilitySettings.FundingSourceEnum": capabilitySettings_1.CapabilitySettings.FundingSourceEnum,
    "CapabilitySettings.IntervalEnum": capabilitySettings_1.CapabilitySettings.IntervalEnum,
    "Card.FormFactorEnum": card_1.Card.FormFactorEnum,
    "CardInfo.FormFactorEnum": cardInfo_1.CardInfo.FormFactorEnum,
    "CreateSweepConfigurationV2.CategoryEnum": createSweepConfigurationV2_1.CreateSweepConfigurationV2.CategoryEnum,
    "CreateSweepConfigurationV2.PrioritiesEnum": createSweepConfigurationV2_1.CreateSweepConfigurationV2.PrioritiesEnum,
    "CreateSweepConfigurationV2.ReasonEnum": createSweepConfigurationV2_1.CreateSweepConfigurationV2.ReasonEnum,
    "CreateSweepConfigurationV2.StatusEnum": createSweepConfigurationV2_1.CreateSweepConfigurationV2.StatusEnum,
    "CreateSweepConfigurationV2.TypeEnum": createSweepConfigurationV2_1.CreateSweepConfigurationV2.TypeEnum,
    "DKLocalAccountIdentification.TypeEnum": dKLocalAccountIdentification_1.DKLocalAccountIdentification.TypeEnum,
    "DayOfWeekRestriction.ValueEnum": dayOfWeekRestriction_1.DayOfWeekRestriction.ValueEnum,
    "Duration.UnitEnum": duration_1.Duration.UnitEnum,
    "EntryModesRestriction.ValueEnum": entryModesRestriction_1.EntryModesRestriction.ValueEnum,
    "GetTaxFormResponse.ContentTypeEnum": getTaxFormResponse_1.GetTaxFormResponse.ContentTypeEnum,
    "GrantOffer.ContractTypeEnum": grantOffer_1.GrantOffer.ContractTypeEnum,
    "HKLocalAccountIdentification.TypeEnum": hKLocalAccountIdentification_1.HKLocalAccountIdentification.TypeEnum,
    "HULocalAccountIdentification.TypeEnum": hULocalAccountIdentification_1.HULocalAccountIdentification.TypeEnum,
    "IbanAccountIdentification.TypeEnum": ibanAccountIdentification_1.IbanAccountIdentification.TypeEnum,
    "NOLocalAccountIdentification.TypeEnum": nOLocalAccountIdentification_1.NOLocalAccountIdentification.TypeEnum,
    "NZLocalAccountIdentification.TypeEnum": nZLocalAccountIdentification_1.NZLocalAccountIdentification.TypeEnum,
    "NetworkToken.StatusEnum": networkToken_1.NetworkToken.StatusEnum,
    "NumberAndBicAccountIdentification.TypeEnum": numberAndBicAccountIdentification_1.NumberAndBicAccountIdentification.TypeEnum,
    "PLLocalAccountIdentification.TypeEnum": pLLocalAccountIdentification_1.PLLocalAccountIdentification.TypeEnum,
    "PaymentInstrument.StatusEnum": paymentInstrument_1.PaymentInstrument.StatusEnum,
    "PaymentInstrument.StatusReasonEnum": paymentInstrument_1.PaymentInstrument.StatusReasonEnum,
    "PaymentInstrument.TypeEnum": paymentInstrument_1.PaymentInstrument.TypeEnum,
    "PaymentInstrumentInfo.StatusEnum": paymentInstrumentInfo_1.PaymentInstrumentInfo.StatusEnum,
    "PaymentInstrumentInfo.StatusReasonEnum": paymentInstrumentInfo_1.PaymentInstrumentInfo.StatusReasonEnum,
    "PaymentInstrumentInfo.TypeEnum": paymentInstrumentInfo_1.PaymentInstrumentInfo.TypeEnum,
    "PaymentInstrumentRequirement.PaymentInstrumentTypeEnum": paymentInstrumentRequirement_1.PaymentInstrumentRequirement.PaymentInstrumentTypeEnum,
    "PaymentInstrumentRequirement.TypeEnum": paymentInstrumentRequirement_1.PaymentInstrumentRequirement.TypeEnum,
    "PaymentInstrumentUpdateRequest.StatusEnum": paymentInstrumentUpdateRequest_1.PaymentInstrumentUpdateRequest.StatusEnum,
    "PaymentInstrumentUpdateRequest.StatusReasonEnum": paymentInstrumentUpdateRequest_1.PaymentInstrumentUpdateRequest.StatusReasonEnum,
    "Phone.TypeEnum": phone_1.Phone.TypeEnum,
    "PhoneNumber.PhoneTypeEnum": phoneNumber_1.PhoneNumber.PhoneTypeEnum,
    "ProcessingTypesRestriction.ValueEnum": processingTypesRestriction_1.ProcessingTypesRestriction.ValueEnum,
    "SELocalAccountIdentification.TypeEnum": sELocalAccountIdentification_1.SELocalAccountIdentification.TypeEnum,
    "SGLocalAccountIdentification.TypeEnum": sGLocalAccountIdentification_1.SGLocalAccountIdentification.TypeEnum,
    "StringMatch.OperationEnum": stringMatch_1.StringMatch.OperationEnum,
    "SweepConfigurationV2.CategoryEnum": sweepConfigurationV2_1.SweepConfigurationV2.CategoryEnum,
    "SweepConfigurationV2.PrioritiesEnum": sweepConfigurationV2_1.SweepConfigurationV2.PrioritiesEnum,
    "SweepConfigurationV2.ReasonEnum": sweepConfigurationV2_1.SweepConfigurationV2.ReasonEnum,
    "SweepConfigurationV2.StatusEnum": sweepConfigurationV2_1.SweepConfigurationV2.StatusEnum,
    "SweepConfigurationV2.TypeEnum": sweepConfigurationV2_1.SweepConfigurationV2.TypeEnum,
    "SweepSchedule.TypeEnum": sweepSchedule_1.SweepSchedule.TypeEnum,
    "TransactionRule.OutcomeTypeEnum": transactionRule_1.TransactionRule.OutcomeTypeEnum,
    "TransactionRule.RequestTypeEnum": transactionRule_1.TransactionRule.RequestTypeEnum,
    "TransactionRule.StatusEnum": transactionRule_1.TransactionRule.StatusEnum,
    "TransactionRule.TypeEnum": transactionRule_1.TransactionRule.TypeEnum,
    "TransactionRuleInfo.OutcomeTypeEnum": transactionRuleInfo_1.TransactionRuleInfo.OutcomeTypeEnum,
    "TransactionRuleInfo.RequestTypeEnum": transactionRuleInfo_1.TransactionRuleInfo.RequestTypeEnum,
    "TransactionRuleInfo.StatusEnum": transactionRuleInfo_1.TransactionRuleInfo.StatusEnum,
    "TransactionRuleInfo.TypeEnum": transactionRuleInfo_1.TransactionRuleInfo.TypeEnum,
    "TransactionRuleInterval.DayOfWeekEnum": transactionRuleInterval_1.TransactionRuleInterval.DayOfWeekEnum,
    "TransactionRuleInterval.TypeEnum": transactionRuleInterval_1.TransactionRuleInterval.TypeEnum,
    "TransferRoute.CategoryEnum": transferRoute_1.TransferRoute.CategoryEnum,
    "TransferRoute.PriorityEnum": transferRoute_1.TransferRoute.PriorityEnum,
    "TransferRouteRequest.CategoryEnum": transferRouteRequest_1.TransferRouteRequest.CategoryEnum,
    "TransferRouteRequest.PrioritiesEnum": transferRouteRequest_1.TransferRouteRequest.PrioritiesEnum,
    "UKLocalAccountIdentification.TypeEnum": uKLocalAccountIdentification_1.UKLocalAccountIdentification.TypeEnum,
    "USLocalAccountIdentification.AccountTypeEnum": uSLocalAccountIdentification_1.USLocalAccountIdentification.AccountTypeEnum,
    "USLocalAccountIdentification.TypeEnum": uSLocalAccountIdentification_1.USLocalAccountIdentification.TypeEnum,
    "UpdateNetworkTokenRequest.StatusEnum": updateNetworkTokenRequest_1.UpdateNetworkTokenRequest.StatusEnum,
    "UpdatePaymentInstrument.StatusEnum": updatePaymentInstrument_1.UpdatePaymentInstrument.StatusEnum,
    "UpdatePaymentInstrument.StatusReasonEnum": updatePaymentInstrument_1.UpdatePaymentInstrument.StatusReasonEnum,
    "UpdatePaymentInstrument.TypeEnum": updatePaymentInstrument_1.UpdatePaymentInstrument.TypeEnum,
    "UpdateSweepConfigurationV2.CategoryEnum": updateSweepConfigurationV2_1.UpdateSweepConfigurationV2.CategoryEnum,
    "UpdateSweepConfigurationV2.PrioritiesEnum": updateSweepConfigurationV2_1.UpdateSweepConfigurationV2.PrioritiesEnum,
    "UpdateSweepConfigurationV2.ReasonEnum": updateSweepConfigurationV2_1.UpdateSweepConfigurationV2.ReasonEnum,
    "UpdateSweepConfigurationV2.StatusEnum": updateSweepConfigurationV2_1.UpdateSweepConfigurationV2.StatusEnum,
    "UpdateSweepConfigurationV2.TypeEnum": updateSweepConfigurationV2_1.UpdateSweepConfigurationV2.TypeEnum,
    "VerificationDeadline.CapabilitiesEnum": verificationDeadline_1.VerificationDeadline.CapabilitiesEnum,
    "VerificationError.CapabilitiesEnum": verificationError_1.VerificationError.CapabilitiesEnum,
    "VerificationError.TypeEnum": verificationError_1.VerificationError.TypeEnum,
    "VerificationErrorRecursive.CapabilitiesEnum": verificationErrorRecursive_1.VerificationErrorRecursive.CapabilitiesEnum,
    "VerificationErrorRecursive.TypeEnum": verificationErrorRecursive_1.VerificationErrorRecursive.TypeEnum,
};
let typeMap = {
    "AULocalAccountIdentification": aULocalAccountIdentification_1.AULocalAccountIdentification,
    "AccountHolder": accountHolder_1.AccountHolder,
    "AccountHolderCapability": accountHolderCapability_1.AccountHolderCapability,
    "AccountHolderInfo": accountHolderInfo_1.AccountHolderInfo,
    "AccountHolderUpdateRequest": accountHolderUpdateRequest_1.AccountHolderUpdateRequest,
    "AccountSupportingEntityCapability": accountSupportingEntityCapability_1.AccountSupportingEntityCapability,
    "ActiveNetworkTokensRestriction": activeNetworkTokensRestriction_1.ActiveNetworkTokensRestriction,
    "AdditionalBankIdentification": additionalBankIdentification_1.AdditionalBankIdentification,
    "Address": address_1.Address,
    "AddressRequirement": addressRequirement_1.AddressRequirement,
    "Amount": amount_1.Amount,
    "AmountMinMaxRequirement": amountMinMaxRequirement_1.AmountMinMaxRequirement,
    "Authentication": authentication_1.Authentication,
    "BRLocalAccountIdentification": bRLocalAccountIdentification_1.BRLocalAccountIdentification,
    "Balance": balance_1.Balance,
    "BalanceAccount": balanceAccount_1.BalanceAccount,
    "BalanceAccountBase": balanceAccountBase_1.BalanceAccountBase,
    "BalanceAccountInfo": balanceAccountInfo_1.BalanceAccountInfo,
    "BalanceAccountUpdateRequest": balanceAccountUpdateRequest_1.BalanceAccountUpdateRequest,
    "BalancePlatform": balancePlatform_1.BalancePlatform,
    "BalanceSweepConfigurationsResponse": balanceSweepConfigurationsResponse_1.BalanceSweepConfigurationsResponse,
    "BankAccount": bankAccount_1.BankAccount,
    "BankAccountIdentificationTypeRequirement": bankAccountIdentificationTypeRequirement_1.BankAccountIdentificationTypeRequirement,
    "BankAccountIdentificationValidationRequest": bankAccountIdentificationValidationRequest_1.BankAccountIdentificationValidationRequest,
    "BankIdentification": bankIdentification_1.BankIdentification,
    "BrandVariantsRestriction": brandVariantsRestriction_1.BrandVariantsRestriction,
    "BulkAddress": bulkAddress_1.BulkAddress,
    "CALocalAccountIdentification": cALocalAccountIdentification_1.CALocalAccountIdentification,
    "CZLocalAccountIdentification": cZLocalAccountIdentification_1.CZLocalAccountIdentification,
    "CapabilityProblem": capabilityProblem_1.CapabilityProblem,
    "CapabilityProblemEntity": capabilityProblemEntity_1.CapabilityProblemEntity,
    "CapabilityProblemEntityRecursive": capabilityProblemEntityRecursive_1.CapabilityProblemEntityRecursive,
    "CapabilitySettings": capabilitySettings_1.CapabilitySettings,
    "CapitalBalance": capitalBalance_1.CapitalBalance,
    "CapitalGrantAccount": capitalGrantAccount_1.CapitalGrantAccount,
    "Card": card_1.Card,
    "CardConfiguration": cardConfiguration_1.CardConfiguration,
    "CardInfo": cardInfo_1.CardInfo,
    "ContactDetails": contactDetails_1.ContactDetails,
    "Counterparty": counterparty_1.Counterparty,
    "CounterpartyBankRestriction": counterpartyBankRestriction_1.CounterpartyBankRestriction,
    "CountriesRestriction": countriesRestriction_1.CountriesRestriction,
    "CreateSweepConfigurationV2": createSweepConfigurationV2_1.CreateSweepConfigurationV2,
    "DKLocalAccountIdentification": dKLocalAccountIdentification_1.DKLocalAccountIdentification,
    "DayOfWeekRestriction": dayOfWeekRestriction_1.DayOfWeekRestriction,
    "DeliveryAddress": deliveryAddress_1.DeliveryAddress,
    "DeliveryContact": deliveryContact_1.DeliveryContact,
    "DeviceInfo": deviceInfo_1.DeviceInfo,
    "DifferentCurrenciesRestriction": differentCurrenciesRestriction_1.DifferentCurrenciesRestriction,
    "Duration": duration_1.Duration,
    "EntryModesRestriction": entryModesRestriction_1.EntryModesRestriction,
    "Expiry": expiry_1.Expiry,
    "Fee": fee_1.Fee,
    "GetNetworkTokenResponse": getNetworkTokenResponse_1.GetNetworkTokenResponse,
    "GetTaxFormResponse": getTaxFormResponse_1.GetTaxFormResponse,
    "GrantLimit": grantLimit_1.GrantLimit,
    "GrantOffer": grantOffer_1.GrantOffer,
    "GrantOffers": grantOffers_1.GrantOffers,
    "HKLocalAccountIdentification": hKLocalAccountIdentification_1.HKLocalAccountIdentification,
    "HULocalAccountIdentification": hULocalAccountIdentification_1.HULocalAccountIdentification,
    "IbanAccountIdentification": ibanAccountIdentification_1.IbanAccountIdentification,
    "InternationalTransactionRestriction": internationalTransactionRestriction_1.InternationalTransactionRestriction,
    "InvalidField": invalidField_1.InvalidField,
    "JSONObject": jSONObject_1.JSONObject,
    "JSONPath": jSONPath_1.JSONPath,
    "ListNetworkTokensResponse": listNetworkTokensResponse_1.ListNetworkTokensResponse,
    "MatchingTransactionsRestriction": matchingTransactionsRestriction_1.MatchingTransactionsRestriction,
    "MccsRestriction": mccsRestriction_1.MccsRestriction,
    "MerchantAcquirerPair": merchantAcquirerPair_1.MerchantAcquirerPair,
    "MerchantNamesRestriction": merchantNamesRestriction_1.MerchantNamesRestriction,
    "MerchantsRestriction": merchantsRestriction_1.MerchantsRestriction,
    "NOLocalAccountIdentification": nOLocalAccountIdentification_1.NOLocalAccountIdentification,
    "NZLocalAccountIdentification": nZLocalAccountIdentification_1.NZLocalAccountIdentification,
    "Name": name_1.Name,
    "NetworkToken": networkToken_1.NetworkToken,
    "NumberAndBicAccountIdentification": numberAndBicAccountIdentification_1.NumberAndBicAccountIdentification,
    "PLLocalAccountIdentification": pLLocalAccountIdentification_1.PLLocalAccountIdentification,
    "PaginatedAccountHoldersResponse": paginatedAccountHoldersResponse_1.PaginatedAccountHoldersResponse,
    "PaginatedBalanceAccountsResponse": paginatedBalanceAccountsResponse_1.PaginatedBalanceAccountsResponse,
    "PaginatedPaymentInstrumentsResponse": paginatedPaymentInstrumentsResponse_1.PaginatedPaymentInstrumentsResponse,
    "PaymentInstrument": paymentInstrument_1.PaymentInstrument,
    "PaymentInstrumentGroup": paymentInstrumentGroup_1.PaymentInstrumentGroup,
    "PaymentInstrumentGroupInfo": paymentInstrumentGroupInfo_1.PaymentInstrumentGroupInfo,
    "PaymentInstrumentInfo": paymentInstrumentInfo_1.PaymentInstrumentInfo,
    "PaymentInstrumentRequirement": paymentInstrumentRequirement_1.PaymentInstrumentRequirement,
    "PaymentInstrumentRevealInfo": paymentInstrumentRevealInfo_1.PaymentInstrumentRevealInfo,
    "PaymentInstrumentUpdateRequest": paymentInstrumentUpdateRequest_1.PaymentInstrumentUpdateRequest,
    "Phone": phone_1.Phone,
    "PhoneNumber": phoneNumber_1.PhoneNumber,
    "PlatformPaymentConfiguration": platformPaymentConfiguration_1.PlatformPaymentConfiguration,
    "ProcessingTypesRestriction": processingTypesRestriction_1.ProcessingTypesRestriction,
    "RemediatingAction": remediatingAction_1.RemediatingAction,
    "Repayment": repayment_1.Repayment,
    "RepaymentTerm": repaymentTerm_1.RepaymentTerm,
    "RestServiceError": restServiceError_1.RestServiceError,
    "SELocalAccountIdentification": sELocalAccountIdentification_1.SELocalAccountIdentification,
    "SGLocalAccountIdentification": sGLocalAccountIdentification_1.SGLocalAccountIdentification,
    "StringMatch": stringMatch_1.StringMatch,
    "SweepConfigurationV2": sweepConfigurationV2_1.SweepConfigurationV2,
    "SweepCounterparty": sweepCounterparty_1.SweepCounterparty,
    "SweepSchedule": sweepSchedule_1.SweepSchedule,
    "ThresholdRepayment": thresholdRepayment_1.ThresholdRepayment,
    "TimeOfDay": timeOfDay_1.TimeOfDay,
    "TimeOfDayRestriction": timeOfDayRestriction_1.TimeOfDayRestriction,
    "TotalAmountRestriction": totalAmountRestriction_1.TotalAmountRestriction,
    "TransactionRule": transactionRule_1.TransactionRule,
    "TransactionRuleEntityKey": transactionRuleEntityKey_1.TransactionRuleEntityKey,
    "TransactionRuleInfo": transactionRuleInfo_1.TransactionRuleInfo,
    "TransactionRuleInterval": transactionRuleInterval_1.TransactionRuleInterval,
    "TransactionRuleResponse": transactionRuleResponse_1.TransactionRuleResponse,
    "TransactionRuleRestrictions": transactionRuleRestrictions_1.TransactionRuleRestrictions,
    "TransactionRulesResponse": transactionRulesResponse_1.TransactionRulesResponse,
    "TransferRoute": transferRoute_1.TransferRoute,
    "TransferRouteRequest": transferRouteRequest_1.TransferRouteRequest,
    "TransferRouteResponse": transferRouteResponse_1.TransferRouteResponse,
    "UKLocalAccountIdentification": uKLocalAccountIdentification_1.UKLocalAccountIdentification,
    "USLocalAccountIdentification": uSLocalAccountIdentification_1.USLocalAccountIdentification,
    "UpdateNetworkTokenRequest": updateNetworkTokenRequest_1.UpdateNetworkTokenRequest,
    "UpdatePaymentInstrument": updatePaymentInstrument_1.UpdatePaymentInstrument,
    "UpdateSweepConfigurationV2": updateSweepConfigurationV2_1.UpdateSweepConfigurationV2,
    "VerificationDeadline": verificationDeadline_1.VerificationDeadline,
    "VerificationError": verificationError_1.VerificationError,
    "VerificationErrorRecursive": verificationErrorRecursive_1.VerificationErrorRecursive,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }
            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType; // use the type given in the discriminator
                    }
                    else {
                        return expectedType; // discriminator did not map to a type
                    }
                }
                else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.serialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return data.toISOString();
        }
        else if (type === "SaleToAcquirerData") {
            const dataString = JSON.stringify(data);
            return Buffer.from(dataString).toString("base64");
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) { // in case we dont know the type
                return data;
            }
            // Get the actual type of this object
            type = this.findCorrectType(data, type);
            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.deserialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) { // is Enum
                return data;
            }
            if (!typeMap[type]) { // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
