// @adyen/api-library/lib/src/typings/balancePlatform/processingTypesRestriction.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProcessingTypesRestriction = void 0;
class ProcessingTypesRestriction {
    static getAttributeTypeMap() {
        return ProcessingTypesRestriction.attributeTypeMap;
    }
}
exports.ProcessingTypesRestriction = ProcessingTypesRestriction;
ProcessingTypesRestriction.discriminator = undefined;
ProcessingTypesRestriction.attributeTypeMap = [
    {
        "name": "operation",
        "baseName": "operation",
        "type": "string"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "Array<ProcessingTypesRestriction.ValueEnum>"
    }
];
(function (ProcessingTypesRestriction) {
    let ValueEnum;
    (function (ValueEnum) {
        ValueEnum["AtmWithdraw"] = "atmWithdraw";
        ValueEnum["BalanceInquiry"] = "balanceInquiry";
        ValueEnum["Ecommerce"] = "ecommerce";
        ValueEnum["Moto"] = "moto";
        ValueEnum["Pos"] = "pos";
        ValueEnum["Recurring"] = "recurring";
        ValueEnum["Token"] = "token";
        ValueEnum["Unknown"] = "unknown";
    })(ValueEnum = ProcessingTypesRestriction.ValueEnum || (ProcessingTypesRestriction.ValueEnum = {}));
})(ProcessingTypesRestriction = exports.ProcessingTypesRestriction || (exports.ProcessingTypesRestriction = {}));
