// @adyen/api-library/lib/src/typings/balancePlatform/transferRouteRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferRouteRequest = void 0;
class TransferRouteRequest {
    static getAttributeTypeMap() {
        return TransferRouteRequest.attributeTypeMap;
    }
}
exports.TransferRouteRequest = TransferRouteRequest;
TransferRouteRequest.discriminator = undefined;
TransferRouteRequest.attributeTypeMap = [
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "balancePlatform",
        "baseName": "balancePlatform",
        "type": "string"
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "TransferRouteRequest.CategoryEnum"
    },
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "Counterparty"
    },
    {
        "name": "country",
        "baseName": "country",
        "type": "string"
    },
    {
        "name": "currency",
        "baseName": "currency",
        "type": "string"
    },
    {
        "name": "priorities",
        "baseName": "priorities",
        "type": "Array<TransferRouteRequest.PrioritiesEnum>"
    }
];
(function (TransferRouteRequest) {
    let CategoryEnum;
    (function (CategoryEnum) {
        CategoryEnum["Bank"] = "bank";
    })(CategoryEnum = TransferRouteRequest.CategoryEnum || (TransferRouteRequest.CategoryEnum = {}));
    let PrioritiesEnum;
    (function (PrioritiesEnum) {
        PrioritiesEnum["CrossBorder"] = "crossBorder";
        PrioritiesEnum["Fast"] = "fast";
        PrioritiesEnum["Instant"] = "instant";
        PrioritiesEnum["Internal"] = "internal";
        PrioritiesEnum["Regular"] = "regular";
        PrioritiesEnum["Wire"] = "wire";
    })(PrioritiesEnum = TransferRouteRequest.PrioritiesEnum || (TransferRouteRequest.PrioritiesEnum = {}));
})(TransferRouteRequest = exports.TransferRouteRequest || (exports.TransferRouteRequest = {}));
