// @adyen/api-library/lib/src/typings/balancePlatform/transferRoute.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferRoute = void 0;
class TransferRoute {
    static getAttributeTypeMap() {
        return TransferRoute.attributeTypeMap;
    }
}
exports.TransferRoute = TransferRoute;
TransferRoute.discriminator = undefined;
TransferRoute.attributeTypeMap = [
    {
        "name": "category",
        "baseName": "category",
        "type": "TransferRoute.CategoryEnum"
    },
    {
        "name": "country",
        "baseName": "country",
        "type": "string"
    },
    {
        "name": "currency",
        "baseName": "currency",
        "type": "string"
    },
    {
        "name": "priority",
        "baseName": "priority",
        "type": "TransferRoute.PriorityEnum"
    },
    {
        "name": "requirements",
        "baseName": "requirements",
        "type": "AddressRequirement | AmountMinMaxRequirement | BankAccountIdentificationTypeRequirement | PaymentInstrumentRequirement"
    }
];
(function (TransferRoute) {
    let CategoryEnum;
    (function (CategoryEnum) {
        CategoryEnum["Bank"] = "bank";
        CategoryEnum["Card"] = "card";
        CategoryEnum["Grants"] = "grants";
        CategoryEnum["Internal"] = "internal";
        CategoryEnum["IssuedCard"] = "issuedCard";
        CategoryEnum["Migration"] = "migration";
        CategoryEnum["PlatformPayment"] = "platformPayment";
        CategoryEnum["Upgrade"] = "upgrade";
    })(CategoryEnum = TransferRoute.CategoryEnum || (TransferRoute.CategoryEnum = {}));
    let PriorityEnum;
    (function (PriorityEnum) {
        PriorityEnum["CrossBorder"] = "crossBorder";
        PriorityEnum["Fast"] = "fast";
        PriorityEnum["Instant"] = "instant";
        PriorityEnum["Internal"] = "internal";
        PriorityEnum["Regular"] = "regular";
        PriorityEnum["Wire"] = "wire";
    })(PriorityEnum = TransferRoute.PriorityEnum || (TransferRoute.PriorityEnum = {}));
})(TransferRoute = exports.TransferRoute || (exports.TransferRoute = {}));
