// @adyen/api-library/lib/src/typings/balancePlatform/verificationErrorRecursive.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerificationErrorRecursive = void 0;
class VerificationErrorRecursive {
    static getAttributeTypeMap() {
        return VerificationErrorRecursive.attributeTypeMap;
    }
}
exports.VerificationErrorRecursive = VerificationErrorRecursive;
VerificationErrorRecursive.discriminator = undefined;
VerificationErrorRecursive.attributeTypeMap = [
    {
        "name": "capabilities",
        "baseName": "capabilities",
        "type": "Array<VerificationErrorRecursive.CapabilitiesEnum>"
    },
    {
        "name": "code",
        "baseName": "code",
        "type": "string"
    },
    {
        "name": "message",
        "baseName": "message",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "VerificationErrorRecursive.TypeEnum"
    },
    {
        "name": "remediatingActions",
        "baseName": "remediatingActions",
        "type": "Array<RemediatingAction>"
    }
];
(function (VerificationErrorRecursive) {
    let CapabilitiesEnum;
    (function (CapabilitiesEnum) {
        CapabilitiesEnum["AcceptExternalFunding"] = "acceptExternalFunding";
        CapabilitiesEnum["AcceptPspFunding"] = "acceptPspFunding";
        CapabilitiesEnum["AcceptTransactionInRestrictedCountries"] = "acceptTransactionInRestrictedCountries";
        CapabilitiesEnum["AcceptTransactionInRestrictedCountriesCommercial"] = "acceptTransactionInRestrictedCountriesCommercial";
        CapabilitiesEnum["AcceptTransactionInRestrictedCountriesConsumer"] = "acceptTransactionInRestrictedCountriesConsumer";
        CapabilitiesEnum["AcceptTransactionInRestrictedIndustries"] = "acceptTransactionInRestrictedIndustries";
        CapabilitiesEnum["AcceptTransactionInRestrictedIndustriesCommercial"] = "acceptTransactionInRestrictedIndustriesCommercial";
        CapabilitiesEnum["AcceptTransactionInRestrictedIndustriesConsumer"] = "acceptTransactionInRestrictedIndustriesConsumer";
        CapabilitiesEnum["Acquiring"] = "acquiring";
        CapabilitiesEnum["AtmWithdrawal"] = "atmWithdrawal";
        CapabilitiesEnum["AtmWithdrawalCommercial"] = "atmWithdrawalCommercial";
        CapabilitiesEnum["AtmWithdrawalConsumer"] = "atmWithdrawalConsumer";
        CapabilitiesEnum["AtmWithdrawalInRestrictedCountries"] = "atmWithdrawalInRestrictedCountries";
        CapabilitiesEnum["AtmWithdrawalInRestrictedCountriesCommercial"] = "atmWithdrawalInRestrictedCountriesCommercial";
        CapabilitiesEnum["AtmWithdrawalInRestrictedCountriesConsumer"] = "atmWithdrawalInRestrictedCountriesConsumer";
        CapabilitiesEnum["AuthorisedPaymentInstrumentUser"] = "authorisedPaymentInstrumentUser";
        CapabilitiesEnum["GetGrantOffers"] = "getGrantOffers";
        CapabilitiesEnum["IssueBankAccount"] = "issueBankAccount";
        CapabilitiesEnum["IssueCard"] = "issueCard";
        CapabilitiesEnum["IssueCardCommercial"] = "issueCardCommercial";
        CapabilitiesEnum["IssueCardConsumer"] = "issueCardConsumer";
        CapabilitiesEnum["LocalAcceptance"] = "localAcceptance";
        CapabilitiesEnum["Payout"] = "payout";
        CapabilitiesEnum["PayoutToTransferInstrument"] = "payoutToTransferInstrument";
        CapabilitiesEnum["Processing"] = "processing";
        CapabilitiesEnum["ReceiveFromBalanceAccount"] = "receiveFromBalanceAccount";
        CapabilitiesEnum["ReceiveFromPlatformPayments"] = "receiveFromPlatformPayments";
        CapabilitiesEnum["ReceiveFromThirdParty"] = "receiveFromThirdParty";
        CapabilitiesEnum["ReceiveFromTransferInstrument"] = "receiveFromTransferInstrument";
        CapabilitiesEnum["ReceiveGrants"] = "receiveGrants";
        CapabilitiesEnum["ReceivePayments"] = "receivePayments";
        CapabilitiesEnum["SendToBalanceAccount"] = "sendToBalanceAccount";
        CapabilitiesEnum["SendToThirdParty"] = "sendToThirdParty";
        CapabilitiesEnum["SendToTransferInstrument"] = "sendToTransferInstrument";
        CapabilitiesEnum["ThirdPartyFunding"] = "thirdPartyFunding";
        CapabilitiesEnum["UseCard"] = "useCard";
        CapabilitiesEnum["UseCardCommercial"] = "useCardCommercial";
        CapabilitiesEnum["UseCardConsumer"] = "useCardConsumer";
        CapabilitiesEnum["UseCardInRestrictedCountries"] = "useCardInRestrictedCountries";
        CapabilitiesEnum["UseCardInRestrictedCountriesCommercial"] = "useCardInRestrictedCountriesCommercial";
        CapabilitiesEnum["UseCardInRestrictedCountriesConsumer"] = "useCardInRestrictedCountriesConsumer";
        CapabilitiesEnum["UseCardInRestrictedIndustries"] = "useCardInRestrictedIndustries";
        CapabilitiesEnum["UseCardInRestrictedIndustriesCommercial"] = "useCardInRestrictedIndustriesCommercial";
        CapabilitiesEnum["UseCardInRestrictedIndustriesConsumer"] = "useCardInRestrictedIndustriesConsumer";
        CapabilitiesEnum["WithdrawFromAtm"] = "withdrawFromAtm";
        CapabilitiesEnum["WithdrawFromAtmCommercial"] = "withdrawFromAtmCommercial";
        CapabilitiesEnum["WithdrawFromAtmConsumer"] = "withdrawFromAtmConsumer";
        CapabilitiesEnum["WithdrawFromAtmInRestrictedCountries"] = "withdrawFromAtmInRestrictedCountries";
        CapabilitiesEnum["WithdrawFromAtmInRestrictedCountriesCommercial"] = "withdrawFromAtmInRestrictedCountriesCommercial";
        CapabilitiesEnum["WithdrawFromAtmInRestrictedCountriesConsumer"] = "withdrawFromAtmInRestrictedCountriesConsumer";
    })(CapabilitiesEnum = VerificationErrorRecursive.CapabilitiesEnum || (VerificationErrorRecursive.CapabilitiesEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["DataMissing"] = "dataMissing";
        TypeEnum["InvalidInput"] = "invalidInput";
        TypeEnum["PendingStatus"] = "pendingStatus";
    })(TypeEnum = VerificationErrorRecursive.TypeEnum || (VerificationErrorRecursive.TypeEnum = {}));
})(VerificationErrorRecursive = exports.VerificationErrorRecursive || (exports.VerificationErrorRecursive = {}));
