// @adyen/api-library/lib/src/typings/platformsFund/payoutAccountHolderRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayoutAccountHolderRequest = void 0;
class PayoutAccountHolderRequest {
    static getAttributeTypeMap() {
        return PayoutAccountHolderRequest.attributeTypeMap;
    }
}
exports.PayoutAccountHolderRequest = PayoutAccountHolderRequest;
PayoutAccountHolderRequest.discriminator = undefined;
PayoutAccountHolderRequest.attributeTypeMap = [
    {
        "name": "accountCode",
        "baseName": "accountCode",
        "type": "string"
    },
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "merchantReference",
        "baseName": "merchantReference",
        "type": "string"
    },
    {
        "name": "payoutMethodCode",
        "baseName": "payoutMethodCode",
        "type": "string"
    },
    {
        "name": "payoutSpeed",
        "baseName": "payoutSpeed",
        "type": "PayoutAccountHolderRequest.PayoutSpeedEnum"
    }
];
(function (PayoutAccountHolderRequest) {
    let PayoutSpeedEnum;
    (function (PayoutSpeedEnum) {
        PayoutSpeedEnum["Instant"] = "INSTANT";
        PayoutSpeedEnum["SameDay"] = "SAME_DAY";
        PayoutSpeedEnum["Standard"] = "STANDARD";
    })(PayoutSpeedEnum = PayoutAccountHolderRequest.PayoutSpeedEnum || (PayoutAccountHolderRequest.PayoutSpeedEnum = {}));
})(PayoutAccountHolderRequest = exports.PayoutAccountHolderRequest || (exports.PayoutAccountHolderRequest = {}));
