// @adyen/api-library/lib/src/typings/management/payAtTable.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayAtTable = void 0;
class PayAtTable {
    static getAttributeTypeMap() {
        return PayAtTable.attributeTypeMap;
    }
}
exports.PayAtTable = PayAtTable;
PayAtTable.discriminator = undefined;
PayAtTable.attributeTypeMap = [
    {
        "name": "authenticationMethod",
        "baseName": "authenticationMethod",
        "type": "PayAtTable.AuthenticationMethodEnum"
    },
    {
        "name": "enablePayAtTable",
        "baseName": "enablePayAtTable",
        "type": "boolean"
    }
];
(function (PayAtTable) {
    let AuthenticationMethodEnum;
    (function (AuthenticationMethodEnum) {
        AuthenticationMethodEnum["Magswipe"] = "MAGSWIPE";
        AuthenticationMethodEnum["Mke"] = "MKE";
    })(AuthenticationMethodEnum = PayAtTable.AuthenticationMethodEnum || (PayAtTable.AuthenticationMethodEnum = {}));
})(PayAtTable = exports.PayAtTable || (exports.PayAtTable = {}));
