// @adyen/api-library/lib/src/typings/management/splitConfigurationLogic.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SplitConfigurationLogic = void 0;
class SplitConfigurationLogic {
    static getAttributeTypeMap() {
        return SplitConfigurationLogic.attributeTypeMap;
    }
}
exports.SplitConfigurationLogic = SplitConfigurationLogic;
SplitConfigurationLogic.discriminator = undefined;
SplitConfigurationLogic.attributeTypeMap = [
    {
        "name": "additionalCommission",
        "baseName": "additionalCommission",
        "type": "AdditionalCommission"
    },
    {
        "name": "chargeback",
        "baseName": "chargeback",
        "type": "SplitConfigurationLogic.ChargebackEnum"
    },
    {
        "name": "commission",
        "baseName": "commission",
        "type": "Commission"
    },
    {
        "name": "paymentFee",
        "baseName": "paymentFee",
        "type": "SplitConfigurationLogic.PaymentFeeEnum"
    },
    {
        "name": "remainder",
        "baseName": "remainder",
        "type": "SplitConfigurationLogic.RemainderEnum"
    },
    {
        "name": "splitLogicId",
        "baseName": "splitLogicId",
        "type": "string"
    },
    {
        "name": "surcharge",
        "baseName": "surcharge",
        "type": "SplitConfigurationLogic.SurchargeEnum"
    },
    {
        "name": "tip",
        "baseName": "tip",
        "type": "SplitConfigurationLogic.TipEnum"
    }
];
(function (SplitConfigurationLogic) {
    let ChargebackEnum;
    (function (ChargebackEnum) {
        ChargebackEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        ChargebackEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
        ChargebackEnum["DeductAccordingToSplitRatio"] = "deductAccordingToSplitRatio";
    })(ChargebackEnum = SplitConfigurationLogic.ChargebackEnum || (SplitConfigurationLogic.ChargebackEnum = {}));
    let PaymentFeeEnum;
    (function (PaymentFeeEnum) {
        PaymentFeeEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        PaymentFeeEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(PaymentFeeEnum = SplitConfigurationLogic.PaymentFeeEnum || (SplitConfigurationLogic.PaymentFeeEnum = {}));
    let RemainderEnum;
    (function (RemainderEnum) {
        RemainderEnum["AddToLiableAccount"] = "addToLiableAccount";
        RemainderEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(RemainderEnum = SplitConfigurationLogic.RemainderEnum || (SplitConfigurationLogic.RemainderEnum = {}));
    let SurchargeEnum;
    (function (SurchargeEnum) {
        SurchargeEnum["AddToLiableAccount"] = "addToLiableAccount";
        SurchargeEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(SurchargeEnum = SplitConfigurationLogic.SurchargeEnum || (SplitConfigurationLogic.SurchargeEnum = {}));
    let TipEnum;
    (function (TipEnum) {
        TipEnum["AddToLiableAccount"] = "addToLiableAccount";
        TipEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(TipEnum = SplitConfigurationLogic.TipEnum || (SplitConfigurationLogic.TipEnum = {}));
})(SplitConfigurationLogic = exports.SplitConfigurationLogic || (exports.SplitConfigurationLogic = {}));
