// @adyen/api-library/lib/src/typings/management/terminalConnectivityCellular.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerminalConnectivityCellular = void 0;
class TerminalConnectivityCellular {
    static getAttributeTypeMap() {
        return TerminalConnectivityCellular.attributeTypeMap;
    }
}
exports.TerminalConnectivityCellular = TerminalConnectivityCellular;
TerminalConnectivityCellular.discriminator = undefined;
TerminalConnectivityCellular.attributeTypeMap = [
    {
        "name": "iccid",
        "baseName": "iccid",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "TerminalConnectivityCellular.StatusEnum"
    }
];
(function (TerminalConnectivityCellular) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Activated"] = "activated";
        StatusEnum["Deactivated"] = "deactivated";
        StatusEnum["Deprecated"] = "deprecated";
        StatusEnum["Inventory"] = "inventory";
        StatusEnum["ReadyForActivation"] = "readyForActivation";
    })(StatusEnum = TerminalConnectivityCellular.StatusEnum || (TerminalConnectivityCellular.StatusEnum = {}));
})(TerminalConnectivityCellular = exports.TerminalConnectivityCellular || (exports.TerminalConnectivityCellular = {}));
