// @adyen/api-library/lib/src/typings/management/transactionDescriptionInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionDescriptionInfo = void 0;
class TransactionDescriptionInfo {
    static getAttributeTypeMap() {
        return TransactionDescriptionInfo.attributeTypeMap;
    }
}
exports.TransactionDescriptionInfo = TransactionDescriptionInfo;
TransactionDescriptionInfo.discriminator = undefined;
TransactionDescriptionInfo.attributeTypeMap = [
    {
        "name": "doingBusinessAsName",
        "baseName": "doingBusinessAsName",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransactionDescriptionInfo.TypeEnum"
    }
];
(function (TransactionDescriptionInfo) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Append"] = "append";
        TypeEnum["Dynamic"] = "dynamic";
        TypeEnum["Fixed"] = "fixed";
    })(TypeEnum = TransactionDescriptionInfo.TypeEnum || (TransactionDescriptionInfo.TypeEnum = {}));
})(TransactionDescriptionInfo = exports.TransactionDescriptionInfo || (exports.TransactionDescriptionInfo = {}));
