// @adyen/api-library/lib/src/typings/management/updateSplitConfigurationLogicRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSplitConfigurationLogicRequest = void 0;
class UpdateSplitConfigurationLogicRequest {
    static getAttributeTypeMap() {
        return UpdateSplitConfigurationLogicRequest.attributeTypeMap;
    }
}
exports.UpdateSplitConfigurationLogicRequest = UpdateSplitConfigurationLogicRequest;
UpdateSplitConfigurationLogicRequest.discriminator = undefined;
UpdateSplitConfigurationLogicRequest.attributeTypeMap = [
    {
        "name": "additionalCommission",
        "baseName": "additionalCommission",
        "type": "AdditionalCommission"
    },
    {
        "name": "chargeback",
        "baseName": "chargeback",
        "type": "UpdateSplitConfigurationLogicRequest.ChargebackEnum"
    },
    {
        "name": "commission",
        "baseName": "commission",
        "type": "Commission"
    },
    {
        "name": "paymentFee",
        "baseName": "paymentFee",
        "type": "UpdateSplitConfigurationLogicRequest.PaymentFeeEnum"
    },
    {
        "name": "remainder",
        "baseName": "remainder",
        "type": "UpdateSplitConfigurationLogicRequest.RemainderEnum"
    },
    {
        "name": "splitLogicId",
        "baseName": "splitLogicId",
        "type": "string"
    },
    {
        "name": "surcharge",
        "baseName": "surcharge",
        "type": "UpdateSplitConfigurationLogicRequest.SurchargeEnum"
    },
    {
        "name": "tip",
        "baseName": "tip",
        "type": "UpdateSplitConfigurationLogicRequest.TipEnum"
    }
];
(function (UpdateSplitConfigurationLogicRequest) {
    let ChargebackEnum;
    (function (ChargebackEnum) {
        ChargebackEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        ChargebackEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
        ChargebackEnum["DeductAccordingToSplitRatio"] = "deductAccordingToSplitRatio";
    })(ChargebackEnum = UpdateSplitConfigurationLogicRequest.ChargebackEnum || (UpdateSplitConfigurationLogicRequest.ChargebackEnum = {}));
    let PaymentFeeEnum;
    (function (PaymentFeeEnum) {
        PaymentFeeEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        PaymentFeeEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(PaymentFeeEnum = UpdateSplitConfigurationLogicRequest.PaymentFeeEnum || (UpdateSplitConfigurationLogicRequest.PaymentFeeEnum = {}));
    let RemainderEnum;
    (function (RemainderEnum) {
        RemainderEnum["AddToLiableAccount"] = "addToLiableAccount";
        RemainderEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(RemainderEnum = UpdateSplitConfigurationLogicRequest.RemainderEnum || (UpdateSplitConfigurationLogicRequest.RemainderEnum = {}));
    let SurchargeEnum;
    (function (SurchargeEnum) {
        SurchargeEnum["AddToLiableAccount"] = "addToLiableAccount";
        SurchargeEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(SurchargeEnum = UpdateSplitConfigurationLogicRequest.SurchargeEnum || (UpdateSplitConfigurationLogicRequest.SurchargeEnum = {}));
    let TipEnum;
    (function (TipEnum) {
        TipEnum["AddToLiableAccount"] = "addToLiableAccount";
        TipEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(TipEnum = UpdateSplitConfigurationLogicRequest.TipEnum || (UpdateSplitConfigurationLogicRequest.TipEnum = {}));
})(UpdateSplitConfigurationLogicRequest = exports.UpdateSplitConfigurationLogicRequest || (exports.UpdateSplitConfigurationLogicRequest = {}));
