// @adyen/api-library/lib/src/typings/management/webhook.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhook = void 0;
class Webhook {
    static getAttributeTypeMap() {
        return Webhook.attributeTypeMap;
    }
}
exports.Webhook = Webhook;
Webhook.discriminator = undefined;
Webhook.attributeTypeMap = [
    {
        "name": "_links",
        "baseName": "_links",
        "type": "WebhookLinks"
    },
    {
        "name": "acceptsExpiredCertificate",
        "baseName": "acceptsExpiredCertificate",
        "type": "boolean"
    },
    {
        "name": "acceptsSelfSignedCertificate",
        "baseName": "acceptsSelfSignedCertificate",
        "type": "boolean"
    },
    {
        "name": "acceptsUntrustedRootCertificate",
        "baseName": "acceptsUntrustedRootCertificate",
        "type": "boolean"
    },
    {
        "name": "accountReference",
        "baseName": "accountReference",
        "type": "string"
    },
    {
        "name": "active",
        "baseName": "active",
        "type": "boolean"
    },
    {
        "name": "additionalSettings",
        "baseName": "additionalSettings",
        "type": "AdditionalSettingsResponse"
    },
    {
        "name": "certificateAlias",
        "baseName": "certificateAlias",
        "type": "string"
    },
    {
        "name": "communicationFormat",
        "baseName": "communicationFormat",
        "type": "Webhook.CommunicationFormatEnum"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "filterMerchantAccountType",
        "baseName": "filterMerchantAccountType",
        "type": "Webhook.FilterMerchantAccountTypeEnum"
    },
    {
        "name": "filterMerchantAccounts",
        "baseName": "filterMerchantAccounts",
        "type": "Array<string>"
    },
    {
        "name": "hasError",
        "baseName": "hasError",
        "type": "boolean"
    },
    {
        "name": "hasPassword",
        "baseName": "hasPassword",
        "type": "boolean"
    },
    {
        "name": "hmacKeyCheckValue",
        "baseName": "hmacKeyCheckValue",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "networkType",
        "baseName": "networkType",
        "type": "Webhook.NetworkTypeEnum"
    },
    {
        "name": "populateSoapActionHeader",
        "baseName": "populateSoapActionHeader",
        "type": "boolean"
    },
    {
        "name": "sslVersion",
        "baseName": "sslVersion",
        "type": "Webhook.SslVersionEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "string"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string"
    },
    {
        "name": "username",
        "baseName": "username",
        "type": "string"
    }
];
(function (Webhook) {
    let CommunicationFormatEnum;
    (function (CommunicationFormatEnum) {
        CommunicationFormatEnum["Http"] = "http";
        CommunicationFormatEnum["Json"] = "json";
        CommunicationFormatEnum["Soap"] = "soap";
    })(CommunicationFormatEnum = Webhook.CommunicationFormatEnum || (Webhook.CommunicationFormatEnum = {}));
    let FilterMerchantAccountTypeEnum;
    (function (FilterMerchantAccountTypeEnum) {
        FilterMerchantAccountTypeEnum["AllAccounts"] = "allAccounts";
        FilterMerchantAccountTypeEnum["ExcludeAccounts"] = "excludeAccounts";
        FilterMerchantAccountTypeEnum["IncludeAccounts"] = "includeAccounts";
    })(FilterMerchantAccountTypeEnum = Webhook.FilterMerchantAccountTypeEnum || (Webhook.FilterMerchantAccountTypeEnum = {}));
    let NetworkTypeEnum;
    (function (NetworkTypeEnum) {
        NetworkTypeEnum["Local"] = "local";
        NetworkTypeEnum["Public"] = "public";
    })(NetworkTypeEnum = Webhook.NetworkTypeEnum || (Webhook.NetworkTypeEnum = {}));
    let SslVersionEnum;
    (function (SslVersionEnum) {
        SslVersionEnum["Http"] = "HTTP";
        SslVersionEnum["Tlsv12"] = "TLSv1.2";
        SslVersionEnum["Tlsv13"] = "TLSv1.3";
    })(SslVersionEnum = Webhook.SslVersionEnum || (Webhook.SslVersionEnum = {}));
})(Webhook = exports.Webhook || (exports.Webhook = {}));
