// @adyen/api-library/lib/src/services/dataProtectionApi.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataProtectionAPI = void 0;
const getJsonResponse_1 = __importDefault(require("../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../service"));
const resource_1 = __importDefault(require("./resource"));
const models_1 = require("../typings/dataProtection/models");
class DataProtectionAPI extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://ca-test.adyen.com/ca/services/DataProtectionService/v1";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Submit a Subject Erasure Request.
    * @param subjectErasureByPspReferenceRequest {@link SubjectErasureByPspReferenceRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link SubjectErasureResponse }
    */
    async requestSubjectErasure(subjectErasureByPspReferenceRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/requestSubjectErasure`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(subjectErasureByPspReferenceRequest, "SubjectErasureByPspReferenceRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "SubjectErasureResponse");
    }
}
exports.DataProtectionAPI = DataProtectionAPI;
exports.default = DataProtectionAPI;
