// @adyen/api-library/lib/src/typings/legalEntityManagement/calculateTermsOfServiceStatusResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CalculateTermsOfServiceStatusResponse = void 0;
class CalculateTermsOfServiceStatusResponse {
    static getAttributeTypeMap() {
        return CalculateTermsOfServiceStatusResponse.attributeTypeMap;
    }
}
exports.CalculateTermsOfServiceStatusResponse = CalculateTermsOfServiceStatusResponse;
CalculateTermsOfServiceStatusResponse.discriminator = undefined;
CalculateTermsOfServiceStatusResponse.attributeTypeMap = [
    {
        "name": "termsOfServiceTypes",
        "baseName": "termsOfServiceTypes",
        "type": "Array<CalculateTermsOfServiceStatusResponse.TermsOfServiceTypesEnum>"
    }
];
(function (CalculateTermsOfServiceStatusResponse) {
    let TermsOfServiceTypesEnum;
    (function (TermsOfServiceTypesEnum) {
        TermsOfServiceTypesEnum["AdyenAccount"] = "adyenAccount";
        TermsOfServiceTypesEnum["AdyenCapital"] = "adyenCapital";
        TermsOfServiceTypesEnum["AdyenCard"] = "adyenCard";
        TermsOfServiceTypesEnum["AdyenForPlatformsAdvanced"] = "adyenForPlatformsAdvanced";
        TermsOfServiceTypesEnum["AdyenForPlatformsManage"] = "adyenForPlatformsManage";
        TermsOfServiceTypesEnum["AdyenFranchisee"] = "adyenFranchisee";
        TermsOfServiceTypesEnum["AdyenIssuing"] = "adyenIssuing";
    })(TermsOfServiceTypesEnum = CalculateTermsOfServiceStatusResponse.TermsOfServiceTypesEnum || (CalculateTermsOfServiceStatusResponse.TermsOfServiceTypesEnum = {}));
})(CalculateTermsOfServiceStatusResponse = exports.CalculateTermsOfServiceStatusResponse || (exports.CalculateTermsOfServiceStatusResponse = {}));
