// @adyen/api-library/lib/src/typings/legalEntityManagement/getTermsOfServiceDocumentRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTermsOfServiceDocumentRequest = void 0;
class GetTermsOfServiceDocumentRequest {
    static getAttributeTypeMap() {
        return GetTermsOfServiceDocumentRequest.attributeTypeMap;
    }
}
exports.GetTermsOfServiceDocumentRequest = GetTermsOfServiceDocumentRequest;
GetTermsOfServiceDocumentRequest.discriminator = undefined;
GetTermsOfServiceDocumentRequest.attributeTypeMap = [
    {
        "name": "language",
        "baseName": "language",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "GetTermsOfServiceDocumentRequest.TypeEnum"
    }
];
(function (GetTermsOfServiceDocumentRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AdyenAccount"] = "adyenAccount";
        TypeEnum["AdyenCapital"] = "adyenCapital";
        TypeEnum["AdyenCard"] = "adyenCard";
        TypeEnum["AdyenForPlatformsAdvanced"] = "adyenForPlatformsAdvanced";
        TypeEnum["AdyenForPlatformsManage"] = "adyenForPlatformsManage";
        TypeEnum["AdyenFranchisee"] = "adyenFranchisee";
        TypeEnum["AdyenIssuing"] = "adyenIssuing";
    })(TypeEnum = GetTermsOfServiceDocumentRequest.TypeEnum || (GetTermsOfServiceDocumentRequest.TypeEnum = {}));
})(GetTermsOfServiceDocumentRequest = exports.GetTermsOfServiceDocumentRequest || (exports.GetTermsOfServiceDocumentRequest = {}));
