// @adyen/api-library/lib/src/typings/legalEntityManagement/legalEntityAssociation.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalEntityAssociation = void 0;
class LegalEntityAssociation {
    static getAttributeTypeMap() {
        return LegalEntityAssociation.attributeTypeMap;
    }
}
exports.LegalEntityAssociation = LegalEntityAssociation;
LegalEntityAssociation.discriminator = undefined;
LegalEntityAssociation.attributeTypeMap = [
    {
        "name": "associatorId",
        "baseName": "associatorId",
        "type": "string"
    },
    {
        "name": "entityType",
        "baseName": "entityType",
        "type": "string"
    },
    {
        "name": "jobTitle",
        "baseName": "jobTitle",
        "type": "string"
    },
    {
        "name": "legalEntityId",
        "baseName": "legalEntityId",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "LegalEntityAssociation.TypeEnum"
    }
];
(function (LegalEntityAssociation) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["PciSignatory"] = "pciSignatory";
        TypeEnum["Signatory"] = "signatory";
        TypeEnum["SoleProprietorship"] = "soleProprietorship";
        TypeEnum["UboThroughControl"] = "uboThroughControl";
        TypeEnum["UboThroughOwnership"] = "uboThroughOwnership";
        TypeEnum["UltimateParentCompany"] = "ultimateParentCompany";
    })(TypeEnum = LegalEntityAssociation.TypeEnum || (LegalEntityAssociation.TypeEnum = {}));
})(LegalEntityAssociation = exports.LegalEntityAssociation || (exports.LegalEntityAssociation = {}));
