// @adyen/api-library/lib/src/typings/legalEntityManagement/legalEntity.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalEntity = void 0;
class LegalEntity {
    static getAttributeTypeMap() {
        return LegalEntity.attributeTypeMap;
    }
}
exports.LegalEntity = LegalEntity;
LegalEntity.discriminator = undefined;
LegalEntity.attributeTypeMap = [
    {
        "name": "capabilities",
        "baseName": "capabilities",
        "type": "{ [key: string]: LegalEntityCapability; }"
    },
    {
        "name": "documentDetails",
        "baseName": "documentDetails",
        "type": "Array<DocumentReference>"
    },
    {
        "name": "documents",
        "baseName": "documents",
        "type": "Array<EntityReference>"
    },
    {
        "name": "entityAssociations",
        "baseName": "entityAssociations",
        "type": "Array<LegalEntityAssociation>"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "individual",
        "baseName": "individual",
        "type": "Individual"
    },
    {
        "name": "organization",
        "baseName": "organization",
        "type": "Organization"
    },
    {
        "name": "problems",
        "baseName": "problems",
        "type": "Array<CapabilityProblem>"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "soleProprietorship",
        "baseName": "soleProprietorship",
        "type": "SoleProprietorship"
    },
    {
        "name": "transferInstruments",
        "baseName": "transferInstruments",
        "type": "Array<TransferInstrumentReference>"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "LegalEntity.TypeEnum"
    },
    {
        "name": "verificationDeadlines",
        "baseName": "verificationDeadlines",
        "type": "Array<VerificationDeadline>"
    }
];
(function (LegalEntity) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Individual"] = "individual";
        TypeEnum["Organization"] = "organization";
        TypeEnum["SoleProprietorship"] = "soleProprietorship";
        TypeEnum["Trust"] = "trust";
        TypeEnum["UnincorporatedPartnership"] = "unincorporatedPartnership";
    })(TypeEnum = LegalEntity.TypeEnum || (LegalEntity.TypeEnum = {}));
})(LegalEntity = exports.LegalEntity || (exports.LegalEntity = {}));
