// @adyen/api-library/lib/src/typings/legalEntityManagement/legalEntityInfoRequiredType.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalEntityInfoRequiredType = void 0;
class LegalEntityInfoRequiredType {
    static getAttributeTypeMap() {
        return LegalEntityInfoRequiredType.attributeTypeMap;
    }
}
exports.LegalEntityInfoRequiredType = LegalEntityInfoRequiredType;
LegalEntityInfoRequiredType.discriminator = undefined;
LegalEntityInfoRequiredType.attributeTypeMap = [
    {
        "name": "capabilities",
        "baseName": "capabilities",
        "type": "{ [key: string]: LegalEntityCapability; }"
    },
    {
        "name": "entityAssociations",
        "baseName": "entityAssociations",
        "type": "Array<LegalEntityAssociation>"
    },
    {
        "name": "individual",
        "baseName": "individual",
        "type": "Individual"
    },
    {
        "name": "organization",
        "baseName": "organization",
        "type": "Organization"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "soleProprietorship",
        "baseName": "soleProprietorship",
        "type": "SoleProprietorship"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "LegalEntityInfoRequiredType.TypeEnum"
    }
];
(function (LegalEntityInfoRequiredType) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Individual"] = "individual";
        TypeEnum["Organization"] = "organization";
        TypeEnum["SoleProprietorship"] = "soleProprietorship";
        TypeEnum["Trust"] = "trust";
        TypeEnum["UnincorporatedPartnership"] = "unincorporatedPartnership";
    })(TypeEnum = LegalEntityInfoRequiredType.TypeEnum || (LegalEntityInfoRequiredType.TypeEnum = {}));
})(LegalEntityInfoRequiredType = exports.LegalEntityInfoRequiredType || (exports.LegalEntityInfoRequiredType = {}));
