// @adyen/api-library/lib/src/typings/legalEntityManagement/legalEntityInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalEntityInfo = void 0;
class LegalEntityInfo {
    static getAttributeTypeMap() {
        return LegalEntityInfo.attributeTypeMap;
    }
}
exports.LegalEntityInfo = LegalEntityInfo;
LegalEntityInfo.discriminator = undefined;
LegalEntityInfo.attributeTypeMap = [
    {
        "name": "capabilities",
        "baseName": "capabilities",
        "type": "{ [key: string]: LegalEntityCapability; }"
    },
    {
        "name": "entityAssociations",
        "baseName": "entityAssociations",
        "type": "Array<LegalEntityAssociation>"
    },
    {
        "name": "individual",
        "baseName": "individual",
        "type": "Individual"
    },
    {
        "name": "organization",
        "baseName": "organization",
        "type": "Organization"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "soleProprietorship",
        "baseName": "soleProprietorship",
        "type": "SoleProprietorship"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "LegalEntityInfo.TypeEnum"
    }
];
(function (LegalEntityInfo) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Individual"] = "individual";
        TypeEnum["Organization"] = "organization";
        TypeEnum["SoleProprietorship"] = "soleProprietorship";
        TypeEnum["Trust"] = "trust";
        TypeEnum["UnincorporatedPartnership"] = "unincorporatedPartnership";
    })(TypeEnum = LegalEntityInfo.TypeEnum || (LegalEntityInfo.TypeEnum = {}));
})(LegalEntityInfo = exports.LegalEntityInfo || (exports.LegalEntityInfo = {}));
