// @adyen/api-library/lib/src/typings/legalEntityManagement/models.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectSerializer = void 0;
__exportStar(require("./aULocalAccountIdentification"), exports);
__exportStar(require("./acceptTermsOfServiceRequest"), exports);
__exportStar(require("./acceptTermsOfServiceResponse"), exports);
__exportStar(require("./additionalBankIdentification"), exports);
__exportStar(require("./address"), exports);
__exportStar(require("./amount"), exports);
__exportStar(require("./attachment"), exports);
__exportStar(require("./bankAccountInfo"), exports);
__exportStar(require("./birthData"), exports);
__exportStar(require("./businessLine"), exports);
__exportStar(require("./businessLineInfo"), exports);
__exportStar(require("./businessLineInfoUpdate"), exports);
__exportStar(require("./businessLines"), exports);
__exportStar(require("./cALocalAccountIdentification"), exports);
__exportStar(require("./cZLocalAccountIdentification"), exports);
__exportStar(require("./calculateTermsOfServiceStatusResponse"), exports);
__exportStar(require("./capabilityProblem"), exports);
__exportStar(require("./capabilityProblemEntity"), exports);
__exportStar(require("./capabilityProblemEntityRecursive"), exports);
__exportStar(require("./capabilitySettings"), exports);
__exportStar(require("./dKLocalAccountIdentification"), exports);
__exportStar(require("./document"), exports);
__exportStar(require("./documentPage"), exports);
__exportStar(require("./documentReference"), exports);
__exportStar(require("./entityReference"), exports);
__exportStar(require("./generatePciDescriptionRequest"), exports);
__exportStar(require("./generatePciDescriptionResponse"), exports);
__exportStar(require("./getPciQuestionnaireInfosResponse"), exports);
__exportStar(require("./getPciQuestionnaireResponse"), exports);
__exportStar(require("./getTermsOfServiceAcceptanceInfosResponse"), exports);
__exportStar(require("./getTermsOfServiceDocumentRequest"), exports);
__exportStar(require("./getTermsOfServiceDocumentResponse"), exports);
__exportStar(require("./hKLocalAccountIdentification"), exports);
__exportStar(require("./hULocalAccountIdentification"), exports);
__exportStar(require("./ibanAccountIdentification"), exports);
__exportStar(require("./identificationData"), exports);
__exportStar(require("./individual"), exports);
__exportStar(require("./legalEntity"), exports);
__exportStar(require("./legalEntityAssociation"), exports);
__exportStar(require("./legalEntityCapability"), exports);
__exportStar(require("./legalEntityInfo"), exports);
__exportStar(require("./legalEntityInfoRequiredType"), exports);
__exportStar(require("./nOLocalAccountIdentification"), exports);
__exportStar(require("./nZLocalAccountIdentification"), exports);
__exportStar(require("./name"), exports);
__exportStar(require("./numberAndBicAccountIdentification"), exports);
__exportStar(require("./onboardingLink"), exports);
__exportStar(require("./onboardingLinkInfo"), exports);
__exportStar(require("./onboardingTheme"), exports);
__exportStar(require("./onboardingThemes"), exports);
__exportStar(require("./organization"), exports);
__exportStar(require("./ownerEntity"), exports);
__exportStar(require("./pLLocalAccountIdentification"), exports);
__exportStar(require("./pciDocumentInfo"), exports);
__exportStar(require("./pciSigningRequest"), exports);
__exportStar(require("./pciSigningResponse"), exports);
__exportStar(require("./phoneNumber"), exports);
__exportStar(require("./remediatingAction"), exports);
__exportStar(require("./sELocalAccountIdentification"), exports);
__exportStar(require("./sGLocalAccountIdentification"), exports);
__exportStar(require("./serviceError"), exports);
__exportStar(require("./soleProprietorship"), exports);
__exportStar(require("./sourceOfFunds"), exports);
__exportStar(require("./stockData"), exports);
__exportStar(require("./supportingEntityCapability"), exports);
__exportStar(require("./taxInformation"), exports);
__exportStar(require("./taxReportingClassification"), exports);
__exportStar(require("./termsOfServiceAcceptanceInfo"), exports);
__exportStar(require("./transferInstrument"), exports);
__exportStar(require("./transferInstrumentInfo"), exports);
__exportStar(require("./transferInstrumentReference"), exports);
__exportStar(require("./uKLocalAccountIdentification"), exports);
__exportStar(require("./uSLocalAccountIdentification"), exports);
__exportStar(require("./verificationDeadline"), exports);
__exportStar(require("./verificationError"), exports);
__exportStar(require("./verificationErrorRecursive"), exports);
__exportStar(require("./verificationErrors"), exports);
__exportStar(require("./webData"), exports);
__exportStar(require("./webDataExemption"), exports);
const aULocalAccountIdentification_1 = require("./aULocalAccountIdentification");
const acceptTermsOfServiceRequest_1 = require("./acceptTermsOfServiceRequest");
const acceptTermsOfServiceResponse_1 = require("./acceptTermsOfServiceResponse");
const additionalBankIdentification_1 = require("./additionalBankIdentification");
const address_1 = require("./address");
const amount_1 = require("./amount");
const attachment_1 = require("./attachment");
const bankAccountInfo_1 = require("./bankAccountInfo");
const birthData_1 = require("./birthData");
const businessLine_1 = require("./businessLine");
const businessLineInfo_1 = require("./businessLineInfo");
const businessLineInfoUpdate_1 = require("./businessLineInfoUpdate");
const businessLines_1 = require("./businessLines");
const cALocalAccountIdentification_1 = require("./cALocalAccountIdentification");
const cZLocalAccountIdentification_1 = require("./cZLocalAccountIdentification");
const calculateTermsOfServiceStatusResponse_1 = require("./calculateTermsOfServiceStatusResponse");
const capabilityProblem_1 = require("./capabilityProblem");
const capabilityProblemEntity_1 = require("./capabilityProblemEntity");
const capabilityProblemEntityRecursive_1 = require("./capabilityProblemEntityRecursive");
const capabilitySettings_1 = require("./capabilitySettings");
const dKLocalAccountIdentification_1 = require("./dKLocalAccountIdentification");
const document_1 = require("./document");
const documentPage_1 = require("./documentPage");
const documentReference_1 = require("./documentReference");
const entityReference_1 = require("./entityReference");
const generatePciDescriptionRequest_1 = require("./generatePciDescriptionRequest");
const generatePciDescriptionResponse_1 = require("./generatePciDescriptionResponse");
const getPciQuestionnaireInfosResponse_1 = require("./getPciQuestionnaireInfosResponse");
const getPciQuestionnaireResponse_1 = require("./getPciQuestionnaireResponse");
const getTermsOfServiceAcceptanceInfosResponse_1 = require("./getTermsOfServiceAcceptanceInfosResponse");
const getTermsOfServiceDocumentRequest_1 = require("./getTermsOfServiceDocumentRequest");
const getTermsOfServiceDocumentResponse_1 = require("./getTermsOfServiceDocumentResponse");
const hKLocalAccountIdentification_1 = require("./hKLocalAccountIdentification");
const hULocalAccountIdentification_1 = require("./hULocalAccountIdentification");
const ibanAccountIdentification_1 = require("./ibanAccountIdentification");
const identificationData_1 = require("./identificationData");
const individual_1 = require("./individual");
const legalEntity_1 = require("./legalEntity");
const legalEntityAssociation_1 = require("./legalEntityAssociation");
const legalEntityCapability_1 = require("./legalEntityCapability");
const legalEntityInfo_1 = require("./legalEntityInfo");
const legalEntityInfoRequiredType_1 = require("./legalEntityInfoRequiredType");
const nOLocalAccountIdentification_1 = require("./nOLocalAccountIdentification");
const nZLocalAccountIdentification_1 = require("./nZLocalAccountIdentification");
const name_1 = require("./name");
const numberAndBicAccountIdentification_1 = require("./numberAndBicAccountIdentification");
const onboardingLink_1 = require("./onboardingLink");
const onboardingLinkInfo_1 = require("./onboardingLinkInfo");
const onboardingTheme_1 = require("./onboardingTheme");
const onboardingThemes_1 = require("./onboardingThemes");
const organization_1 = require("./organization");
const ownerEntity_1 = require("./ownerEntity");
const pLLocalAccountIdentification_1 = require("./pLLocalAccountIdentification");
const pciDocumentInfo_1 = require("./pciDocumentInfo");
const pciSigningRequest_1 = require("./pciSigningRequest");
const pciSigningResponse_1 = require("./pciSigningResponse");
const phoneNumber_1 = require("./phoneNumber");
const remediatingAction_1 = require("./remediatingAction");
const sELocalAccountIdentification_1 = require("./sELocalAccountIdentification");
const sGLocalAccountIdentification_1 = require("./sGLocalAccountIdentification");
const serviceError_1 = require("./serviceError");
const soleProprietorship_1 = require("./soleProprietorship");
const sourceOfFunds_1 = require("./sourceOfFunds");
const stockData_1 = require("./stockData");
const supportingEntityCapability_1 = require("./supportingEntityCapability");
const taxInformation_1 = require("./taxInformation");
const taxReportingClassification_1 = require("./taxReportingClassification");
const termsOfServiceAcceptanceInfo_1 = require("./termsOfServiceAcceptanceInfo");
const transferInstrument_1 = require("./transferInstrument");
const transferInstrumentInfo_1 = require("./transferInstrumentInfo");
const transferInstrumentReference_1 = require("./transferInstrumentReference");
const uKLocalAccountIdentification_1 = require("./uKLocalAccountIdentification");
const uSLocalAccountIdentification_1 = require("./uSLocalAccountIdentification");
const verificationDeadline_1 = require("./verificationDeadline");
const verificationError_1 = require("./verificationError");
const verificationErrorRecursive_1 = require("./verificationErrorRecursive");
const verificationErrors_1 = require("./verificationErrors");
const webData_1 = require("./webData");
const webDataExemption_1 = require("./webDataExemption");
/* tslint:disable:no-unused-variable */
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
let enumsMap = {
    "AULocalAccountIdentification.TypeEnum": aULocalAccountIdentification_1.AULocalAccountIdentification.TypeEnum,
    "AcceptTermsOfServiceResponse.TypeEnum": acceptTermsOfServiceResponse_1.AcceptTermsOfServiceResponse.TypeEnum,
    "AdditionalBankIdentification.TypeEnum": additionalBankIdentification_1.AdditionalBankIdentification.TypeEnum,
    "BusinessLine.CapabilityEnum": businessLine_1.BusinessLine.CapabilityEnum,
    "BusinessLine.ServiceEnum": businessLine_1.BusinessLine.ServiceEnum,
    "BusinessLineInfo.CapabilityEnum": businessLineInfo_1.BusinessLineInfo.CapabilityEnum,
    "BusinessLineInfo.ServiceEnum": businessLineInfo_1.BusinessLineInfo.ServiceEnum,
    "BusinessLineInfoUpdate.CapabilityEnum": businessLineInfoUpdate_1.BusinessLineInfoUpdate.CapabilityEnum,
    "BusinessLineInfoUpdate.ServiceEnum": businessLineInfoUpdate_1.BusinessLineInfoUpdate.ServiceEnum,
    "CALocalAccountIdentification.AccountTypeEnum": cALocalAccountIdentification_1.CALocalAccountIdentification.AccountTypeEnum,
    "CALocalAccountIdentification.TypeEnum": cALocalAccountIdentification_1.CALocalAccountIdentification.TypeEnum,
    "CZLocalAccountIdentification.TypeEnum": cZLocalAccountIdentification_1.CZLocalAccountIdentification.TypeEnum,
    "CalculateTermsOfServiceStatusResponse.TermsOfServiceTypesEnum": calculateTermsOfServiceStatusResponse_1.CalculateTermsOfServiceStatusResponse.TermsOfServiceTypesEnum,
    "CapabilityProblemEntity.TypeEnum": capabilityProblemEntity_1.CapabilityProblemEntity.TypeEnum,
    "CapabilityProblemEntityRecursive.TypeEnum": capabilityProblemEntityRecursive_1.CapabilityProblemEntityRecursive.TypeEnum,
    "CapabilitySettings.FundingSourceEnum": capabilitySettings_1.CapabilitySettings.FundingSourceEnum,
    "CapabilitySettings.IntervalEnum": capabilitySettings_1.CapabilitySettings.IntervalEnum,
    "DKLocalAccountIdentification.TypeEnum": dKLocalAccountIdentification_1.DKLocalAccountIdentification.TypeEnum,
    "Document.TypeEnum": document_1.Document.TypeEnum,
    "DocumentPage.TypeEnum": documentPage_1.DocumentPage.TypeEnum,
    "GeneratePciDescriptionRequest.AdditionalSalesChannelsEnum": generatePciDescriptionRequest_1.GeneratePciDescriptionRequest.AdditionalSalesChannelsEnum,
    "GetTermsOfServiceDocumentRequest.TypeEnum": getTermsOfServiceDocumentRequest_1.GetTermsOfServiceDocumentRequest.TypeEnum,
    "GetTermsOfServiceDocumentResponse.TypeEnum": getTermsOfServiceDocumentResponse_1.GetTermsOfServiceDocumentResponse.TypeEnum,
    "HKLocalAccountIdentification.TypeEnum": hKLocalAccountIdentification_1.HKLocalAccountIdentification.TypeEnum,
    "HULocalAccountIdentification.TypeEnum": hULocalAccountIdentification_1.HULocalAccountIdentification.TypeEnum,
    "IbanAccountIdentification.TypeEnum": ibanAccountIdentification_1.IbanAccountIdentification.TypeEnum,
    "IdentificationData.TypeEnum": identificationData_1.IdentificationData.TypeEnum,
    "LegalEntity.TypeEnum": legalEntity_1.LegalEntity.TypeEnum,
    "LegalEntityAssociation.TypeEnum": legalEntityAssociation_1.LegalEntityAssociation.TypeEnum,
    "LegalEntityCapability.AllowedLevelEnum": legalEntityCapability_1.LegalEntityCapability.AllowedLevelEnum,
    "LegalEntityCapability.RequestedLevelEnum": legalEntityCapability_1.LegalEntityCapability.RequestedLevelEnum,
    "LegalEntityInfo.TypeEnum": legalEntityInfo_1.LegalEntityInfo.TypeEnum,
    "LegalEntityInfoRequiredType.TypeEnum": legalEntityInfoRequiredType_1.LegalEntityInfoRequiredType.TypeEnum,
    "NOLocalAccountIdentification.TypeEnum": nOLocalAccountIdentification_1.NOLocalAccountIdentification.TypeEnum,
    "NZLocalAccountIdentification.TypeEnum": nZLocalAccountIdentification_1.NZLocalAccountIdentification.TypeEnum,
    "NumberAndBicAccountIdentification.TypeEnum": numberAndBicAccountIdentification_1.NumberAndBicAccountIdentification.TypeEnum,
    "Organization.TypeEnum": organization_1.Organization.TypeEnum,
    "Organization.VatAbsenceReasonEnum": organization_1.Organization.VatAbsenceReasonEnum,
    "PLLocalAccountIdentification.TypeEnum": pLLocalAccountIdentification_1.PLLocalAccountIdentification.TypeEnum,
    "SELocalAccountIdentification.TypeEnum": sELocalAccountIdentification_1.SELocalAccountIdentification.TypeEnum,
    "SGLocalAccountIdentification.TypeEnum": sGLocalAccountIdentification_1.SGLocalAccountIdentification.TypeEnum,
    "SoleProprietorship.VatAbsenceReasonEnum": soleProprietorship_1.SoleProprietorship.VatAbsenceReasonEnum,
    "SourceOfFunds.TypeEnum": sourceOfFunds_1.SourceOfFunds.TypeEnum,
    "TaxReportingClassification.BusinessTypeEnum": taxReportingClassification_1.TaxReportingClassification.BusinessTypeEnum,
    "TaxReportingClassification.MainSourceOfIncomeEnum": taxReportingClassification_1.TaxReportingClassification.MainSourceOfIncomeEnum,
    "TaxReportingClassification.TypeEnum": taxReportingClassification_1.TaxReportingClassification.TypeEnum,
    "TermsOfServiceAcceptanceInfo.TypeEnum": termsOfServiceAcceptanceInfo_1.TermsOfServiceAcceptanceInfo.TypeEnum,
    "TransferInstrument.TypeEnum": transferInstrument_1.TransferInstrument.TypeEnum,
    "TransferInstrumentInfo.TypeEnum": transferInstrumentInfo_1.TransferInstrumentInfo.TypeEnum,
    "UKLocalAccountIdentification.TypeEnum": uKLocalAccountIdentification_1.UKLocalAccountIdentification.TypeEnum,
    "USLocalAccountIdentification.AccountTypeEnum": uSLocalAccountIdentification_1.USLocalAccountIdentification.AccountTypeEnum,
    "USLocalAccountIdentification.TypeEnum": uSLocalAccountIdentification_1.USLocalAccountIdentification.TypeEnum,
    "VerificationDeadline.CapabilitiesEnum": verificationDeadline_1.VerificationDeadline.CapabilitiesEnum,
    "VerificationError.CapabilitiesEnum": verificationError_1.VerificationError.CapabilitiesEnum,
    "VerificationError.TypeEnum": verificationError_1.VerificationError.TypeEnum,
    "VerificationErrorRecursive.CapabilitiesEnum": verificationErrorRecursive_1.VerificationErrorRecursive.CapabilitiesEnum,
    "VerificationErrorRecursive.TypeEnum": verificationErrorRecursive_1.VerificationErrorRecursive.TypeEnum,
    "WebDataExemption.ReasonEnum": webDataExemption_1.WebDataExemption.ReasonEnum,
};
let typeMap = {
    "AULocalAccountIdentification": aULocalAccountIdentification_1.AULocalAccountIdentification,
    "AcceptTermsOfServiceRequest": acceptTermsOfServiceRequest_1.AcceptTermsOfServiceRequest,
    "AcceptTermsOfServiceResponse": acceptTermsOfServiceResponse_1.AcceptTermsOfServiceResponse,
    "AdditionalBankIdentification": additionalBankIdentification_1.AdditionalBankIdentification,
    "Address": address_1.Address,
    "Amount": amount_1.Amount,
    "Attachment": attachment_1.Attachment,
    "BankAccountInfo": bankAccountInfo_1.BankAccountInfo,
    "BirthData": birthData_1.BirthData,
    "BusinessLine": businessLine_1.BusinessLine,
    "BusinessLineInfo": businessLineInfo_1.BusinessLineInfo,
    "BusinessLineInfoUpdate": businessLineInfoUpdate_1.BusinessLineInfoUpdate,
    "BusinessLines": businessLines_1.BusinessLines,
    "CALocalAccountIdentification": cALocalAccountIdentification_1.CALocalAccountIdentification,
    "CZLocalAccountIdentification": cZLocalAccountIdentification_1.CZLocalAccountIdentification,
    "CalculateTermsOfServiceStatusResponse": calculateTermsOfServiceStatusResponse_1.CalculateTermsOfServiceStatusResponse,
    "CapabilityProblem": capabilityProblem_1.CapabilityProblem,
    "CapabilityProblemEntity": capabilityProblemEntity_1.CapabilityProblemEntity,
    "CapabilityProblemEntityRecursive": capabilityProblemEntityRecursive_1.CapabilityProblemEntityRecursive,
    "CapabilitySettings": capabilitySettings_1.CapabilitySettings,
    "DKLocalAccountIdentification": dKLocalAccountIdentification_1.DKLocalAccountIdentification,
    "Document": document_1.Document,
    "DocumentPage": documentPage_1.DocumentPage,
    "DocumentReference": documentReference_1.DocumentReference,
    "EntityReference": entityReference_1.EntityReference,
    "GeneratePciDescriptionRequest": generatePciDescriptionRequest_1.GeneratePciDescriptionRequest,
    "GeneratePciDescriptionResponse": generatePciDescriptionResponse_1.GeneratePciDescriptionResponse,
    "GetPciQuestionnaireInfosResponse": getPciQuestionnaireInfosResponse_1.GetPciQuestionnaireInfosResponse,
    "GetPciQuestionnaireResponse": getPciQuestionnaireResponse_1.GetPciQuestionnaireResponse,
    "GetTermsOfServiceAcceptanceInfosResponse": getTermsOfServiceAcceptanceInfosResponse_1.GetTermsOfServiceAcceptanceInfosResponse,
    "GetTermsOfServiceDocumentRequest": getTermsOfServiceDocumentRequest_1.GetTermsOfServiceDocumentRequest,
    "GetTermsOfServiceDocumentResponse": getTermsOfServiceDocumentResponse_1.GetTermsOfServiceDocumentResponse,
    "HKLocalAccountIdentification": hKLocalAccountIdentification_1.HKLocalAccountIdentification,
    "HULocalAccountIdentification": hULocalAccountIdentification_1.HULocalAccountIdentification,
    "IbanAccountIdentification": ibanAccountIdentification_1.IbanAccountIdentification,
    "IdentificationData": identificationData_1.IdentificationData,
    "Individual": individual_1.Individual,
    "LegalEntity": legalEntity_1.LegalEntity,
    "LegalEntityAssociation": legalEntityAssociation_1.LegalEntityAssociation,
    "LegalEntityCapability": legalEntityCapability_1.LegalEntityCapability,
    "LegalEntityInfo": legalEntityInfo_1.LegalEntityInfo,
    "LegalEntityInfoRequiredType": legalEntityInfoRequiredType_1.LegalEntityInfoRequiredType,
    "NOLocalAccountIdentification": nOLocalAccountIdentification_1.NOLocalAccountIdentification,
    "NZLocalAccountIdentification": nZLocalAccountIdentification_1.NZLocalAccountIdentification,
    "Name": name_1.Name,
    "NumberAndBicAccountIdentification": numberAndBicAccountIdentification_1.NumberAndBicAccountIdentification,
    "OnboardingLink": onboardingLink_1.OnboardingLink,
    "OnboardingLinkInfo": onboardingLinkInfo_1.OnboardingLinkInfo,
    "OnboardingTheme": onboardingTheme_1.OnboardingTheme,
    "OnboardingThemes": onboardingThemes_1.OnboardingThemes,
    "Organization": organization_1.Organization,
    "OwnerEntity": ownerEntity_1.OwnerEntity,
    "PLLocalAccountIdentification": pLLocalAccountIdentification_1.PLLocalAccountIdentification,
    "PciDocumentInfo": pciDocumentInfo_1.PciDocumentInfo,
    "PciSigningRequest": pciSigningRequest_1.PciSigningRequest,
    "PciSigningResponse": pciSigningResponse_1.PciSigningResponse,
    "PhoneNumber": phoneNumber_1.PhoneNumber,
    "RemediatingAction": remediatingAction_1.RemediatingAction,
    "SELocalAccountIdentification": sELocalAccountIdentification_1.SELocalAccountIdentification,
    "SGLocalAccountIdentification": sGLocalAccountIdentification_1.SGLocalAccountIdentification,
    "ServiceError": serviceError_1.ServiceError,
    "SoleProprietorship": soleProprietorship_1.SoleProprietorship,
    "SourceOfFunds": sourceOfFunds_1.SourceOfFunds,
    "StockData": stockData_1.StockData,
    "SupportingEntityCapability": supportingEntityCapability_1.SupportingEntityCapability,
    "TaxInformation": taxInformation_1.TaxInformation,
    "TaxReportingClassification": taxReportingClassification_1.TaxReportingClassification,
    "TermsOfServiceAcceptanceInfo": termsOfServiceAcceptanceInfo_1.TermsOfServiceAcceptanceInfo,
    "TransferInstrument": transferInstrument_1.TransferInstrument,
    "TransferInstrumentInfo": transferInstrumentInfo_1.TransferInstrumentInfo,
    "TransferInstrumentReference": transferInstrumentReference_1.TransferInstrumentReference,
    "UKLocalAccountIdentification": uKLocalAccountIdentification_1.UKLocalAccountIdentification,
    "USLocalAccountIdentification": uSLocalAccountIdentification_1.USLocalAccountIdentification,
    "VerificationDeadline": verificationDeadline_1.VerificationDeadline,
    "VerificationError": verificationError_1.VerificationError,
    "VerificationErrorRecursive": verificationErrorRecursive_1.VerificationErrorRecursive,
    "VerificationErrors": verificationErrors_1.VerificationErrors,
    "WebData": webData_1.WebData,
    "WebDataExemption": webDataExemption_1.WebDataExemption,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }
            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType; // use the type given in the discriminator
                    }
                    else {
                        return expectedType; // discriminator did not map to a type
                    }
                }
                else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.serialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return data.toISOString();
        }
        else if (type === "SaleToAcquirerData") {
            const dataString = JSON.stringify(data);
            return Buffer.from(dataString).toString("base64");
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) { // in case we dont know the type
                return data;
            }
            // Get the actual type of this object
            type = this.findCorrectType(data, type);
            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.deserialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) { // is Enum
                return data;
            }
            if (!typeMap[type]) { // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
