// @adyen/api-library/lib/src/typings/legalEntityManagement/organization.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Organization = void 0;
class Organization {
    static getAttributeTypeMap() {
        return Organization.attributeTypeMap;
    }
}
exports.Organization = Organization;
Organization.discriminator = undefined;
Organization.attributeTypeMap = [
    {
        "name": "dateOfIncorporation",
        "baseName": "dateOfIncorporation",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "doingBusinessAs",
        "baseName": "doingBusinessAs",
        "type": "string"
    },
    {
        "name": "email",
        "baseName": "email",
        "type": "string"
    },
    {
        "name": "legalName",
        "baseName": "legalName",
        "type": "string"
    },
    {
        "name": "phone",
        "baseName": "phone",
        "type": "PhoneNumber"
    },
    {
        "name": "principalPlaceOfBusiness",
        "baseName": "principalPlaceOfBusiness",
        "type": "Address"
    },
    {
        "name": "registeredAddress",
        "baseName": "registeredAddress",
        "type": "Address"
    },
    {
        "name": "registrationNumber",
        "baseName": "registrationNumber",
        "type": "string"
    },
    {
        "name": "stockData",
        "baseName": "stockData",
        "type": "StockData"
    },
    {
        "name": "taxInformation",
        "baseName": "taxInformation",
        "type": "Array<TaxInformation>"
    },
    {
        "name": "taxReportingClassification",
        "baseName": "taxReportingClassification",
        "type": "TaxReportingClassification"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Organization.TypeEnum"
    },
    {
        "name": "vatAbsenceReason",
        "baseName": "vatAbsenceReason",
        "type": "Organization.VatAbsenceReasonEnum"
    },
    {
        "name": "vatNumber",
        "baseName": "vatNumber",
        "type": "string"
    },
    {
        "name": "webData",
        "baseName": "webData",
        "type": "WebData"
    }
];
(function (Organization) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AssociationIncorporated"] = "associationIncorporated";
        TypeEnum["GovernmentalOrganization"] = "governmentalOrganization";
        TypeEnum["ListedPublicCompany"] = "listedPublicCompany";
        TypeEnum["NonProfit"] = "nonProfit";
        TypeEnum["PartnershipIncorporated"] = "partnershipIncorporated";
        TypeEnum["PrivateCompany"] = "privateCompany";
    })(TypeEnum = Organization.TypeEnum || (Organization.TypeEnum = {}));
    let VatAbsenceReasonEnum;
    (function (VatAbsenceReasonEnum) {
        VatAbsenceReasonEnum["IndustryExemption"] = "industryExemption";
        VatAbsenceReasonEnum["BelowTaxThreshold"] = "belowTaxThreshold";
    })(VatAbsenceReasonEnum = Organization.VatAbsenceReasonEnum || (Organization.VatAbsenceReasonEnum = {}));
})(Organization = exports.Organization || (exports.Organization = {}));
