// @adyen/api-library/lib/src/typings/legalEntityManagement/soleProprietorship.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SoleProprietorship = void 0;
class SoleProprietorship {
    static getAttributeTypeMap() {
        return SoleProprietorship.attributeTypeMap;
    }
}
exports.SoleProprietorship = SoleProprietorship;
SoleProprietorship.discriminator = undefined;
SoleProprietorship.attributeTypeMap = [
    {
        "name": "countryOfGoverningLaw",
        "baseName": "countryOfGoverningLaw",
        "type": "string"
    },
    {
        "name": "dateOfIncorporation",
        "baseName": "dateOfIncorporation",
        "type": "string"
    },
    {
        "name": "doingBusinessAs",
        "baseName": "doingBusinessAs",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "principalPlaceOfBusiness",
        "baseName": "principalPlaceOfBusiness",
        "type": "Address"
    },
    {
        "name": "registeredAddress",
        "baseName": "registeredAddress",
        "type": "Address"
    },
    {
        "name": "registrationNumber",
        "baseName": "registrationNumber",
        "type": "string"
    },
    {
        "name": "vatAbsenceReason",
        "baseName": "vatAbsenceReason",
        "type": "SoleProprietorship.VatAbsenceReasonEnum"
    },
    {
        "name": "vatNumber",
        "baseName": "vatNumber",
        "type": "string"
    }
];
(function (SoleProprietorship) {
    let VatAbsenceReasonEnum;
    (function (VatAbsenceReasonEnum) {
        VatAbsenceReasonEnum["IndustryExemption"] = "industryExemption";
        VatAbsenceReasonEnum["BelowTaxThreshold"] = "belowTaxThreshold";
    })(VatAbsenceReasonEnum = SoleProprietorship.VatAbsenceReasonEnum || (SoleProprietorship.VatAbsenceReasonEnum = {}));
})(SoleProprietorship = exports.SoleProprietorship || (exports.SoleProprietorship = {}));
