// @adyen/api-library/lib/src/typings/legalEntityManagement/unincorporatedPartnership.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnincorporatedPartnership = void 0;
class UnincorporatedPartnership {
    static getAttributeTypeMap() {
        return UnincorporatedPartnership.attributeTypeMap;
    }
}
exports.UnincorporatedPartnership = UnincorporatedPartnership;
UnincorporatedPartnership.discriminator = undefined;
UnincorporatedPartnership.attributeTypeMap = [
    {
        "name": "countryOfGoverningLaw",
        "baseName": "countryOfGoverningLaw",
        "type": "string"
    },
    {
        "name": "dateOfIncorporation",
        "baseName": "dateOfIncorporation",
        "type": "string"
    },
    {
        "name": "doingBusinessAs",
        "baseName": "doingBusinessAs",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "principalPlaceOfBusiness",
        "baseName": "principalPlaceOfBusiness",
        "type": "Address"
    },
    {
        "name": "registeredAddress",
        "baseName": "registeredAddress",
        "type": "Address"
    },
    {
        "name": "registrationNumber",
        "baseName": "registrationNumber",
        "type": "string"
    },
    {
        "name": "taxInformation",
        "baseName": "taxInformation",
        "type": "Array<TaxInformation>"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "UnincorporatedPartnership.TypeEnum"
    },
    {
        "name": "vatAbsenceReason",
        "baseName": "vatAbsenceReason",
        "type": "UnincorporatedPartnership.VatAbsenceReasonEnum"
    },
    {
        "name": "vatNumber",
        "baseName": "vatNumber",
        "type": "string"
    }
];
(function (UnincorporatedPartnership) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["LimitedPartnership"] = "limitedPartnership";
        TypeEnum["GeneralPartnership"] = "generalPartnership";
        TypeEnum["FamilyPartnership"] = "familyPartnership";
        TypeEnum["CommercialPartnership"] = "commercialPartnership";
        TypeEnum["PublicPartnership"] = "publicPartnership";
        TypeEnum["OtherPartnership"] = "otherPartnership";
        TypeEnum["Gbr"] = "gbr";
        TypeEnum["Gmbh"] = "gmbh";
        TypeEnum["Kgaa"] = "kgaa";
        TypeEnum["Cv"] = "cv";
        TypeEnum["Vof"] = "vof";
        TypeEnum["Maatschap"] = "maatschap";
        TypeEnum["PrivateFundLimitedPartnership"] = "privateFundLimitedPartnership";
        TypeEnum["BusinessTrustEntity"] = "businessTrustEntity";
        TypeEnum["BusinessPartnership"] = "businessPartnership";
    })(TypeEnum = UnincorporatedPartnership.TypeEnum || (UnincorporatedPartnership.TypeEnum = {}));
    let VatAbsenceReasonEnum;
    (function (VatAbsenceReasonEnum) {
        VatAbsenceReasonEnum["IndustryExemption"] = "industryExemption";
        VatAbsenceReasonEnum["BelowTaxThreshold"] = "belowTaxThreshold";
    })(VatAbsenceReasonEnum = UnincorporatedPartnership.VatAbsenceReasonEnum || (UnincorporatedPartnership.VatAbsenceReasonEnum = {}));
})(UnincorporatedPartnership = exports.UnincorporatedPartnership || (exports.UnincorporatedPartnership = {}));
