// @adyen/api-library/lib/src/typings/terminal/accountType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var AccountType;
(function (AccountType) {
    AccountType[AccountType["CardTotals"] = 'CardTotals'] = "CardTotals";
    AccountType[AccountType["Checking"] = 'Checking'] = "Checking";
    AccountType[AccountType["CreditCard"] = 'CreditCard'] = "CreditCard";
    AccountType[AccountType["Default"] = 'Default'] = "Default";
    AccountType[AccountType["EpurseCard"] = 'EpurseCard'] = "EpurseCard";
    AccountType[AccountType["Investment"] = 'Investment'] = "Investment";
    AccountType[AccountType["Savings"] = 'Savings'] = "Savings";
    AccountType[AccountType["Universal"] = 'Universal'] = "Universal";
})(AccountType = exports.AccountType || (exports.AccountType = {}));
