// @adyen/api-library/lib/src/typings/terminal/barcodeType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BarcodeType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var BarcodeType;
(function (BarcodeType) {
    BarcodeType[BarcodeType["Code128"] = 'Code128'] = "Code128";
    BarcodeType[BarcodeType["Code25"] = 'Code25'] = "Code25";
    BarcodeType[BarcodeType["Ean13"] = 'EAN13'] = "Ean13";
    BarcodeType[BarcodeType["Ean8"] = 'EAN8'] = "Ean8";
    BarcodeType[BarcodeType["Pdf417"] = 'PDF417'] = "Pdf417";
    BarcodeType[BarcodeType["Qrcode"] = 'QRCODE'] = "Qrcode";
    BarcodeType[BarcodeType["Upca"] = 'UPCA'] = "Upca";
})(BarcodeType = exports.BarcodeType || (exports.BarcodeType = {}));
