// @adyen/api-library/lib/src/typings/terminal/colorType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ColorType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var ColorType;
(function (ColorType) {
    ColorType[ColorType["Black"] = 'Black'] = "Black";
    ColorType[ColorType["Blue"] = 'Blue'] = "Blue";
    ColorType[ColorType["Cyan"] = 'Cyan'] = "Cyan";
    ColorType[ColorType["Green"] = 'Green'] = "Green";
    ColorType[ColorType["Magenta"] = 'Magenta'] = "Magenta";
    ColorType[ColorType["Red"] = 'Red'] = "Red";
    ColorType[ColorType["White"] = 'White'] = "White";
    ColorType[ColorType["Yellow"] = 'Yellow'] = "Yellow";
})(ColorType = exports.ColorType || (exports.ColorType = {}));
