// @adyen/api-library/lib/src/typings/terminal/contentType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var ContentType;
(function (ContentType) {
    ContentType[ContentType["CtAuthData"] = 'Id-ct-authData'] = "CtAuthData";
    ContentType[ContentType["Data"] = 'Id-data'] = "Data";
    ContentType[ContentType["DigestedData"] = 'Id-digestedData'] = "DigestedData";
    ContentType[ContentType["EncryptedData"] = 'Id-encryptedData'] = "EncryptedData";
    ContentType[ContentType["EnvelopedData"] = 'Id-envelopedData'] = "EnvelopedData";
    ContentType[ContentType["SignedData"] = 'Id-signedData'] = "SignedData";
})(ContentType = exports.ContentType || (exports.ContentType = {}));
