// @adyen/api-library/lib/src/typings/terminal/eventToNotifyType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventToNotifyType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var EventToNotifyType;
(function (EventToNotifyType) {
    EventToNotifyType[EventToNotifyType["Abort"] = 'Abort'] = "Abort";
    EventToNotifyType[EventToNotifyType["BeginMaintenance"] = 'BeginMaintenance'] = "BeginMaintenance";
    EventToNotifyType[EventToNotifyType["CardInserted"] = 'CardInserted'] = "CardInserted";
    EventToNotifyType[EventToNotifyType["CardRemoved"] = 'CardRemoved'] = "CardRemoved";
    EventToNotifyType[EventToNotifyType["Completed"] = 'Completed'] = "Completed";
    EventToNotifyType[EventToNotifyType["CustomerLanguage"] = 'CustomerLanguage'] = "CustomerLanguage";
    EventToNotifyType[EventToNotifyType["EndMaintenance"] = 'EndMaintenance'] = "EndMaintenance";
    EventToNotifyType[EventToNotifyType["Initialised"] = 'Initialised'] = "Initialised";
    EventToNotifyType[EventToNotifyType["KeyPressed"] = 'KeyPressed'] = "KeyPressed";
    EventToNotifyType[EventToNotifyType["OutOfOrder"] = 'OutOfOrder'] = "OutOfOrder";
    EventToNotifyType[EventToNotifyType["Reject"] = 'Reject'] = "Reject";
    EventToNotifyType[EventToNotifyType["SaleAdmin"] = 'SaleAdmin'] = "SaleAdmin";
    EventToNotifyType[EventToNotifyType["SaleWakeUp"] = 'SaleWakeUp'] = "SaleWakeUp";
    EventToNotifyType[EventToNotifyType["SecurityAlarm"] = 'SecurityAlarm'] = "SecurityAlarm";
    EventToNotifyType[EventToNotifyType["Shutdown"] = 'Shutdown'] = "Shutdown";
    EventToNotifyType[EventToNotifyType["StopAssistance"] = 'StopAssistance'] = "StopAssistance";
})(EventToNotifyType = exports.EventToNotifyType || (exports.EventToNotifyType = {}));
