// @adyen/api-library/lib/src/typings/terminal/genericProfileType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenericProfileType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var GenericProfileType;
(function (GenericProfileType) {
    GenericProfileType[GenericProfileType["Basic"] = 'Basic'] = "Basic";
    GenericProfileType[GenericProfileType["Extended"] = 'Extended'] = "Extended";
    GenericProfileType[GenericProfileType["Standard"] = 'Standard'] = "Standard";
})(GenericProfileType = exports.GenericProfileType || (exports.GenericProfileType = {}));
