// @adyen/api-library/lib/src/typings/terminal/identificationType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentificationType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var IdentificationType;
(function (IdentificationType) {
    IdentificationType[IdentificationType["AccountNumber"] = 'AccountNumber'] = "AccountNumber";
    IdentificationType[IdentificationType["BarCode"] = 'BarCode'] = "BarCode";
    IdentificationType[IdentificationType["IsoTrack2"] = 'ISOTrack2'] = "IsoTrack2";
    IdentificationType[IdentificationType["Pan"] = 'PAN'] = "Pan";
    IdentificationType[IdentificationType["PhoneNumber"] = 'PhoneNumber'] = "PhoneNumber";
})(IdentificationType = exports.IdentificationType || (exports.IdentificationType = {}));
