// @adyen/api-library/lib/src/typings/terminal/inputCommandType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InputCommandType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var InputCommandType;
(function (InputCommandType) {
    InputCommandType[InputCommandType["DecimalString"] = 'DecimalString'] = "DecimalString";
    InputCommandType[InputCommandType["DigitString"] = 'DigitString'] = "DigitString";
    InputCommandType[InputCommandType["GetAnyKey"] = 'GetAnyKey'] = "GetAnyKey";
    InputCommandType[InputCommandType["GetConfirmation"] = 'GetConfirmation'] = "GetConfirmation";
    InputCommandType[InputCommandType["GetFunctionKey"] = 'GetFunctionKey'] = "GetFunctionKey";
    InputCommandType[InputCommandType["GetMenuEntry"] = 'GetMenuEntry'] = "GetMenuEntry";
    InputCommandType[InputCommandType["Password"] = 'Password'] = "Password";
    InputCommandType[InputCommandType["SiteManager"] = 'SiteManager'] = "SiteManager";
    InputCommandType[InputCommandType["TextString"] = 'TextString'] = "TextString";
})(InputCommandType = exports.InputCommandType || (exports.InputCommandType = {}));
