// @adyen/api-library/lib/src/typings/terminal/messageCategoryType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageCategoryType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var MessageCategoryType;
(function (MessageCategoryType) {
    MessageCategoryType[MessageCategoryType["Abort"] = 'Abort'] = "Abort";
    MessageCategoryType[MessageCategoryType["Admin"] = 'Admin'] = "Admin";
    MessageCategoryType[MessageCategoryType["BalanceInquiry"] = 'BalanceInquiry'] = "BalanceInquiry";
    MessageCategoryType[MessageCategoryType["Batch"] = 'Batch'] = "Batch";
    MessageCategoryType[MessageCategoryType["CardAcquisition"] = 'CardAcquisition'] = "CardAcquisition";
    MessageCategoryType[MessageCategoryType["CardReaderApdu"] = 'CardReaderAPDU'] = "CardReaderApdu";
    MessageCategoryType[MessageCategoryType["CardReaderInit"] = 'CardReaderInit'] = "CardReaderInit";
    MessageCategoryType[MessageCategoryType["CardReaderPowerOff"] = 'CardReaderPowerOff'] = "CardReaderPowerOff";
    MessageCategoryType[MessageCategoryType["Diagnosis"] = 'Diagnosis'] = "Diagnosis";
    MessageCategoryType[MessageCategoryType["Display"] = 'Display'] = "Display";
    MessageCategoryType[MessageCategoryType["EnableService"] = 'EnableService'] = "EnableService";
    MessageCategoryType[MessageCategoryType["Event"] = 'Event'] = "Event";
    MessageCategoryType[MessageCategoryType["GetTotals"] = 'GetTotals'] = "GetTotals";
    MessageCategoryType[MessageCategoryType["Input"] = 'Input'] = "Input";
    MessageCategoryType[MessageCategoryType["InputUpdate"] = 'InputUpdate'] = "InputUpdate";
    MessageCategoryType[MessageCategoryType["Login"] = 'Login'] = "Login";
    MessageCategoryType[MessageCategoryType["Logout"] = 'Logout'] = "Logout";
    MessageCategoryType[MessageCategoryType["Loyalty"] = 'Loyalty'] = "Loyalty";
    MessageCategoryType[MessageCategoryType["Pin"] = 'PIN'] = "Pin";
    MessageCategoryType[MessageCategoryType["Payment"] = 'Payment'] = "Payment";
    MessageCategoryType[MessageCategoryType["Print"] = 'Print'] = "Print";
    MessageCategoryType[MessageCategoryType["Reconciliation"] = 'Reconciliation'] = "Reconciliation";
    MessageCategoryType[MessageCategoryType["Reversal"] = 'Reversal'] = "Reversal";
    MessageCategoryType[MessageCategoryType["Sound"] = 'Sound'] = "Sound";
    MessageCategoryType[MessageCategoryType["StoredValue"] = 'StoredValue'] = "StoredValue";
    MessageCategoryType[MessageCategoryType["TransactionStatus"] = 'TransactionStatus'] = "TransactionStatus";
    MessageCategoryType[MessageCategoryType["Transmit"] = 'Transmit'] = "Transmit";
})(MessageCategoryType = exports.MessageCategoryType || (exports.MessageCategoryType = {}));
