// @adyen/api-library/lib/src/typings/terminal/outputBarcode.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutputBarcode = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
class OutputBarcode {
    static getAttributeTypeMap() {
        return OutputBarcode.attributeTypeMap;
    }
}
exports.OutputBarcode = OutputBarcode;
OutputBarcode.discriminator = undefined;
OutputBarcode.attributeTypeMap = [
    {
        "name": "BarcodeType",
        "baseName": "BarcodeType",
        "type": "OutputBarcode.BarcodeTypeEnum"
    },
    {
        "name": "BarcodeValue",
        "baseName": "BarcodeValue",
        "type": "string"
    }
];
(function (OutputBarcode) {
    let BarcodeTypeEnum;
    (function (BarcodeTypeEnum) {
        BarcodeTypeEnum[BarcodeTypeEnum["Code128"] = 'Code128'] = "Code128";
        BarcodeTypeEnum[BarcodeTypeEnum["Code25"] = 'Code25'] = "Code25";
        BarcodeTypeEnum[BarcodeTypeEnum["Ean13"] = 'EAN13'] = "Ean13";
        BarcodeTypeEnum[BarcodeTypeEnum["Ean8"] = 'EAN8'] = "Ean8";
        BarcodeTypeEnum[BarcodeTypeEnum["Pdf417"] = 'PDF417'] = "Pdf417";
        BarcodeTypeEnum[BarcodeTypeEnum["Qrcode"] = 'QRCode'] = "Qrcode";
        BarcodeTypeEnum[BarcodeTypeEnum["Upca"] = 'UPCA'] = "Upca";
    })(BarcodeTypeEnum = OutputBarcode.BarcodeTypeEnum || (OutputBarcode.BarcodeTypeEnum = {}));
})(OutputBarcode = exports.OutputBarcode || (exports.OutputBarcode = {}));
