// @adyen/api-library/lib/src/typings/terminal/pINFormatType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PINFormatType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var PINFormatType;
(function (PINFormatType) {
    PINFormatType[PINFormatType["Iso0"] = 'ISO0'] = "Iso0";
    PINFormatType[PINFormatType["Iso1"] = 'ISO1'] = "Iso1";
    PINFormatType[PINFormatType["Iso2"] = 'ISO2'] = "Iso2";
    PINFormatType[PINFormatType["Iso3"] = 'ISO3'] = "Iso3";
})(PINFormatType = exports.PINFormatType || (exports.PINFormatType = {}));
