// @adyen/api-library/lib/src/typings/balancePlatform/accountSupportingEntityCapability.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountSupportingEntityCapability = void 0;
class AccountSupportingEntityCapability {
    static getAttributeTypeMap() {
        return AccountSupportingEntityCapability.attributeTypeMap;
    }
}
exports.AccountSupportingEntityCapability = AccountSupportingEntityCapability;
AccountSupportingEntityCapability.discriminator = undefined;
AccountSupportingEntityCapability.attributeTypeMap = [
    {
        "name": "allowed",
        "baseName": "allowed",
        "type": "boolean"
    },
    {
        "name": "allowedLevel",
        "baseName": "allowedLevel",
        "type": "AccountSupportingEntityCapability.AllowedLevelEnum"
    },
    {
        "name": "enabled",
        "baseName": "enabled",
        "type": "boolean"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "requested",
        "baseName": "requested",
        "type": "boolean"
    },
    {
        "name": "requestedLevel",
        "baseName": "requestedLevel",
        "type": "AccountSupportingEntityCapability.RequestedLevelEnum"
    },
    {
        "name": "verificationStatus",
        "baseName": "verificationStatus",
        "type": "AccountSupportingEntityCapability.VerificationStatusEnum"
    }
];
(function (AccountSupportingEntityCapability) {
    let AllowedLevelEnum;
    (function (AllowedLevelEnum) {
        AllowedLevelEnum["High"] = "high";
        AllowedLevelEnum["Low"] = "low";
        AllowedLevelEnum["Medium"] = "medium";
        AllowedLevelEnum["NotApplicable"] = "notApplicable";
    })(AllowedLevelEnum = AccountSupportingEntityCapability.AllowedLevelEnum || (AccountSupportingEntityCapability.AllowedLevelEnum = {}));
    let RequestedLevelEnum;
    (function (RequestedLevelEnum) {
        RequestedLevelEnum["High"] = "high";
        RequestedLevelEnum["Low"] = "low";
        RequestedLevelEnum["Medium"] = "medium";
        RequestedLevelEnum["NotApplicable"] = "notApplicable";
    })(RequestedLevelEnum = AccountSupportingEntityCapability.RequestedLevelEnum || (AccountSupportingEntityCapability.RequestedLevelEnum = {}));
    let VerificationStatusEnum;
    (function (VerificationStatusEnum) {
        VerificationStatusEnum["Invalid"] = "invalid";
        VerificationStatusEnum["Pending"] = "pending";
        VerificationStatusEnum["Rejected"] = "rejected";
        VerificationStatusEnum["Valid"] = "valid";
    })(VerificationStatusEnum = AccountSupportingEntityCapability.VerificationStatusEnum || (AccountSupportingEntityCapability.VerificationStatusEnum = {}));
})(AccountSupportingEntityCapability = exports.AccountSupportingEntityCapability || (exports.AccountSupportingEntityCapability = {}));
