// @adyen/api-library/lib/src/typings/balancePlatform/addressRequirement.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddressRequirement = void 0;
class AddressRequirement {
    static getAttributeTypeMap() {
        return AddressRequirement.attributeTypeMap;
    }
}
exports.AddressRequirement = AddressRequirement;
AddressRequirement.discriminator = undefined;
AddressRequirement.attributeTypeMap = [
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "requiredAddressFields",
        "baseName": "requiredAddressFields",
        "type": "Array<AddressRequirement.RequiredAddressFieldsEnum>"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "AddressRequirement.TypeEnum"
    }
];
(function (AddressRequirement) {
    let RequiredAddressFieldsEnum;
    (function (RequiredAddressFieldsEnum) {
        RequiredAddressFieldsEnum["City"] = "city";
        RequiredAddressFieldsEnum["Country"] = "country";
        RequiredAddressFieldsEnum["Line1"] = "line1";
        RequiredAddressFieldsEnum["PostalCode"] = "postalCode";
        RequiredAddressFieldsEnum["StateOrProvince"] = "stateOrProvince";
    })(RequiredAddressFieldsEnum = AddressRequirement.RequiredAddressFieldsEnum || (AddressRequirement.RequiredAddressFieldsEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AddressRequirement"] = "addressRequirement";
    })(TypeEnum = AddressRequirement.TypeEnum || (AddressRequirement.TypeEnum = {}));
})(AddressRequirement = exports.AddressRequirement || (exports.AddressRequirement = {}));
