// @adyen/api-library/lib/src/typings/balancePlatform/cardInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardInfo = void 0;
class CardInfo {
    static getAttributeTypeMap() {
        return CardInfo.attributeTypeMap;
    }
}
exports.CardInfo = CardInfo;
CardInfo.discriminator = undefined;
CardInfo.attributeTypeMap = [
    {
        "name": "authentication",
        "baseName": "authentication",
        "type": "Authentication"
    },
    {
        "name": "brand",
        "baseName": "brand",
        "type": "string"
    },
    {
        "name": "brandVariant",
        "baseName": "brandVariant",
        "type": "string"
    },
    {
        "name": "cardholderName",
        "baseName": "cardholderName",
        "type": "string"
    },
    {
        "name": "configuration",
        "baseName": "configuration",
        "type": "CardConfiguration"
    },
    {
        "name": "deliveryContact",
        "baseName": "deliveryContact",
        "type": "DeliveryContact"
    },
    {
        "name": "formFactor",
        "baseName": "formFactor",
        "type": "CardInfo.FormFactorEnum"
    },
    {
        "name": "threeDSecure",
        "baseName": "threeDSecure",
        "type": "string"
    }
];
(function (CardInfo) {
    let FormFactorEnum;
    (function (FormFactorEnum) {
        FormFactorEnum["Physical"] = "physical";
        FormFactorEnum["Unknown"] = "unknown";
        FormFactorEnum["Virtual"] = "virtual";
    })(FormFactorEnum = CardInfo.FormFactorEnum || (CardInfo.FormFactorEnum = {}));
})(CardInfo = exports.CardInfo || (exports.CardInfo = {}));
