// @adyen/api-library/lib/src/typings/balancePlatform/sweepSchedule.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SweepSchedule = void 0;
class SweepSchedule {
    static getAttributeTypeMap() {
        return SweepSchedule.attributeTypeMap;
    }
}
exports.SweepSchedule = SweepSchedule;
SweepSchedule.discriminator = undefined;
SweepSchedule.attributeTypeMap = [
    {
        "name": "cronExpression",
        "baseName": "cronExpression",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "SweepSchedule.TypeEnum"
    }
];
(function (SweepSchedule) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Daily"] = "daily";
        TypeEnum["Weekly"] = "weekly";
        TypeEnum["Monthly"] = "monthly";
        TypeEnum["Balance"] = "balance";
        TypeEnum["Cron"] = "cron";
    })(TypeEnum = SweepSchedule.TypeEnum || (SweepSchedule.TypeEnum = {}));
})(SweepSchedule = exports.SweepSchedule || (exports.SweepSchedule = {}));
