// @adyen/api-library/lib/src/typings/balancePlatform/transactionRuleInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionRuleInfo = void 0;
class TransactionRuleInfo {
    static getAttributeTypeMap() {
        return TransactionRuleInfo.attributeTypeMap;
    }
}
exports.TransactionRuleInfo = TransactionRuleInfo;
TransactionRuleInfo.discriminator = undefined;
TransactionRuleInfo.attributeTypeMap = [
    {
        "name": "aggregationLevel",
        "baseName": "aggregationLevel",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "endDate",
        "baseName": "endDate",
        "type": "string"
    },
    {
        "name": "entityKey",
        "baseName": "entityKey",
        "type": "TransactionRuleEntityKey"
    },
    {
        "name": "interval",
        "baseName": "interval",
        "type": "TransactionRuleInterval"
    },
    {
        "name": "outcomeType",
        "baseName": "outcomeType",
        "type": "TransactionRuleInfo.OutcomeTypeEnum"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "requestType",
        "baseName": "requestType",
        "type": "TransactionRuleInfo.RequestTypeEnum"
    },
    {
        "name": "ruleRestrictions",
        "baseName": "ruleRestrictions",
        "type": "TransactionRuleRestrictions"
    },
    {
        "name": "score",
        "baseName": "score",
        "type": "number"
    },
    {
        "name": "startDate",
        "baseName": "startDate",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "TransactionRuleInfo.StatusEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransactionRuleInfo.TypeEnum"
    }
];
(function (TransactionRuleInfo) {
    let OutcomeTypeEnum;
    (function (OutcomeTypeEnum) {
        OutcomeTypeEnum["EnforceSca"] = "enforceSCA";
        OutcomeTypeEnum["HardBlock"] = "hardBlock";
        OutcomeTypeEnum["ScoreBased"] = "scoreBased";
    })(OutcomeTypeEnum = TransactionRuleInfo.OutcomeTypeEnum || (TransactionRuleInfo.OutcomeTypeEnum = {}));
    let RequestTypeEnum;
    (function (RequestTypeEnum) {
        RequestTypeEnum["Authentication"] = "authentication";
        RequestTypeEnum["Authorization"] = "authorization";
        RequestTypeEnum["BankTransfer"] = "bankTransfer";
        RequestTypeEnum["Tokenization"] = "tokenization";
    })(RequestTypeEnum = TransactionRuleInfo.RequestTypeEnum || (TransactionRuleInfo.RequestTypeEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Inactive"] = "inactive";
    })(StatusEnum = TransactionRuleInfo.StatusEnum || (TransactionRuleInfo.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AllowList"] = "allowList";
        TypeEnum["BlockList"] = "blockList";
        TypeEnum["MaxUsage"] = "maxUsage";
        TypeEnum["Velocity"] = "velocity";
    })(TypeEnum = TransactionRuleInfo.TypeEnum || (TransactionRuleInfo.TypeEnum = {}));
})(TransactionRuleInfo = exports.TransactionRuleInfo || (exports.TransactionRuleInfo = {}));
