// @adyen/api-library/lib/src/typings/balancePlatform/updatePaymentInstrument.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatePaymentInstrument = void 0;
class UpdatePaymentInstrument {
    static getAttributeTypeMap() {
        return UpdatePaymentInstrument.attributeTypeMap;
    }
}
exports.UpdatePaymentInstrument = UpdatePaymentInstrument;
UpdatePaymentInstrument.discriminator = undefined;
UpdatePaymentInstrument.attributeTypeMap = [
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "bankAccount",
        "baseName": "bankAccount",
        "type": "IbanAccountIdentification | USLocalAccountIdentification"
    },
    {
        "name": "card",
        "baseName": "card",
        "type": "Card"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "issuingCountryCode",
        "baseName": "issuingCountryCode",
        "type": "string"
    },
    {
        "name": "paymentInstrumentGroupId",
        "baseName": "paymentInstrumentGroupId",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "UpdatePaymentInstrument.StatusEnum"
    },
    {
        "name": "statusComment",
        "baseName": "statusComment",
        "type": "string"
    },
    {
        "name": "statusReason",
        "baseName": "statusReason",
        "type": "UpdatePaymentInstrument.StatusReasonEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "UpdatePaymentInstrument.TypeEnum"
    }
];
(function (UpdatePaymentInstrument) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Closed"] = "closed";
        StatusEnum["Inactive"] = "inactive";
        StatusEnum["Suspended"] = "suspended";
    })(StatusEnum = UpdatePaymentInstrument.StatusEnum || (UpdatePaymentInstrument.StatusEnum = {}));
    let StatusReasonEnum;
    (function (StatusReasonEnum) {
        StatusReasonEnum["AccountClosure"] = "accountClosure";
        StatusReasonEnum["Damaged"] = "damaged";
        StatusReasonEnum["EndOfLife"] = "endOfLife";
        StatusReasonEnum["Expired"] = "expired";
        StatusReasonEnum["Lost"] = "lost";
        StatusReasonEnum["Other"] = "other";
        StatusReasonEnum["Stolen"] = "stolen";
        StatusReasonEnum["SuspectedFraud"] = "suspectedFraud";
        StatusReasonEnum["TransactionRule"] = "transactionRule";
    })(StatusReasonEnum = UpdatePaymentInstrument.StatusReasonEnum || (UpdatePaymentInstrument.StatusReasonEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BankAccount"] = "bankAccount";
        TypeEnum["Card"] = "card";
    })(TypeEnum = UpdatePaymentInstrument.TypeEnum || (UpdatePaymentInstrument.TypeEnum = {}));
})(UpdatePaymentInstrument = exports.UpdatePaymentInstrument || (exports.UpdatePaymentInstrument = {}));
