// @adyen/api-library/lib/src/typings/balancePlatform/updateSweepConfigurationV2.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSweepConfigurationV2 = void 0;
class UpdateSweepConfigurationV2 {
    static getAttributeTypeMap() {
        return UpdateSweepConfigurationV2.attributeTypeMap;
    }
}
exports.UpdateSweepConfigurationV2 = UpdateSweepConfigurationV2;
UpdateSweepConfigurationV2.discriminator = undefined;
UpdateSweepConfigurationV2.attributeTypeMap = [
    {
        "name": "category",
        "baseName": "category",
        "type": "UpdateSweepConfigurationV2.CategoryEnum"
    },
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "SweepCounterparty"
    },
    {
        "name": "currency",
        "baseName": "currency",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "priorities",
        "baseName": "priorities",
        "type": "Array<UpdateSweepConfigurationV2.PrioritiesEnum>"
    },
    {
        "name": "reason",
        "baseName": "reason",
        "type": "UpdateSweepConfigurationV2.ReasonEnum"
    },
    {
        "name": "schedule",
        "baseName": "schedule",
        "type": "SweepSchedule"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "UpdateSweepConfigurationV2.StatusEnum"
    },
    {
        "name": "sweepAmount",
        "baseName": "sweepAmount",
        "type": "Amount"
    },
    {
        "name": "targetAmount",
        "baseName": "targetAmount",
        "type": "Amount"
    },
    {
        "name": "triggerAmount",
        "baseName": "triggerAmount",
        "type": "Amount"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "UpdateSweepConfigurationV2.TypeEnum"
    }
];
(function (UpdateSweepConfigurationV2) {
    let CategoryEnum;
    (function (CategoryEnum) {
        CategoryEnum["Bank"] = "bank";
        CategoryEnum["Internal"] = "internal";
        CategoryEnum["PlatformPayment"] = "platformPayment";
    })(CategoryEnum = UpdateSweepConfigurationV2.CategoryEnum || (UpdateSweepConfigurationV2.CategoryEnum = {}));
    let PrioritiesEnum;
    (function (PrioritiesEnum) {
        PrioritiesEnum["CrossBorder"] = "crossBorder";
        PrioritiesEnum["Fast"] = "fast";
        PrioritiesEnum["Instant"] = "instant";
        PrioritiesEnum["Internal"] = "internal";
        PrioritiesEnum["Regular"] = "regular";
        PrioritiesEnum["Wire"] = "wire";
    })(PrioritiesEnum = UpdateSweepConfigurationV2.PrioritiesEnum || (UpdateSweepConfigurationV2.PrioritiesEnum = {}));
    let ReasonEnum;
    (function (ReasonEnum) {
        ReasonEnum["AmountLimitExceeded"] = "amountLimitExceeded";
        ReasonEnum["Approved"] = "approved";
        ReasonEnum["BalanceAccountTemporarilyBlockedByTransactionRule"] = "balanceAccountTemporarilyBlockedByTransactionRule";
        ReasonEnum["CounterpartyAccountBlocked"] = "counterpartyAccountBlocked";
        ReasonEnum["CounterpartyAccountClosed"] = "counterpartyAccountClosed";
        ReasonEnum["CounterpartyAccountNotFound"] = "counterpartyAccountNotFound";
        ReasonEnum["CounterpartyAddressRequired"] = "counterpartyAddressRequired";
        ReasonEnum["CounterpartyBankTimedOut"] = "counterpartyBankTimedOut";
        ReasonEnum["CounterpartyBankUnavailable"] = "counterpartyBankUnavailable";
        ReasonEnum["DeclinedByTransactionRule"] = "declinedByTransactionRule";
        ReasonEnum["Error"] = "error";
        ReasonEnum["NotEnoughBalance"] = "notEnoughBalance";
        ReasonEnum["RefusedByCounterpartyBank"] = "refusedByCounterpartyBank";
        ReasonEnum["RouteNotFound"] = "routeNotFound";
        ReasonEnum["ScaFailed"] = "scaFailed";
        ReasonEnum["Unknown"] = "unknown";
    })(ReasonEnum = UpdateSweepConfigurationV2.ReasonEnum || (UpdateSweepConfigurationV2.ReasonEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Inactive"] = "inactive";
    })(StatusEnum = UpdateSweepConfigurationV2.StatusEnum || (UpdateSweepConfigurationV2.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Pull"] = "pull";
        TypeEnum["Push"] = "push";
    })(TypeEnum = UpdateSweepConfigurationV2.TypeEnum || (UpdateSweepConfigurationV2.TypeEnum = {}));
})(UpdateSweepConfigurationV2 = exports.UpdateSweepConfigurationV2 || (exports.UpdateSweepConfigurationV2 = {}));
