// @adyen/api-library/lib/src/typings/balancePlatform/updateNetworkTokenRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateNetworkTokenRequest = void 0;
class UpdateNetworkTokenRequest {
    static getAttributeTypeMap() {
        return UpdateNetworkTokenRequest.attributeTypeMap;
    }
}
exports.UpdateNetworkTokenRequest = UpdateNetworkTokenRequest;
UpdateNetworkTokenRequest.discriminator = undefined;
UpdateNetworkTokenRequest.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "UpdateNetworkTokenRequest.StatusEnum"
    }
];
(function (UpdateNetworkTokenRequest) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Suspended"] = "suspended";
        StatusEnum["Closed"] = "closed";
    })(StatusEnum = UpdateNetworkTokenRequest.StatusEnum || (UpdateNetworkTokenRequest.StatusEnum = {}));
})(UpdateNetworkTokenRequest = exports.UpdateNetworkTokenRequest || (exports.UpdateNetworkTokenRequest = {}));
