// @adyen/api-library/lib/src/typings/balancePlatform/verificationDeadline.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerificationDeadline = void 0;
class VerificationDeadline {
    static getAttributeTypeMap() {
        return VerificationDeadline.attributeTypeMap;
    }
}
exports.VerificationDeadline = VerificationDeadline;
VerificationDeadline.discriminator = undefined;
VerificationDeadline.attributeTypeMap = [
    {
        "name": "capabilities",
        "baseName": "capabilities",
        "type": "Array<VerificationDeadline.CapabilitiesEnum>"
    },
    {
        "name": "entityIds",
        "baseName": "entityIds",
        "type": "Array<string>"
    },
    {
        "name": "expiresAt",
        "baseName": "expiresAt",
        "type": "Date"
    }
];
(function (VerificationDeadline) {
    let CapabilitiesEnum;
    (function (CapabilitiesEnum) {
        CapabilitiesEnum["AcceptExternalFunding"] = "acceptExternalFunding";
        CapabilitiesEnum["AcceptPspFunding"] = "acceptPspFunding";
        CapabilitiesEnum["AcceptTransactionInRestrictedCountries"] = "acceptTransactionInRestrictedCountries";
        CapabilitiesEnum["AcceptTransactionInRestrictedCountriesCommercial"] = "acceptTransactionInRestrictedCountriesCommercial";
        CapabilitiesEnum["AcceptTransactionInRestrictedCountriesConsumer"] = "acceptTransactionInRestrictedCountriesConsumer";
        CapabilitiesEnum["AcceptTransactionInRestrictedIndustries"] = "acceptTransactionInRestrictedIndustries";
        CapabilitiesEnum["AcceptTransactionInRestrictedIndustriesCommercial"] = "acceptTransactionInRestrictedIndustriesCommercial";
        CapabilitiesEnum["AcceptTransactionInRestrictedIndustriesConsumer"] = "acceptTransactionInRestrictedIndustriesConsumer";
        CapabilitiesEnum["Acquiring"] = "acquiring";
        CapabilitiesEnum["AtmWithdrawal"] = "atmWithdrawal";
        CapabilitiesEnum["AtmWithdrawalCommercial"] = "atmWithdrawalCommercial";
        CapabilitiesEnum["AtmWithdrawalConsumer"] = "atmWithdrawalConsumer";
        CapabilitiesEnum["AtmWithdrawalInRestrictedCountries"] = "atmWithdrawalInRestrictedCountries";
        CapabilitiesEnum["AtmWithdrawalInRestrictedCountriesCommercial"] = "atmWithdrawalInRestrictedCountriesCommercial";
        CapabilitiesEnum["AtmWithdrawalInRestrictedCountriesConsumer"] = "atmWithdrawalInRestrictedCountriesConsumer";
        CapabilitiesEnum["AuthorisedPaymentInstrumentUser"] = "authorisedPaymentInstrumentUser";
        CapabilitiesEnum["GetGrantOffers"] = "getGrantOffers";
        CapabilitiesEnum["IssueBankAccount"] = "issueBankAccount";
        CapabilitiesEnum["IssueCard"] = "issueCard";
        CapabilitiesEnum["IssueCardCommercial"] = "issueCardCommercial";
        CapabilitiesEnum["IssueCardConsumer"] = "issueCardConsumer";
        CapabilitiesEnum["LocalAcceptance"] = "localAcceptance";
        CapabilitiesEnum["Payout"] = "payout";
        CapabilitiesEnum["PayoutToTransferInstrument"] = "payoutToTransferInstrument";
        CapabilitiesEnum["Processing"] = "processing";
        CapabilitiesEnum["ReceiveFromBalanceAccount"] = "receiveFromBalanceAccount";
        CapabilitiesEnum["ReceiveFromPlatformPayments"] = "receiveFromPlatformPayments";
        CapabilitiesEnum["ReceiveFromThirdParty"] = "receiveFromThirdParty";
        CapabilitiesEnum["ReceiveFromTransferInstrument"] = "receiveFromTransferInstrument";
        CapabilitiesEnum["ReceiveGrants"] = "receiveGrants";
        CapabilitiesEnum["ReceivePayments"] = "receivePayments";
        CapabilitiesEnum["SendToBalanceAccount"] = "sendToBalanceAccount";
        CapabilitiesEnum["SendToThirdParty"] = "sendToThirdParty";
        CapabilitiesEnum["SendToTransferInstrument"] = "sendToTransferInstrument";
        CapabilitiesEnum["ThirdPartyFunding"] = "thirdPartyFunding";
        CapabilitiesEnum["UseCard"] = "useCard";
        CapabilitiesEnum["UseCardCommercial"] = "useCardCommercial";
        CapabilitiesEnum["UseCardConsumer"] = "useCardConsumer";
        CapabilitiesEnum["UseCardInRestrictedCountries"] = "useCardInRestrictedCountries";
        CapabilitiesEnum["UseCardInRestrictedCountriesCommercial"] = "useCardInRestrictedCountriesCommercial";
        CapabilitiesEnum["UseCardInRestrictedCountriesConsumer"] = "useCardInRestrictedCountriesConsumer";
        CapabilitiesEnum["UseCardInRestrictedIndustries"] = "useCardInRestrictedIndustries";
        CapabilitiesEnum["UseCardInRestrictedIndustriesCommercial"] = "useCardInRestrictedIndustriesCommercial";
        CapabilitiesEnum["UseCardInRestrictedIndustriesConsumer"] = "useCardInRestrictedIndustriesConsumer";
        CapabilitiesEnum["WithdrawFromAtm"] = "withdrawFromAtm";
        CapabilitiesEnum["WithdrawFromAtmCommercial"] = "withdrawFromAtmCommercial";
        CapabilitiesEnum["WithdrawFromAtmConsumer"] = "withdrawFromAtmConsumer";
        CapabilitiesEnum["WithdrawFromAtmInRestrictedCountries"] = "withdrawFromAtmInRestrictedCountries";
        CapabilitiesEnum["WithdrawFromAtmInRestrictedCountriesCommercial"] = "withdrawFromAtmInRestrictedCountriesCommercial";
        CapabilitiesEnum["WithdrawFromAtmInRestrictedCountriesConsumer"] = "withdrawFromAtmInRestrictedCountriesConsumer";
    })(CapabilitiesEnum = VerificationDeadline.CapabilitiesEnum || (VerificationDeadline.CapabilitiesEnum = {}));
})(VerificationDeadline = exports.VerificationDeadline || (exports.VerificationDeadline = {}));
