// @adyen/api-library/lib/src/typings/transfers/bankCategoryData.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankCategoryData = void 0;
class BankCategoryData {
    static getAttributeTypeMap() {
        return BankCategoryData.attributeTypeMap;
    }
}
exports.BankCategoryData = BankCategoryData;
BankCategoryData.discriminator = undefined;
BankCategoryData.attributeTypeMap = [
    {
        "name": "priority",
        "baseName": "priority",
        "type": "BankCategoryData.PriorityEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "BankCategoryData.TypeEnum"
    }
];
(function (BankCategoryData) {
    let PriorityEnum;
    (function (PriorityEnum) {
        PriorityEnum["CrossBorder"] = "crossBorder";
        PriorityEnum["Fast"] = "fast";
        PriorityEnum["Instant"] = "instant";
        PriorityEnum["Internal"] = "internal";
        PriorityEnum["Regular"] = "regular";
        PriorityEnum["Wire"] = "wire";
    })(PriorityEnum = BankCategoryData.PriorityEnum || (BankCategoryData.PriorityEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Bank"] = "bank";
    })(TypeEnum = BankCategoryData.TypeEnum || (BankCategoryData.TypeEnum = {}));
})(BankCategoryData = exports.BankCategoryData || (exports.BankCategoryData = {}));
