// @adyen/api-library/lib/src/typings/transfers/capitalGrant.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapitalGrant = void 0;
class CapitalGrant {
    static getAttributeTypeMap() {
        return CapitalGrant.attributeTypeMap;
    }
}
exports.CapitalGrant = CapitalGrant;
CapitalGrant.discriminator = undefined;
CapitalGrant.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "balances",
        "baseName": "balances",
        "type": "CapitalBalance"
    },
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "Counterparty"
    },
    {
        "name": "fee",
        "baseName": "fee",
        "type": "Fee"
    },
    {
        "name": "grantAccountId",
        "baseName": "grantAccountId",
        "type": "string"
    },
    {
        "name": "grantOfferId",
        "baseName": "grantOfferId",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "repayment",
        "baseName": "repayment",
        "type": "Repayment"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "CapitalGrant.StatusEnum"
    }
];
(function (CapitalGrant) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Pending"] = "Pending";
        StatusEnum["Active"] = "Active";
        StatusEnum["Repaid"] = "Repaid";
    })(StatusEnum = CapitalGrant.StatusEnum || (CapitalGrant.StatusEnum = {}));
})(CapitalGrant = exports.CapitalGrant || (exports.CapitalGrant = {}));
