// @adyen/api-library/lib/src/typings/transfers/platformPayment.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlatformPayment = void 0;
class PlatformPayment {
    static getAttributeTypeMap() {
        return PlatformPayment.attributeTypeMap;
    }
}
exports.PlatformPayment = PlatformPayment;
PlatformPayment.discriminator = undefined;
PlatformPayment.attributeTypeMap = [
    {
        "name": "modificationMerchantReference",
        "baseName": "modificationMerchantReference",
        "type": "string"
    },
    {
        "name": "modificationPspReference",
        "baseName": "modificationPspReference",
        "type": "string"
    },
    {
        "name": "paymentMerchantReference",
        "baseName": "paymentMerchantReference",
        "type": "string"
    },
    {
        "name": "platformPaymentType",
        "baseName": "platformPaymentType",
        "type": "PlatformPayment.PlatformPaymentTypeEnum"
    },
    {
        "name": "pspPaymentReference",
        "baseName": "pspPaymentReference",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PlatformPayment.TypeEnum"
    }
];
(function (PlatformPayment) {
    let PlatformPaymentTypeEnum;
    (function (PlatformPaymentTypeEnum) {
        PlatformPaymentTypeEnum["AcquiringFees"] = "AcquiringFees";
        PlatformPaymentTypeEnum["AdyenCommission"] = "AdyenCommission";
        PlatformPaymentTypeEnum["AdyenFees"] = "AdyenFees";
        PlatformPaymentTypeEnum["AdyenMarkup"] = "AdyenMarkup";
        PlatformPaymentTypeEnum["BalanceAccount"] = "BalanceAccount";
        PlatformPaymentTypeEnum["Commission"] = "Commission";
        PlatformPaymentTypeEnum["Default"] = "Default";
        PlatformPaymentTypeEnum["Interchange"] = "Interchange";
        PlatformPaymentTypeEnum["PaymentFee"] = "PaymentFee";
        PlatformPaymentTypeEnum["Remainder"] = "Remainder";
        PlatformPaymentTypeEnum["SchemeFee"] = "SchemeFee";
        PlatformPaymentTypeEnum["TopUp"] = "TopUp";
        PlatformPaymentTypeEnum["Vat"] = "VAT";
    })(PlatformPaymentTypeEnum = PlatformPayment.PlatformPaymentTypeEnum || (PlatformPayment.PlatformPaymentTypeEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["PlatformPayment"] = "platformPayment";
    })(TypeEnum = PlatformPayment.TypeEnum || (PlatformPayment.TypeEnum = {}));
})(PlatformPayment = exports.PlatformPayment || (exports.PlatformPayment = {}));
