// @adyen/api-library/lib/src/typings/transfers/returnTransferResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReturnTransferResponse = void 0;
class ReturnTransferResponse {
    static getAttributeTypeMap() {
        return ReturnTransferResponse.attributeTypeMap;
    }
}
exports.ReturnTransferResponse = ReturnTransferResponse;
ReturnTransferResponse.discriminator = undefined;
ReturnTransferResponse.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "ReturnTransferResponse.StatusEnum"
    },
    {
        "name": "transferId",
        "baseName": "transferId",
        "type": "string"
    }
];
(function (ReturnTransferResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Authorised"] = "Authorised";
        StatusEnum["Declined"] = "Declined";
    })(StatusEnum = ReturnTransferResponse.StatusEnum || (ReturnTransferResponse.StatusEnum = {}));
})(ReturnTransferResponse = exports.ReturnTransferResponse || (exports.ReturnTransferResponse = {}));
