// @adyen/api-library/lib/src/services/binLookupApi.js
"use strict";
/*
 * The version of the OpenAPI document: v54
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BinLookupAPI = void 0;
const getJsonResponse_1 = __importDefault(require("../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../service"));
const resource_1 = __importDefault(require("./resource"));
const models_1 = require("../typings/binLookup/models");
class BinLookupAPI extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://pal-test.adyen.com/pal/servlet/BinLookup/v54";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Check if 3D Secure is available
    * @param threeDSAvailabilityRequest {@link ThreeDSAvailabilityRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ThreeDSAvailabilityResponse }
    */
    async get3dsAvailability(threeDSAvailabilityRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/get3dsAvailability`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(threeDSAvailabilityRequest, "ThreeDSAvailabilityRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "ThreeDSAvailabilityResponse");
    }
    /**
    * @summary Get a fees cost estimate
    * @param costEstimateRequest {@link CostEstimateRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link CostEstimateResponse }
    */
    async getCostEstimate(costEstimateRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/getCostEstimate`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(costEstimateRequest, "CostEstimateRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "CostEstimateResponse");
    }
}
exports.BinLookupAPI = BinLookupAPI;
exports.default = BinLookupAPI;
