// @adyen/api-library/lib/src/typings/checkout/donationPaymentResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DonationPaymentResponse = void 0;
class DonationPaymentResponse {
    static getAttributeTypeMap() {
        return DonationPaymentResponse.attributeTypeMap;
    }
}
exports.DonationPaymentResponse = DonationPaymentResponse;
DonationPaymentResponse.discriminator = undefined;
DonationPaymentResponse.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "donationAccount",
        "baseName": "donationAccount",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "payment",
        "baseName": "payment",
        "type": "PaymentResponse"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "DonationPaymentResponse.StatusEnum"
    }
];
(function (DonationPaymentResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Completed"] = "completed";
        StatusEnum["Pending"] = "pending";
        StatusEnum["Refused"] = "refused";
    })(StatusEnum = DonationPaymentResponse.StatusEnum || (DonationPaymentResponse.StatusEnum = {}));
})(DonationPaymentResponse = exports.DonationPaymentResponse || (exports.DonationPaymentResponse = {}));
