// @adyen/api-library/lib/src/typings/checkout/genericIssuerPaymentMethodDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenericIssuerPaymentMethodDetails = void 0;
class GenericIssuerPaymentMethodDetails {
    static getAttributeTypeMap() {
        return GenericIssuerPaymentMethodDetails.attributeTypeMap;
    }
}
exports.GenericIssuerPaymentMethodDetails = GenericIssuerPaymentMethodDetails;
GenericIssuerPaymentMethodDetails.discriminator = undefined;
GenericIssuerPaymentMethodDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "issuer",
        "baseName": "issuer",
        "type": "string"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "GenericIssuerPaymentMethodDetails.TypeEnum"
    }
];
(function (GenericIssuerPaymentMethodDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["OnlineBankingPl"] = "onlineBanking_PL";
        TypeEnum["Eps"] = "eps";
        TypeEnum["OnlineBankingSk"] = "onlineBanking_SK";
        TypeEnum["OnlineBankingCz"] = "onlineBanking_CZ";
    })(TypeEnum = GenericIssuerPaymentMethodDetails.TypeEnum || (GenericIssuerPaymentMethodDetails.TypeEnum = {}));
})(GenericIssuerPaymentMethodDetails = exports.GenericIssuerPaymentMethodDetails || (exports.GenericIssuerPaymentMethodDetails = {}));
