// @adyen/api-library/lib/src/typings/checkout/masterpassDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MasterpassDetails = void 0;
class MasterpassDetails {
    static getAttributeTypeMap() {
        return MasterpassDetails.attributeTypeMap;
    }
}
exports.MasterpassDetails = MasterpassDetails;
MasterpassDetails.discriminator = undefined;
MasterpassDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "MasterpassDetails.FundingSourceEnum"
    },
    {
        "name": "masterpassTransactionId",
        "baseName": "masterpassTransactionId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "MasterpassDetails.TypeEnum"
    }
];
(function (MasterpassDetails) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = MasterpassDetails.FundingSourceEnum || (MasterpassDetails.FundingSourceEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Masterpass"] = "masterpass";
    })(TypeEnum = MasterpassDetails.TypeEnum || (MasterpassDetails.TypeEnum = {}));
})(MasterpassDetails = exports.MasterpassDetails || (exports.MasterpassDetails = {}));
