// @adyen/api-library/lib/src/typings/checkout/paymentAmountUpdateRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentAmountUpdateRequest = void 0;
class PaymentAmountUpdateRequest {
    static getAttributeTypeMap() {
        return PaymentAmountUpdateRequest.attributeTypeMap;
    }
}
exports.PaymentAmountUpdateRequest = PaymentAmountUpdateRequest;
PaymentAmountUpdateRequest.discriminator = undefined;
PaymentAmountUpdateRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "applicationInfo",
        "baseName": "applicationInfo",
        "type": "ApplicationInfo"
    },
    {
        "name": "industryUsage",
        "baseName": "industryUsage",
        "type": "PaymentAmountUpdateRequest.IndustryUsageEnum"
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>"
    }
];
(function (PaymentAmountUpdateRequest) {
    let IndustryUsageEnum;
    (function (IndustryUsageEnum) {
        IndustryUsageEnum["DelayedCharge"] = "delayedCharge";
        IndustryUsageEnum["Installment"] = "installment";
        IndustryUsageEnum["NoShow"] = "noShow";
    })(IndustryUsageEnum = PaymentAmountUpdateRequest.IndustryUsageEnum || (PaymentAmountUpdateRequest.IndustryUsageEnum = {}));
})(PaymentAmountUpdateRequest = exports.PaymentAmountUpdateRequest || (exports.PaymentAmountUpdateRequest = {}));
