// @adyen/api-library/lib/src/typings/checkout/paymentVerificationResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentVerificationResponse = void 0;
class PaymentVerificationResponse {
    static getAttributeTypeMap() {
        return PaymentVerificationResponse.attributeTypeMap;
    }
}
exports.PaymentVerificationResponse = PaymentVerificationResponse;
PaymentVerificationResponse.discriminator = undefined;
PaymentVerificationResponse.attributeTypeMap = [
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "fraudResult",
        "baseName": "fraudResult",
        "type": "FraudResult"
    },
    {
        "name": "merchantReference",
        "baseName": "merchantReference",
        "type": "string"
    },
    {
        "name": "order",
        "baseName": "order",
        "type": "CheckoutOrderResponse"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string"
    },
    {
        "name": "refusalReasonCode",
        "baseName": "refusalReasonCode",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "PaymentVerificationResponse.ResultCodeEnum"
    },
    {
        "name": "serviceError",
        "baseName": "serviceError",
        "type": "ServiceErrorDetails"
    },
    {
        "name": "shopperLocale",
        "baseName": "shopperLocale",
        "type": "string"
    }
];
(function (PaymentVerificationResponse) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["AuthenticationFinished"] = "AuthenticationFinished";
        ResultCodeEnum["AuthenticationNotRequired"] = "AuthenticationNotRequired";
        ResultCodeEnum["Authorised"] = "Authorised";
        ResultCodeEnum["Cancelled"] = "Cancelled";
        ResultCodeEnum["ChallengeShopper"] = "ChallengeShopper";
        ResultCodeEnum["Error"] = "Error";
        ResultCodeEnum["IdentifyShopper"] = "IdentifyShopper";
        ResultCodeEnum["PartiallyAuthorised"] = "PartiallyAuthorised";
        ResultCodeEnum["Pending"] = "Pending";
        ResultCodeEnum["PresentToShopper"] = "PresentToShopper";
        ResultCodeEnum["Received"] = "Received";
        ResultCodeEnum["RedirectShopper"] = "RedirectShopper";
        ResultCodeEnum["Refused"] = "Refused";
        ResultCodeEnum["Success"] = "Success";
    })(ResultCodeEnum = PaymentVerificationResponse.ResultCodeEnum || (PaymentVerificationResponse.ResultCodeEnum = {}));
})(PaymentVerificationResponse = exports.PaymentVerificationResponse || (exports.PaymentVerificationResponse = {}));
