// @adyen/api-library/lib/src/typings/balancePlatform/balanceAccountUpdateRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceAccountUpdateRequest = void 0;
class BalanceAccountUpdateRequest {
    static getAttributeTypeMap() {
        return BalanceAccountUpdateRequest.attributeTypeMap;
    }
}
exports.BalanceAccountUpdateRequest = BalanceAccountUpdateRequest;
BalanceAccountUpdateRequest.discriminator = undefined;
BalanceAccountUpdateRequest.attributeTypeMap = [
    {
        "name": "accountHolderId",
        "baseName": "accountHolderId",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "platformPaymentConfiguration",
        "baseName": "platformPaymentConfiguration",
        "type": "PlatformPaymentConfiguration"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BalanceAccountUpdateRequest.StatusEnum"
    },
    {
        "name": "timeZone",
        "baseName": "timeZone",
        "type": "string"
    }
];
(function (BalanceAccountUpdateRequest) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Closed"] = "closed";
        StatusEnum["Inactive"] = "inactive";
        StatusEnum["Suspended"] = "suspended";
    })(StatusEnum = BalanceAccountUpdateRequest.StatusEnum || (BalanceAccountUpdateRequest.StatusEnum = {}));
})(BalanceAccountUpdateRequest = exports.BalanceAccountUpdateRequest || (exports.BalanceAccountUpdateRequest = {}));
