// @adyen/api-library/lib/src/typings/balancePlatform/balanceAccountBase.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceAccountBase = void 0;
class BalanceAccountBase {
    static getAttributeTypeMap() {
        return BalanceAccountBase.attributeTypeMap;
    }
}
exports.BalanceAccountBase = BalanceAccountBase;
BalanceAccountBase.discriminator = undefined;
BalanceAccountBase.attributeTypeMap = [
    {
        "name": "accountHolderId",
        "baseName": "accountHolderId",
        "type": "string"
    },
    {
        "name": "defaultCurrencyCode",
        "baseName": "defaultCurrencyCode",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "migratedAccountCode",
        "baseName": "migratedAccountCode",
        "type": "string"
    },
    {
        "name": "platformPaymentConfiguration",
        "baseName": "platformPaymentConfiguration",
        "type": "PlatformPaymentConfiguration"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BalanceAccountBase.StatusEnum"
    },
    {
        "name": "timeZone",
        "baseName": "timeZone",
        "type": "string"
    }
];
(function (BalanceAccountBase) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Closed"] = "closed";
        StatusEnum["Inactive"] = "inactive";
        StatusEnum["Suspended"] = "suspended";
    })(StatusEnum = BalanceAccountBase.StatusEnum || (BalanceAccountBase.StatusEnum = {}));
})(BalanceAccountBase = exports.BalanceAccountBase || (exports.BalanceAccountBase = {}));
