// @adyen/api-library/lib/src/typings/balancePlatform/bankAccountIdentificationTypeRequirement.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankAccountIdentificationTypeRequirement = void 0;
class BankAccountIdentificationTypeRequirement {
    static getAttributeTypeMap() {
        return BankAccountIdentificationTypeRequirement.attributeTypeMap;
    }
}
exports.BankAccountIdentificationTypeRequirement = BankAccountIdentificationTypeRequirement;
BankAccountIdentificationTypeRequirement.discriminator = undefined;
BankAccountIdentificationTypeRequirement.attributeTypeMap = [
    {
        "name": "bankAccountIdentificationTypes",
        "baseName": "bankAccountIdentificationTypes",
        "type": "Array<BankAccountIdentificationTypeRequirement.BankAccountIdentificationTypesEnum>"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "BankAccountIdentificationTypeRequirement.TypeEnum"
    }
];
(function (BankAccountIdentificationTypeRequirement) {
    let BankAccountIdentificationTypesEnum;
    (function (BankAccountIdentificationTypesEnum) {
        BankAccountIdentificationTypesEnum["AuLocal"] = "auLocal";
        BankAccountIdentificationTypesEnum["BrLocal"] = "brLocal";
        BankAccountIdentificationTypesEnum["CaLocal"] = "caLocal";
        BankAccountIdentificationTypesEnum["CzLocal"] = "czLocal";
        BankAccountIdentificationTypesEnum["DkLocal"] = "dkLocal";
        BankAccountIdentificationTypesEnum["HkLocal"] = "hkLocal";
        BankAccountIdentificationTypesEnum["HuLocal"] = "huLocal";
        BankAccountIdentificationTypesEnum["Iban"] = "iban";
        BankAccountIdentificationTypesEnum["Legacy"] = "legacy";
        BankAccountIdentificationTypesEnum["NoLocal"] = "noLocal";
        BankAccountIdentificationTypesEnum["NumberAndBic"] = "numberAndBic";
        BankAccountIdentificationTypesEnum["NzLocal"] = "nzLocal";
        BankAccountIdentificationTypesEnum["PlLocal"] = "plLocal";
        BankAccountIdentificationTypesEnum["SeLocal"] = "seLocal";
        BankAccountIdentificationTypesEnum["SgLocal"] = "sgLocal";
        BankAccountIdentificationTypesEnum["UkLocal"] = "ukLocal";
        BankAccountIdentificationTypesEnum["UsLocal"] = "usLocal";
    })(BankAccountIdentificationTypesEnum = BankAccountIdentificationTypeRequirement.BankAccountIdentificationTypesEnum || (BankAccountIdentificationTypeRequirement.BankAccountIdentificationTypesEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BankAccountIdentificationTypeRequirement"] = "bankAccountIdentificationTypeRequirement";
    })(TypeEnum = BankAccountIdentificationTypeRequirement.TypeEnum || (BankAccountIdentificationTypeRequirement.TypeEnum = {}));
})(BankAccountIdentificationTypeRequirement = exports.BankAccountIdentificationTypeRequirement || (exports.BankAccountIdentificationTypeRequirement = {}));
