// @adyen/api-library/lib/src/typings/balancePlatform/bankIdentification.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankIdentification = void 0;
class BankIdentification {
    static getAttributeTypeMap() {
        return BankIdentification.attributeTypeMap;
    }
}
exports.BankIdentification = BankIdentification;
BankIdentification.discriminator = undefined;
BankIdentification.attributeTypeMap = [
    {
        "name": "country",
        "baseName": "country",
        "type": "string"
    },
    {
        "name": "identification",
        "baseName": "identification",
        "type": "string"
    },
    {
        "name": "identificationType",
        "baseName": "identificationType",
        "type": "BankIdentification.IdentificationTypeEnum"
    }
];
(function (BankIdentification) {
    let IdentificationTypeEnum;
    (function (IdentificationTypeEnum) {
        IdentificationTypeEnum["Iban"] = "iban";
        IdentificationTypeEnum["RoutingNumber"] = "routingNumber";
    })(IdentificationTypeEnum = BankIdentification.IdentificationTypeEnum || (BankIdentification.IdentificationTypeEnum = {}));
})(BankIdentification = exports.BankIdentification || (exports.BankIdentification = {}));
