// @adyen/api-library/lib/src/typings/balancePlatform/capabilitySettings.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapabilitySettings = void 0;
class CapabilitySettings {
    static getAttributeTypeMap() {
        return CapabilitySettings.attributeTypeMap;
    }
}
exports.CapabilitySettings = CapabilitySettings;
CapabilitySettings.discriminator = undefined;
CapabilitySettings.attributeTypeMap = [
    {
        "name": "amountPerIndustry",
        "baseName": "amountPerIndustry",
        "type": "{ [key: string]: Amount; }"
    },
    {
        "name": "authorizedCardUsers",
        "baseName": "authorizedCardUsers",
        "type": "boolean"
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "Array<CapabilitySettings.FundingSourceEnum>"
    },
    {
        "name": "interval",
        "baseName": "interval",
        "type": "CapabilitySettings.IntervalEnum"
    },
    {
        "name": "maxAmount",
        "baseName": "maxAmount",
        "type": "Amount"
    }
];
(function (CapabilitySettings) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
        FundingSourceEnum["Prepaid"] = "prepaid";
    })(FundingSourceEnum = CapabilitySettings.FundingSourceEnum || (CapabilitySettings.FundingSourceEnum = {}));
    let IntervalEnum;
    (function (IntervalEnum) {
        IntervalEnum["Daily"] = "daily";
        IntervalEnum["Monthly"] = "monthly";
        IntervalEnum["Weekly"] = "weekly";
    })(IntervalEnum = CapabilitySettings.IntervalEnum || (CapabilitySettings.IntervalEnum = {}));
})(CapabilitySettings = exports.CapabilitySettings || (exports.CapabilitySettings = {}));
