// @adyen/api-library/lib/src/typings/balancePlatform/card.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Card = void 0;
class Card {
    static getAttributeTypeMap() {
        return Card.attributeTypeMap;
    }
}
exports.Card = Card;
Card.discriminator = undefined;
Card.attributeTypeMap = [
    {
        "name": "authentication",
        "baseName": "authentication",
        "type": "Authentication"
    },
    {
        "name": "bin",
        "baseName": "bin",
        "type": "string"
    },
    {
        "name": "brand",
        "baseName": "brand",
        "type": "string"
    },
    {
        "name": "brandVariant",
        "baseName": "brandVariant",
        "type": "string"
    },
    {
        "name": "cardholderName",
        "baseName": "cardholderName",
        "type": "string"
    },
    {
        "name": "configuration",
        "baseName": "configuration",
        "type": "CardConfiguration"
    },
    {
        "name": "cvc",
        "baseName": "cvc",
        "type": "string"
    },
    {
        "name": "deliveryContact",
        "baseName": "deliveryContact",
        "type": "DeliveryContact"
    },
    {
        "name": "expiration",
        "baseName": "expiration",
        "type": "Expiry"
    },
    {
        "name": "formFactor",
        "baseName": "formFactor",
        "type": "Card.FormFactorEnum"
    },
    {
        "name": "lastFour",
        "baseName": "lastFour",
        "type": "string"
    },
    {
        "name": "number",
        "baseName": "number",
        "type": "string"
    },
    {
        "name": "threeDSecure",
        "baseName": "threeDSecure",
        "type": "string"
    }
];
(function (Card) {
    let FormFactorEnum;
    (function (FormFactorEnum) {
        FormFactorEnum["Physical"] = "physical";
        FormFactorEnum["Unknown"] = "unknown";
        FormFactorEnum["Virtual"] = "virtual";
    })(FormFactorEnum = Card.FormFactorEnum || (Card.FormFactorEnum = {}));
})(Card = exports.Card || (exports.Card = {}));
