// @adyen/api-library/lib/src/typings/balancePlatform/dayOfWeekRestriction.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DayOfWeekRestriction = void 0;
class DayOfWeekRestriction {
    static getAttributeTypeMap() {
        return DayOfWeekRestriction.attributeTypeMap;
    }
}
exports.DayOfWeekRestriction = DayOfWeekRestriction;
DayOfWeekRestriction.discriminator = undefined;
DayOfWeekRestriction.attributeTypeMap = [
    {
        "name": "operation",
        "baseName": "operation",
        "type": "string"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "Array<DayOfWeekRestriction.ValueEnum>"
    }
];
(function (DayOfWeekRestriction) {
    let ValueEnum;
    (function (ValueEnum) {
        ValueEnum["Friday"] = "friday";
        ValueEnum["Monday"] = "monday";
        ValueEnum["Saturday"] = "saturday";
        ValueEnum["Sunday"] = "sunday";
        ValueEnum["Thursday"] = "thursday";
        ValueEnum["Tuesday"] = "tuesday";
        ValueEnum["Wednesday"] = "wednesday";
    })(ValueEnum = DayOfWeekRestriction.ValueEnum || (DayOfWeekRestriction.ValueEnum = {}));
})(DayOfWeekRestriction = exports.DayOfWeekRestriction || (exports.DayOfWeekRestriction = {}));
