// @adyen/api-library/lib/src/typings/balancePlatform/duration.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Duration = void 0;
class Duration {
    static getAttributeTypeMap() {
        return Duration.attributeTypeMap;
    }
}
exports.Duration = Duration;
Duration.discriminator = undefined;
Duration.attributeTypeMap = [
    {
        "name": "unit",
        "baseName": "unit",
        "type": "Duration.UnitEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "number"
    }
];
(function (Duration) {
    let UnitEnum;
    (function (UnitEnum) {
        UnitEnum["Days"] = "days";
        UnitEnum["Hours"] = "hours";
        UnitEnum["Minutes"] = "minutes";
        UnitEnum["Months"] = "months";
        UnitEnum["Weeks"] = "weeks";
    })(UnitEnum = Duration.UnitEnum || (Duration.UnitEnum = {}));
})(Duration = exports.Duration || (exports.Duration = {}));
