// @adyen/api-library/lib/src/typings/balancePlatform/grantOffer.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrantOffer = void 0;
class GrantOffer {
    static getAttributeTypeMap() {
        return GrantOffer.attributeTypeMap;
    }
}
exports.GrantOffer = GrantOffer;
GrantOffer.discriminator = undefined;
GrantOffer.attributeTypeMap = [
    {
        "name": "accountHolderId",
        "baseName": "accountHolderId",
        "type": "string"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "contractType",
        "baseName": "contractType",
        "type": "GrantOffer.ContractTypeEnum"
    },
    {
        "name": "expiresAt",
        "baseName": "expiresAt",
        "type": "object"
    },
    {
        "name": "fee",
        "baseName": "fee",
        "type": "Fee"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "repayment",
        "baseName": "repayment",
        "type": "Repayment"
    },
    {
        "name": "startsAt",
        "baseName": "startsAt",
        "type": "object"
    }
];
(function (GrantOffer) {
    let ContractTypeEnum;
    (function (ContractTypeEnum) {
        ContractTypeEnum["CashAdvance"] = "cashAdvance";
        ContractTypeEnum["Loan"] = "loan";
    })(ContractTypeEnum = GrantOffer.ContractTypeEnum || (GrantOffer.ContractTypeEnum = {}));
})(GrantOffer = exports.GrantOffer || (exports.GrantOffer = {}));
