// @adyen/api-library/lib/src/typings/balancePlatform/networkToken.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkToken = void 0;
class NetworkToken {
    static getAttributeTypeMap() {
        return NetworkToken.attributeTypeMap;
    }
}
exports.NetworkToken = NetworkToken;
NetworkToken.discriminator = undefined;
NetworkToken.attributeTypeMap = [
    {
        "name": "brandVariant",
        "baseName": "brandVariant",
        "type": "string"
    },
    {
        "name": "creationDate",
        "baseName": "creationDate",
        "type": "Date"
    },
    {
        "name": "device",
        "baseName": "device",
        "type": "DeviceInfo"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "paymentInstrumentId",
        "baseName": "paymentInstrumentId",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "NetworkToken.StatusEnum"
    },
    {
        "name": "tokenLastFour",
        "baseName": "tokenLastFour",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "string"
    }
];
(function (NetworkToken) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Inactive"] = "inactive";
        StatusEnum["Suspended"] = "suspended";
        StatusEnum["Closed"] = "closed";
    })(StatusEnum = NetworkToken.StatusEnum || (NetworkToken.StatusEnum = {}));
})(NetworkToken = exports.NetworkToken || (exports.NetworkToken = {}));
