// @adyen/api-library/lib/src/typings/balancePlatform/paymentInstrumentUpdateRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentInstrumentUpdateRequest = void 0;
class PaymentInstrumentUpdateRequest {
    static getAttributeTypeMap() {
        return PaymentInstrumentUpdateRequest.attributeTypeMap;
    }
}
exports.PaymentInstrumentUpdateRequest = PaymentInstrumentUpdateRequest;
PaymentInstrumentUpdateRequest.discriminator = undefined;
PaymentInstrumentUpdateRequest.attributeTypeMap = [
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "card",
        "baseName": "card",
        "type": "CardInfo"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentInstrumentUpdateRequest.StatusEnum"
    },
    {
        "name": "statusComment",
        "baseName": "statusComment",
        "type": "string"
    },
    {
        "name": "statusReason",
        "baseName": "statusReason",
        "type": "PaymentInstrumentUpdateRequest.StatusReasonEnum"
    }
];
(function (PaymentInstrumentUpdateRequest) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Closed"] = "closed";
        StatusEnum["Inactive"] = "inactive";
        StatusEnum["Suspended"] = "suspended";
    })(StatusEnum = PaymentInstrumentUpdateRequest.StatusEnum || (PaymentInstrumentUpdateRequest.StatusEnum = {}));
    let StatusReasonEnum;
    (function (StatusReasonEnum) {
        StatusReasonEnum["AccountClosure"] = "accountClosure";
        StatusReasonEnum["Damaged"] = "damaged";
        StatusReasonEnum["EndOfLife"] = "endOfLife";
        StatusReasonEnum["Expired"] = "expired";
        StatusReasonEnum["Lost"] = "lost";
        StatusReasonEnum["Other"] = "other";
        StatusReasonEnum["Stolen"] = "stolen";
        StatusReasonEnum["SuspectedFraud"] = "suspectedFraud";
        StatusReasonEnum["TransactionRule"] = "transactionRule";
    })(StatusReasonEnum = PaymentInstrumentUpdateRequest.StatusReasonEnum || (PaymentInstrumentUpdateRequest.StatusReasonEnum = {}));
})(PaymentInstrumentUpdateRequest = exports.PaymentInstrumentUpdateRequest || (exports.PaymentInstrumentUpdateRequest = {}));
