// @adyen/api-library/lib/src/typings/balancePlatform/paymentInstrumentInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentInstrumentInfo = void 0;
class PaymentInstrumentInfo {
    static getAttributeTypeMap() {
        return PaymentInstrumentInfo.attributeTypeMap;
    }
}
exports.PaymentInstrumentInfo = PaymentInstrumentInfo;
PaymentInstrumentInfo.discriminator = undefined;
PaymentInstrumentInfo.attributeTypeMap = [
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "card",
        "baseName": "card",
        "type": "CardInfo"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "issuingCountryCode",
        "baseName": "issuingCountryCode",
        "type": "string"
    },
    {
        "name": "paymentInstrumentGroupId",
        "baseName": "paymentInstrumentGroupId",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentInstrumentInfo.StatusEnum"
    },
    {
        "name": "statusReason",
        "baseName": "statusReason",
        "type": "PaymentInstrumentInfo.StatusReasonEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PaymentInstrumentInfo.TypeEnum"
    }
];
(function (PaymentInstrumentInfo) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Closed"] = "closed";
        StatusEnum["Inactive"] = "inactive";
        StatusEnum["Suspended"] = "suspended";
    })(StatusEnum = PaymentInstrumentInfo.StatusEnum || (PaymentInstrumentInfo.StatusEnum = {}));
    let StatusReasonEnum;
    (function (StatusReasonEnum) {
        StatusReasonEnum["AccountClosure"] = "accountClosure";
        StatusReasonEnum["Damaged"] = "damaged";
        StatusReasonEnum["EndOfLife"] = "endOfLife";
        StatusReasonEnum["Expired"] = "expired";
        StatusReasonEnum["Lost"] = "lost";
        StatusReasonEnum["Other"] = "other";
        StatusReasonEnum["Stolen"] = "stolen";
        StatusReasonEnum["SuspectedFraud"] = "suspectedFraud";
        StatusReasonEnum["TransactionRule"] = "transactionRule";
    })(StatusReasonEnum = PaymentInstrumentInfo.StatusReasonEnum || (PaymentInstrumentInfo.StatusReasonEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BankAccount"] = "bankAccount";
        TypeEnum["Card"] = "card";
    })(TypeEnum = PaymentInstrumentInfo.TypeEnum || (PaymentInstrumentInfo.TypeEnum = {}));
})(PaymentInstrumentInfo = exports.PaymentInstrumentInfo || (exports.PaymentInstrumentInfo = {}));
