// @adyen/api-library/lib/src/typings/balancePlatform/stringMatch.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StringMatch = void 0;
class StringMatch {
    static getAttributeTypeMap() {
        return StringMatch.attributeTypeMap;
    }
}
exports.StringMatch = StringMatch;
StringMatch.discriminator = undefined;
StringMatch.attributeTypeMap = [
    {
        "name": "operation",
        "baseName": "operation",
        "type": "StringMatch.OperationEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "string"
    }
];
(function (StringMatch) {
    let OperationEnum;
    (function (OperationEnum) {
        OperationEnum["Contains"] = "contains";
        OperationEnum["EndsWith"] = "endsWith";
        OperationEnum["IsEqualTo"] = "isEqualTo";
        OperationEnum["StartsWith"] = "startsWith";
    })(OperationEnum = StringMatch.OperationEnum || (StringMatch.OperationEnum = {}));
})(StringMatch = exports.StringMatch || (exports.StringMatch = {}));
