// @adyen/api-library/lib/src/typings/balancePlatform/transactionRuleInterval.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionRuleInterval = void 0;
class TransactionRuleInterval {
    static getAttributeTypeMap() {
        return TransactionRuleInterval.attributeTypeMap;
    }
}
exports.TransactionRuleInterval = TransactionRuleInterval;
TransactionRuleInterval.discriminator = undefined;
TransactionRuleInterval.attributeTypeMap = [
    {
        "name": "dayOfMonth",
        "baseName": "dayOfMonth",
        "type": "number"
    },
    {
        "name": "dayOfWeek",
        "baseName": "dayOfWeek",
        "type": "TransactionRuleInterval.DayOfWeekEnum"
    },
    {
        "name": "duration",
        "baseName": "duration",
        "type": "Duration"
    },
    {
        "name": "timeOfDay",
        "baseName": "timeOfDay",
        "type": "string"
    },
    {
        "name": "timeZone",
        "baseName": "timeZone",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransactionRuleInterval.TypeEnum"
    }
];
(function (TransactionRuleInterval) {
    let DayOfWeekEnum;
    (function (DayOfWeekEnum) {
        DayOfWeekEnum["Friday"] = "friday";
        DayOfWeekEnum["Monday"] = "monday";
        DayOfWeekEnum["Saturday"] = "saturday";
        DayOfWeekEnum["Sunday"] = "sunday";
        DayOfWeekEnum["Thursday"] = "thursday";
        DayOfWeekEnum["Tuesday"] = "tuesday";
        DayOfWeekEnum["Wednesday"] = "wednesday";
    })(DayOfWeekEnum = TransactionRuleInterval.DayOfWeekEnum || (TransactionRuleInterval.DayOfWeekEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Daily"] = "daily";
        TypeEnum["Lifetime"] = "lifetime";
        TypeEnum["Monthly"] = "monthly";
        TypeEnum["PerTransaction"] = "perTransaction";
        TypeEnum["Rolling"] = "rolling";
        TypeEnum["Sliding"] = "sliding";
        TypeEnum["Weekly"] = "weekly";
    })(TypeEnum = TransactionRuleInterval.TypeEnum || (TransactionRuleInterval.TypeEnum = {}));
})(TransactionRuleInterval = exports.TransactionRuleInterval || (exports.TransactionRuleInterval = {}));
