// @adyen/api-library/lib/src/typings/balancePlatform/transactionRule.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionRule = void 0;
class TransactionRule {
    static getAttributeTypeMap() {
        return TransactionRule.attributeTypeMap;
    }
}
exports.TransactionRule = TransactionRule;
TransactionRule.discriminator = undefined;
TransactionRule.attributeTypeMap = [
    {
        "name": "aggregationLevel",
        "baseName": "aggregationLevel",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "endDate",
        "baseName": "endDate",
        "type": "string"
    },
    {
        "name": "entityKey",
        "baseName": "entityKey",
        "type": "TransactionRuleEntityKey"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "interval",
        "baseName": "interval",
        "type": "TransactionRuleInterval"
    },
    {
        "name": "outcomeType",
        "baseName": "outcomeType",
        "type": "TransactionRule.OutcomeTypeEnum"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "requestType",
        "baseName": "requestType",
        "type": "TransactionRule.RequestTypeEnum"
    },
    {
        "name": "ruleRestrictions",
        "baseName": "ruleRestrictions",
        "type": "TransactionRuleRestrictions"
    },
    {
        "name": "score",
        "baseName": "score",
        "type": "number"
    },
    {
        "name": "startDate",
        "baseName": "startDate",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "TransactionRule.StatusEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransactionRule.TypeEnum"
    }
];
(function (TransactionRule) {
    let OutcomeTypeEnum;
    (function (OutcomeTypeEnum) {
        OutcomeTypeEnum["EnforceSca"] = "enforceSCA";
        OutcomeTypeEnum["HardBlock"] = "hardBlock";
        OutcomeTypeEnum["ScoreBased"] = "scoreBased";
    })(OutcomeTypeEnum = TransactionRule.OutcomeTypeEnum || (TransactionRule.OutcomeTypeEnum = {}));
    let RequestTypeEnum;
    (function (RequestTypeEnum) {
        RequestTypeEnum["Authentication"] = "authentication";
        RequestTypeEnum["Authorization"] = "authorization";
        RequestTypeEnum["BankTransfer"] = "bankTransfer";
        RequestTypeEnum["Tokenization"] = "tokenization";
    })(RequestTypeEnum = TransactionRule.RequestTypeEnum || (TransactionRule.RequestTypeEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Inactive"] = "inactive";
    })(StatusEnum = TransactionRule.StatusEnum || (TransactionRule.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AllowList"] = "allowList";
        TypeEnum["BlockList"] = "blockList";
        TypeEnum["MaxUsage"] = "maxUsage";
        TypeEnum["Velocity"] = "velocity";
    })(TypeEnum = TransactionRule.TypeEnum || (TransactionRule.TypeEnum = {}));
})(TransactionRule = exports.TransactionRule || (exports.TransactionRule = {}));
