// @adyen/api-library/lib/src/services/balancePlatform/networkTokensApi.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkTokensApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/balancePlatform/models");
class NetworkTokensApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://balanceplatform-api-test.adyen.com/bcl/v2";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get a network token
    * @param networkTokenId {@link string } The unique identifier of the network token.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link GetNetworkTokenResponse }
    */
    async getNetworkToken(networkTokenId, requestOptions) {
        const endpoint = `${this.baseUrl}/networkTokens/{networkTokenId}`
            .replace("{" + "networkTokenId" + "}", encodeURIComponent(String(networkTokenId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "GetNetworkTokenResponse");
    }
    /**
    * @summary Update a network token
    * @param networkTokenId {@link string } The unique identifier of the network token.
    * @param updateNetworkTokenRequest {@link UpdateNetworkTokenRequest }
    * @param requestOptions {@link IRequest.Options}
    */
    async updateNetworkToken(networkTokenId, updateNetworkTokenRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/networkTokens/{networkTokenId}`
            .replace("{" + "networkTokenId" + "}", encodeURIComponent(String(networkTokenId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(updateNetworkTokenRequest, "UpdateNetworkTokenRequest");
        await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
    }
}
exports.NetworkTokensApi = NetworkTokensApi;
