// @adyen/api-library/lib/src/typings/transfers/ultimatePartyIdentification.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UltimatePartyIdentification = void 0;
class UltimatePartyIdentification {
    static getAttributeTypeMap() {
        return UltimatePartyIdentification.attributeTypeMap;
    }
}
exports.UltimatePartyIdentification = UltimatePartyIdentification;
UltimatePartyIdentification.discriminator = undefined;
UltimatePartyIdentification.attributeTypeMap = [
    {
        "name": "address",
        "baseName": "address",
        "type": "Address"
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "string"
    },
    {
        "name": "firstName",
        "baseName": "firstName",
        "type": "string"
    },
    {
        "name": "fullName",
        "baseName": "fullName",
        "type": "string"
    },
    {
        "name": "lastName",
        "baseName": "lastName",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "UltimatePartyIdentification.TypeEnum"
    }
];
(function (UltimatePartyIdentification) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Individual"] = "individual";
        TypeEnum["Organization"] = "organization";
        TypeEnum["Unknown"] = "unknown";
    })(TypeEnum = UltimatePartyIdentification.TypeEnum || (UltimatePartyIdentification.TypeEnum = {}));
})(UltimatePartyIdentification = exports.UltimatePartyIdentification || (exports.UltimatePartyIdentification = {}));
