// @adyen/api-library/lib/src/services/management/accountStoreLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountStoreLevelApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/management/models");
class AccountStoreLevelApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://management-test.adyen.com/v3";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get a list of stores
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param pageNumber {@link number } The number of the page to fetch.
    * @param pageSize {@link number } The number of items to have on a page, maximum 100. The default is 10 items on a page.
    * @param reference {@link string } The reference of the store.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ListStoresResponse }
    */
    async listStoresByMerchantId(merchantId, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/stores`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ListStoresResponse");
    }
    /**
    * @summary Get a store
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param storeId {@link string } The unique identifier of the store.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Store }
    */
    async getStore(merchantId, storeId, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/stores/{storeId}`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)))
            .replace("{" + "storeId" + "}", encodeURIComponent(String(storeId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
    /**
    * @summary Get a list of stores
    * @param pageNumber {@link number } The number of the page to fetch.
    * @param pageSize {@link number } The number of items to have on a page, maximum 100. The default is 10 items on a page.
    * @param reference {@link string } The reference of the store.
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ListStoresResponse }
    */
    async listStores(requestOptions) {
        const endpoint = `${this.baseUrl}/stores`;
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ListStoresResponse");
    }
    /**
    * @summary Get a store
    * @param storeId {@link string } The unique identifier of the store.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Store }
    */
    async getStoreById(storeId, requestOptions) {
        const endpoint = `${this.baseUrl}/stores/{storeId}`
            .replace("{" + "storeId" + "}", encodeURIComponent(String(storeId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
    /**
    * @summary Update a store
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param storeId {@link string } The unique identifier of the store.
    * @param updateStoreRequest {@link UpdateStoreRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Store }
    */
    async updateStore(merchantId, storeId, updateStoreRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/stores/{storeId}`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)))
            .replace("{" + "storeId" + "}", encodeURIComponent(String(storeId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(updateStoreRequest, "UpdateStoreRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
    /**
    * @summary Update a store
    * @param storeId {@link string } The unique identifier of the store.
    * @param updateStoreRequest {@link UpdateStoreRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Store }
    */
    async updateStoreById(storeId, updateStoreRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/stores/{storeId}`
            .replace("{" + "storeId" + "}", encodeURIComponent(String(storeId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(updateStoreRequest, "UpdateStoreRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
    /**
    * @summary Create a store
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param storeCreationRequest {@link StoreCreationRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Store }
    */
    async createStoreByMerchantId(merchantId, storeCreationRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/stores`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(storeCreationRequest, "StoreCreationRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
    /**
    * @summary Create a store
    * @param storeCreationWithMerchantCodeRequest {@link StoreCreationWithMerchantCodeRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Store }
    */
    async createStore(storeCreationWithMerchantCodeRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/stores`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(storeCreationWithMerchantCodeRequest, "StoreCreationWithMerchantCodeRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "Store");
    }
}
exports.AccountStoreLevelApi = AccountStoreLevelApi;
