// @adyen/api-library/lib/src/services/management/terminalSettingsMerchantLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerminalSettingsMerchantLevelApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/management/models");
class TerminalSettingsMerchantLevelApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://management-test.adyen.com/v3";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get the terminal logo
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param model {@link string } The terminal model. Possible values: E355, VX675WIFIBT, VX680, VX690, VX700, VX820, M400, MX925, P400Plus, UX300, UX410, V200cPlus, V240mPlus, V400cPlus, V400m, e280, e285, e285p, S1E, S1EL, S1F2, S1L, S1U, S7T.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Logo }
    */
    async getTerminalLogo(merchantId, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/terminalLogos`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "Logo");
    }
    /**
    * @summary Get terminal settings
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link TerminalSettings }
    */
    async getTerminalSettings(merchantId, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/terminalSettings`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "TerminalSettings");
    }
    /**
    * @summary Update the terminal logo
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param model {@link string } The terminal model. Allowed values: E355, VX675WIFIBT, VX680, VX690, VX700, VX820, M400, MX925, P400Plus, UX300, UX410, V200cPlus, V240mPlus, V400cPlus, V400m, e280, e285, e285p, S1E, S1EL, S1F2, S1L, S1U, S7T.
    * @param logo {@link Logo }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Logo }
    */
    async updateTerminalLogo(merchantId, logo, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/terminalLogos`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(logo, "Logo");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "Logo");
    }
    /**
    * @summary Update terminal settings
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param terminalSettings {@link TerminalSettings }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link TerminalSettings }
    */
    async updateTerminalSettings(merchantId, terminalSettings, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/terminalSettings`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(terminalSettings, "TerminalSettings");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "TerminalSettings");
    }
}
exports.TerminalSettingsMerchantLevelApi = TerminalSettingsMerchantLevelApi;
