// @adyen/api-library/lib/src/services/management/terminalsTerminalLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerminalsTerminalLevelApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/management/models");
class TerminalsTerminalLevelApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://management-test.adyen.com/v3";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get a list of terminals
    * @param searchQuery {@link string } Returns terminals with an ID that contains the specified string. If present, other query parameters are ignored.
    * @param otpQuery {@link string } Returns one or more terminals associated with the one-time passwords specified in the request. If this query parameter is used, other query parameters are ignored.
    * @param countries {@link string } Returns terminals located in the countries specified by their [two-letter country code](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2).
    * @param merchantIds {@link string } Returns terminals that belong to the merchant accounts specified by their unique merchant account ID.
    * @param storeIds {@link string } Returns terminals that are assigned to the [stores](https://docs.adyen.com/api-explorer/#/ManagementService/latest/get/stores) specified by their unique store ID.
    * @param brandModels {@link string } Returns terminals of the [models](https://docs.adyen.com/api-explorer/#/ManagementService/latest/get/companies/{companyId}/terminalModels) specified in the format *brand.model*.
    * @param pageNumber {@link number } The number of the page to fetch.
    * @param pageSize {@link number } The number of items to have on a page, maximum 100. The default is 20 items on a page.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ListTerminalsResponse }
    */
    async listTerminals(requestOptions) {
        const endpoint = `${this.baseUrl}/terminals`;
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ListTerminalsResponse");
    }
    /**
    * @summary Reassign a terminal
    * @param terminalId {@link string } The unique identifier of the payment terminal.
    * @param terminalReassignmentRequest {@link TerminalReassignmentRequest }
    * @param requestOptions {@link IRequest.Options}
    */
    async reassignTerminal(terminalId, terminalReassignmentRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/terminals/{terminalId}/reassign`
            .replace("{" + "terminalId" + "}", encodeURIComponent(String(terminalId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(terminalReassignmentRequest, "TerminalReassignmentRequest");
        await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
    }
}
exports.TerminalsTerminalLevelApi = TerminalsTerminalLevelApi;
