// @adyen/api-library/lib/src/typings/management/klarnaInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KlarnaInfo = void 0;
class KlarnaInfo {
    static getAttributeTypeMap() {
        return KlarnaInfo.attributeTypeMap;
    }
}
exports.KlarnaInfo = KlarnaInfo;
KlarnaInfo.discriminator = undefined;
KlarnaInfo.attributeTypeMap = [
    {
        "name": "autoCapture",
        "baseName": "autoCapture",
        "type": "boolean"
    },
    {
        "name": "disputeEmail",
        "baseName": "disputeEmail",
        "type": "string"
    },
    {
        "name": "region",
        "baseName": "region",
        "type": "KlarnaInfo.RegionEnum"
    },
    {
        "name": "supportEmail",
        "baseName": "supportEmail",
        "type": "string"
    }
];
(function (KlarnaInfo) {
    let RegionEnum;
    (function (RegionEnum) {
        RegionEnum["Na"] = "NA";
        RegionEnum["Eu"] = "EU";
        RegionEnum["Ch"] = "CH";
        RegionEnum["Au"] = "AU";
    })(RegionEnum = KlarnaInfo.RegionEnum || (KlarnaInfo.RegionEnum = {}));
})(KlarnaInfo = exports.KlarnaInfo || (exports.KlarnaInfo = {}));
