// @adyen/api-library/lib/src/typings/management/splitConfigurationRule.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SplitConfigurationRule = void 0;
class SplitConfigurationRule {
    static getAttributeTypeMap() {
        return SplitConfigurationRule.attributeTypeMap;
    }
}
exports.SplitConfigurationRule = SplitConfigurationRule;
SplitConfigurationRule.discriminator = undefined;
SplitConfigurationRule.attributeTypeMap = [
    {
        "name": "currency",
        "baseName": "currency",
        "type": "string"
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "SplitConfigurationRule.FundingSourceEnum"
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "string"
    },
    {
        "name": "ruleId",
        "baseName": "ruleId",
        "type": "string"
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "SplitConfigurationRule.ShopperInteractionEnum"
    },
    {
        "name": "splitLogic",
        "baseName": "splitLogic",
        "type": "SplitConfigurationLogic"
    }
];
(function (SplitConfigurationRule) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
        FundingSourceEnum["Any"] = "ANY";
    })(FundingSourceEnum = SplitConfigurationRule.FundingSourceEnum || (SplitConfigurationRule.FundingSourceEnum = {}));
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
        ShopperInteractionEnum["Any"] = "ANY";
    })(ShopperInteractionEnum = SplitConfigurationRule.ShopperInteractionEnum || (SplitConfigurationRule.ShopperInteractionEnum = {}));
})(SplitConfigurationRule = exports.SplitConfigurationRule || (exports.SplitConfigurationRule = {}));
