// @adyen/api-library/lib/src/services/management/myAPICredentialApi.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MyAPICredentialApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/management/models");
class MyAPICredentialApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://management-test.adyen.com/v3";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Remove allowed origin
    * @param originId {@link string } Unique identifier of the allowed origin.
    * @param requestOptions {@link IRequest.Options}
    */
    async removeAllowedOrigin(originId, requestOptions) {
        const endpoint = `${this.baseUrl}/me/allowedOrigins/{originId}`
            .replace("{" + "originId" + "}", encodeURIComponent(String(originId)));
        const resource = new resource_1.default(this, endpoint);
        await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "DELETE" });
    }
    /**
    * @summary Get API credential details
    * @param requestOptions {@link IRequest.Options}
    * @return {@link MeApiCredential }
    */
    async getApiCredentialDetails(requestOptions) {
        const endpoint = `${this.baseUrl}/me`;
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "MeApiCredential");
    }
    /**
    * @summary Get allowed origins
    * @param requestOptions {@link IRequest.Options}
    * @return {@link AllowedOriginsResponse }
    */
    async getAllowedOrigins(requestOptions) {
        const endpoint = `${this.baseUrl}/me/allowedOrigins`;
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "AllowedOriginsResponse");
    }
    /**
    * @summary Get allowed origin details
    * @param originId {@link string } Unique identifier of the allowed origin.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link AllowedOrigin }
    */
    async getAllowedOriginDetails(originId, requestOptions) {
        const endpoint = `${this.baseUrl}/me/allowedOrigins/{originId}`
            .replace("{" + "originId" + "}", encodeURIComponent(String(originId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "AllowedOrigin");
    }
    /**
    * @summary Add allowed origin
    * @param createAllowedOriginRequest {@link CreateAllowedOriginRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link AllowedOrigin }
    */
    async addAllowedOrigin(createAllowedOriginRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/me/allowedOrigins`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(createAllowedOriginRequest, "CreateAllowedOriginRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "AllowedOrigin");
    }
    /**
    * @summary Generate new client key for self
    * @param requestOptions {@link IRequest.Options}
    * @return {@link GenerateClientKeyResponse }
    */
    async generateNewClientKeyForSelf(requestOptions) {
        const endpoint = `${this.baseUrl}/me/generateClientKey`;
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "GenerateClientKeyResponse");
    }
}
exports.MyAPICredentialApi = MyAPICredentialApi;
