// @adyen/api-library/lib/src/services/management/terminalActionsCompanyLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerminalActionsCompanyLevelApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/management/models");
class TerminalActionsCompanyLevelApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://management-test.adyen.com/v3";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get a list of terminal actions
    * @param companyId {@link string } The unique identifier of the company account.
    * @param pageNumber {@link number } The number of the page to fetch.
    * @param pageSize {@link number } The number of items to have on a page, maximum 100. The default is 20 items on a page.
    * @param status {@link string } Returns terminal actions with the specified status.  Allowed values: **pending**, **successful**, **failed**, **cancelled**, **tryLater**.
    * @param type {@link string } Returns terminal actions of the specified type.  Allowed values: **InstallAndroidApp**, **UninstallAndroidApp**, **InstallAndroidCertificate**, **UninstallAndroidCertificate**.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ListExternalTerminalActionsResponse }
    */
    async listTerminalActions(companyId, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/terminalActions`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ListExternalTerminalActionsResponse");
    }
    /**
    * @summary Get terminal action
    * @param companyId {@link string } The unique identifier of the company account.
    * @param actionId {@link string } The unique identifier of the terminal action.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ExternalTerminalAction }
    */
    async getTerminalAction(companyId, actionId, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/terminalActions/{actionId}`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)))
            .replace("{" + "actionId" + "}", encodeURIComponent(String(actionId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ExternalTerminalAction");
    }
}
exports.TerminalActionsCompanyLevelApi = TerminalActionsCompanyLevelApi;
