// @adyen/api-library/lib/src/services/management/aPICredentialsMerchantLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.APICredentialsMerchantLevelApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/management/models");
class APICredentialsMerchantLevelApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://management-test.adyen.com/v3";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get a list of API credentials
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param pageNumber {@link number } The number of the page to fetch.
    * @param pageSize {@link number } The number of items to have on a page, maximum 100. The default is 10 items on a page.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ListMerchantApiCredentialsResponse }
    */
    async listApiCredentials(merchantId, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/apiCredentials`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ListMerchantApiCredentialsResponse");
    }
    /**
    * @summary Get an API credential
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param apiCredentialId {@link string } Unique identifier of the API credential.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ApiCredential }
    */
    async getApiCredential(merchantId, apiCredentialId, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/apiCredentials/{apiCredentialId}`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)))
            .replace("{" + "apiCredentialId" + "}", encodeURIComponent(String(apiCredentialId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ApiCredential");
    }
    /**
    * @summary Update an API credential
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param apiCredentialId {@link string } Unique identifier of the API credential.
    * @param updateMerchantApiCredentialRequest {@link UpdateMerchantApiCredentialRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ApiCredential }
    */
    async updateApiCredential(merchantId, apiCredentialId, updateMerchantApiCredentialRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/apiCredentials/{apiCredentialId}`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)))
            .replace("{" + "apiCredentialId" + "}", encodeURIComponent(String(apiCredentialId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(updateMerchantApiCredentialRequest, "UpdateMerchantApiCredentialRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "ApiCredential");
    }
    /**
    * @summary Create an API credential
    * @param merchantId {@link string } The unique identifier of the merchant account.
    * @param createMerchantApiCredentialRequest {@link CreateMerchantApiCredentialRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link CreateApiCredentialResponse }
    */
    async createApiCredential(merchantId, createMerchantApiCredentialRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/merchants/{merchantId}/apiCredentials`
            .replace("{" + "merchantId" + "}", encodeURIComponent(String(merchantId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(createMerchantApiCredentialRequest, "CreateMerchantApiCredentialRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "CreateApiCredentialResponse");
    }
}
exports.APICredentialsMerchantLevelApi = APICredentialsMerchantLevelApi;
