// @adyen/api-library/lib/src/services/management/terminalOrdersCompanyLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerminalOrdersCompanyLevelApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/management/models");
class TerminalOrdersCompanyLevelApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://management-test.adyen.com/v3";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get a list of billing entities
    * @param companyId {@link string } The unique identifier of the company account.
    * @param name {@link string } The name of the billing entity.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link BillingEntitiesResponse }
    */
    async listBillingEntities(companyId, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/billingEntities`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "BillingEntitiesResponse");
    }
    /**
    * @summary Get a list of shipping locations
    * @param companyId {@link string } The unique identifier of the company account.
    * @param name {@link string } The name of the shipping location.
    * @param offset {@link number } The number of locations to skip.
    * @param limit {@link number } The number of locations to return.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ShippingLocationsResponse }
    */
    async listShippingLocations(companyId, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/shippingLocations`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "ShippingLocationsResponse");
    }
    /**
    * @summary Get a list of terminal models
    * @param companyId {@link string } The unique identifier of the company account.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link TerminalModelsResponse }
    */
    async listTerminalModels(companyId, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/terminalModels`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "TerminalModelsResponse");
    }
    /**
    * @summary Get a list of orders
    * @param companyId {@link string } The unique identifier of the company account.
    * @param customerOrderReference {@link string } Your purchase order number.
    * @param status {@link string } The order status. Possible values (not case-sensitive): Placed, Confirmed, Cancelled, Shipped, Delivered.
    * @param offset {@link number } The number of orders to skip.
    * @param limit {@link number } The number of orders to return.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link TerminalOrdersResponse }
    */
    async listOrders(companyId, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/terminalOrders`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "TerminalOrdersResponse");
    }
    /**
    * @summary Get an order
    * @param companyId {@link string } The unique identifier of the company account.
    * @param orderId {@link string } The unique identifier of the order.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link TerminalOrder }
    */
    async getOrder(companyId, orderId, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/terminalOrders/{orderId}`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)))
            .replace("{" + "orderId" + "}", encodeURIComponent(String(orderId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "TerminalOrder");
    }
    /**
    * @summary Get a list of terminal products
    * @param companyId {@link string } The unique identifier of the company account.
    * @param country {@link string } The country to return products for, in [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) format. For example, **US**
    * @param terminalModelId {@link string } The terminal model to return products for. Use the ID returned in the [GET &#x60;/terminalModels&#x60;](https://docs.adyen.com/api-explorer/#/ManagementService/latest/get/companies/{companyId}/terminalModels) response. For example, **Verifone.M400**
    * @param offset {@link number } The number of products to skip.
    * @param limit {@link number } The number of products to return.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link TerminalProductsResponse }
    */
    async listTerminalProducts(companyId, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/terminalProducts`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "TerminalProductsResponse");
    }
    /**
    * @summary Update an order
    * @param companyId {@link string } The unique identifier of the company account.
    * @param orderId {@link string } The unique identifier of the order.
    * @param terminalOrderRequest {@link TerminalOrderRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link TerminalOrder }
    */
    async updateOrder(companyId, orderId, terminalOrderRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/terminalOrders/{orderId}`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)))
            .replace("{" + "orderId" + "}", encodeURIComponent(String(orderId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(terminalOrderRequest, "TerminalOrderRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "TerminalOrder");
    }
    /**
    * @summary Create a shipping location
    * @param companyId {@link string } The unique identifier of the company account.
    * @param shippingLocation {@link ShippingLocation }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ShippingLocation }
    */
    async createShippingLocation(companyId, shippingLocation, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/shippingLocations`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(shippingLocation, "ShippingLocation");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "ShippingLocation");
    }
    /**
    * @summary Create an order
    * @param companyId {@link string } The unique identifier of the company account.
    * @param terminalOrderRequest {@link TerminalOrderRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link TerminalOrder }
    */
    async createOrder(companyId, terminalOrderRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/terminalOrders`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(terminalOrderRequest, "TerminalOrderRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "TerminalOrder");
    }
    /**
    * @summary Cancel an order
    * @param companyId {@link string } The unique identifier of the company account.
    * @param orderId {@link string } The unique identifier of the order.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link TerminalOrder }
    */
    async cancelOrder(companyId, orderId, requestOptions) {
        const endpoint = `${this.baseUrl}/companies/{companyId}/terminalOrders/{orderId}/cancel`
            .replace("{" + "companyId" + "}", encodeURIComponent(String(companyId)))
            .replace("{" + "orderId" + "}", encodeURIComponent(String(orderId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "TerminalOrder");
    }
}
exports.TerminalOrdersCompanyLevelApi = TerminalOrdersCompanyLevelApi;
