// @adyen/api-library/lib/src/services/management/terminalActionsTerminalLevelApi.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerminalActionsTerminalLevelApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/management/models");
class TerminalActionsTerminalLevelApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://management-test.adyen.com/v3";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Create a terminal action
    * @param scheduleTerminalActionsRequest {@link ScheduleTerminalActionsRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ScheduleTerminalActionsResponse }
    */
    async createTerminalAction(scheduleTerminalActionsRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/terminals/scheduleActions`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(scheduleTerminalActionsRequest, "ScheduleTerminalActionsRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "ScheduleTerminalActionsResponse");
    }
}
exports.TerminalActionsTerminalLevelApi = TerminalActionsTerminalLevelApi;
