// @adyen/api-library/lib/src/services/transfers/index.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const capitalApi_1 = require("./capitalApi");
const transactionsApi_1 = require("./transactionsApi");
const transfersApi_1 = require("./transfersApi");
const service_1 = __importDefault(require("../../service"));
class TransfersAPI extends service_1.default {
    constructor(client) {
        super(client);
    }
    get CapitalApi() {
        return new capitalApi_1.CapitalApi(this.client);
    }
    get TransactionsApi() {
        return new transactionsApi_1.TransactionsApi(this.client);
    }
    get TransfersApi() {
        return new transfersApi_1.TransfersApi(this.client);
    }
}
exports.default = TransfersAPI;
