// @adyen/api-library/lib/src/typings/acsWebhooks/authenticationInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticationInfo = void 0;
class AuthenticationInfo {
    static getAttributeTypeMap() {
        return AuthenticationInfo.attributeTypeMap;
    }
}
exports.AuthenticationInfo = AuthenticationInfo;
AuthenticationInfo.discriminator = undefined;
AuthenticationInfo.attributeTypeMap = [
    {
        "name": "acsTransId",
        "baseName": "acsTransId",
        "type": "string"
    },
    {
        "name": "challenge",
        "baseName": "challenge",
        "type": "ChallengeInfo"
    },
    {
        "name": "challengeIndicator",
        "baseName": "challengeIndicator",
        "type": "AuthenticationInfo.ChallengeIndicatorEnum"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date"
    },
    {
        "name": "deviceChannel",
        "baseName": "deviceChannel",
        "type": "AuthenticationInfo.DeviceChannelEnum"
    },
    {
        "name": "dsTransID",
        "baseName": "dsTransID",
        "type": "string"
    },
    {
        "name": "exemptionIndicator",
        "baseName": "exemptionIndicator",
        "type": "AuthenticationInfo.ExemptionIndicatorEnum"
    },
    {
        "name": "inPSD2Scope",
        "baseName": "inPSD2Scope",
        "type": "boolean"
    },
    {
        "name": "messageCategory",
        "baseName": "messageCategory",
        "type": "AuthenticationInfo.MessageCategoryEnum"
    },
    {
        "name": "messageVersion",
        "baseName": "messageVersion",
        "type": "string"
    },
    {
        "name": "riskScore",
        "baseName": "riskScore",
        "type": "number"
    },
    {
        "name": "threeDSServerTransID",
        "baseName": "threeDSServerTransID",
        "type": "string"
    },
    {
        "name": "transStatus",
        "baseName": "transStatus",
        "type": "AuthenticationInfo.TransStatusEnum"
    },
    {
        "name": "transStatusReason",
        "baseName": "transStatusReason",
        "type": "AuthenticationInfo.TransStatusReasonEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "AuthenticationInfo.TypeEnum"
    }
];
(function (AuthenticationInfo) {
    let ChallengeIndicatorEnum;
    (function (ChallengeIndicatorEnum) {
        ChallengeIndicatorEnum["_01"] = "01";
        ChallengeIndicatorEnum["_02"] = "02";
        ChallengeIndicatorEnum["_03"] = "03";
        ChallengeIndicatorEnum["_04"] = "04";
        ChallengeIndicatorEnum["_05"] = "05";
        ChallengeIndicatorEnum["_07"] = "07";
        ChallengeIndicatorEnum["_08"] = "08";
        ChallengeIndicatorEnum["_09"] = "09";
        ChallengeIndicatorEnum["_80"] = "80";
        ChallengeIndicatorEnum["_82"] = "82";
    })(ChallengeIndicatorEnum = AuthenticationInfo.ChallengeIndicatorEnum || (AuthenticationInfo.ChallengeIndicatorEnum = {}));
    let DeviceChannelEnum;
    (function (DeviceChannelEnum) {
        DeviceChannelEnum["App"] = "app";
        DeviceChannelEnum["Browser"] = "browser";
        DeviceChannelEnum["ThreeDsRequestorInitiated"] = "ThreeDSRequestorInitiated";
    })(DeviceChannelEnum = AuthenticationInfo.DeviceChannelEnum || (AuthenticationInfo.DeviceChannelEnum = {}));
    let ExemptionIndicatorEnum;
    (function (ExemptionIndicatorEnum) {
        ExemptionIndicatorEnum["LowValue"] = "lowValue";
        ExemptionIndicatorEnum["SecureCorporate"] = "secureCorporate";
        ExemptionIndicatorEnum["TrustedBeneficiary"] = "trustedBeneficiary";
        ExemptionIndicatorEnum["TransactionRiskAnalysis"] = "transactionRiskAnalysis";
        ExemptionIndicatorEnum["AcquirerExemption"] = "acquirerExemption";
        ExemptionIndicatorEnum["NoExemptionApplied"] = "noExemptionApplied";
        ExemptionIndicatorEnum["VisaDafExemption"] = "visaDAFExemption";
    })(ExemptionIndicatorEnum = AuthenticationInfo.ExemptionIndicatorEnum || (AuthenticationInfo.ExemptionIndicatorEnum = {}));
    let MessageCategoryEnum;
    (function (MessageCategoryEnum) {
        MessageCategoryEnum["Payment"] = "payment";
        MessageCategoryEnum["NonPayment"] = "nonPayment";
    })(MessageCategoryEnum = AuthenticationInfo.MessageCategoryEnum || (AuthenticationInfo.MessageCategoryEnum = {}));
    let TransStatusEnum;
    (function (TransStatusEnum) {
        TransStatusEnum["Y"] = "Y";
        TransStatusEnum["N"] = "N";
        TransStatusEnum["R"] = "R";
        TransStatusEnum["I"] = "I";
        TransStatusEnum["U"] = "U";
    })(TransStatusEnum = AuthenticationInfo.TransStatusEnum || (AuthenticationInfo.TransStatusEnum = {}));
    let TransStatusReasonEnum;
    (function (TransStatusReasonEnum) {
        TransStatusReasonEnum["_01"] = "01";
        TransStatusReasonEnum["_02"] = "02";
        TransStatusReasonEnum["_03"] = "03";
        TransStatusReasonEnum["_04"] = "04";
        TransStatusReasonEnum["_05"] = "05";
        TransStatusReasonEnum["_06"] = "06";
        TransStatusReasonEnum["_07"] = "07";
        TransStatusReasonEnum["_08"] = "08";
        TransStatusReasonEnum["_09"] = "09";
        TransStatusReasonEnum["_10"] = "10";
        TransStatusReasonEnum["_11"] = "11";
        TransStatusReasonEnum["_12"] = "12";
        TransStatusReasonEnum["_13"] = "13";
        TransStatusReasonEnum["_14"] = "14";
        TransStatusReasonEnum["_15"] = "15";
        TransStatusReasonEnum["_16"] = "16";
        TransStatusReasonEnum["_17"] = "17";
        TransStatusReasonEnum["_18"] = "18";
        TransStatusReasonEnum["_19"] = "19";
        TransStatusReasonEnum["_20"] = "20";
        TransStatusReasonEnum["_21"] = "21";
        TransStatusReasonEnum["_22"] = "22";
        TransStatusReasonEnum["_23"] = "23";
        TransStatusReasonEnum["_24"] = "24";
        TransStatusReasonEnum["_25"] = "25";
        TransStatusReasonEnum["_26"] = "26";
        TransStatusReasonEnum["_80"] = "80";
        TransStatusReasonEnum["_81"] = "81";
        TransStatusReasonEnum["_82"] = "82";
        TransStatusReasonEnum["_83"] = "83";
        TransStatusReasonEnum["_84"] = "84";
        TransStatusReasonEnum["_85"] = "85";
        TransStatusReasonEnum["_86"] = "86";
        TransStatusReasonEnum["_87"] = "87";
        TransStatusReasonEnum["_88"] = "88";
    })(TransStatusReasonEnum = AuthenticationInfo.TransStatusReasonEnum || (AuthenticationInfo.TransStatusReasonEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Frictionless"] = "frictionless";
        TypeEnum["Challenge"] = "challenge";
    })(TypeEnum = AuthenticationInfo.TypeEnum || (AuthenticationInfo.TypeEnum = {}));
})(AuthenticationInfo = exports.AuthenticationInfo || (exports.AuthenticationInfo = {}));
