// @adyen/api-library/lib/src/typings/acsWebhooks/challengeInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChallengeInfo = void 0;
class ChallengeInfo {
    static getAttributeTypeMap() {
        return ChallengeInfo.attributeTypeMap;
    }
}
exports.ChallengeInfo = ChallengeInfo;
ChallengeInfo.discriminator = undefined;
ChallengeInfo.attributeTypeMap = [
    {
        "name": "challengeCancel",
        "baseName": "challengeCancel",
        "type": "ChallengeInfo.ChallengeCancelEnum"
    },
    {
        "name": "flow",
        "baseName": "flow",
        "type": "ChallengeInfo.FlowEnum"
    },
    {
        "name": "lastInteraction",
        "baseName": "lastInteraction",
        "type": "Date"
    },
    {
        "name": "phoneNumber",
        "baseName": "phoneNumber",
        "type": "string"
    },
    {
        "name": "resends",
        "baseName": "resends",
        "type": "number"
    },
    {
        "name": "retries",
        "baseName": "retries",
        "type": "number"
    }
];
(function (ChallengeInfo) {
    let ChallengeCancelEnum;
    (function (ChallengeCancelEnum) {
        ChallengeCancelEnum["_01"] = "01";
        ChallengeCancelEnum["_02"] = "02";
        ChallengeCancelEnum["_03"] = "03";
        ChallengeCancelEnum["_04"] = "04";
        ChallengeCancelEnum["_05"] = "05";
        ChallengeCancelEnum["_06"] = "06";
        ChallengeCancelEnum["_07"] = "07";
    })(ChallengeCancelEnum = ChallengeInfo.ChallengeCancelEnum || (ChallengeInfo.ChallengeCancelEnum = {}));
    let FlowEnum;
    (function (FlowEnum) {
        FlowEnum["OtpSms"] = "OTP_SMS";
        FlowEnum["Oob"] = "OOB";
    })(FlowEnum = ChallengeInfo.FlowEnum || (ChallengeInfo.FlowEnum = {}));
})(ChallengeInfo = exports.ChallengeInfo || (exports.ChallengeInfo = {}));
