// @adyen/api-library/lib/src/typings/checkout/additionalDataCommon.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdditionalDataCommon = void 0;
class AdditionalDataCommon {
    static getAttributeTypeMap() {
        return AdditionalDataCommon.attributeTypeMap;
    }
}
exports.AdditionalDataCommon = AdditionalDataCommon;
AdditionalDataCommon.discriminator = undefined;
AdditionalDataCommon.attributeTypeMap = [
    {
        "name": "RequestedTestErrorResponseCode",
        "baseName": "RequestedTestErrorResponseCode",
        "type": "string"
    },
    {
        "name": "allowPartialAuth",
        "baseName": "allowPartialAuth",
        "type": "string"
    },
    {
        "name": "authorisationType",
        "baseName": "authorisationType",
        "type": "string"
    },
    {
        "name": "customRoutingFlag",
        "baseName": "customRoutingFlag",
        "type": "string"
    },
    {
        "name": "industryUsage",
        "baseName": "industryUsage",
        "type": "AdditionalDataCommon.IndustryUsageEnum"
    },
    {
        "name": "manualCapture",
        "baseName": "manualCapture",
        "type": "string"
    },
    {
        "name": "networkTxReference",
        "baseName": "networkTxReference",
        "type": "string"
    },
    {
        "name": "overwriteBrand",
        "baseName": "overwriteBrand",
        "type": "string"
    },
    {
        "name": "subMerchantCity",
        "baseName": "subMerchantCity",
        "type": "string"
    },
    {
        "name": "subMerchantCountry",
        "baseName": "subMerchantCountry",
        "type": "string"
    },
    {
        "name": "subMerchantID",
        "baseName": "subMerchantID",
        "type": "string"
    },
    {
        "name": "subMerchantName",
        "baseName": "subMerchantName",
        "type": "string"
    },
    {
        "name": "subMerchantPostalCode",
        "baseName": "subMerchantPostalCode",
        "type": "string"
    },
    {
        "name": "subMerchantState",
        "baseName": "subMerchantState",
        "type": "string"
    },
    {
        "name": "subMerchantStreet",
        "baseName": "subMerchantStreet",
        "type": "string"
    },
    {
        "name": "subMerchantTaxId",
        "baseName": "subMerchantTaxId",
        "type": "string"
    }
];
(function (AdditionalDataCommon) {
    let IndustryUsageEnum;
    (function (IndustryUsageEnum) {
        IndustryUsageEnum["NoShow"] = "NoShow";
        IndustryUsageEnum["DelayedCharge"] = "DelayedCharge";
    })(IndustryUsageEnum = AdditionalDataCommon.IndustryUsageEnum || (AdditionalDataCommon.IndustryUsageEnum = {}));
})(AdditionalDataCommon = exports.AdditionalDataCommon || (exports.AdditionalDataCommon = {}));
