// @adyen/api-library/lib/src/typings/checkout/balanceCheckResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceCheckResponse = void 0;
class BalanceCheckResponse {
    static getAttributeTypeMap() {
        return BalanceCheckResponse.attributeTypeMap;
    }
}
exports.BalanceCheckResponse = BalanceCheckResponse;
BalanceCheckResponse.discriminator = undefined;
BalanceCheckResponse.attributeTypeMap = [
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "balance",
        "baseName": "balance",
        "type": "Amount"
    },
    {
        "name": "fraudResult",
        "baseName": "fraudResult",
        "type": "FraudResult"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "BalanceCheckResponse.ResultCodeEnum"
    },
    {
        "name": "transactionLimit",
        "baseName": "transactionLimit",
        "type": "Amount"
    }
];
(function (BalanceCheckResponse) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["Success"] = "Success";
        ResultCodeEnum["NotEnoughBalance"] = "NotEnoughBalance";
        ResultCodeEnum["Failed"] = "Failed";
    })(ResultCodeEnum = BalanceCheckResponse.ResultCodeEnum || (BalanceCheckResponse.ResultCodeEnum = {}));
})(BalanceCheckResponse = exports.BalanceCheckResponse || (exports.BalanceCheckResponse = {}));
