// @adyen/api-library/lib/src/typings/checkout/configuration.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Configuration = void 0;
class Configuration {
    static getAttributeTypeMap() {
        return Configuration.attributeTypeMap;
    }
}
exports.Configuration = Configuration;
Configuration.discriminator = undefined;
Configuration.attributeTypeMap = [
    {
        "name": "avs",
        "baseName": "avs",
        "type": "Avs"
    },
    {
        "name": "cardHolderName",
        "baseName": "cardHolderName",
        "type": "Configuration.CardHolderNameEnum"
    },
    {
        "name": "installments",
        "baseName": "installments",
        "type": "InstallmentsNumber"
    },
    {
        "name": "shopperInput",
        "baseName": "shopperInput",
        "type": "ShopperInput"
    }
];
(function (Configuration) {
    let CardHolderNameEnum;
    (function (CardHolderNameEnum) {
        CardHolderNameEnum["None"] = "NONE";
        CardHolderNameEnum["Optional"] = "OPTIONAL";
        CardHolderNameEnum["Required"] = "REQUIRED";
    })(CardHolderNameEnum = Configuration.CardHolderNameEnum || (Configuration.CardHolderNameEnum = {}));
})(Configuration = exports.Configuration || (exports.Configuration = {}));
