// @adyen/api-library/lib/src/typings/checkout/deviceRenderOptions.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceRenderOptions = void 0;
class DeviceRenderOptions {
    static getAttributeTypeMap() {
        return DeviceRenderOptions.attributeTypeMap;
    }
}
exports.DeviceRenderOptions = DeviceRenderOptions;
DeviceRenderOptions.discriminator = undefined;
DeviceRenderOptions.attributeTypeMap = [
    {
        "name": "sdkInterface",
        "baseName": "sdkInterface",
        "type": "DeviceRenderOptions.SdkInterfaceEnum"
    },
    {
        "name": "sdkUiType",
        "baseName": "sdkUiType",
        "type": "Array<DeviceRenderOptions.SdkUiTypeEnum>"
    }
];
(function (DeviceRenderOptions) {
    let SdkInterfaceEnum;
    (function (SdkInterfaceEnum) {
        SdkInterfaceEnum["Native"] = "native";
        SdkInterfaceEnum["Html"] = "html";
        SdkInterfaceEnum["Both"] = "both";
    })(SdkInterfaceEnum = DeviceRenderOptions.SdkInterfaceEnum || (DeviceRenderOptions.SdkInterfaceEnum = {}));
    let SdkUiTypeEnum;
    (function (SdkUiTypeEnum) {
        SdkUiTypeEnum["MultiSelect"] = "multiSelect";
        SdkUiTypeEnum["OtherHtml"] = "otherHtml";
        SdkUiTypeEnum["OutOfBand"] = "outOfBand";
        SdkUiTypeEnum["SingleSelect"] = "singleSelect";
        SdkUiTypeEnum["Text"] = "text";
    })(SdkUiTypeEnum = DeviceRenderOptions.SdkUiTypeEnum || (DeviceRenderOptions.SdkUiTypeEnum = {}));
})(DeviceRenderOptions = exports.DeviceRenderOptions || (exports.DeviceRenderOptions = {}));
