// @adyen/api-library/lib/src/typings/checkout/openInvoiceDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenInvoiceDetails = void 0;
class OpenInvoiceDetails {
    static getAttributeTypeMap() {
        return OpenInvoiceDetails.attributeTypeMap;
    }
}
exports.OpenInvoiceDetails = OpenInvoiceDetails;
OpenInvoiceDetails.discriminator = undefined;
OpenInvoiceDetails.attributeTypeMap = [
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "string"
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "string"
    },
    {
        "name": "personalDetails",
        "baseName": "personalDetails",
        "type": "string"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "OpenInvoiceDetails.TypeEnum"
    }
];
(function (OpenInvoiceDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Openinvoice"] = "openinvoice";
        TypeEnum["AfterpayDirectdebit"] = "afterpay_directdebit";
        TypeEnum["AtomePos"] = "atome_pos";
    })(TypeEnum = OpenInvoiceDetails.TypeEnum || (OpenInvoiceDetails.TypeEnum = {}));
})(OpenInvoiceDetails = exports.OpenInvoiceDetails || (exports.OpenInvoiceDetails = {}));
