// @adyen/api-library/lib/src/typings/checkout/paymentAmountUpdateResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentAmountUpdateResponse = void 0;
class PaymentAmountUpdateResponse {
    static getAttributeTypeMap() {
        return PaymentAmountUpdateResponse.attributeTypeMap;
    }
}
exports.PaymentAmountUpdateResponse = PaymentAmountUpdateResponse;
PaymentAmountUpdateResponse.discriminator = undefined;
PaymentAmountUpdateResponse.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "industryUsage",
        "baseName": "industryUsage",
        "type": "PaymentAmountUpdateResponse.IndustryUsageEnum"
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "paymentPspReference",
        "baseName": "paymentPspReference",
        "type": "string"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentAmountUpdateResponse.StatusEnum"
    }
];
(function (PaymentAmountUpdateResponse) {
    let IndustryUsageEnum;
    (function (IndustryUsageEnum) {
        IndustryUsageEnum["DelayedCharge"] = "delayedCharge";
        IndustryUsageEnum["Installment"] = "installment";
        IndustryUsageEnum["NoShow"] = "noShow";
    })(IndustryUsageEnum = PaymentAmountUpdateResponse.IndustryUsageEnum || (PaymentAmountUpdateResponse.IndustryUsageEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Received"] = "received";
    })(StatusEnum = PaymentAmountUpdateResponse.StatusEnum || (PaymentAmountUpdateResponse.StatusEnum = {}));
})(PaymentAmountUpdateResponse = exports.PaymentAmountUpdateResponse || (exports.PaymentAmountUpdateResponse = {}));
