// @adyen/api-library/lib/src/typings/checkout/ratepayDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RatepayDetails = void 0;
class RatepayDetails {
    static getAttributeTypeMap() {
        return RatepayDetails.attributeTypeMap;
    }
}
exports.RatepayDetails = RatepayDetails;
RatepayDetails.discriminator = undefined;
RatepayDetails.attributeTypeMap = [
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "string"
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string"
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "string"
    },
    {
        "name": "personalDetails",
        "baseName": "personalDetails",
        "type": "string"
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string"
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "RatepayDetails.TypeEnum"
    }
];
(function (RatepayDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Ratepay"] = "ratepay";
        TypeEnum["RatepayDirectdebit"] = "ratepay_directdebit";
    })(TypeEnum = RatepayDetails.TypeEnum || (RatepayDetails.TypeEnum = {}));
})(RatepayDetails = exports.RatepayDetails || (exports.RatepayDetails = {}));
