// @adyen/api-library/lib/src/typings/checkout/sessionResultResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionResultResponse = void 0;
class SessionResultResponse {
    static getAttributeTypeMap() {
        return SessionResultResponse.attributeTypeMap;
    }
}
exports.SessionResultResponse = SessionResultResponse;
SessionResultResponse.discriminator = undefined;
SessionResultResponse.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "SessionResultResponse.StatusEnum"
    }
];
(function (SessionResultResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Canceled"] = "canceled";
        StatusEnum["Completed"] = "completed";
        StatusEnum["Expired"] = "expired";
        StatusEnum["PaymentPending"] = "paymentPending";
        StatusEnum["Refused"] = "refused";
    })(StatusEnum = SessionResultResponse.StatusEnum || (SessionResultResponse.StatusEnum = {}));
})(SessionResultResponse = exports.SessionResultResponse || (exports.SessionResultResponse = {}));
