// @adyen/api-library/lib/src/typings/checkout/threeDSRequestData.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreeDSRequestData = void 0;
class ThreeDSRequestData {
    static getAttributeTypeMap() {
        return ThreeDSRequestData.attributeTypeMap;
    }
}
exports.ThreeDSRequestData = ThreeDSRequestData;
ThreeDSRequestData.discriminator = undefined;
ThreeDSRequestData.attributeTypeMap = [
    {
        "name": "challengeWindowSize",
        "baseName": "challengeWindowSize",
        "type": "ThreeDSRequestData.ChallengeWindowSizeEnum"
    },
    {
        "name": "dataOnly",
        "baseName": "dataOnly",
        "type": "ThreeDSRequestData.DataOnlyEnum"
    },
    {
        "name": "nativeThreeDS",
        "baseName": "nativeThreeDS",
        "type": "ThreeDSRequestData.NativeThreeDSEnum"
    },
    {
        "name": "threeDSVersion",
        "baseName": "threeDSVersion",
        "type": "ThreeDSRequestData.ThreeDSVersionEnum"
    }
];
(function (ThreeDSRequestData) {
    let ChallengeWindowSizeEnum;
    (function (ChallengeWindowSizeEnum) {
        ChallengeWindowSizeEnum["_01"] = "01";
        ChallengeWindowSizeEnum["_02"] = "02";
        ChallengeWindowSizeEnum["_03"] = "03";
        ChallengeWindowSizeEnum["_04"] = "04";
        ChallengeWindowSizeEnum["_05"] = "05";
    })(ChallengeWindowSizeEnum = ThreeDSRequestData.ChallengeWindowSizeEnum || (ThreeDSRequestData.ChallengeWindowSizeEnum = {}));
    let DataOnlyEnum;
    (function (DataOnlyEnum) {
        DataOnlyEnum["False"] = "false";
        DataOnlyEnum["True"] = "true";
    })(DataOnlyEnum = ThreeDSRequestData.DataOnlyEnum || (ThreeDSRequestData.DataOnlyEnum = {}));
    let NativeThreeDSEnum;
    (function (NativeThreeDSEnum) {
        NativeThreeDSEnum["Preferred"] = "preferred";
    })(NativeThreeDSEnum = ThreeDSRequestData.NativeThreeDSEnum || (ThreeDSRequestData.NativeThreeDSEnum = {}));
    let ThreeDSVersionEnum;
    (function (ThreeDSVersionEnum) {
        ThreeDSVersionEnum["_10"] = "2.1.0";
        ThreeDSVersionEnum["_20"] = "2.2.0";
    })(ThreeDSVersionEnum = ThreeDSRequestData.ThreeDSVersionEnum || (ThreeDSRequestData.ThreeDSVersionEnum = {}));
})(ThreeDSRequestData = exports.ThreeDSRequestData || (exports.ThreeDSRequestData = {}));
