// @adyen/api-library/lib/src/notification/bankingWebhookHandler.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/*
 * Adyen NodeJS API Library
 * Copyright (c) 2023 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
const typings_1 = require("../typings");
const typings_2 = require("../typings");
const typings_3 = require("../typings");
const typings_4 = require("../typings");
class BankingWebhookHandler {
    constructor(jsonPayload) {
        this.payload = JSON.parse(jsonPayload);
    }
    // Return generic webhook type
    getGenericWebhook() {
        const type = this.payload['type'];
        if (Object.values(typings_2.acsWebhooks.AuthenticationNotificationRequest.TypeEnum).includes(type)) {
            return this.getAuthenticationNotificationRequest();
        }
        if (Object.values(typings_1.configurationWebhooks.AccountHolderNotificationRequest.TypeEnum).includes(type)) {
            return this.getAccountHolderNotificationRequest();
        }
        if (Object.values(typings_1.configurationWebhooks.BalanceAccountNotificationRequest.TypeEnum).includes(type)) {
            return this.getBalanceAccountNotificationRequest();
        }
        if (Object.values(typings_1.configurationWebhooks.CardOrderNotificationRequest.TypeEnum).includes(type)) {
            return this.getCardOrderNotificationRequest();
        }
        if (Object.values(typings_1.configurationWebhooks.PaymentNotificationRequest.TypeEnum).includes(type)) {
            return this.getPaymentNotificationRequest();
        }
        if (Object.values(typings_1.configurationWebhooks.SweepConfigurationNotificationRequest.TypeEnum).includes(type)) {
            return this.getSweepConfigurationNotificationRequest();
        }
        if (Object.values(typings_3.reportWebhooks.ReportNotificationRequest.TypeEnum).includes(type)) {
            return this.getReportNotificationRequest();
        }
        if (Object.values(typings_4.transferWebhooks.TransferNotificationRequest.TypeEnum).includes(type)) {
            return this.getTransferNotificationRequest();
        }
        throw new Error("Could not parse the json payload: " + this.payload);
    }
    getAuthenticationNotificationRequest() {
        return typings_2.acsWebhooks.ObjectSerializer.deserialize(this.payload, "AuthenticationNotificationRequest");
    }
    getAccountHolderNotificationRequest() {
        return typings_1.configurationWebhooks.ObjectSerializer.deserialize(this.payload, "AccountHolderNotificationRequest");
    }
    getBalanceAccountNotificationRequest() {
        return typings_1.configurationWebhooks.ObjectSerializer.deserialize(this.payload, "BalanceAccountNotificationRequest");
    }
    getCardOrderNotificationRequest() {
        return typings_1.configurationWebhooks.ObjectSerializer.deserialize(this.payload, "CardOrderNotificationRequest");
    }
    getPaymentNotificationRequest() {
        return typings_1.configurationWebhooks.ObjectSerializer.deserialize(this.payload, "PaymentNotificationRequest");
    }
    getSweepConfigurationNotificationRequest() {
        return typings_1.configurationWebhooks.ObjectSerializer.deserialize(this.payload, "SweepConfigurationNotificationRequest");
    }
    getReportNotificationRequest() {
        return typings_3.reportWebhooks.ObjectSerializer.deserialize(this.payload, "ReportNotificationRequest");
    }
    getTransferNotificationRequest() {
        return typings_4.transferWebhooks.ObjectSerializer.deserialize(this.payload, "TransferNotificationRequest");
    }
}
exports.default = BankingWebhookHandler;
