// @adyen/api-library/lib/src/typings/balanceControl/balanceTransferRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceTransferRequest = void 0;
class BalanceTransferRequest {
    static getAttributeTypeMap() {
        return BalanceTransferRequest.attributeTypeMap;
    }
}
exports.BalanceTransferRequest = BalanceTransferRequest;
BalanceTransferRequest.discriminator = undefined;
BalanceTransferRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "fromMerchant",
        "baseName": "fromMerchant",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "toMerchant",
        "baseName": "toMerchant",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "BalanceTransferRequest.TypeEnum"
    }
];
(function (BalanceTransferRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Tax"] = "tax";
        TypeEnum["Fee"] = "fee";
        TypeEnum["TerminalSale"] = "terminalSale";
        TypeEnum["Credit"] = "credit";
        TypeEnum["Debit"] = "debit";
        TypeEnum["Adjustment"] = "adjustment";
    })(TypeEnum = BalanceTransferRequest.TypeEnum || (BalanceTransferRequest.TypeEnum = {}));
})(BalanceTransferRequest = exports.BalanceTransferRequest || (exports.BalanceTransferRequest = {}));
