// @adyen/api-library/lib/src/typings/balanceControl/balanceTransferResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceTransferResponse = void 0;
class BalanceTransferResponse {
    static getAttributeTypeMap() {
        return BalanceTransferResponse.attributeTypeMap;
    }
}
exports.BalanceTransferResponse = BalanceTransferResponse;
BalanceTransferResponse.discriminator = undefined;
BalanceTransferResponse.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "fromMerchant",
        "baseName": "fromMerchant",
        "type": "string"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BalanceTransferResponse.StatusEnum"
    },
    {
        "name": "toMerchant",
        "baseName": "toMerchant",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "BalanceTransferResponse.TypeEnum"
    }
];
(function (BalanceTransferResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Error"] = "error";
        StatusEnum["Failed"] = "failed";
        StatusEnum["NotEnoughBalance"] = "notEnoughBalance";
        StatusEnum["Transferred"] = "transferred";
    })(StatusEnum = BalanceTransferResponse.StatusEnum || (BalanceTransferResponse.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Tax"] = "tax";
        TypeEnum["Fee"] = "fee";
        TypeEnum["TerminalSale"] = "terminalSale";
        TypeEnum["Credit"] = "credit";
        TypeEnum["Debit"] = "debit";
        TypeEnum["Adjustment"] = "adjustment";
    })(TypeEnum = BalanceTransferResponse.TypeEnum || (BalanceTransferResponse.TypeEnum = {}));
})(BalanceTransferResponse = exports.BalanceTransferResponse || (exports.BalanceTransferResponse = {}));
