// @adyen/api-library/lib/src/typings/management/connectivity.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Connectivity = void 0;
class Connectivity {
    static getAttributeTypeMap() {
        return Connectivity.attributeTypeMap;
    }
}
exports.Connectivity = Connectivity;
Connectivity.discriminator = undefined;
Connectivity.attributeTypeMap = [
    {
        "name": "simcardStatus",
        "baseName": "simcardStatus",
        "type": "Connectivity.SimcardStatusEnum"
    }
];
(function (Connectivity) {
    let SimcardStatusEnum;
    (function (SimcardStatusEnum) {
        SimcardStatusEnum["Activated"] = "ACTIVATED";
        SimcardStatusEnum["Inventory"] = "INVENTORY";
    })(SimcardStatusEnum = Connectivity.SimcardStatusEnum || (Connectivity.SimcardStatusEnum = {}));
})(Connectivity = exports.Connectivity || (exports.Connectivity = {}));
