// @adyen/api-library/lib/src/typings/management/paymentMethod.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethod = void 0;
class PaymentMethod {
    static getAttributeTypeMap() {
        return PaymentMethod.attributeTypeMap;
    }
}
exports.PaymentMethod = PaymentMethod;
PaymentMethod.discriminator = undefined;
PaymentMethod.attributeTypeMap = [
    {
        "name": "afterpayTouch",
        "baseName": "afterpayTouch",
        "type": "AfterpayTouchInfo"
    },
    {
        "name": "allowed",
        "baseName": "allowed",
        "type": "boolean"
    },
    {
        "name": "applePay",
        "baseName": "applePay",
        "type": "ApplePayInfo"
    },
    {
        "name": "bcmc",
        "baseName": "bcmc",
        "type": "BcmcInfo"
    },
    {
        "name": "businessLineId",
        "baseName": "businessLineId",
        "type": "string"
    },
    {
        "name": "cartesBancaires",
        "baseName": "cartesBancaires",
        "type": "CartesBancairesInfo"
    },
    {
        "name": "clearpay",
        "baseName": "clearpay",
        "type": "ClearpayInfo"
    },
    {
        "name": "countries",
        "baseName": "countries",
        "type": "Array<string>"
    },
    {
        "name": "cup",
        "baseName": "cup",
        "type": "GenericPmWithTdiInfo"
    },
    {
        "name": "currencies",
        "baseName": "currencies",
        "type": "Array<string>"
    },
    {
        "name": "customRoutingFlags",
        "baseName": "customRoutingFlags",
        "type": "Array<string>"
    },
    {
        "name": "diners",
        "baseName": "diners",
        "type": "GenericPmWithTdiInfo"
    },
    {
        "name": "discover",
        "baseName": "discover",
        "type": "GenericPmWithTdiInfo"
    },
    {
        "name": "eftpos_australia",
        "baseName": "eftpos_australia",
        "type": "GenericPmWithTdiInfo"
    },
    {
        "name": "enabled",
        "baseName": "enabled",
        "type": "boolean"
    },
    {
        "name": "giroPay",
        "baseName": "giroPay",
        "type": "GiroPayInfo"
    },
    {
        "name": "girocard",
        "baseName": "girocard",
        "type": "GenericPmWithTdiInfo"
    },
    {
        "name": "googlePay",
        "baseName": "googlePay",
        "type": "GooglePayInfo"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "ideal",
        "baseName": "ideal",
        "type": "GenericPmWithTdiInfo"
    },
    {
        "name": "interac_card",
        "baseName": "interac_card",
        "type": "GenericPmWithTdiInfo"
    },
    {
        "name": "jcb",
        "baseName": "jcb",
        "type": "GenericPmWithTdiInfo"
    },
    {
        "name": "klarna",
        "baseName": "klarna",
        "type": "KlarnaInfo"
    },
    {
        "name": "maestro",
        "baseName": "maestro",
        "type": "GenericPmWithTdiInfo"
    },
    {
        "name": "mc",
        "baseName": "mc",
        "type": "GenericPmWithTdiInfo"
    },
    {
        "name": "mealVoucher_FR",
        "baseName": "mealVoucher_FR",
        "type": "MealVoucherFRInfo"
    },
    {
        "name": "paypal",
        "baseName": "paypal",
        "type": "PayPalInfo"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "string"
    },
    {
        "name": "sofort",
        "baseName": "sofort",
        "type": "SofortInfo"
    },
    {
        "name": "storeIds",
        "baseName": "storeIds",
        "type": "Array<string>"
    },
    {
        "name": "swish",
        "baseName": "swish",
        "type": "SwishInfo"
    },
    {
        "name": "twint",
        "baseName": "twint",
        "type": "TwintInfo"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "string"
    },
    {
        "name": "verificationStatus",
        "baseName": "verificationStatus",
        "type": "PaymentMethod.VerificationStatusEnum"
    },
    {
        "name": "vipps",
        "baseName": "vipps",
        "type": "VippsInfo"
    },
    {
        "name": "visa",
        "baseName": "visa",
        "type": "GenericPmWithTdiInfo"
    }
];
(function (PaymentMethod) {
    let VerificationStatusEnum;
    (function (VerificationStatusEnum) {
        VerificationStatusEnum["Valid"] = "valid";
        VerificationStatusEnum["Pending"] = "pending";
        VerificationStatusEnum["Invalid"] = "invalid";
        VerificationStatusEnum["Rejected"] = "rejected";
    })(VerificationStatusEnum = PaymentMethod.VerificationStatusEnum || (PaymentMethod.VerificationStatusEnum = {}));
})(PaymentMethod = exports.PaymentMethod || (exports.PaymentMethod = {}));
