// @adyen/api-library/lib/src/typings/management/paymentMethodResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodResponse = void 0;
class PaymentMethodResponse {
    static getAttributeTypeMap() {
        return PaymentMethodResponse.attributeTypeMap;
    }
}
exports.PaymentMethodResponse = PaymentMethodResponse;
PaymentMethodResponse.discriminator = undefined;
PaymentMethodResponse.attributeTypeMap = [
    {
        "name": "_links",
        "baseName": "_links",
        "type": "PaginationLinks"
    },
    {
        "name": "data",
        "baseName": "data",
        "type": "Array<PaymentMethod>"
    },
    {
        "name": "itemsTotal",
        "baseName": "itemsTotal",
        "type": "number"
    },
    {
        "name": "pagesTotal",
        "baseName": "pagesTotal",
        "type": "number"
    },
    {
        "name": "typesWithErrors",
        "baseName": "typesWithErrors",
        "type": "Array<PaymentMethodResponse.TypesWithErrorsEnum>"
    }
];
(function (PaymentMethodResponse) {
    let TypesWithErrorsEnum;
    (function (TypesWithErrorsEnum) {
        TypesWithErrorsEnum["Afterpaytouch"] = "afterpaytouch";
        TypesWithErrorsEnum["Alipay"] = "alipay";
        TypesWithErrorsEnum["AlipayHk"] = "alipay_hk";
        TypesWithErrorsEnum["Amex"] = "amex";
        TypesWithErrorsEnum["Applepay"] = "applepay";
        TypesWithErrorsEnum["Bcmc"] = "bcmc";
        TypesWithErrorsEnum["Blik"] = "blik";
        TypesWithErrorsEnum["Cartebancaire"] = "cartebancaire";
        TypesWithErrorsEnum["Clearpay"] = "clearpay";
        TypesWithErrorsEnum["Cup"] = "cup";
        TypesWithErrorsEnum["Diners"] = "diners";
        TypesWithErrorsEnum["DirectdebitGb"] = "directdebit_GB";
        TypesWithErrorsEnum["Discover"] = "discover";
        TypesWithErrorsEnum["EbankingFi"] = "ebanking_FI";
        TypesWithErrorsEnum["EftposAustralia"] = "eftpos_australia";
        TypesWithErrorsEnum["Elo"] = "elo";
        TypesWithErrorsEnum["Elocredit"] = "elocredit";
        TypesWithErrorsEnum["Elodebit"] = "elodebit";
        TypesWithErrorsEnum["Girocard"] = "girocard";
        TypesWithErrorsEnum["Googlepay"] = "googlepay";
        TypesWithErrorsEnum["Hiper"] = "hiper";
        TypesWithErrorsEnum["Hipercard"] = "hipercard";
        TypesWithErrorsEnum["Ideal"] = "ideal";
        TypesWithErrorsEnum["InteracCard"] = "interac_card";
        TypesWithErrorsEnum["Jcb"] = "jcb";
        TypesWithErrorsEnum["Klarna"] = "klarna";
        TypesWithErrorsEnum["KlarnaAccount"] = "klarna_account";
        TypesWithErrorsEnum["KlarnaPaynow"] = "klarna_paynow";
        TypesWithErrorsEnum["Maestro"] = "maestro";
        TypesWithErrorsEnum["Mbway"] = "mbway";
        TypesWithErrorsEnum["Mc"] = "mc";
        TypesWithErrorsEnum["Mcdebit"] = "mcdebit";
        TypesWithErrorsEnum["MealVoucherFr"] = "mealVoucher_FR";
        TypesWithErrorsEnum["Mobilepay"] = "mobilepay";
        TypesWithErrorsEnum["Multibanco"] = "multibanco";
        TypesWithErrorsEnum["OnlineBankingPl"] = "onlineBanking_PL";
        TypesWithErrorsEnum["Paybybank"] = "paybybank";
        TypesWithErrorsEnum["Paypal"] = "paypal";
        TypesWithErrorsEnum["Payshop"] = "payshop";
        TypesWithErrorsEnum["Swish"] = "swish";
        TypesWithErrorsEnum["Trustly"] = "trustly";
        TypesWithErrorsEnum["Twint"] = "twint";
        TypesWithErrorsEnum["TwintPos"] = "twint_pos";
        TypesWithErrorsEnum["Vipps"] = "vipps";
        TypesWithErrorsEnum["Visa"] = "visa";
        TypesWithErrorsEnum["Visadebit"] = "visadebit";
        TypesWithErrorsEnum["Vpay"] = "vpay";
        TypesWithErrorsEnum["Wechatpay"] = "wechatpay";
        TypesWithErrorsEnum["WechatpayPos"] = "wechatpay_pos";
    })(TypesWithErrorsEnum = PaymentMethodResponse.TypesWithErrorsEnum || (PaymentMethodResponse.TypesWithErrorsEnum = {}));
})(PaymentMethodResponse = exports.PaymentMethodResponse || (exports.PaymentMethodResponse = {}));
