// @adyen/api-library/lib/src/typings/management/store.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Store = void 0;
class Store {
    static getAttributeTypeMap() {
        return Store.attributeTypeMap;
    }
}
exports.Store = Store;
Store.discriminator = undefined;
Store.attributeTypeMap = [
    {
        "name": "_links",
        "baseName": "_links",
        "type": "Links"
    },
    {
        "name": "address",
        "baseName": "address",
        "type": "StoreLocation"
    },
    {
        "name": "businessLineIds",
        "baseName": "businessLineIds",
        "type": "Array<string>"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "externalReferenceId",
        "baseName": "externalReferenceId",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "merchantId",
        "baseName": "merchantId",
        "type": "string"
    },
    {
        "name": "phoneNumber",
        "baseName": "phoneNumber",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "shopperStatement",
        "baseName": "shopperStatement",
        "type": "string"
    },
    {
        "name": "splitConfiguration",
        "baseName": "splitConfiguration",
        "type": "StoreSplitConfiguration"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "Store.StatusEnum"
    }
];
(function (Store) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Closed"] = "closed";
        StatusEnum["Inactive"] = "inactive";
    })(StatusEnum = Store.StatusEnum || (Store.StatusEnum = {}));
})(Store = exports.Store || (exports.Store = {}));
