// @adyen/api-library/lib/src/typings/management/updateSplitConfigurationLogicRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSplitConfigurationLogicRequest = void 0;
class UpdateSplitConfigurationLogicRequest {
    static getAttributeTypeMap() {
        return UpdateSplitConfigurationLogicRequest.attributeTypeMap;
    }
}
exports.UpdateSplitConfigurationLogicRequest = UpdateSplitConfigurationLogicRequest;
UpdateSplitConfigurationLogicRequest.discriminator = undefined;
UpdateSplitConfigurationLogicRequest.attributeTypeMap = [
    {
        "name": "acquiringFees",
        "baseName": "acquiringFees",
        "type": "UpdateSplitConfigurationLogicRequest.AcquiringFeesEnum"
    },
    {
        "name": "additionalCommission",
        "baseName": "additionalCommission",
        "type": "AdditionalCommission"
    },
    {
        "name": "adyenCommission",
        "baseName": "adyenCommission",
        "type": "UpdateSplitConfigurationLogicRequest.AdyenCommissionEnum"
    },
    {
        "name": "adyenFees",
        "baseName": "adyenFees",
        "type": "UpdateSplitConfigurationLogicRequest.AdyenFeesEnum"
    },
    {
        "name": "adyenMarkup",
        "baseName": "adyenMarkup",
        "type": "UpdateSplitConfigurationLogicRequest.AdyenMarkupEnum"
    },
    {
        "name": "chargeback",
        "baseName": "chargeback",
        "type": "UpdateSplitConfigurationLogicRequest.ChargebackEnum"
    },
    {
        "name": "chargebackCostAllocation",
        "baseName": "chargebackCostAllocation",
        "type": "UpdateSplitConfigurationLogicRequest.ChargebackCostAllocationEnum"
    },
    {
        "name": "commission",
        "baseName": "commission",
        "type": "Commission"
    },
    {
        "name": "interchange",
        "baseName": "interchange",
        "type": "UpdateSplitConfigurationLogicRequest.InterchangeEnum"
    },
    {
        "name": "paymentFee",
        "baseName": "paymentFee",
        "type": "UpdateSplitConfigurationLogicRequest.PaymentFeeEnum"
    },
    {
        "name": "remainder",
        "baseName": "remainder",
        "type": "UpdateSplitConfigurationLogicRequest.RemainderEnum"
    },
    {
        "name": "schemeFee",
        "baseName": "schemeFee",
        "type": "UpdateSplitConfigurationLogicRequest.SchemeFeeEnum"
    },
    {
        "name": "splitLogicId",
        "baseName": "splitLogicId",
        "type": "string"
    },
    {
        "name": "surcharge",
        "baseName": "surcharge",
        "type": "UpdateSplitConfigurationLogicRequest.SurchargeEnum"
    },
    {
        "name": "tip",
        "baseName": "tip",
        "type": "UpdateSplitConfigurationLogicRequest.TipEnum"
    }
];
(function (UpdateSplitConfigurationLogicRequest) {
    let AcquiringFeesEnum;
    (function (AcquiringFeesEnum) {
        AcquiringFeesEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        AcquiringFeesEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(AcquiringFeesEnum = UpdateSplitConfigurationLogicRequest.AcquiringFeesEnum || (UpdateSplitConfigurationLogicRequest.AcquiringFeesEnum = {}));
    let AdyenCommissionEnum;
    (function (AdyenCommissionEnum) {
        AdyenCommissionEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        AdyenCommissionEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(AdyenCommissionEnum = UpdateSplitConfigurationLogicRequest.AdyenCommissionEnum || (UpdateSplitConfigurationLogicRequest.AdyenCommissionEnum = {}));
    let AdyenFeesEnum;
    (function (AdyenFeesEnum) {
        AdyenFeesEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        AdyenFeesEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(AdyenFeesEnum = UpdateSplitConfigurationLogicRequest.AdyenFeesEnum || (UpdateSplitConfigurationLogicRequest.AdyenFeesEnum = {}));
    let AdyenMarkupEnum;
    (function (AdyenMarkupEnum) {
        AdyenMarkupEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        AdyenMarkupEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(AdyenMarkupEnum = UpdateSplitConfigurationLogicRequest.AdyenMarkupEnum || (UpdateSplitConfigurationLogicRequest.AdyenMarkupEnum = {}));
    let ChargebackEnum;
    (function (ChargebackEnum) {
        ChargebackEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        ChargebackEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
        ChargebackEnum["DeductAccordingToSplitRatio"] = "deductAccordingToSplitRatio";
    })(ChargebackEnum = UpdateSplitConfigurationLogicRequest.ChargebackEnum || (UpdateSplitConfigurationLogicRequest.ChargebackEnum = {}));
    let ChargebackCostAllocationEnum;
    (function (ChargebackCostAllocationEnum) {
        ChargebackCostAllocationEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        ChargebackCostAllocationEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(ChargebackCostAllocationEnum = UpdateSplitConfigurationLogicRequest.ChargebackCostAllocationEnum || (UpdateSplitConfigurationLogicRequest.ChargebackCostAllocationEnum = {}));
    let InterchangeEnum;
    (function (InterchangeEnum) {
        InterchangeEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        InterchangeEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(InterchangeEnum = UpdateSplitConfigurationLogicRequest.InterchangeEnum || (UpdateSplitConfigurationLogicRequest.InterchangeEnum = {}));
    let PaymentFeeEnum;
    (function (PaymentFeeEnum) {
        PaymentFeeEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        PaymentFeeEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(PaymentFeeEnum = UpdateSplitConfigurationLogicRequest.PaymentFeeEnum || (UpdateSplitConfigurationLogicRequest.PaymentFeeEnum = {}));
    let RemainderEnum;
    (function (RemainderEnum) {
        RemainderEnum["AddToLiableAccount"] = "addToLiableAccount";
        RemainderEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(RemainderEnum = UpdateSplitConfigurationLogicRequest.RemainderEnum || (UpdateSplitConfigurationLogicRequest.RemainderEnum = {}));
    let SchemeFeeEnum;
    (function (SchemeFeeEnum) {
        SchemeFeeEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        SchemeFeeEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(SchemeFeeEnum = UpdateSplitConfigurationLogicRequest.SchemeFeeEnum || (UpdateSplitConfigurationLogicRequest.SchemeFeeEnum = {}));
    let SurchargeEnum;
    (function (SurchargeEnum) {
        SurchargeEnum["AddToLiableAccount"] = "addToLiableAccount";
        SurchargeEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(SurchargeEnum = UpdateSplitConfigurationLogicRequest.SurchargeEnum || (UpdateSplitConfigurationLogicRequest.SurchargeEnum = {}));
    let TipEnum;
    (function (TipEnum) {
        TipEnum["AddToLiableAccount"] = "addToLiableAccount";
        TipEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(TipEnum = UpdateSplitConfigurationLogicRequest.TipEnum || (UpdateSplitConfigurationLogicRequest.TipEnum = {}));
})(UpdateSplitConfigurationLogicRequest = exports.UpdateSplitConfigurationLogicRequest || (exports.UpdateSplitConfigurationLogicRequest = {}));
