// @adyen/api-library/lib/src/typings/transferWebhooks/amountAdjustment.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AmountAdjustment = void 0;
class AmountAdjustment {
    static getAttributeTypeMap() {
        return AmountAdjustment.attributeTypeMap;
    }
}
exports.AmountAdjustment = AmountAdjustment;
AmountAdjustment.discriminator = undefined;
AmountAdjustment.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "amountAdjustmentType",
        "baseName": "amountAdjustmentType",
        "type": "AmountAdjustment.AmountAdjustmentTypeEnum"
    },
    {
        "name": "basepoints",
        "baseName": "basepoints",
        "type": "number"
    }
];
(function (AmountAdjustment) {
    let AmountAdjustmentTypeEnum;
    (function (AmountAdjustmentTypeEnum) {
        AmountAdjustmentTypeEnum["AtmMarkup"] = "atmMarkup";
        AmountAdjustmentTypeEnum["AuthHoldReserve"] = "authHoldReserve";
        AmountAdjustmentTypeEnum["Exchange"] = "exchange";
        AmountAdjustmentTypeEnum["ForexMarkup"] = "forexMarkup";
    })(AmountAdjustmentTypeEnum = AmountAdjustment.AmountAdjustmentTypeEnum || (AmountAdjustment.AmountAdjustmentTypeEnum = {}));
})(AmountAdjustment = exports.AmountAdjustment || (exports.AmountAdjustment = {}));
