// @adyen/api-library/lib/src/typings/transferWebhooks/partyIdentification.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartyIdentification = void 0;
class PartyIdentification {
    static getAttributeTypeMap() {
        return PartyIdentification.attributeTypeMap;
    }
}
exports.PartyIdentification = PartyIdentification;
PartyIdentification.discriminator = undefined;
PartyIdentification.attributeTypeMap = [
    {
        "name": "address",
        "baseName": "address",
        "type": "Address"
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "string"
    },
    {
        "name": "firstName",
        "baseName": "firstName",
        "type": "string"
    },
    {
        "name": "fullName",
        "baseName": "fullName",
        "type": "string"
    },
    {
        "name": "lastName",
        "baseName": "lastName",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PartyIdentification.TypeEnum"
    }
];
(function (PartyIdentification) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Individual"] = "individual";
        TypeEnum["Organization"] = "organization";
        TypeEnum["Unknown"] = "unknown";
    })(TypeEnum = PartyIdentification.TypeEnum || (PartyIdentification.TypeEnum = {}));
})(PartyIdentification = exports.PartyIdentification || (exports.PartyIdentification = {}));
