// @adyen/api-library/lib/src/typings/transferWebhooks/transferNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferNotificationRequest = void 0;
class TransferNotificationRequest {
    static getAttributeTypeMap() {
        return TransferNotificationRequest.attributeTypeMap;
    }
}
exports.TransferNotificationRequest = TransferNotificationRequest;
TransferNotificationRequest.discriminator = undefined;
TransferNotificationRequest.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "TransferData"
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransferNotificationRequest.TypeEnum"
    }
];
(function (TransferNotificationRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Created"] = "balancePlatform.transfer.created";
        TypeEnum["Updated"] = "balancePlatform.transfer.updated";
    })(TypeEnum = TransferNotificationRequest.TypeEnum || (TransferNotificationRequest.TypeEnum = {}));
})(TransferNotificationRequest = exports.TransferNotificationRequest || (exports.TransferNotificationRequest = {}));
