// @adyen/api-library/lib/src/typings/transferWebhooks/transferEvent.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferEvent = void 0;
class TransferEvent {
    static getAttributeTypeMap() {
        return TransferEvent.attributeTypeMap;
    }
}
exports.TransferEvent = TransferEvent;
TransferEvent.discriminator = undefined;
TransferEvent.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "amountAdjustments",
        "baseName": "amountAdjustments",
        "type": "Array<AmountAdjustment>"
    },
    {
        "name": "bookingDate",
        "baseName": "bookingDate",
        "type": "Date"
    },
    {
        "name": "estimatedArrivalTime",
        "baseName": "estimatedArrivalTime",
        "type": "Date"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "modification",
        "baseName": "modification",
        "type": "Modification"
    },
    {
        "name": "mutations",
        "baseName": "mutations",
        "type": "Array<BalanceMutation>"
    },
    {
        "name": "originalAmount",
        "baseName": "originalAmount",
        "type": "Amount"
    },
    {
        "name": "reason",
        "baseName": "reason",
        "type": "TransferEvent.ReasonEnum"
    },
    {
        "name": "schemeTraceID",
        "baseName": "schemeTraceID",
        "type": "string"
    },
    {
        "name": "schemeUniqueTransactionID",
        "baseName": "schemeUniqueTransactionID",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "TransferEvent.StatusEnum"
    },
    {
        "name": "transactionId",
        "baseName": "transactionId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransferEvent.TypeEnum"
    },
    {
        "name": "updateDate",
        "baseName": "updateDate",
        "type": "Date"
    },
    {
        "name": "valueDate",
        "baseName": "valueDate",
        "type": "Date"
    }
];
(function (TransferEvent) {
    let ReasonEnum;
    (function (ReasonEnum) {
        ReasonEnum["AmountLimitExceeded"] = "amountLimitExceeded";
        ReasonEnum["Approved"] = "approved";
        ReasonEnum["BalanceAccountTemporarilyBlockedByTransactionRule"] = "balanceAccountTemporarilyBlockedByTransactionRule";
        ReasonEnum["CounterpartyAccountBlocked"] = "counterpartyAccountBlocked";
        ReasonEnum["CounterpartyAccountClosed"] = "counterpartyAccountClosed";
        ReasonEnum["CounterpartyAccountNotFound"] = "counterpartyAccountNotFound";
        ReasonEnum["CounterpartyAddressRequired"] = "counterpartyAddressRequired";
        ReasonEnum["CounterpartyBankTimedOut"] = "counterpartyBankTimedOut";
        ReasonEnum["CounterpartyBankUnavailable"] = "counterpartyBankUnavailable";
        ReasonEnum["DeclinedByTransactionRule"] = "declinedByTransactionRule";
        ReasonEnum["Error"] = "error";
        ReasonEnum["NotEnoughBalance"] = "notEnoughBalance";
        ReasonEnum["RefusedByCounterpartyBank"] = "refusedByCounterpartyBank";
        ReasonEnum["RouteNotFound"] = "routeNotFound";
        ReasonEnum["ScaFailed"] = "scaFailed";
        ReasonEnum["Unknown"] = "unknown";
    })(ReasonEnum = TransferEvent.ReasonEnum || (TransferEvent.ReasonEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["ApprovalPending"] = "approvalPending";
        StatusEnum["AtmWithdrawal"] = "atmWithdrawal";
        StatusEnum["AtmWithdrawalReversalPending"] = "atmWithdrawalReversalPending";
        StatusEnum["AtmWithdrawalReversed"] = "atmWithdrawalReversed";
        StatusEnum["AuthAdjustmentAuthorised"] = "authAdjustmentAuthorised";
        StatusEnum["AuthAdjustmentError"] = "authAdjustmentError";
        StatusEnum["AuthAdjustmentRefused"] = "authAdjustmentRefused";
        StatusEnum["Authorised"] = "authorised";
        StatusEnum["BankTransfer"] = "bankTransfer";
        StatusEnum["BankTransferPending"] = "bankTransferPending";
        StatusEnum["Booked"] = "booked";
        StatusEnum["BookingPending"] = "bookingPending";
        StatusEnum["Cancelled"] = "cancelled";
        StatusEnum["CapturePending"] = "capturePending";
        StatusEnum["CaptureReversalPending"] = "captureReversalPending";
        StatusEnum["CaptureReversed"] = "captureReversed";
        StatusEnum["Captured"] = "captured";
        StatusEnum["CapturedExternally"] = "capturedExternally";
        StatusEnum["Chargeback"] = "chargeback";
        StatusEnum["ChargebackExternally"] = "chargebackExternally";
        StatusEnum["ChargebackPending"] = "chargebackPending";
        StatusEnum["ChargebackReversalPending"] = "chargebackReversalPending";
        StatusEnum["ChargebackReversed"] = "chargebackReversed";
        StatusEnum["Credited"] = "credited";
        StatusEnum["DepositCorrection"] = "depositCorrection";
        StatusEnum["DepositCorrectionPending"] = "depositCorrectionPending";
        StatusEnum["Dispute"] = "dispute";
        StatusEnum["DisputeClosed"] = "disputeClosed";
        StatusEnum["DisputeExpired"] = "disputeExpired";
        StatusEnum["DisputeNeedsReview"] = "disputeNeedsReview";
        StatusEnum["Error"] = "error";
        StatusEnum["Expired"] = "expired";
        StatusEnum["Failed"] = "failed";
        StatusEnum["Fee"] = "fee";
        StatusEnum["FeePending"] = "feePending";
        StatusEnum["InternalTransfer"] = "internalTransfer";
        StatusEnum["InternalTransferPending"] = "internalTransferPending";
        StatusEnum["InvoiceDeduction"] = "invoiceDeduction";
        StatusEnum["InvoiceDeductionPending"] = "invoiceDeductionPending";
        StatusEnum["ManualCorrectionPending"] = "manualCorrectionPending";
        StatusEnum["ManuallyCorrected"] = "manuallyCorrected";
        StatusEnum["MatchedStatement"] = "matchedStatement";
        StatusEnum["MatchedStatementPending"] = "matchedStatementPending";
        StatusEnum["MerchantPayin"] = "merchantPayin";
        StatusEnum["MerchantPayinPending"] = "merchantPayinPending";
        StatusEnum["MerchantPayinReversed"] = "merchantPayinReversed";
        StatusEnum["MerchantPayinReversedPending"] = "merchantPayinReversedPending";
        StatusEnum["MiscCost"] = "miscCost";
        StatusEnum["MiscCostPending"] = "miscCostPending";
        StatusEnum["PaymentCost"] = "paymentCost";
        StatusEnum["PaymentCostPending"] = "paymentCostPending";
        StatusEnum["Received"] = "received";
        StatusEnum["RefundPending"] = "refundPending";
        StatusEnum["RefundReversalPending"] = "refundReversalPending";
        StatusEnum["RefundReversed"] = "refundReversed";
        StatusEnum["Refunded"] = "refunded";
        StatusEnum["RefundedExternally"] = "refundedExternally";
        StatusEnum["Refused"] = "refused";
        StatusEnum["ReserveAdjustment"] = "reserveAdjustment";
        StatusEnum["ReserveAdjustmentPending"] = "reserveAdjustmentPending";
        StatusEnum["Returned"] = "returned";
        StatusEnum["SecondChargeback"] = "secondChargeback";
        StatusEnum["SecondChargebackPending"] = "secondChargebackPending";
        StatusEnum["Undefined"] = "undefined";
    })(StatusEnum = TransferEvent.StatusEnum || (TransferEvent.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Accounting"] = "accounting";
        TypeEnum["Tracking"] = "tracking";
    })(TypeEnum = TransferEvent.TypeEnum || (TransferEvent.TypeEnum = {}));
})(TransferEvent = exports.TransferEvent || (exports.TransferEvent = {}));
