// @adyen/api-library/lib/src/typings/platformsNotificationConfiguration/notificationConfigurationDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationConfigurationDetails = void 0;
class NotificationConfigurationDetails {
    static getAttributeTypeMap() {
        return NotificationConfigurationDetails.attributeTypeMap;
    }
}
exports.NotificationConfigurationDetails = NotificationConfigurationDetails;
NotificationConfigurationDetails.discriminator = undefined;
NotificationConfigurationDetails.attributeTypeMap = [
    {
        "name": "active",
        "baseName": "active",
        "type": "boolean"
    },
    {
        "name": "apiVersion",
        "baseName": "apiVersion",
        "type": "number"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "eventConfigs",
        "baseName": "eventConfigs",
        "type": "Array<NotificationEventConfiguration>"
    },
    {
        "name": "hmacSignatureKey",
        "baseName": "hmacSignatureKey",
        "type": "string"
    },
    {
        "name": "notificationId",
        "baseName": "notificationId",
        "type": "number"
    },
    {
        "name": "notifyPassword",
        "baseName": "notifyPassword",
        "type": "string"
    },
    {
        "name": "notifyURL",
        "baseName": "notifyURL",
        "type": "string"
    },
    {
        "name": "notifyUsername",
        "baseName": "notifyUsername",
        "type": "string"
    },
    {
        "name": "sslProtocol",
        "baseName": "sslProtocol",
        "type": "NotificationConfigurationDetails.SslProtocolEnum"
    }
];
(function (NotificationConfigurationDetails) {
    let SslProtocolEnum;
    (function (SslProtocolEnum) {
        SslProtocolEnum["Tlsv12"] = "TLSv12";
        SslProtocolEnum["Tlsv13"] = "TLSv13";
    })(SslProtocolEnum = NotificationConfigurationDetails.SslProtocolEnum || (NotificationConfigurationDetails.SslProtocolEnum = {}));
})(NotificationConfigurationDetails = exports.NotificationConfigurationDetails || (exports.NotificationConfigurationDetails = {}));
