// @adyen/api-library/lib/src/typings/configurationWebhooks/sweepConfigurationV2.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SweepConfigurationV2 = void 0;
class SweepConfigurationV2 {
    static getAttributeTypeMap() {
        return SweepConfigurationV2.attributeTypeMap;
    }
}
exports.SweepConfigurationV2 = SweepConfigurationV2;
SweepConfigurationV2.discriminator = undefined;
SweepConfigurationV2.attributeTypeMap = [
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "SweepCounterparty"
    },
    {
        "name": "currency",
        "baseName": "currency",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "reason",
        "baseName": "reason",
        "type": "SweepConfigurationV2.ReasonEnum"
    },
    {
        "name": "schedule",
        "baseName": "schedule",
        "type": "SweepSchedule"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "SweepConfigurationV2.StatusEnum"
    },
    {
        "name": "sweepAmount",
        "baseName": "sweepAmount",
        "type": "Amount"
    },
    {
        "name": "targetAmount",
        "baseName": "targetAmount",
        "type": "Amount"
    },
    {
        "name": "triggerAmount",
        "baseName": "triggerAmount",
        "type": "Amount"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "SweepConfigurationV2.TypeEnum"
    }
];
(function (SweepConfigurationV2) {
    let ReasonEnum;
    (function (ReasonEnum) {
        ReasonEnum["AmountLimitExceeded"] = "amountLimitExceeded";
        ReasonEnum["Approved"] = "approved";
        ReasonEnum["CounterpartyAccountBlocked"] = "counterpartyAccountBlocked";
        ReasonEnum["CounterpartyAccountClosed"] = "counterpartyAccountClosed";
        ReasonEnum["CounterpartyAccountNotFound"] = "counterpartyAccountNotFound";
        ReasonEnum["CounterpartyAddressRequired"] = "counterpartyAddressRequired";
        ReasonEnum["CounterpartyBankTimedOut"] = "counterpartyBankTimedOut";
        ReasonEnum["CounterpartyBankUnavailable"] = "counterpartyBankUnavailable";
        ReasonEnum["Error"] = "error";
        ReasonEnum["NotEnoughBalance"] = "notEnoughBalance";
        ReasonEnum["RefusedByCounterpartyBank"] = "refusedByCounterpartyBank";
        ReasonEnum["RouteNotFound"] = "routeNotFound";
        ReasonEnum["Unknown"] = "unknown";
    })(ReasonEnum = SweepConfigurationV2.ReasonEnum || (SweepConfigurationV2.ReasonEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Inactive"] = "inactive";
    })(StatusEnum = SweepConfigurationV2.StatusEnum || (SweepConfigurationV2.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Pull"] = "pull";
        TypeEnum["Push"] = "push";
    })(TypeEnum = SweepConfigurationV2.TypeEnum || (SweepConfigurationV2.TypeEnum = {}));
})(SweepConfigurationV2 = exports.SweepConfigurationV2 || (exports.SweepConfigurationV2 = {}));
