// @adyen/api-library/lib/src/typings/configurationWebhooks/sweepConfigurationNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SweepConfigurationNotificationRequest = void 0;
class SweepConfigurationNotificationRequest {
    static getAttributeTypeMap() {
        return SweepConfigurationNotificationRequest.attributeTypeMap;
    }
}
exports.SweepConfigurationNotificationRequest = SweepConfigurationNotificationRequest;
SweepConfigurationNotificationRequest.discriminator = undefined;
SweepConfigurationNotificationRequest.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "SweepConfigurationNotificationData"
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "SweepConfigurationNotificationRequest.TypeEnum"
    }
];
(function (SweepConfigurationNotificationRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Created"] = "balancePlatform.balanceAccountSweep.created";
        TypeEnum["Updated"] = "balancePlatform.balanceAccountSweep.updated";
        TypeEnum["Deleted"] = "balancePlatform.balanceAccountSweep.deleted";
    })(TypeEnum = SweepConfigurationNotificationRequest.TypeEnum || (SweepConfigurationNotificationRequest.TypeEnum = {}));
})(SweepConfigurationNotificationRequest = exports.SweepConfigurationNotificationRequest || (exports.SweepConfigurationNotificationRequest = {}));
