// @adyen/api-library/lib/src/typings/transferWebhooks/transferData.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferData = void 0;
class TransferData {
    static getAttributeTypeMap() {
        return TransferData.attributeTypeMap;
    }
}
exports.TransferData = TransferData;
TransferData.discriminator = undefined;
TransferData.attributeTypeMap = [
    {
        "name": "accountHolder",
        "baseName": "accountHolder",
        "type": "ResourceReference"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "balanceAccount",
        "baseName": "balanceAccount",
        "type": "ResourceReference"
    },
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "balancePlatform",
        "baseName": "balancePlatform",
        "type": "string"
    },
    {
        "name": "balances",
        "baseName": "balances",
        "type": "Array<BalanceMutation>"
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "TransferData.CategoryEnum"
    },
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "CounterpartyV3"
    },
    {
        "name": "creationDate",
        "baseName": "creationDate",
        "type": "Date"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "direction",
        "baseName": "direction",
        "type": "TransferData.DirectionEnum"
    },
    {
        "name": "events",
        "baseName": "events",
        "type": "Array<TransferEvent>"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "modificationMerchantReference",
        "baseName": "modificationMerchantReference",
        "type": "string"
    },
    {
        "name": "modificationPspReference",
        "baseName": "modificationPspReference",
        "type": "string"
    },
    {
        "name": "panEntryMode",
        "baseName": "panEntryMode",
        "type": "TransferData.PanEntryModeEnum"
    },
    {
        "name": "paymentInstrument",
        "baseName": "paymentInstrument",
        "type": "PaymentInstrument"
    },
    {
        "name": "paymentInstrumentId",
        "baseName": "paymentInstrumentId",
        "type": "string"
    },
    {
        "name": "paymentMerchantReference",
        "baseName": "paymentMerchantReference",
        "type": "string"
    },
    {
        "name": "platformPaymentType",
        "baseName": "platformPaymentType",
        "type": "TransferData.PlatformPaymentTypeEnum"
    },
    {
        "name": "priority",
        "baseName": "priority",
        "type": "TransferData.PriorityEnum"
    },
    {
        "name": "processingType",
        "baseName": "processingType",
        "type": "TransferData.ProcessingTypeEnum"
    },
    {
        "name": "pspPaymentReference",
        "baseName": "pspPaymentReference",
        "type": "string"
    },
    {
        "name": "reason",
        "baseName": "reason",
        "type": "TransferData.ReasonEnum"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "referenceForBeneficiary",
        "baseName": "referenceForBeneficiary",
        "type": "string"
    },
    {
        "name": "relayedAuthorisationData",
        "baseName": "relayedAuthorisationData",
        "type": "RelayedAuthorisationData"
    },
    {
        "name": "sequenceNumber",
        "baseName": "sequenceNumber",
        "type": "number"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "TransferData.StatusEnum"
    },
    {
        "name": "tracking",
        "baseName": "tracking",
        "type": "TransferNotificationTransferTracking"
    },
    {
        "name": "transactionId",
        "baseName": "transactionId",
        "type": "string"
    },
    {
        "name": "transactionRulesResult",
        "baseName": "transactionRulesResult",
        "type": "TransactionRulesResult"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransferData.TypeEnum"
    },
    {
        "name": "validationFacts",
        "baseName": "validationFacts",
        "type": "Array<TransferNotificationValidationFact>"
    }
];
(function (TransferData) {
    let CategoryEnum;
    (function (CategoryEnum) {
        CategoryEnum["Bank"] = "bank";
        CategoryEnum["Internal"] = "internal";
        CategoryEnum["IssuedCard"] = "issuedCard";
        CategoryEnum["PlatformPayment"] = "platformPayment";
    })(CategoryEnum = TransferData.CategoryEnum || (TransferData.CategoryEnum = {}));
    let DirectionEnum;
    (function (DirectionEnum) {
        DirectionEnum["Incoming"] = "incoming";
        DirectionEnum["Outgoing"] = "outgoing";
    })(DirectionEnum = TransferData.DirectionEnum || (TransferData.DirectionEnum = {}));
    let PanEntryModeEnum;
    (function (PanEntryModeEnum) {
        PanEntryModeEnum["Chip"] = "chip";
        PanEntryModeEnum["Cof"] = "cof";
        PanEntryModeEnum["Contactless"] = "contactless";
        PanEntryModeEnum["Ecommerce"] = "ecommerce";
        PanEntryModeEnum["Magstripe"] = "magstripe";
        PanEntryModeEnum["Manual"] = "manual";
        PanEntryModeEnum["Token"] = "token";
    })(PanEntryModeEnum = TransferData.PanEntryModeEnum || (TransferData.PanEntryModeEnum = {}));
    let PlatformPaymentTypeEnum;
    (function (PlatformPaymentTypeEnum) {
        PlatformPaymentTypeEnum["BalanceAccount"] = "BalanceAccount";
        PlatformPaymentTypeEnum["Commission"] = "Commission";
        PlatformPaymentTypeEnum["Default"] = "Default";
        PlatformPaymentTypeEnum["PaymentFee"] = "PaymentFee";
        PlatformPaymentTypeEnum["Vat"] = "VAT";
    })(PlatformPaymentTypeEnum = TransferData.PlatformPaymentTypeEnum || (TransferData.PlatformPaymentTypeEnum = {}));
    let PriorityEnum;
    (function (PriorityEnum) {
        PriorityEnum["CrossBorder"] = "crossBorder";
        PriorityEnum["DirectDebit"] = "directDebit";
        PriorityEnum["Fast"] = "fast";
        PriorityEnum["Instant"] = "instant";
        PriorityEnum["Internal"] = "internal";
        PriorityEnum["Regular"] = "regular";
        PriorityEnum["Wire"] = "wire";
    })(PriorityEnum = TransferData.PriorityEnum || (TransferData.PriorityEnum = {}));
    let ProcessingTypeEnum;
    (function (ProcessingTypeEnum) {
        ProcessingTypeEnum["AtmWithdraw"] = "atmWithdraw";
        ProcessingTypeEnum["BalanceInquiry"] = "balanceInquiry";
        ProcessingTypeEnum["Ecommerce"] = "ecommerce";
        ProcessingTypeEnum["Moto"] = "moto";
        ProcessingTypeEnum["Pos"] = "pos";
        ProcessingTypeEnum["PurchaseWithCashback"] = "purchaseWithCashback";
        ProcessingTypeEnum["Recurring"] = "recurring";
        ProcessingTypeEnum["Token"] = "token";
    })(ProcessingTypeEnum = TransferData.ProcessingTypeEnum || (TransferData.ProcessingTypeEnum = {}));
    let ReasonEnum;
    (function (ReasonEnum) {
        ReasonEnum["AmountLimitExceeded"] = "amountLimitExceeded";
        ReasonEnum["Approved"] = "approved";
        ReasonEnum["CounterpartyAccountBlocked"] = "counterpartyAccountBlocked";
        ReasonEnum["CounterpartyAccountClosed"] = "counterpartyAccountClosed";
        ReasonEnum["CounterpartyAccountNotFound"] = "counterpartyAccountNotFound";
        ReasonEnum["CounterpartyAddressRequired"] = "counterpartyAddressRequired";
        ReasonEnum["CounterpartyBankTimedOut"] = "counterpartyBankTimedOut";
        ReasonEnum["CounterpartyBankUnavailable"] = "counterpartyBankUnavailable";
        ReasonEnum["Error"] = "error";
        ReasonEnum["NotEnoughBalance"] = "notEnoughBalance";
        ReasonEnum["RefusedByCounterpartyBank"] = "refusedByCounterpartyBank";
        ReasonEnum["RouteNotFound"] = "routeNotFound";
        ReasonEnum["Unknown"] = "unknown";
    })(ReasonEnum = TransferData.ReasonEnum || (TransferData.ReasonEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["ApprovalPending"] = "approvalPending";
        StatusEnum["AtmWithdrawal"] = "atmWithdrawal";
        StatusEnum["AtmWithdrawalReversalPending"] = "atmWithdrawalReversalPending";
        StatusEnum["AtmWithdrawalReversed"] = "atmWithdrawalReversed";
        StatusEnum["AuthAdjustmentAuthorised"] = "authAdjustmentAuthorised";
        StatusEnum["AuthAdjustmentError"] = "authAdjustmentError";
        StatusEnum["AuthAdjustmentRefused"] = "authAdjustmentRefused";
        StatusEnum["Authorised"] = "authorised";
        StatusEnum["BankTransfer"] = "bankTransfer";
        StatusEnum["BankTransferPending"] = "bankTransferPending";
        StatusEnum["Booked"] = "booked";
        StatusEnum["BookingPending"] = "bookingPending";
        StatusEnum["Cancelled"] = "cancelled";
        StatusEnum["CapturePending"] = "capturePending";
        StatusEnum["CaptureReversalPending"] = "captureReversalPending";
        StatusEnum["CaptureReversed"] = "captureReversed";
        StatusEnum["Captured"] = "captured";
        StatusEnum["CapturedExternally"] = "capturedExternally";
        StatusEnum["Chargeback"] = "chargeback";
        StatusEnum["ChargebackExternally"] = "chargebackExternally";
        StatusEnum["ChargebackPending"] = "chargebackPending";
        StatusEnum["ChargebackReversalPending"] = "chargebackReversalPending";
        StatusEnum["ChargebackReversed"] = "chargebackReversed";
        StatusEnum["Credited"] = "credited";
        StatusEnum["DepositCorrection"] = "depositCorrection";
        StatusEnum["DepositCorrectionPending"] = "depositCorrectionPending";
        StatusEnum["Dispute"] = "dispute";
        StatusEnum["DisputeClosed"] = "disputeClosed";
        StatusEnum["DisputeExpired"] = "disputeExpired";
        StatusEnum["DisputeNeedsReview"] = "disputeNeedsReview";
        StatusEnum["Error"] = "error";
        StatusEnum["Expired"] = "expired";
        StatusEnum["Failed"] = "failed";
        StatusEnum["Fee"] = "fee";
        StatusEnum["FeePending"] = "feePending";
        StatusEnum["InternalTransfer"] = "internalTransfer";
        StatusEnum["InternalTransferPending"] = "internalTransferPending";
        StatusEnum["InvoiceDeduction"] = "invoiceDeduction";
        StatusEnum["InvoiceDeductionPending"] = "invoiceDeductionPending";
        StatusEnum["ManualCorrectionPending"] = "manualCorrectionPending";
        StatusEnum["ManuallyCorrected"] = "manuallyCorrected";
        StatusEnum["MatchedStatement"] = "matchedStatement";
        StatusEnum["MatchedStatementPending"] = "matchedStatementPending";
        StatusEnum["MerchantPayin"] = "merchantPayin";
        StatusEnum["MerchantPayinPending"] = "merchantPayinPending";
        StatusEnum["MerchantPayinReversed"] = "merchantPayinReversed";
        StatusEnum["MerchantPayinReversedPending"] = "merchantPayinReversedPending";
        StatusEnum["MiscCost"] = "miscCost";
        StatusEnum["MiscCostPending"] = "miscCostPending";
        StatusEnum["OperationAuthorized"] = "operationAuthorized";
        StatusEnum["OperationBooked"] = "operationBooked";
        StatusEnum["OperationPending"] = "operationPending";
        StatusEnum["OperationReceived"] = "operationReceived";
        StatusEnum["PaymentCost"] = "paymentCost";
        StatusEnum["PaymentCostPending"] = "paymentCostPending";
        StatusEnum["Received"] = "received";
        StatusEnum["RefundPending"] = "refundPending";
        StatusEnum["RefundReversalPending"] = "refundReversalPending";
        StatusEnum["RefundReversed"] = "refundReversed";
        StatusEnum["Refunded"] = "refunded";
        StatusEnum["RefundedExternally"] = "refundedExternally";
        StatusEnum["Refused"] = "refused";
        StatusEnum["ReserveAdjustment"] = "reserveAdjustment";
        StatusEnum["ReserveAdjustmentPending"] = "reserveAdjustmentPending";
        StatusEnum["Returned"] = "returned";
        StatusEnum["SecondChargeback"] = "secondChargeback";
        StatusEnum["SecondChargebackPending"] = "secondChargebackPending";
        StatusEnum["Undefined"] = "undefined";
    })(StatusEnum = TransferData.StatusEnum || (TransferData.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AtmWithdrawal"] = "atmWithdrawal";
        TypeEnum["AtmWithdrawalReversal"] = "atmWithdrawalReversal";
        TypeEnum["BalanceAdjustment"] = "balanceAdjustment";
        TypeEnum["BalanceRollover"] = "balanceRollover";
        TypeEnum["BankTransfer"] = "bankTransfer";
        TypeEnum["Capture"] = "capture";
        TypeEnum["CaptureReversal"] = "captureReversal";
        TypeEnum["CardTransfer"] = "cardTransfer";
        TypeEnum["Chargeback"] = "chargeback";
        TypeEnum["ChargebackReversal"] = "chargebackReversal";
        TypeEnum["DepositCorrection"] = "depositCorrection";
        TypeEnum["Fee"] = "fee";
        TypeEnum["Grant"] = "grant";
        TypeEnum["Installment"] = "installment";
        TypeEnum["InstallmentReversal"] = "installmentReversal";
        TypeEnum["InternalTransfer"] = "internalTransfer";
        TypeEnum["InvoiceDeduction"] = "invoiceDeduction";
        TypeEnum["Leftover"] = "leftover";
        TypeEnum["ManualCorrection"] = "manualCorrection";
        TypeEnum["MiscCost"] = "miscCost";
        TypeEnum["Payment"] = "payment";
        TypeEnum["PaymentCost"] = "paymentCost";
        TypeEnum["Refund"] = "refund";
        TypeEnum["RefundReversal"] = "refundReversal";
        TypeEnum["Repayment"] = "repayment";
        TypeEnum["ReserveAdjustment"] = "reserveAdjustment";
        TypeEnum["SecondChargeback"] = "secondChargeback";
    })(TypeEnum = TransferData.TypeEnum || (TransferData.TypeEnum = {}));
})(TransferData = exports.TransferData || (exports.TransferData = {}));
