// @adyen/api-library/lib/src/services/payout/initializationApi.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InitializationApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/payout/models");
class InitializationApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://pal-test.adyen.com/pal/servlet/Payout/v68";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Store payout details
    * @param storeDetailRequest {@link StoreDetailRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link StoreDetailResponse }
    */
    async storeDetail(storeDetailRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/storeDetail`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(storeDetailRequest, "StoreDetailRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "StoreDetailResponse");
    }
    /**
    * @summary Store details and submit a payout
    * @param storeDetailAndSubmitRequest {@link StoreDetailAndSubmitRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link StoreDetailAndSubmitResponse }
    */
    async storeDetailAndSubmitThirdParty(storeDetailAndSubmitRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/storeDetailAndSubmitThirdParty`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(storeDetailAndSubmitRequest, "StoreDetailAndSubmitRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "StoreDetailAndSubmitResponse");
    }
    /**
    * @summary Submit a payout
    * @param submitRequest {@link SubmitRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link SubmitResponse }
    */
    async submitThirdParty(submitRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/submitThirdParty`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(submitRequest, "SubmitRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "SubmitResponse");
    }
}
exports.InitializationApi = InitializationApi;
