// @adyen/api-library/lib/src/services/payout/instantPayoutsApi.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstantPayoutsApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/payout/models");
class InstantPayoutsApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://pal-test.adyen.com/pal/servlet/Payout/v68";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Make an instant card payout
    * @param payoutRequest {@link PayoutRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link PayoutResponse }
    */
    async payout(payoutRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/payout`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(payoutRequest, "PayoutRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "PayoutResponse");
    }
}
exports.InstantPayoutsApi = InstantPayoutsApi;
