// @adyen/api-library/lib/src/typings/payout/storeDetailRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoreDetailRequest = void 0;
class StoreDetailRequest {
    static getAttributeTypeMap() {
        return StoreDetailRequest.attributeTypeMap;
    }
}
exports.StoreDetailRequest = StoreDetailRequest;
StoreDetailRequest.discriminator = undefined;
StoreDetailRequest.attributeTypeMap = [
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "bank",
        "baseName": "bank",
        "type": "BankAccount"
    },
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "Address"
    },
    {
        "name": "card",
        "baseName": "card",
        "type": "Card"
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "string"
    },
    {
        "name": "entityType",
        "baseName": "entityType",
        "type": "StoreDetailRequest.EntityTypeEnum"
    },
    {
        "name": "fraudOffset",
        "baseName": "fraudOffset",
        "type": "number"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "nationality",
        "baseName": "nationality",
        "type": "string"
    },
    {
        "name": "recurring",
        "baseName": "recurring",
        "type": "Recurring"
    },
    {
        "name": "selectedBrand",
        "baseName": "selectedBrand",
        "type": "string"
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string"
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name"
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string"
    },
    {
        "name": "socialSecurityNumber",
        "baseName": "socialSecurityNumber",
        "type": "string"
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string"
    }
];
(function (StoreDetailRequest) {
    let EntityTypeEnum;
    (function (EntityTypeEnum) {
        EntityTypeEnum["NaturalPerson"] = "NaturalPerson";
        EntityTypeEnum["Company"] = "Company";
    })(EntityTypeEnum = StoreDetailRequest.EntityTypeEnum || (StoreDetailRequest.EntityTypeEnum = {}));
})(StoreDetailRequest = exports.StoreDetailRequest || (exports.StoreDetailRequest = {}));
