// @adyen/api-library/lib/src/typings/managementWebhooks/verificationError.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerificationError = void 0;
class VerificationError {
    static getAttributeTypeMap() {
        return VerificationError.attributeTypeMap;
    }
}
exports.VerificationError = VerificationError;
VerificationError.discriminator = undefined;
VerificationError.attributeTypeMap = [
    {
        "name": "code",
        "baseName": "code",
        "type": "string"
    },
    {
        "name": "message",
        "baseName": "message",
        "type": "string"
    },
    {
        "name": "remediatingActions",
        "baseName": "remediatingActions",
        "type": "Array<RemediatingAction>"
    },
    {
        "name": "subErrors",
        "baseName": "subErrors",
        "type": "Array<VerificationErrorRecursive>"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "VerificationError.TypeEnum"
    }
];
(function (VerificationError) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["DataMissing"] = "dataMissing";
        TypeEnum["InvalidInput"] = "invalidInput";
        TypeEnum["PendingStatus"] = "pendingStatus";
    })(TypeEnum = VerificationError.TypeEnum || (VerificationError.TypeEnum = {}));
})(VerificationError = exports.VerificationError || (exports.VerificationError = {}));
