// @adyen/api-library/lib/src/typings/platformsFund/fieldType.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FieldType = void 0;
class FieldType {
    static getAttributeTypeMap() {
        return FieldType.attributeTypeMap;
    }
}
exports.FieldType = FieldType;
FieldType.discriminator = undefined;
FieldType.attributeTypeMap = [
    {
        "name": "field",
        "baseName": "field",
        "type": "string"
    },
    {
        "name": "fieldName",
        "baseName": "fieldName",
        "type": "FieldType.FieldNameEnum"
    },
    {
        "name": "shareholderCode",
        "baseName": "shareholderCode",
        "type": "string"
    }
];
(function (FieldType) {
    let FieldNameEnum;
    (function (FieldNameEnum) {
        FieldNameEnum["AccountCode"] = "accountCode";
        FieldNameEnum["AccountHolderCode"] = "accountHolderCode";
        FieldNameEnum["AccountHolderDetails"] = "accountHolderDetails";
        FieldNameEnum["AccountNumber"] = "accountNumber";
        FieldNameEnum["AccountStateType"] = "accountStateType";
        FieldNameEnum["AccountStatus"] = "accountStatus";
        FieldNameEnum["AccountType"] = "accountType";
        FieldNameEnum["Address"] = "address";
        FieldNameEnum["BalanceAccount"] = "balanceAccount";
        FieldNameEnum["BalanceAccountActive"] = "balanceAccountActive";
        FieldNameEnum["BalanceAccountCode"] = "balanceAccountCode";
        FieldNameEnum["BalanceAccountId"] = "balanceAccountId";
        FieldNameEnum["BankAccount"] = "bankAccount";
        FieldNameEnum["BankAccountCode"] = "bankAccountCode";
        FieldNameEnum["BankAccountName"] = "bankAccountName";
        FieldNameEnum["BankAccountUuid"] = "bankAccountUUID";
        FieldNameEnum["BankBicSwift"] = "bankBicSwift";
        FieldNameEnum["BankCity"] = "bankCity";
        FieldNameEnum["BankCode"] = "bankCode";
        FieldNameEnum["BankName"] = "bankName";
        FieldNameEnum["BankStatement"] = "bankStatement";
        FieldNameEnum["BranchCode"] = "branchCode";
        FieldNameEnum["BusinessContact"] = "businessContact";
        FieldNameEnum["CardToken"] = "cardToken";
        FieldNameEnum["CheckCode"] = "checkCode";
        FieldNameEnum["City"] = "city";
        FieldNameEnum["CompanyRegistration"] = "companyRegistration";
        FieldNameEnum["ConstitutionalDocument"] = "constitutionalDocument";
        FieldNameEnum["Controller"] = "controller";
        FieldNameEnum["Country"] = "country";
        FieldNameEnum["CountryCode"] = "countryCode";
        FieldNameEnum["Currency"] = "currency";
        FieldNameEnum["CurrencyCode"] = "currencyCode";
        FieldNameEnum["DateOfBirth"] = "dateOfBirth";
        FieldNameEnum["Description"] = "description";
        FieldNameEnum["DestinationAccountCode"] = "destinationAccountCode";
        FieldNameEnum["Document"] = "document";
        FieldNameEnum["DocumentContent"] = "documentContent";
        FieldNameEnum["DocumentExpirationDate"] = "documentExpirationDate";
        FieldNameEnum["DocumentIssuerCountry"] = "documentIssuerCountry";
        FieldNameEnum["DocumentIssuerState"] = "documentIssuerState";
        FieldNameEnum["DocumentName"] = "documentName";
        FieldNameEnum["DocumentNumber"] = "documentNumber";
        FieldNameEnum["DocumentType"] = "documentType";
        FieldNameEnum["DoingBusinessAs"] = "doingBusinessAs";
        FieldNameEnum["DrivingLicence"] = "drivingLicence";
        FieldNameEnum["DrivingLicenceBack"] = "drivingLicenceBack";
        FieldNameEnum["DrivingLicenceFront"] = "drivingLicenceFront";
        FieldNameEnum["DrivingLicense"] = "drivingLicense";
        FieldNameEnum["Email"] = "email";
        FieldNameEnum["FirstName"] = "firstName";
        FieldNameEnum["FormType"] = "formType";
        FieldNameEnum["FullPhoneNumber"] = "fullPhoneNumber";
        FieldNameEnum["Gender"] = "gender";
        FieldNameEnum["HopWebserviceUser"] = "hopWebserviceUser";
        FieldNameEnum["HouseNumberOrName"] = "houseNumberOrName";
        FieldNameEnum["Iban"] = "iban";
        FieldNameEnum["IdCard"] = "idCard";
        FieldNameEnum["IdCardBack"] = "idCardBack";
        FieldNameEnum["IdCardFront"] = "idCardFront";
        FieldNameEnum["IdNumber"] = "idNumber";
        FieldNameEnum["IdentityDocument"] = "identityDocument";
        FieldNameEnum["IndividualDetails"] = "individualDetails";
        FieldNameEnum["Infix"] = "infix";
        FieldNameEnum["JobTitle"] = "jobTitle";
        FieldNameEnum["LastName"] = "lastName";
        FieldNameEnum["LastReviewDate"] = "lastReviewDate";
        FieldNameEnum["LegalArrangement"] = "legalArrangement";
        FieldNameEnum["LegalArrangementCode"] = "legalArrangementCode";
        FieldNameEnum["LegalArrangementEntity"] = "legalArrangementEntity";
        FieldNameEnum["LegalArrangementEntityCode"] = "legalArrangementEntityCode";
        FieldNameEnum["LegalArrangementLegalForm"] = "legalArrangementLegalForm";
        FieldNameEnum["LegalArrangementMember"] = "legalArrangementMember";
        FieldNameEnum["LegalArrangementMembers"] = "legalArrangementMembers";
        FieldNameEnum["LegalArrangementName"] = "legalArrangementName";
        FieldNameEnum["LegalArrangementReference"] = "legalArrangementReference";
        FieldNameEnum["LegalArrangementRegistrationNumber"] = "legalArrangementRegistrationNumber";
        FieldNameEnum["LegalArrangementTaxNumber"] = "legalArrangementTaxNumber";
        FieldNameEnum["LegalArrangementType"] = "legalArrangementType";
        FieldNameEnum["LegalBusinessName"] = "legalBusinessName";
        FieldNameEnum["LegalEntity"] = "legalEntity";
        FieldNameEnum["LegalEntityType"] = "legalEntityType";
        FieldNameEnum["Logo"] = "logo";
        FieldNameEnum["MerchantAccount"] = "merchantAccount";
        FieldNameEnum["MerchantCategoryCode"] = "merchantCategoryCode";
        FieldNameEnum["MerchantHouseNumber"] = "merchantHouseNumber";
        FieldNameEnum["MerchantReference"] = "merchantReference";
        FieldNameEnum["MicroDeposit"] = "microDeposit";
        FieldNameEnum["Name"] = "name";
        FieldNameEnum["Nationality"] = "nationality";
        FieldNameEnum["OriginalReference"] = "originalReference";
        FieldNameEnum["OwnerCity"] = "ownerCity";
        FieldNameEnum["OwnerCountryCode"] = "ownerCountryCode";
        FieldNameEnum["OwnerDateOfBirth"] = "ownerDateOfBirth";
        FieldNameEnum["OwnerHouseNumberOrName"] = "ownerHouseNumberOrName";
        FieldNameEnum["OwnerName"] = "ownerName";
        FieldNameEnum["OwnerPostalCode"] = "ownerPostalCode";
        FieldNameEnum["OwnerState"] = "ownerState";
        FieldNameEnum["OwnerStreet"] = "ownerStreet";
        FieldNameEnum["Passport"] = "passport";
        FieldNameEnum["PassportNumber"] = "passportNumber";
        FieldNameEnum["PayoutMethodCode"] = "payoutMethodCode";
        FieldNameEnum["PayoutSchedule"] = "payoutSchedule";
        FieldNameEnum["PciSelfAssessment"] = "pciSelfAssessment";
        FieldNameEnum["PersonalData"] = "personalData";
        FieldNameEnum["PhoneCountryCode"] = "phoneCountryCode";
        FieldNameEnum["PhoneNumber"] = "phoneNumber";
        FieldNameEnum["PostalCode"] = "postalCode";
        FieldNameEnum["PrimaryCurrency"] = "primaryCurrency";
        FieldNameEnum["Reason"] = "reason";
        FieldNameEnum["RegistrationNumber"] = "registrationNumber";
        FieldNameEnum["ReturnUrl"] = "returnUrl";
        FieldNameEnum["Schedule"] = "schedule";
        FieldNameEnum["Shareholder"] = "shareholder";
        FieldNameEnum["ShareholderCode"] = "shareholderCode";
        FieldNameEnum["ShareholderCodeAndSignatoryCode"] = "shareholderCodeAndSignatoryCode";
        FieldNameEnum["ShareholderCodeOrSignatoryCode"] = "shareholderCodeOrSignatoryCode";
        FieldNameEnum["ShareholderType"] = "shareholderType";
        FieldNameEnum["ShareholderTypes"] = "shareholderTypes";
        FieldNameEnum["ShopperInteraction"] = "shopperInteraction";
        FieldNameEnum["Signatory"] = "signatory";
        FieldNameEnum["SignatoryCode"] = "signatoryCode";
        FieldNameEnum["SocialSecurityNumber"] = "socialSecurityNumber";
        FieldNameEnum["SourceAccountCode"] = "sourceAccountCode";
        FieldNameEnum["SplitAccount"] = "splitAccount";
        FieldNameEnum["SplitConfigurationUuid"] = "splitConfigurationUUID";
        FieldNameEnum["SplitCurrency"] = "splitCurrency";
        FieldNameEnum["SplitValue"] = "splitValue";
        FieldNameEnum["Splits"] = "splits";
        FieldNameEnum["StateOrProvince"] = "stateOrProvince";
        FieldNameEnum["Status"] = "status";
        FieldNameEnum["StockExchange"] = "stockExchange";
        FieldNameEnum["StockNumber"] = "stockNumber";
        FieldNameEnum["StockTicker"] = "stockTicker";
        FieldNameEnum["Store"] = "store";
        FieldNameEnum["StoreDetail"] = "storeDetail";
        FieldNameEnum["StoreName"] = "storeName";
        FieldNameEnum["StoreReference"] = "storeReference";
        FieldNameEnum["Street"] = "street";
        FieldNameEnum["TaxId"] = "taxId";
        FieldNameEnum["Tier"] = "tier";
        FieldNameEnum["TierNumber"] = "tierNumber";
        FieldNameEnum["TransferCode"] = "transferCode";
        FieldNameEnum["UltimateParentCompany"] = "ultimateParentCompany";
        FieldNameEnum["UltimateParentCompanyAddressDetails"] = "ultimateParentCompanyAddressDetails";
        FieldNameEnum["UltimateParentCompanyAddressDetailsCountry"] = "ultimateParentCompanyAddressDetailsCountry";
        FieldNameEnum["UltimateParentCompanyBusinessDetails"] = "ultimateParentCompanyBusinessDetails";
        FieldNameEnum["UltimateParentCompanyBusinessDetailsLegalBusinessName"] = "ultimateParentCompanyBusinessDetailsLegalBusinessName";
        FieldNameEnum["UltimateParentCompanyBusinessDetailsRegistrationNumber"] = "ultimateParentCompanyBusinessDetailsRegistrationNumber";
        FieldNameEnum["UltimateParentCompanyCode"] = "ultimateParentCompanyCode";
        FieldNameEnum["UltimateParentCompanyStockExchange"] = "ultimateParentCompanyStockExchange";
        FieldNameEnum["UltimateParentCompanyStockNumber"] = "ultimateParentCompanyStockNumber";
        FieldNameEnum["UltimateParentCompanyStockNumberOrStockTicker"] = "ultimateParentCompanyStockNumberOrStockTicker";
        FieldNameEnum["UltimateParentCompanyStockTicker"] = "ultimateParentCompanyStockTicker";
        FieldNameEnum["Unknown"] = "unknown";
        FieldNameEnum["Value"] = "value";
        FieldNameEnum["VerificationType"] = "verificationType";
        FieldNameEnum["VirtualAccount"] = "virtualAccount";
        FieldNameEnum["VisaNumber"] = "visaNumber";
        FieldNameEnum["WebAddress"] = "webAddress";
        FieldNameEnum["Year"] = "year";
    })(FieldNameEnum = FieldType.FieldNameEnum || (FieldType.FieldNameEnum = {}));
})(FieldType = exports.FieldType || (exports.FieldType = {}));
