// @adyen/api-library/lib/src/typings/platformsFund/split.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Split = void 0;
class Split {
    static getAttributeTypeMap() {
        return Split.attributeTypeMap;
    }
}
exports.Split = Split;
Split.discriminator = undefined;
Split.attributeTypeMap = [
    {
        "name": "account",
        "baseName": "account",
        "type": "string"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "SplitAmount"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Split.TypeEnum"
    }
];
(function (Split) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BalanceAccount"] = "BalanceAccount";
        TypeEnum["Commission"] = "Commission";
        TypeEnum["Default"] = "Default";
        TypeEnum["MarketPlace"] = "MarketPlace";
        TypeEnum["PaymentFee"] = "PaymentFee";
        TypeEnum["Remainder"] = "Remainder";
        TypeEnum["Surcharge"] = "Surcharge";
        TypeEnum["Tip"] = "Tip";
        TypeEnum["Vat"] = "VAT";
        TypeEnum["Verification"] = "Verification";
    })(TypeEnum = Split.TypeEnum || (Split.TypeEnum = {}));
})(Split = exports.Split || (exports.Split = {}));
