// @adyen/api-library/lib/src/typings/configurationWebhooks/accountHolderNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountHolderNotificationRequest = void 0;
class AccountHolderNotificationRequest {
    static getAttributeTypeMap() {
        return AccountHolderNotificationRequest.attributeTypeMap;
    }
}
exports.AccountHolderNotificationRequest = AccountHolderNotificationRequest;
AccountHolderNotificationRequest.discriminator = undefined;
AccountHolderNotificationRequest.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "AccountHolderNotificationData"
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "AccountHolderNotificationRequest.TypeEnum"
    }
];
(function (AccountHolderNotificationRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Updated"] = "balancePlatform.accountHolder.updated";
        TypeEnum["Created"] = "balancePlatform.accountHolder.created";
    })(TypeEnum = AccountHolderNotificationRequest.TypeEnum || (AccountHolderNotificationRequest.TypeEnum = {}));
})(AccountHolderNotificationRequest = exports.AccountHolderNotificationRequest || (exports.AccountHolderNotificationRequest = {}));
