// @adyen/api-library/lib/src/typings/configurationWebhooks/paymentNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentNotificationRequest = void 0;
class PaymentNotificationRequest {
    static getAttributeTypeMap() {
        return PaymentNotificationRequest.attributeTypeMap;
    }
}
exports.PaymentNotificationRequest = PaymentNotificationRequest;
PaymentNotificationRequest.discriminator = undefined;
PaymentNotificationRequest.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "PaymentInstrumentNotificationData"
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PaymentNotificationRequest.TypeEnum"
    }
];
(function (PaymentNotificationRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Created"] = "balancePlatform.paymentInstrument.created";
        TypeEnum["Updated"] = "balancePlatform.paymentInstrument.updated";
    })(TypeEnum = PaymentNotificationRequest.TypeEnum || (PaymentNotificationRequest.TypeEnum = {}));
})(PaymentNotificationRequest = exports.PaymentNotificationRequest || (exports.PaymentNotificationRequest = {}));
