// @adyen/api-library/lib/src/services/payout/index.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const initializationApi_1 = require("./initializationApi");
const instantPayoutsApi_1 = require("./instantPayoutsApi");
const reviewingApi_1 = require("./reviewingApi");
const service_1 = __importDefault(require("../../service"));
class PayoutAPI extends service_1.default {
    constructor(client) {
        super(client);
    }
    get InitializationApi() {
        return new initializationApi_1.InitializationApi(this.client);
    }
    get InstantPayoutsApi() {
        return new instantPayoutsApi_1.InstantPayoutsApi(this.client);
    }
    get ReviewingApi() {
        return new reviewingApi_1.ReviewingApi(this.client);
    }
}
exports.default = PayoutAPI;
