// @adyen/api-library/lib/src/services/balancePlatform/accountHoldersApi.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountHoldersApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/balancePlatform/models");
class AccountHoldersApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://balanceplatform-api-test.adyen.com/bcl/v2";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get an account holder
    * @param id {@link string } The unique identifier of the account holder.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link AccountHolder }
    */
    async getAccountHolder(id, requestOptions) {
        const endpoint = `${this.baseUrl}/accountHolders/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "AccountHolder");
    }
    /**
    * @summary Get all balance accounts of an account holder
    * @param id {@link string } The unique identifier of the account holder.
    * @param offset {@link number } The number of items that you want to skip.
    * @param limit {@link number } The number of items returned per page, maximum 100 items. By default, the response returns 10 items per page.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link PaginatedBalanceAccountsResponse }
    */
    async getAllBalanceAccountsOfAccountHolder(id, requestOptions) {
        const endpoint = `${this.baseUrl}/accountHolders/{id}/balanceAccounts`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "PaginatedBalanceAccountsResponse");
    }
    /**
    * @summary Get a tax form
    * @param id {@link string } The unique identifier of the account holder.
    * @param formType {@link &#39;US1099k&#39; | &#39;US1099nec&#39; } The type of tax form you want to retrieve. Accepted values are **us1099k** and **us1099nec**
    * @param year {@link number } The tax year in YYYY format for the tax form you want to retrieve
    * @param requestOptions {@link IRequest.Options}
    * @return {@link GetTaxFormResponse }
    */
    async getTaxForm(id, requestOptions) {
        const endpoint = `${this.baseUrl}/accountHolders/{id}/taxForms`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "GetTaxFormResponse");
    }
    /**
    * @summary Update an account holder
    * @param id {@link string } The unique identifier of the account holder.
    * @param accountHolderUpdateRequest {@link AccountHolderUpdateRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link AccountHolder }
    */
    async updateAccountHolder(id, accountHolderUpdateRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/accountHolders/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(accountHolderUpdateRequest, "AccountHolderUpdateRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "AccountHolder");
    }
    /**
    * @summary Create an account holder
    * @param accountHolderInfo {@link AccountHolderInfo }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link AccountHolder }
    */
    async createAccountHolder(accountHolderInfo, requestOptions) {
        const endpoint = `${this.baseUrl}/accountHolders`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(accountHolderInfo, "AccountHolderInfo");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "AccountHolder");
    }
}
exports.AccountHoldersApi = AccountHoldersApi;
