// @adyen/api-library/lib/src/services/balancePlatform/grantOffersApi.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrantOffersApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/balancePlatform/models");
class GrantOffersApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://balanceplatform-api-test.adyen.com/bcl/v2";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get all available grant offers
    * @param accountHolderId {@link string } The unique identifier of the grant account.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link GrantOffers }
    */
    async getAllAvailableGrantOffers(requestOptions) {
        const endpoint = `${this.baseUrl}/grantOffers`;
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "GrantOffers");
    }
    /**
    * @summary Get a grant offer
    * @param grantOfferId {@link string } The unique identifier of the grant offer.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link GrantOffer }
    */
    async getGrantOffer(grantOfferId, requestOptions) {
        const endpoint = `${this.baseUrl}/grantOffers/{grantOfferId}`
            .replace("{" + "grantOfferId" + "}", encodeURIComponent(String(grantOfferId)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "GrantOffer");
    }
}
exports.GrantOffersApi = GrantOffersApi;
