// @adyen/api-library/lib/src/services/balancePlatform/index.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const accountHoldersApi_1 = require("./accountHoldersApi");
const balanceAccountsApi_1 = require("./balanceAccountsApi");
const bankAccountValidationApi_1 = require("./bankAccountValidationApi");
const grantAccountsApi_1 = require("./grantAccountsApi");
const grantOffersApi_1 = require("./grantOffersApi");
const networkTokensApi_1 = require("./networkTokensApi");
const paymentInstrumentGroupsApi_1 = require("./paymentInstrumentGroupsApi");
const paymentInstrumentsApi_1 = require("./paymentInstrumentsApi");
const platformApi_1 = require("./platformApi");
const transactionRulesApi_1 = require("./transactionRulesApi");
const transferRoutesApi_1 = require("./transferRoutesApi");
const service_1 = __importDefault(require("../../service"));
class BalancePlatformAPI extends service_1.default {
    constructor(client) {
        super(client);
    }
    get AccountHoldersApi() {
        return new accountHoldersApi_1.AccountHoldersApi(this.client);
    }
    get BalanceAccountsApi() {
        return new balanceAccountsApi_1.BalanceAccountsApi(this.client);
    }
    get BankAccountValidationApi() {
        return new bankAccountValidationApi_1.BankAccountValidationApi(this.client);
    }
    get GrantAccountsApi() {
        return new grantAccountsApi_1.GrantAccountsApi(this.client);
    }
    get GrantOffersApi() {
        return new grantOffersApi_1.GrantOffersApi(this.client);
    }
    get NetworkTokensApi() {
        return new networkTokensApi_1.NetworkTokensApi(this.client);
    }
    get PaymentInstrumentGroupsApi() {
        return new paymentInstrumentGroupsApi_1.PaymentInstrumentGroupsApi(this.client);
    }
    get PaymentInstrumentsApi() {
        return new paymentInstrumentsApi_1.PaymentInstrumentsApi(this.client);
    }
    get PlatformApi() {
        return new platformApi_1.PlatformApi(this.client);
    }
    get TransactionRulesApi() {
        return new transactionRulesApi_1.TransactionRulesApi(this.client);
    }
    get TransferRoutesApi() {
        return new transferRoutesApi_1.TransferRoutesApi(this.client);
    }
}
exports.default = BalancePlatformAPI;
