// @adyen/api-library/lib/src/services/balancePlatform/paymentInstrumentGroupsApi.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentInstrumentGroupsApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/balancePlatform/models");
class PaymentInstrumentGroupsApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://balanceplatform-api-test.adyen.com/bcl/v2";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get a payment instrument group
    * @param id {@link string } The unique identifier of the payment instrument group.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link PaymentInstrumentGroup }
    */
    async getPaymentInstrumentGroup(id, requestOptions) {
        const endpoint = `${this.baseUrl}/paymentInstrumentGroups/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "PaymentInstrumentGroup");
    }
    /**
    * @summary Get all transaction rules for a payment instrument group
    * @param id {@link string } The unique identifier of the payment instrument group.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link TransactionRulesResponse }
    */
    async getAllTransactionRulesForPaymentInstrumentGroup(id, requestOptions) {
        const endpoint = `${this.baseUrl}/paymentInstrumentGroups/{id}/transactionRules`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "TransactionRulesResponse");
    }
    /**
    * @summary Create a payment instrument group
    * @param paymentInstrumentGroupInfo {@link PaymentInstrumentGroupInfo }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link PaymentInstrumentGroup }
    */
    async createPaymentInstrumentGroup(paymentInstrumentGroupInfo, requestOptions) {
        const endpoint = `${this.baseUrl}/paymentInstrumentGroups`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(paymentInstrumentGroupInfo, "PaymentInstrumentGroupInfo");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "PaymentInstrumentGroup");
    }
}
exports.PaymentInstrumentGroupsApi = PaymentInstrumentGroupsApi;
