// @adyen/api-library/lib/src/services/balancePlatform/transferRoutesApi.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferRoutesApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/balancePlatform/models");
class TransferRoutesApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://balanceplatform-api-test.adyen.com/bcl/v2";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Calculate transfer routes
    * @param transferRouteRequest {@link TransferRouteRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link TransferRouteResponse }
    */
    async calculateTransferRoutes(transferRouteRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/transferRoutes/calculate`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(transferRouteRequest, "TransferRouteRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "TransferRouteResponse");
    }
}
exports.TransferRoutesApi = TransferRoutesApi;
