// @adyen/api-library/lib/src/services/transfers/capitalApi.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapitalApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/transfers/models");
class CapitalApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://balanceplatform-api-test.adyen.com/btl/v4";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get a capital account
    * @param counterpartyAccountHolderId {@link string } The counterparty account holder id.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link CapitalGrants }
    */
    async getCapitalAccount(requestOptions) {
        const endpoint = `${this.baseUrl}/grants`;
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "CapitalGrants");
    }
    /**
    * @summary Get grant reference details
    * @param id {@link string } The unique identifier of the grant.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link CapitalGrant }
    */
    async getGrantReferenceDetails(id, requestOptions) {
        const endpoint = `${this.baseUrl}/grants/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "CapitalGrant");
    }
    /**
    * @summary Request a grant payout
    * @param capitalGrantInfo {@link CapitalGrantInfo }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link CapitalGrant }
    */
    async requestGrantPayout(capitalGrantInfo, requestOptions) {
        const endpoint = `${this.baseUrl}/grants`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(capitalGrantInfo, "CapitalGrantInfo");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "CapitalGrant");
    }
}
exports.CapitalApi = CapitalApi;
