// @adyen/api-library/lib/src/services/transfers/transfersApi.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransfersApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/transfers/models");
class TransfersApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://balanceplatform-api-test.adyen.com/btl/v4";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Transfer funds
    * @param wWWAuthenticate {@link string } Header for authenticating through SCA
    * @param transferInfo {@link TransferInfo }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Transfer }
    */
    async transferFunds(transferInfo, requestOptions) {
        const endpoint = `${this.baseUrl}/transfers`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(transferInfo, "TransferInfo");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "Transfer");
    }
    /**
    * @summary Return a transfer
    * @param id {@link string } The unique identifier of the transfer to be returned.
    * @param returnTransferRequest {@link ReturnTransferRequest }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link ReturnTransferResponse }
    */
    async returnTransfer(id, returnTransferRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/transfers/{id}/returns`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(returnTransferRequest, "ReturnTransferRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "ReturnTransferResponse");
    }
}
exports.TransfersApi = TransfersApi;
