// @adyen/api-library/lib/src/typings/platformsAccount/account.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Account = void 0;
class Account {
    static getAttributeTypeMap() {
        return Account.attributeTypeMap;
    }
}
exports.Account = Account;
Account.discriminator = undefined;
Account.attributeTypeMap = [
    {
        "name": "accountCode",
        "baseName": "accountCode",
        "type": "string"
    },
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "beneficiaryAccount",
        "baseName": "beneficiaryAccount",
        "type": "string"
    },
    {
        "name": "beneficiaryMerchantReference",
        "baseName": "beneficiaryMerchantReference",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "payoutMethodCode",
        "baseName": "payoutMethodCode",
        "type": "string"
    },
    {
        "name": "payoutSchedule",
        "baseName": "payoutSchedule",
        "type": "PayoutScheduleResponse"
    },
    {
        "name": "payoutSpeed",
        "baseName": "payoutSpeed",
        "type": "Account.PayoutSpeedEnum"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "string"
    }
];
(function (Account) {
    let PayoutSpeedEnum;
    (function (PayoutSpeedEnum) {
        PayoutSpeedEnum["Instant"] = "INSTANT";
        PayoutSpeedEnum["SameDay"] = "SAME_DAY";
        PayoutSpeedEnum["Standard"] = "STANDARD";
    })(PayoutSpeedEnum = Account.PayoutSpeedEnum || (Account.PayoutSpeedEnum = {}));
})(Account = exports.Account || (exports.Account = {}));
