// @adyen/api-library/lib/src/typings/platformsAccount/kYCCheckStatusData.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KYCCheckStatusData = void 0;
class KYCCheckStatusData {
    static getAttributeTypeMap() {
        return KYCCheckStatusData.attributeTypeMap;
    }
}
exports.KYCCheckStatusData = KYCCheckStatusData;
KYCCheckStatusData.discriminator = undefined;
KYCCheckStatusData.attributeTypeMap = [
    {
        "name": "requiredFields",
        "baseName": "requiredFields",
        "type": "Array<string>"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "KYCCheckStatusData.StatusEnum"
    },
    {
        "name": "summary",
        "baseName": "summary",
        "type": "KYCCheckSummary"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "KYCCheckStatusData.TypeEnum"
    }
];
(function (KYCCheckStatusData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["AwaitingData"] = "AWAITING_DATA";
        StatusEnum["DataProvided"] = "DATA_PROVIDED";
        StatusEnum["Failed"] = "FAILED";
        StatusEnum["InvalidData"] = "INVALID_DATA";
        StatusEnum["Passed"] = "PASSED";
        StatusEnum["Pending"] = "PENDING";
        StatusEnum["PendingReview"] = "PENDING_REVIEW";
        StatusEnum["RetryLimitReached"] = "RETRY_LIMIT_REACHED";
        StatusEnum["Unchecked"] = "UNCHECKED";
    })(StatusEnum = KYCCheckStatusData.StatusEnum || (KYCCheckStatusData.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BankAccountVerification"] = "BANK_ACCOUNT_VERIFICATION";
        TypeEnum["CardVerification"] = "CARD_VERIFICATION";
        TypeEnum["CompanyVerification"] = "COMPANY_VERIFICATION";
        TypeEnum["IdentityVerification"] = "IDENTITY_VERIFICATION";
        TypeEnum["LegalArrangementVerification"] = "LEGAL_ARRANGEMENT_VERIFICATION";
        TypeEnum["NonprofitVerification"] = "NONPROFIT_VERIFICATION";
        TypeEnum["PassportVerification"] = "PASSPORT_VERIFICATION";
        TypeEnum["PayoutMethodVerification"] = "PAYOUT_METHOD_VERIFICATION";
        TypeEnum["PciVerification"] = "PCI_VERIFICATION";
    })(TypeEnum = KYCCheckStatusData.TypeEnum || (KYCCheckStatusData.TypeEnum = {}));
})(KYCCheckStatusData = exports.KYCCheckStatusData || (exports.KYCCheckStatusData = {}));
