// @adyen/api-library/lib/src/typings/platformsAccount/performVerificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PerformVerificationRequest = void 0;
class PerformVerificationRequest {
    static getAttributeTypeMap() {
        return PerformVerificationRequest.attributeTypeMap;
    }
}
exports.PerformVerificationRequest = PerformVerificationRequest;
PerformVerificationRequest.discriminator = undefined;
PerformVerificationRequest.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "accountStateType",
        "baseName": "accountStateType",
        "type": "PerformVerificationRequest.AccountStateTypeEnum"
    },
    {
        "name": "tier",
        "baseName": "tier",
        "type": "number"
    }
];
(function (PerformVerificationRequest) {
    let AccountStateTypeEnum;
    (function (AccountStateTypeEnum) {
        AccountStateTypeEnum["LimitedPayout"] = "LimitedPayout";
        AccountStateTypeEnum["LimitedProcessing"] = "LimitedProcessing";
        AccountStateTypeEnum["LimitlessPayout"] = "LimitlessPayout";
        AccountStateTypeEnum["LimitlessProcessing"] = "LimitlessProcessing";
        AccountStateTypeEnum["Payout"] = "Payout";
        AccountStateTypeEnum["Processing"] = "Processing";
    })(AccountStateTypeEnum = PerformVerificationRequest.AccountStateTypeEnum || (PerformVerificationRequest.AccountStateTypeEnum = {}));
})(PerformVerificationRequest = exports.PerformVerificationRequest || (exports.PerformVerificationRequest = {}));
