// @adyen/api-library/lib/src/typings/platformsAccount/personalDocumentData.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PersonalDocumentData = void 0;
class PersonalDocumentData {
    static getAttributeTypeMap() {
        return PersonalDocumentData.attributeTypeMap;
    }
}
exports.PersonalDocumentData = PersonalDocumentData;
PersonalDocumentData.discriminator = undefined;
PersonalDocumentData.attributeTypeMap = [
    {
        "name": "expirationDate",
        "baseName": "expirationDate",
        "type": "string"
    },
    {
        "name": "issuerCountry",
        "baseName": "issuerCountry",
        "type": "string"
    },
    {
        "name": "issuerState",
        "baseName": "issuerState",
        "type": "string"
    },
    {
        "name": "number",
        "baseName": "number",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PersonalDocumentData.TypeEnum"
    }
];
(function (PersonalDocumentData) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Drivinglicense"] = "DRIVINGLICENSE";
        TypeEnum["Id"] = "ID";
        TypeEnum["Passport"] = "PASSPORT";
        TypeEnum["Socialsecurity"] = "SOCIALSECURITY";
        TypeEnum["Visa"] = "VISA";
    })(TypeEnum = PersonalDocumentData.TypeEnum || (PersonalDocumentData.TypeEnum = {}));
})(PersonalDocumentData = exports.PersonalDocumentData || (exports.PersonalDocumentData = {}));
