// @adyen/api-library/lib/src/typings/platformsAccount/updateAccountHolderResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAccountHolderResponse = void 0;
class UpdateAccountHolderResponse {
    static getAttributeTypeMap() {
        return UpdateAccountHolderResponse.attributeTypeMap;
    }
}
exports.UpdateAccountHolderResponse = UpdateAccountHolderResponse;
UpdateAccountHolderResponse.discriminator = undefined;
UpdateAccountHolderResponse.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "accountHolderDetails",
        "baseName": "accountHolderDetails",
        "type": "AccountHolderDetails"
    },
    {
        "name": "accountHolderStatus",
        "baseName": "accountHolderStatus",
        "type": "AccountHolderStatus"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "invalidFields",
        "baseName": "invalidFields",
        "type": "Array<ErrorFieldType>"
    },
    {
        "name": "legalEntity",
        "baseName": "legalEntity",
        "type": "UpdateAccountHolderResponse.LegalEntityEnum"
    },
    {
        "name": "primaryCurrency",
        "baseName": "primaryCurrency",
        "type": "string"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "string"
    },
    {
        "name": "verification",
        "baseName": "verification",
        "type": "KYCVerificationResult"
    },
    {
        "name": "verificationProfile",
        "baseName": "verificationProfile",
        "type": "string"
    }
];
(function (UpdateAccountHolderResponse) {
    let LegalEntityEnum;
    (function (LegalEntityEnum) {
        LegalEntityEnum["Business"] = "Business";
        LegalEntityEnum["Individual"] = "Individual";
        LegalEntityEnum["NonProfit"] = "NonProfit";
        LegalEntityEnum["Partnership"] = "Partnership";
        LegalEntityEnum["PublicCompany"] = "PublicCompany";
    })(LegalEntityEnum = UpdateAccountHolderResponse.LegalEntityEnum || (UpdateAccountHolderResponse.LegalEntityEnum = {}));
})(UpdateAccountHolderResponse = exports.UpdateAccountHolderResponse || (exports.UpdateAccountHolderResponse = {}));
