// @adyen/api-library/lib/src/typings/platformsAccount/updateAccountHolderStateRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAccountHolderStateRequest = void 0;
class UpdateAccountHolderStateRequest {
    static getAttributeTypeMap() {
        return UpdateAccountHolderStateRequest.attributeTypeMap;
    }
}
exports.UpdateAccountHolderStateRequest = UpdateAccountHolderStateRequest;
UpdateAccountHolderStateRequest.discriminator = undefined;
UpdateAccountHolderStateRequest.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "disable",
        "baseName": "disable",
        "type": "boolean"
    },
    {
        "name": "reason",
        "baseName": "reason",
        "type": "string"
    },
    {
        "name": "stateType",
        "baseName": "stateType",
        "type": "UpdateAccountHolderStateRequest.StateTypeEnum"
    }
];
(function (UpdateAccountHolderStateRequest) {
    let StateTypeEnum;
    (function (StateTypeEnum) {
        StateTypeEnum["LimitedPayout"] = "LimitedPayout";
        StateTypeEnum["LimitedProcessing"] = "LimitedProcessing";
        StateTypeEnum["LimitlessPayout"] = "LimitlessPayout";
        StateTypeEnum["LimitlessProcessing"] = "LimitlessProcessing";
        StateTypeEnum["Payout"] = "Payout";
        StateTypeEnum["Processing"] = "Processing";
    })(StateTypeEnum = UpdateAccountHolderStateRequest.StateTypeEnum || (UpdateAccountHolderStateRequest.StateTypeEnum = {}));
})(UpdateAccountHolderStateRequest = exports.UpdateAccountHolderStateRequest || (exports.UpdateAccountHolderStateRequest = {}));
