// @adyen/api-library/lib/src/typings/platformsAccount/viasName.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViasName = void 0;
class ViasName {
    static getAttributeTypeMap() {
        return ViasName.attributeTypeMap;
    }
}
exports.ViasName = ViasName;
ViasName.discriminator = undefined;
ViasName.attributeTypeMap = [
    {
        "name": "firstName",
        "baseName": "firstName",
        "type": "string"
    },
    {
        "name": "gender",
        "baseName": "gender",
        "type": "ViasName.GenderEnum"
    },
    {
        "name": "infix",
        "baseName": "infix",
        "type": "string"
    },
    {
        "name": "lastName",
        "baseName": "lastName",
        "type": "string"
    }
];
(function (ViasName) {
    let GenderEnum;
    (function (GenderEnum) {
        GenderEnum["Male"] = "MALE";
        GenderEnum["Female"] = "FEMALE";
        GenderEnum["Unknown"] = "UNKNOWN";
    })(GenderEnum = ViasName.GenderEnum || (ViasName.GenderEnum = {}));
})(ViasName = exports.ViasName || (exports.ViasName = {}));
