// @adyen/api-library/lib/src/typings/payout/payoutRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayoutRequest = void 0;
class PayoutRequest {
    static getAttributeTypeMap() {
        return PayoutRequest.attributeTypeMap;
    }
}
exports.PayoutRequest = PayoutRequest;
PayoutRequest.discriminator = undefined;
PayoutRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "Address"
    },
    {
        "name": "card",
        "baseName": "card",
        "type": "Card"
    },
    {
        "name": "fraudOffset",
        "baseName": "fraudOffset",
        "type": "number"
    },
    {
        "name": "fundSource",
        "baseName": "fundSource",
        "type": "FundSource"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "recurring",
        "baseName": "recurring",
        "type": "Recurring"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "selectedRecurringDetailReference",
        "baseName": "selectedRecurringDetailReference",
        "type": "string"
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string"
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "PayoutRequest.ShopperInteractionEnum"
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name"
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string"
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string"
    }
];
(function (PayoutRequest) {
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = PayoutRequest.ShopperInteractionEnum || (PayoutRequest.ShopperInteractionEnum = {}));
})(PayoutRequest = exports.PayoutRequest || (exports.PayoutRequest = {}));
