// @adyen/api-library/lib/src/typings/payout/storeDetailAndSubmitRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoreDetailAndSubmitRequest = void 0;
class StoreDetailAndSubmitRequest {
    static getAttributeTypeMap() {
        return StoreDetailAndSubmitRequest.attributeTypeMap;
    }
}
exports.StoreDetailAndSubmitRequest = StoreDetailAndSubmitRequest;
StoreDetailAndSubmitRequest.discriminator = undefined;
StoreDetailAndSubmitRequest.attributeTypeMap = [
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "bank",
        "baseName": "bank",
        "type": "BankAccount"
    },
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "Address"
    },
    {
        "name": "card",
        "baseName": "card",
        "type": "Card"
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "string"
    },
    {
        "name": "entityType",
        "baseName": "entityType",
        "type": "StoreDetailAndSubmitRequest.EntityTypeEnum"
    },
    {
        "name": "fraudOffset",
        "baseName": "fraudOffset",
        "type": "number"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "nationality",
        "baseName": "nationality",
        "type": "string"
    },
    {
        "name": "recurring",
        "baseName": "recurring",
        "type": "Recurring"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "selectedBrand",
        "baseName": "selectedBrand",
        "type": "string"
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string"
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name"
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string"
    },
    {
        "name": "shopperStatement",
        "baseName": "shopperStatement",
        "type": "string"
    },
    {
        "name": "socialSecurityNumber",
        "baseName": "socialSecurityNumber",
        "type": "string"
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string"
    }
];
(function (StoreDetailAndSubmitRequest) {
    let EntityTypeEnum;
    (function (EntityTypeEnum) {
        EntityTypeEnum["NaturalPerson"] = "NaturalPerson";
        EntityTypeEnum["Company"] = "Company";
    })(EntityTypeEnum = StoreDetailAndSubmitRequest.EntityTypeEnum || (StoreDetailAndSubmitRequest.EntityTypeEnum = {}));
})(StoreDetailAndSubmitRequest = exports.StoreDetailAndSubmitRequest || (exports.StoreDetailAndSubmitRequest = {}));
