// @adyen/api-library/lib/src/typings/disputes/models.js
"use strict";
/*
 * The version of the OpenAPI document: v30
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectSerializer = void 0;
__exportStar(require("./acceptDisputeRequest"), exports);
__exportStar(require("./acceptDisputeResponse"), exports);
__exportStar(require("./defendDisputeRequest"), exports);
__exportStar(require("./defendDisputeResponse"), exports);
__exportStar(require("./defenseDocument"), exports);
__exportStar(require("./defenseDocumentType"), exports);
__exportStar(require("./defenseReason"), exports);
__exportStar(require("./defenseReasonsRequest"), exports);
__exportStar(require("./defenseReasonsResponse"), exports);
__exportStar(require("./deleteDefenseDocumentRequest"), exports);
__exportStar(require("./deleteDefenseDocumentResponse"), exports);
__exportStar(require("./disputeServiceResult"), exports);
__exportStar(require("./serviceError"), exports);
__exportStar(require("./supplyDefenseDocumentRequest"), exports);
__exportStar(require("./supplyDefenseDocumentResponse"), exports);
const acceptDisputeRequest_1 = require("./acceptDisputeRequest");
const acceptDisputeResponse_1 = require("./acceptDisputeResponse");
const defendDisputeRequest_1 = require("./defendDisputeRequest");
const defendDisputeResponse_1 = require("./defendDisputeResponse");
const defenseDocument_1 = require("./defenseDocument");
const defenseDocumentType_1 = require("./defenseDocumentType");
const defenseReason_1 = require("./defenseReason");
const defenseReasonsRequest_1 = require("./defenseReasonsRequest");
const defenseReasonsResponse_1 = require("./defenseReasonsResponse");
const deleteDefenseDocumentRequest_1 = require("./deleteDefenseDocumentRequest");
const deleteDefenseDocumentResponse_1 = require("./deleteDefenseDocumentResponse");
const disputeServiceResult_1 = require("./disputeServiceResult");
const serviceError_1 = require("./serviceError");
const supplyDefenseDocumentRequest_1 = require("./supplyDefenseDocumentRequest");
const supplyDefenseDocumentResponse_1 = require("./supplyDefenseDocumentResponse");
/* tslint:disable:no-unused-variable */
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
let enumsMap = {};
let typeMap = {
    "AcceptDisputeRequest": acceptDisputeRequest_1.AcceptDisputeRequest,
    "AcceptDisputeResponse": acceptDisputeResponse_1.AcceptDisputeResponse,
    "DefendDisputeRequest": defendDisputeRequest_1.DefendDisputeRequest,
    "DefendDisputeResponse": defendDisputeResponse_1.DefendDisputeResponse,
    "DefenseDocument": defenseDocument_1.DefenseDocument,
    "DefenseDocumentType": defenseDocumentType_1.DefenseDocumentType,
    "DefenseReason": defenseReason_1.DefenseReason,
    "DefenseReasonsRequest": defenseReasonsRequest_1.DefenseReasonsRequest,
    "DefenseReasonsResponse": defenseReasonsResponse_1.DefenseReasonsResponse,
    "DeleteDefenseDocumentRequest": deleteDefenseDocumentRequest_1.DeleteDefenseDocumentRequest,
    "DeleteDefenseDocumentResponse": deleteDefenseDocumentResponse_1.DeleteDefenseDocumentResponse,
    "DisputeServiceResult": disputeServiceResult_1.DisputeServiceResult,
    "ServiceError": serviceError_1.ServiceError,
    "SupplyDefenseDocumentRequest": supplyDefenseDocumentRequest_1.SupplyDefenseDocumentRequest,
    "SupplyDefenseDocumentResponse": supplyDefenseDocumentResponse_1.SupplyDefenseDocumentResponse,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }
            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType; // use the type given in the discriminator
                    }
                    else {
                        return expectedType; // discriminator did not map to a type
                    }
                }
                else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.serialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return data.toISOString();
        }
        else if (type === "SaleToAcquirerData") {
            const dataString = JSON.stringify(data);
            return Buffer.from(dataString).toString("base64");
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) { // in case we dont know the type
                return data;
            }
            // Get the actual type of this object
            type = this.findCorrectType(data, type);
            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.deserialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) { // is Enum
                return data;
            }
            if (!typeMap[type]) { // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
