// @adyen/api-library/lib/src/typings/storedValue/storedValueBalanceMergeResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueBalanceMergeResponse = void 0;
class StoredValueBalanceMergeResponse {
    static getAttributeTypeMap() {
        return StoredValueBalanceMergeResponse.attributeTypeMap;
    }
}
exports.StoredValueBalanceMergeResponse = StoredValueBalanceMergeResponse;
StoredValueBalanceMergeResponse.discriminator = undefined;
StoredValueBalanceMergeResponse.attributeTypeMap = [
    {
        "name": "authCode",
        "baseName": "authCode",
        "type": "string"
    },
    {
        "name": "currentBalance",
        "baseName": "currentBalance",
        "type": "Amount"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "StoredValueBalanceMergeResponse.ResultCodeEnum"
    },
    {
        "name": "thirdPartyRefusalReason",
        "baseName": "thirdPartyRefusalReason",
        "type": "string"
    }
];
(function (StoredValueBalanceMergeResponse) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["Success"] = "Success";
        ResultCodeEnum["Refused"] = "Refused";
        ResultCodeEnum["Error"] = "Error";
        ResultCodeEnum["NotEnoughBalance"] = "NotEnoughBalance";
    })(ResultCodeEnum = StoredValueBalanceMergeResponse.ResultCodeEnum || (StoredValueBalanceMergeResponse.ResultCodeEnum = {}));
})(StoredValueBalanceMergeResponse = exports.StoredValueBalanceMergeResponse || (exports.StoredValueBalanceMergeResponse = {}));
