// @adyen/api-library/lib/src/typings/platformsAccount/accountEvent.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountEvent = void 0;
class AccountEvent {
    static getAttributeTypeMap() {
        return AccountEvent.attributeTypeMap;
    }
}
exports.AccountEvent = AccountEvent;
AccountEvent.discriminator = undefined;
AccountEvent.attributeTypeMap = [
    {
        "name": "event",
        "baseName": "event",
        "type": "AccountEvent.EventEnum"
    },
    {
        "name": "executionDate",
        "baseName": "executionDate",
        "type": "Date"
    },
    {
        "name": "reason",
        "baseName": "reason",
        "type": "string"
    }
];
(function (AccountEvent) {
    let EventEnum;
    (function (EventEnum) {
        EventEnum["InactivateAccount"] = "InactivateAccount";
        EventEnum["RefundNotPaidOutTransfers"] = "RefundNotPaidOutTransfers";
    })(EventEnum = AccountEvent.EventEnum || (AccountEvent.EventEnum = {}));
})(AccountEvent = exports.AccountEvent || (exports.AccountEvent = {}));
