// @adyen/api-library/lib/src/typings/platformsAccount/createAccountResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAccountResponse = void 0;
class CreateAccountResponse {
    static getAttributeTypeMap() {
        return CreateAccountResponse.attributeTypeMap;
    }
}
exports.CreateAccountResponse = CreateAccountResponse;
CreateAccountResponse.discriminator = undefined;
CreateAccountResponse.attributeTypeMap = [
    {
        "name": "accountCode",
        "baseName": "accountCode",
        "type": "string"
    },
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "invalidFields",
        "baseName": "invalidFields",
        "type": "Array<ErrorFieldType>"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "payoutMethodCode",
        "baseName": "payoutMethodCode",
        "type": "string"
    },
    {
        "name": "payoutSchedule",
        "baseName": "payoutSchedule",
        "type": "PayoutScheduleResponse"
    },
    {
        "name": "payoutSpeed",
        "baseName": "payoutSpeed",
        "type": "CreateAccountResponse.PayoutSpeedEnum"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "CreateAccountResponse.StatusEnum"
    }
];
(function (CreateAccountResponse) {
    let PayoutSpeedEnum;
    (function (PayoutSpeedEnum) {
        PayoutSpeedEnum["Instant"] = "INSTANT";
        PayoutSpeedEnum["SameDay"] = "SAME_DAY";
        PayoutSpeedEnum["Standard"] = "STANDARD";
    })(PayoutSpeedEnum = CreateAccountResponse.PayoutSpeedEnum || (CreateAccountResponse.PayoutSpeedEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "Active";
        StatusEnum["Closed"] = "Closed";
        StatusEnum["Inactive"] = "Inactive";
        StatusEnum["Suspended"] = "Suspended";
    })(StatusEnum = CreateAccountResponse.StatusEnum || (CreateAccountResponse.StatusEnum = {}));
})(CreateAccountResponse = exports.CreateAccountResponse || (exports.CreateAccountResponse = {}));
