// @adyen/api-library/lib/src/typings/platformsAccount/createAccountHolderRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAccountHolderRequest = void 0;
class CreateAccountHolderRequest {
    static getAttributeTypeMap() {
        return CreateAccountHolderRequest.attributeTypeMap;
    }
}
exports.CreateAccountHolderRequest = CreateAccountHolderRequest;
CreateAccountHolderRequest.discriminator = undefined;
CreateAccountHolderRequest.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "accountHolderDetails",
        "baseName": "accountHolderDetails",
        "type": "AccountHolderDetails"
    },
    {
        "name": "createDefaultAccount",
        "baseName": "createDefaultAccount",
        "type": "boolean"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "legalEntity",
        "baseName": "legalEntity",
        "type": "CreateAccountHolderRequest.LegalEntityEnum"
    },
    {
        "name": "primaryCurrency",
        "baseName": "primaryCurrency",
        "type": "string"
    },
    {
        "name": "processingTier",
        "baseName": "processingTier",
        "type": "number"
    },
    {
        "name": "verificationProfile",
        "baseName": "verificationProfile",
        "type": "string"
    }
];
(function (CreateAccountHolderRequest) {
    let LegalEntityEnum;
    (function (LegalEntityEnum) {
        LegalEntityEnum["Business"] = "Business";
        LegalEntityEnum["Individual"] = "Individual";
        LegalEntityEnum["NonProfit"] = "NonProfit";
        LegalEntityEnum["Partnership"] = "Partnership";
        LegalEntityEnum["PublicCompany"] = "PublicCompany";
    })(LegalEntityEnum = CreateAccountHolderRequest.LegalEntityEnum || (CreateAccountHolderRequest.LegalEntityEnum = {}));
})(CreateAccountHolderRequest = exports.CreateAccountHolderRequest || (exports.CreateAccountHolderRequest = {}));
