// @adyen/api-library/lib/src/typings/platformsAccount/documentDetail.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentDetail = void 0;
class DocumentDetail {
    static getAttributeTypeMap() {
        return DocumentDetail.attributeTypeMap;
    }
}
exports.DocumentDetail = DocumentDetail;
DocumentDetail.discriminator = undefined;
DocumentDetail.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "documentType",
        "baseName": "documentType",
        "type": "DocumentDetail.DocumentTypeEnum"
    },
    {
        "name": "filename",
        "baseName": "filename",
        "type": "string"
    },
    {
        "name": "legalArrangementCode",
        "baseName": "legalArrangementCode",
        "type": "string"
    },
    {
        "name": "legalArrangementEntityCode",
        "baseName": "legalArrangementEntityCode",
        "type": "string"
    },
    {
        "name": "shareholderCode",
        "baseName": "shareholderCode",
        "type": "string"
    },
    {
        "name": "signatoryCode",
        "baseName": "signatoryCode",
        "type": "string"
    }
];
(function (DocumentDetail) {
    let DocumentTypeEnum;
    (function (DocumentTypeEnum) {
        DocumentTypeEnum["BankStatement"] = "BANK_STATEMENT";
        DocumentTypeEnum["Bsn"] = "BSN";
        DocumentTypeEnum["CompanyRegistrationScreening"] = "COMPANY_REGISTRATION_SCREENING";
        DocumentTypeEnum["ConstitutionalDocument"] = "CONSTITUTIONAL_DOCUMENT";
        DocumentTypeEnum["DrivingLicence"] = "DRIVING_LICENCE";
        DocumentTypeEnum["DrivingLicenceBack"] = "DRIVING_LICENCE_BACK";
        DocumentTypeEnum["DrivingLicenceFront"] = "DRIVING_LICENCE_FRONT";
        DocumentTypeEnum["IdCard"] = "ID_CARD";
        DocumentTypeEnum["IdCardBack"] = "ID_CARD_BACK";
        DocumentTypeEnum["IdCardFront"] = "ID_CARD_FRONT";
        DocumentTypeEnum["Passport"] = "PASSPORT";
        DocumentTypeEnum["ProofOfResidency"] = "PROOF_OF_RESIDENCY";
        DocumentTypeEnum["Ssn"] = "SSN";
        DocumentTypeEnum["SupportingDocuments"] = "SUPPORTING_DOCUMENTS";
    })(DocumentTypeEnum = DocumentDetail.DocumentTypeEnum || (DocumentDetail.DocumentTypeEnum = {}));
})(DocumentDetail = exports.DocumentDetail || (exports.DocumentDetail = {}));
