// @adyen/api-library/lib/src/typings/platformsAccount/getAccountHolderResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAccountHolderResponse = void 0;
class GetAccountHolderResponse {
    static getAttributeTypeMap() {
        return GetAccountHolderResponse.attributeTypeMap;
    }
}
exports.GetAccountHolderResponse = GetAccountHolderResponse;
GetAccountHolderResponse.discriminator = undefined;
GetAccountHolderResponse.attributeTypeMap = [
    {
        "name": "accountHolderCode",
        "baseName": "accountHolderCode",
        "type": "string"
    },
    {
        "name": "accountHolderDetails",
        "baseName": "accountHolderDetails",
        "type": "AccountHolderDetails"
    },
    {
        "name": "accountHolderStatus",
        "baseName": "accountHolderStatus",
        "type": "AccountHolderStatus"
    },
    {
        "name": "accounts",
        "baseName": "accounts",
        "type": "Array<Account>"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "invalidFields",
        "baseName": "invalidFields",
        "type": "Array<ErrorFieldType>"
    },
    {
        "name": "legalEntity",
        "baseName": "legalEntity",
        "type": "GetAccountHolderResponse.LegalEntityEnum"
    },
    {
        "name": "migrationData",
        "baseName": "migrationData",
        "type": "MigrationData"
    },
    {
        "name": "primaryCurrency",
        "baseName": "primaryCurrency",
        "type": "string"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "string"
    },
    {
        "name": "systemUpToDateTime",
        "baseName": "systemUpToDateTime",
        "type": "Date"
    },
    {
        "name": "verification",
        "baseName": "verification",
        "type": "KYCVerificationResult"
    },
    {
        "name": "verificationProfile",
        "baseName": "verificationProfile",
        "type": "string"
    }
];
(function (GetAccountHolderResponse) {
    let LegalEntityEnum;
    (function (LegalEntityEnum) {
        LegalEntityEnum["Business"] = "Business";
        LegalEntityEnum["Individual"] = "Individual";
        LegalEntityEnum["NonProfit"] = "NonProfit";
        LegalEntityEnum["Partnership"] = "Partnership";
        LegalEntityEnum["PublicCompany"] = "PublicCompany";
    })(LegalEntityEnum = GetAccountHolderResponse.LegalEntityEnum || (GetAccountHolderResponse.LegalEntityEnum = {}));
})(GetAccountHolderResponse = exports.GetAccountHolderResponse || (exports.GetAccountHolderResponse = {}));
