// @adyen/api-library/lib/src/typings/platformsAccount/legalArrangementDetail.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalArrangementDetail = void 0;
class LegalArrangementDetail {
    static getAttributeTypeMap() {
        return LegalArrangementDetail.attributeTypeMap;
    }
}
exports.LegalArrangementDetail = LegalArrangementDetail;
LegalArrangementDetail.discriminator = undefined;
LegalArrangementDetail.attributeTypeMap = [
    {
        "name": "address",
        "baseName": "address",
        "type": "ViasAddress"
    },
    {
        "name": "legalArrangementCode",
        "baseName": "legalArrangementCode",
        "type": "string"
    },
    {
        "name": "legalArrangementEntities",
        "baseName": "legalArrangementEntities",
        "type": "Array<LegalArrangementEntityDetail>"
    },
    {
        "name": "legalArrangementReference",
        "baseName": "legalArrangementReference",
        "type": "string"
    },
    {
        "name": "legalForm",
        "baseName": "legalForm",
        "type": "LegalArrangementDetail.LegalFormEnum"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "registrationNumber",
        "baseName": "registrationNumber",
        "type": "string"
    },
    {
        "name": "taxNumber",
        "baseName": "taxNumber",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "LegalArrangementDetail.TypeEnum"
    }
];
(function (LegalArrangementDetail) {
    let LegalFormEnum;
    (function (LegalFormEnum) {
        LegalFormEnum["CashManagementTrust"] = "CashManagementTrust";
        LegalFormEnum["CorporateUnitTrust"] = "CorporateUnitTrust";
        LegalFormEnum["DeceasedEstate"] = "DeceasedEstate";
        LegalFormEnum["DiscretionaryInvestmentTrust"] = "DiscretionaryInvestmentTrust";
        LegalFormEnum["DiscretionaryServicesManagementTrust"] = "DiscretionaryServicesManagementTrust";
        LegalFormEnum["DiscretionaryTradingTrust"] = "DiscretionaryTradingTrust";
        LegalFormEnum["FirstHomeSaverAccountsTrust"] = "FirstHomeSaverAccountsTrust";
        LegalFormEnum["FixedTrust"] = "FixedTrust";
        LegalFormEnum["FixedUnitTrust"] = "FixedUnitTrust";
        LegalFormEnum["HybridTrust"] = "HybridTrust";
        LegalFormEnum["ListedPublicUnitTrust"] = "ListedPublicUnitTrust";
        LegalFormEnum["OtherTrust"] = "OtherTrust";
        LegalFormEnum["PooledSuperannuationTrust"] = "PooledSuperannuationTrust";
        LegalFormEnum["PublicTradingTrust"] = "PublicTradingTrust";
        LegalFormEnum["UnlistedPublicUnitTrust"] = "UnlistedPublicUnitTrust";
        LegalFormEnum["LimitedPartnership"] = "LimitedPartnership";
        LegalFormEnum["FamilyPartnership"] = "FamilyPartnership";
        LegalFormEnum["OtherPartnership"] = "OtherPartnership";
    })(LegalFormEnum = LegalArrangementDetail.LegalFormEnum || (LegalArrangementDetail.LegalFormEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Association"] = "Association";
        TypeEnum["Partnership"] = "Partnership";
        TypeEnum["SoleProprietorship"] = "SoleProprietorship";
        TypeEnum["Trust"] = "Trust";
    })(TypeEnum = LegalArrangementDetail.TypeEnum || (LegalArrangementDetail.TypeEnum = {}));
})(LegalArrangementDetail = exports.LegalArrangementDetail || (exports.LegalArrangementDetail = {}));
