// @adyen/api-library/lib/src/typings/platformsAccount/shareholderContact.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShareholderContact = void 0;
class ShareholderContact {
    static getAttributeTypeMap() {
        return ShareholderContact.attributeTypeMap;
    }
}
exports.ShareholderContact = ShareholderContact;
ShareholderContact.discriminator = undefined;
ShareholderContact.attributeTypeMap = [
    {
        "name": "address",
        "baseName": "address",
        "type": "ViasAddress"
    },
    {
        "name": "email",
        "baseName": "email",
        "type": "string"
    },
    {
        "name": "fullPhoneNumber",
        "baseName": "fullPhoneNumber",
        "type": "string"
    },
    {
        "name": "jobTitle",
        "baseName": "jobTitle",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "ViasName"
    },
    {
        "name": "personalData",
        "baseName": "personalData",
        "type": "ViasPersonalData"
    },
    {
        "name": "phoneNumber",
        "baseName": "phoneNumber",
        "type": "ViasPhoneNumber"
    },
    {
        "name": "shareholderCode",
        "baseName": "shareholderCode",
        "type": "string"
    },
    {
        "name": "shareholderReference",
        "baseName": "shareholderReference",
        "type": "string"
    },
    {
        "name": "shareholderType",
        "baseName": "shareholderType",
        "type": "ShareholderContact.ShareholderTypeEnum"
    },
    {
        "name": "webAddress",
        "baseName": "webAddress",
        "type": "string"
    }
];
(function (ShareholderContact) {
    let ShareholderTypeEnum;
    (function (ShareholderTypeEnum) {
        ShareholderTypeEnum["Controller"] = "Controller";
        ShareholderTypeEnum["Owner"] = "Owner";
        ShareholderTypeEnum["Signatory"] = "Signatory";
    })(ShareholderTypeEnum = ShareholderContact.ShareholderTypeEnum || (ShareholderContact.ShareholderTypeEnum = {}));
})(ShareholderContact = exports.ShareholderContact || (exports.ShareholderContact = {}));
