// @adyen/api-library/lib/src/typings/platformsAccount/updateAccountResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAccountResponse = void 0;
class UpdateAccountResponse {
    static getAttributeTypeMap() {
        return UpdateAccountResponse.attributeTypeMap;
    }
}
exports.UpdateAccountResponse = UpdateAccountResponse;
UpdateAccountResponse.discriminator = undefined;
UpdateAccountResponse.attributeTypeMap = [
    {
        "name": "accountCode",
        "baseName": "accountCode",
        "type": "string"
    },
    {
        "name": "bankAccountUUID",
        "baseName": "bankAccountUUID",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "invalidFields",
        "baseName": "invalidFields",
        "type": "Array<ErrorFieldType>"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "payoutMethodCode",
        "baseName": "payoutMethodCode",
        "type": "string"
    },
    {
        "name": "payoutSchedule",
        "baseName": "payoutSchedule",
        "type": "PayoutScheduleResponse"
    },
    {
        "name": "payoutSpeed",
        "baseName": "payoutSpeed",
        "type": "UpdateAccountResponse.PayoutSpeedEnum"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "string"
    }
];
(function (UpdateAccountResponse) {
    let PayoutSpeedEnum;
    (function (PayoutSpeedEnum) {
        PayoutSpeedEnum["Instant"] = "INSTANT";
        PayoutSpeedEnum["SameDay"] = "SAME_DAY";
        PayoutSpeedEnum["Standard"] = "STANDARD";
    })(PayoutSpeedEnum = UpdateAccountResponse.PayoutSpeedEnum || (UpdateAccountResponse.PayoutSpeedEnum = {}));
})(UpdateAccountResponse = exports.UpdateAccountResponse || (exports.UpdateAccountResponse = {}));
