// @adyen/api-library/lib/src/services/legalEntityManagement/businessLinesApi.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BusinessLinesApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/legalEntityManagement/models");
class BusinessLinesApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://kyc-test.adyen.com/lem/v3";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Delete a business line
    * @param id {@link string } The unique identifier of the business line to be deleted.
    * @param requestOptions {@link IRequest.Options}
    */
    async deleteBusinessLine(id, requestOptions) {
        const endpoint = `${this.baseUrl}/businessLines/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "DELETE" });
    }
    /**
    * @summary Get a business line
    * @param id {@link string } The unique identifier of the business line.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link BusinessLine }
    */
    async getBusinessLine(id, requestOptions) {
        const endpoint = `${this.baseUrl}/businessLines/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "BusinessLine");
    }
    /**
    * @summary Update a business line
    * @param id {@link string } The unique identifier of the business line.
    * @param businessLineInfoUpdate {@link BusinessLineInfoUpdate }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link BusinessLine }
    */
    async updateBusinessLine(id, businessLineInfoUpdate, requestOptions) {
        const endpoint = `${this.baseUrl}/businessLines/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(businessLineInfoUpdate, "BusinessLineInfoUpdate");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "BusinessLine");
    }
    /**
    * @summary Create a business line
    * @param businessLineInfo {@link BusinessLineInfo }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link BusinessLine }
    */
    async createBusinessLine(businessLineInfo, requestOptions) {
        const endpoint = `${this.baseUrl}/businessLines`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(businessLineInfo, "BusinessLineInfo");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "BusinessLine");
    }
}
exports.BusinessLinesApi = BusinessLinesApi;
