// @adyen/api-library/lib/src/services/legalEntityManagement/documentsApi.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentsApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const models_1 = require("../../typings/legalEntityManagement/models");
class DocumentsApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://kyc-test.adyen.com/lem/v3";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Delete a document
    * @param id {@link string } The unique identifier of the document to be deleted.
    * @param requestOptions {@link IRequest.Options}
    */
    async deleteDocument(id, requestOptions) {
        const endpoint = `${this.baseUrl}/documents/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "DELETE" });
    }
    /**
    * @summary Get a document
    * @param id {@link string } The unique identifier of the document.
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Document }
    */
    async getDocument(id, requestOptions) {
        const endpoint = `${this.baseUrl}/documents/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return models_1.ObjectSerializer.deserialize(response, "Document");
    }
    /**
    * @summary Update a document
    * @param id {@link string } The unique identifier of the document to be updated.
    * @param xRequestedVerificationCode {@link string } Use the requested verification code 0_0001 to resolve any suberrors associated with the document. Requested verification codes can only be used in your test environment.
    * @param document {@link Document }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Document }
    */
    async updateDocument(id, document, requestOptions) {
        const endpoint = `${this.baseUrl}/documents/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(document, "Document");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return models_1.ObjectSerializer.deserialize(response, "Document");
    }
    /**
    * @summary Upload a document for verification checks
    * @param xRequestedVerificationCode {@link string } Use a suberror code as your requested verification code. You can include one code at a time in your request header. Requested verification codes can only be used in your test environment.
    * @param document {@link Document }
    * @param requestOptions {@link IRequest.Options}
    * @return {@link Document }
    */
    async uploadDocumentForVerificationChecks(document, requestOptions) {
        const endpoint = `${this.baseUrl}/documents`;
        const resource = new resource_1.default(this, endpoint);
        const request = models_1.ObjectSerializer.serialize(document, "Document");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return models_1.ObjectSerializer.deserialize(response, "Document");
    }
}
exports.DocumentsApi = DocumentsApi;
