// @adyen/api-library/lib/src/typings/payment/modificationResult.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModificationResult = void 0;
class ModificationResult {
    static getAttributeTypeMap() {
        return ModificationResult.attributeTypeMap;
    }
}
exports.ModificationResult = ModificationResult;
ModificationResult.discriminator = undefined;
ModificationResult.attributeTypeMap = [
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string"
    },
    {
        "name": "response",
        "baseName": "response",
        "type": "ModificationResult.ResponseEnum"
    }
];
(function (ModificationResult) {
    let ResponseEnum;
    (function (ResponseEnum) {
        ResponseEnum["CaptureReceived"] = "[capture-received]";
        ResponseEnum["CancelReceived"] = "[cancel-received]";
        ResponseEnum["RefundReceived"] = "[refund-received]";
        ResponseEnum["CancelOrRefundReceived"] = "[cancelOrRefund-received]";
        ResponseEnum["AdjustAuthorisationReceived"] = "[adjustAuthorisation-received]";
        ResponseEnum["DonationReceived"] = "[donation-received]";
        ResponseEnum["TechnicalCancelReceived"] = "[technical-cancel-received]";
        ResponseEnum["VoidPendingRefundReceived"] = "[voidPendingRefund-received]";
    })(ResponseEnum = ModificationResult.ResponseEnum || (ModificationResult.ResponseEnum = {}));
})(ModificationResult = exports.ModificationResult || (exports.ModificationResult = {}));
