// @adyen/api-library/lib/src/typings/payment/threeDS2Result.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreeDS2Result = void 0;
class ThreeDS2Result {
    static getAttributeTypeMap() {
        return ThreeDS2Result.attributeTypeMap;
    }
}
exports.ThreeDS2Result = ThreeDS2Result;
ThreeDS2Result.discriminator = undefined;
ThreeDS2Result.attributeTypeMap = [
    {
        "name": "authenticationValue",
        "baseName": "authenticationValue",
        "type": "string"
    },
    {
        "name": "cavvAlgorithm",
        "baseName": "cavvAlgorithm",
        "type": "string"
    },
    {
        "name": "challengeCancel",
        "baseName": "challengeCancel",
        "type": "ThreeDS2Result.ChallengeCancelEnum"
    },
    {
        "name": "dsTransID",
        "baseName": "dsTransID",
        "type": "string"
    },
    {
        "name": "eci",
        "baseName": "eci",
        "type": "string"
    },
    {
        "name": "exemptionIndicator",
        "baseName": "exemptionIndicator",
        "type": "ThreeDS2Result.ExemptionIndicatorEnum"
    },
    {
        "name": "messageVersion",
        "baseName": "messageVersion",
        "type": "string"
    },
    {
        "name": "riskScore",
        "baseName": "riskScore",
        "type": "string"
    },
    {
        "name": "threeDSRequestorChallengeInd",
        "baseName": "threeDSRequestorChallengeInd",
        "type": "ThreeDS2Result.ThreeDSRequestorChallengeIndEnum"
    },
    {
        "name": "threeDSServerTransID",
        "baseName": "threeDSServerTransID",
        "type": "string"
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "string"
    },
    {
        "name": "transStatus",
        "baseName": "transStatus",
        "type": "string"
    },
    {
        "name": "transStatusReason",
        "baseName": "transStatusReason",
        "type": "string"
    },
    {
        "name": "whiteListStatus",
        "baseName": "whiteListStatus",
        "type": "string"
    }
];
(function (ThreeDS2Result) {
    let ChallengeCancelEnum;
    (function (ChallengeCancelEnum) {
        ChallengeCancelEnum["_01"] = "01";
        ChallengeCancelEnum["_02"] = "02";
        ChallengeCancelEnum["_03"] = "03";
        ChallengeCancelEnum["_04"] = "04";
        ChallengeCancelEnum["_05"] = "05";
        ChallengeCancelEnum["_06"] = "06";
        ChallengeCancelEnum["_07"] = "07";
    })(ChallengeCancelEnum = ThreeDS2Result.ChallengeCancelEnum || (ThreeDS2Result.ChallengeCancelEnum = {}));
    let ExemptionIndicatorEnum;
    (function (ExemptionIndicatorEnum) {
        ExemptionIndicatorEnum["LowValue"] = "lowValue";
        ExemptionIndicatorEnum["SecureCorporate"] = "secureCorporate";
        ExemptionIndicatorEnum["TrustedBeneficiary"] = "trustedBeneficiary";
        ExemptionIndicatorEnum["TransactionRiskAnalysis"] = "transactionRiskAnalysis";
    })(ExemptionIndicatorEnum = ThreeDS2Result.ExemptionIndicatorEnum || (ThreeDS2Result.ExemptionIndicatorEnum = {}));
    let ThreeDSRequestorChallengeIndEnum;
    (function (ThreeDSRequestorChallengeIndEnum) {
        ThreeDSRequestorChallengeIndEnum["_01"] = "01";
        ThreeDSRequestorChallengeIndEnum["_02"] = "02";
        ThreeDSRequestorChallengeIndEnum["_03"] = "03";
        ThreeDSRequestorChallengeIndEnum["_04"] = "04";
        ThreeDSRequestorChallengeIndEnum["_05"] = "05";
        ThreeDSRequestorChallengeIndEnum["_06"] = "06";
    })(ThreeDSRequestorChallengeIndEnum = ThreeDS2Result.ThreeDSRequestorChallengeIndEnum || (ThreeDS2Result.ThreeDSRequestorChallengeIndEnum = {}));
})(ThreeDS2Result = exports.ThreeDS2Result || (exports.ThreeDS2Result = {}));
