// @adyen/api-library/lib/src/typings/transfers/additionalBankIdentification.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdditionalBankIdentification = void 0;
class AdditionalBankIdentification {
    static getAttributeTypeMap() {
        return AdditionalBankIdentification.attributeTypeMap;
    }
}
exports.AdditionalBankIdentification = AdditionalBankIdentification;
AdditionalBankIdentification.discriminator = undefined;
AdditionalBankIdentification.attributeTypeMap = [
    {
        "name": "code",
        "baseName": "code",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "AdditionalBankIdentification.TypeEnum"
    }
];
(function (AdditionalBankIdentification) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["GbSortCode"] = "gbSortCode";
        TypeEnum["UsRoutingNumber"] = "usRoutingNumber";
    })(TypeEnum = AdditionalBankIdentification.TypeEnum || (AdditionalBankIdentification.TypeEnum = {}));
})(AdditionalBankIdentification = exports.AdditionalBankIdentification || (exports.AdditionalBankIdentification = {}));
