// @adyen/api-library/lib/src/typings/transfers/partyIdentification2.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartyIdentification2 = void 0;
class PartyIdentification2 {
    static getAttributeTypeMap() {
        return PartyIdentification2.attributeTypeMap;
    }
}
exports.PartyIdentification2 = PartyIdentification2;
PartyIdentification2.discriminator = undefined;
PartyIdentification2.attributeTypeMap = [
    {
        "name": "address",
        "baseName": "address",
        "type": "Address2"
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "string"
    },
    {
        "name": "firstName",
        "baseName": "firstName",
        "type": "string"
    },
    {
        "name": "fullName",
        "baseName": "fullName",
        "type": "string"
    },
    {
        "name": "lastName",
        "baseName": "lastName",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PartyIdentification2.TypeEnum"
    }
];
(function (PartyIdentification2) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Individual"] = "individual";
        TypeEnum["Organization"] = "organization";
        TypeEnum["Unknown"] = "unknown";
    })(TypeEnum = PartyIdentification2.TypeEnum || (PartyIdentification2.TypeEnum = {}));
})(PartyIdentification2 = exports.PartyIdentification2 || (exports.PartyIdentification2 = {}));
