// @adyen/api-library/lib/src/typings/transfers/transferInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferInfo = void 0;
class TransferInfo {
    static getAttributeTypeMap() {
        return TransferInfo.attributeTypeMap;
    }
}
exports.TransferInfo = TransferInfo;
TransferInfo.discriminator = undefined;
TransferInfo.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "TransferInfo.CategoryEnum"
    },
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "CounterpartyInfoV3"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "paymentInstrumentId",
        "baseName": "paymentInstrumentId",
        "type": "string"
    },
    {
        "name": "priority",
        "baseName": "priority",
        "type": "TransferInfo.PriorityEnum"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "referenceForBeneficiary",
        "baseName": "referenceForBeneficiary",
        "type": "string"
    },
    {
        "name": "ultimateParty",
        "baseName": "ultimateParty",
        "type": "UltimatePartyIdentification"
    }
];
(function (TransferInfo) {
    let CategoryEnum;
    (function (CategoryEnum) {
        CategoryEnum["Bank"] = "bank";
        CategoryEnum["Internal"] = "internal";
        CategoryEnum["IssuedCard"] = "issuedCard";
        CategoryEnum["PlatformPayment"] = "platformPayment";
        CategoryEnum["Card"] = "card";
    })(CategoryEnum = TransferInfo.CategoryEnum || (TransferInfo.CategoryEnum = {}));
    let PriorityEnum;
    (function (PriorityEnum) {
        PriorityEnum["CrossBorder"] = "crossBorder";
        PriorityEnum["Fast"] = "fast";
        PriorityEnum["Instant"] = "instant";
        PriorityEnum["Internal"] = "internal";
        PriorityEnum["Regular"] = "regular";
        PriorityEnum["Wire"] = "wire";
    })(PriorityEnum = TransferInfo.PriorityEnum || (TransferInfo.PriorityEnum = {}));
})(TransferInfo = exports.TransferInfo || (exports.TransferInfo = {}));
