// @adyen/api-library/lib/src/typings/terminal/authenticationMethodType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticationMethodType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var AuthenticationMethodType;
(function (AuthenticationMethodType) {
    AuthenticationMethodType[AuthenticationMethodType["Bypass"] = 'Bypass'] = "Bypass";
    AuthenticationMethodType[AuthenticationMethodType["ManualVerification"] = 'ManualVerification'] = "ManualVerification";
    AuthenticationMethodType[AuthenticationMethodType["MerchantAuthentication"] = 'MerchantAuthentication'] = "MerchantAuthentication";
    AuthenticationMethodType[AuthenticationMethodType["OfflinePin"] = 'OfflinePIN'] = "OfflinePin";
    AuthenticationMethodType[AuthenticationMethodType["OnlinePin"] = 'OnlinePIN'] = "OnlinePin";
    AuthenticationMethodType[AuthenticationMethodType["PaperSignature"] = 'PaperSignature'] = "PaperSignature";
    AuthenticationMethodType[AuthenticationMethodType["SecureCertificate"] = 'SecureCertificate'] = "SecureCertificate";
    AuthenticationMethodType[AuthenticationMethodType["SecureNoCertificate"] = 'SecureNoCertificate'] = "SecureNoCertificate";
    AuthenticationMethodType[AuthenticationMethodType["SecuredChannel"] = 'SecuredChannel'] = "SecuredChannel";
    AuthenticationMethodType[AuthenticationMethodType["SignatureCapture"] = 'SignatureCapture'] = "SignatureCapture";
    AuthenticationMethodType[AuthenticationMethodType["UnknownMethod"] = 'UnknownMethod'] = "UnknownMethod";
})(AuthenticationMethodType = exports.AuthenticationMethodType || (exports.AuthenticationMethodType = {}));
