// @adyen/api-library/lib/src/typings/terminal/forceEntryModeType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ForceEntryModeType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var ForceEntryModeType;
(function (ForceEntryModeType) {
    ForceEntryModeType[ForceEntryModeType["CheckReader"] = 'CheckReader'] = "CheckReader";
    ForceEntryModeType[ForceEntryModeType["Contactless"] = 'Contactless'] = "Contactless";
    ForceEntryModeType[ForceEntryModeType["File"] = 'File'] = "File";
    ForceEntryModeType[ForceEntryModeType["Icc"] = 'ICC'] = "Icc";
    ForceEntryModeType[ForceEntryModeType["Keyed"] = 'Keyed'] = "Keyed";
    ForceEntryModeType[ForceEntryModeType["MagStripe"] = 'MagStripe'] = "MagStripe";
    ForceEntryModeType[ForceEntryModeType["Manual"] = 'Manual'] = "Manual";
    ForceEntryModeType[ForceEntryModeType["Rfid"] = 'RFID'] = "Rfid";
    ForceEntryModeType[ForceEntryModeType["Scanned"] = 'Scanned'] = "Scanned";
    ForceEntryModeType[ForceEntryModeType["SynchronousIcc"] = 'SynchronousICC'] = "SynchronousIcc";
    ForceEntryModeType[ForceEntryModeType["Tapped"] = 'Tapped'] = "Tapped";
})(ForceEntryModeType = exports.ForceEntryModeType || (exports.ForceEntryModeType = {}));
