// @adyen/api-library/lib/src/typings/terminal/infoQualifyType.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InfoQualifyType = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var InfoQualifyType;
(function (InfoQualifyType) {
    InfoQualifyType[InfoQualifyType["CustomerAssistance"] = 'CustomerAssistance'] = "CustomerAssistance";
    InfoQualifyType[InfoQualifyType["Display"] = 'Display'] = "Display";
    InfoQualifyType[InfoQualifyType["Document"] = 'Document'] = "Document";
    InfoQualifyType[InfoQualifyType["Error"] = 'Error'] = "Error";
    InfoQualifyType[InfoQualifyType["Input"] = 'Input'] = "Input";
    InfoQualifyType[InfoQualifyType["PoiReplication"] = 'POIReplication'] = "PoiReplication";
    InfoQualifyType[InfoQualifyType["Receipt"] = 'Receipt'] = "Receipt";
    InfoQualifyType[InfoQualifyType["Sound"] = 'Sound'] = "Sound";
    InfoQualifyType[InfoQualifyType["Status"] = 'Status'] = "Status";
    InfoQualifyType[InfoQualifyType["Voucher"] = 'Voucher'] = "Voucher";
})(InfoQualifyType = exports.InfoQualifyType || (exports.InfoQualifyType = {}));
