// @adyen/api-library/lib/src/typings/terminal/loyaltyAmount.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoyaltyAmount = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
class LoyaltyAmount {
    static getAttributeTypeMap() {
        return LoyaltyAmount.attributeTypeMap;
    }
}
exports.LoyaltyAmount = LoyaltyAmount;
LoyaltyAmount.discriminator = undefined;
LoyaltyAmount.attributeTypeMap = [
    {
        "name": "Currency",
        "baseName": "Currency",
        "type": "string"
    },
    {
        "name": "LoyaltyUnit",
        "baseName": "LoyaltyUnit",
        "type": "LoyaltyAmount.LoyaltyUnitEnum"
    },
    {
        "name": "Value",
        "baseName": "Value",
        "type": "number"
    }
];
(function (LoyaltyAmount) {
    let LoyaltyUnitEnum;
    (function (LoyaltyUnitEnum) {
        LoyaltyUnitEnum[LoyaltyUnitEnum["Monetary"] = 'Monetary'] = "Monetary";
        LoyaltyUnitEnum[LoyaltyUnitEnum["Point"] = 'Point'] = "Point";
    })(LoyaltyUnitEnum = LoyaltyAmount.LoyaltyUnitEnum || (LoyaltyAmount.LoyaltyUnitEnum = {}));
})(LoyaltyAmount = exports.LoyaltyAmount || (exports.LoyaltyAmount = {}));
