// @adyen/api-library/lib/src/typings/terminal/messageReference.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageReference = void 0;
/**
 * Terminal API
 * Definition of Terminal API Schema
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
class MessageReference {
    static getAttributeTypeMap() {
        return MessageReference.attributeTypeMap;
    }
}
exports.MessageReference = MessageReference;
MessageReference.discriminator = undefined;
MessageReference.attributeTypeMap = [
    {
        "name": "DeviceID",
        "baseName": "DeviceID",
        "type": "string"
    },
    {
        "name": "MessageCategory",
        "baseName": "MessageCategory",
        "type": "MessageReference.MessageCategoryEnum"
    },
    {
        "name": "POIID",
        "baseName": "POIID",
        "type": "string"
    },
    {
        "name": "SaleID",
        "baseName": "SaleID",
        "type": "string"
    },
    {
        "name": "ServiceID",
        "baseName": "ServiceID",
        "type": "string"
    }
];
(function (MessageReference) {
    let MessageCategoryEnum;
    (function (MessageCategoryEnum) {
        MessageCategoryEnum[MessageCategoryEnum["Abort"] = 'Abort'] = "Abort";
        MessageCategoryEnum[MessageCategoryEnum["Admin"] = 'Admin'] = "Admin";
        MessageCategoryEnum[MessageCategoryEnum["BalanceInquiry"] = 'BalanceInquiry'] = "BalanceInquiry";
        MessageCategoryEnum[MessageCategoryEnum["Batch"] = 'Batch'] = "Batch";
        MessageCategoryEnum[MessageCategoryEnum["CardAcquisition"] = 'CardAcquisition'] = "CardAcquisition";
        MessageCategoryEnum[MessageCategoryEnum["CardReaderApdu"] = 'CardReaderAPDU'] = "CardReaderApdu";
        MessageCategoryEnum[MessageCategoryEnum["CardReaderInit"] = 'CardReaderInit'] = "CardReaderInit";
        MessageCategoryEnum[MessageCategoryEnum["CardReaderPowerOff"] = 'CardReaderPowerOff'] = "CardReaderPowerOff";
        MessageCategoryEnum[MessageCategoryEnum["Diagnosis"] = 'Diagnosis'] = "Diagnosis";
        MessageCategoryEnum[MessageCategoryEnum["Display"] = 'Display'] = "Display";
        MessageCategoryEnum[MessageCategoryEnum["EnableService"] = 'EnableService'] = "EnableService";
        MessageCategoryEnum[MessageCategoryEnum["Event"] = 'Event'] = "Event";
        MessageCategoryEnum[MessageCategoryEnum["GetTotals"] = 'GetTotals'] = "GetTotals";
        MessageCategoryEnum[MessageCategoryEnum["Input"] = 'Input'] = "Input";
        MessageCategoryEnum[MessageCategoryEnum["InputUpdate"] = 'InputUpdate'] = "InputUpdate";
        MessageCategoryEnum[MessageCategoryEnum["Login"] = 'Login'] = "Login";
        MessageCategoryEnum[MessageCategoryEnum["Logout"] = 'Logout'] = "Logout";
        MessageCategoryEnum[MessageCategoryEnum["Loyalty"] = 'Loyalty'] = "Loyalty";
        MessageCategoryEnum[MessageCategoryEnum["Pin"] = 'PIN'] = "Pin";
        MessageCategoryEnum[MessageCategoryEnum["Payment"] = 'Payment'] = "Payment";
        MessageCategoryEnum[MessageCategoryEnum["Print"] = 'Print'] = "Print";
        MessageCategoryEnum[MessageCategoryEnum["Reconciliation"] = 'Reconciliation'] = "Reconciliation";
        MessageCategoryEnum[MessageCategoryEnum["Reversal"] = 'Reversal'] = "Reversal";
        MessageCategoryEnum[MessageCategoryEnum["Sound"] = 'Sound'] = "Sound";
        MessageCategoryEnum[MessageCategoryEnum["StoredValue"] = 'StoredValue'] = "StoredValue";
        MessageCategoryEnum[MessageCategoryEnum["TransactionStatus"] = 'TransactionStatus'] = "TransactionStatus";
        MessageCategoryEnum[MessageCategoryEnum["Transmit"] = 'Transmit'] = "Transmit";
    })(MessageCategoryEnum = MessageReference.MessageCategoryEnum || (MessageReference.MessageCategoryEnum = {}));
})(MessageReference = exports.MessageReference || (exports.MessageReference = {}));
