// @adyen/api-library/lib/src/typings/legalEntityManagement/businessLineInfoUpdate.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BusinessLineInfoUpdate = void 0;
class BusinessLineInfoUpdate {
    static getAttributeTypeMap() {
        return BusinessLineInfoUpdate.attributeTypeMap;
    }
}
exports.BusinessLineInfoUpdate = BusinessLineInfoUpdate;
BusinessLineInfoUpdate.discriminator = undefined;
BusinessLineInfoUpdate.attributeTypeMap = [
    {
        "name": "capability",
        "baseName": "capability",
        "type": "BusinessLineInfoUpdate.CapabilityEnum"
    },
    {
        "name": "industryCode",
        "baseName": "industryCode",
        "type": "string"
    },
    {
        "name": "legalEntityId",
        "baseName": "legalEntityId",
        "type": "string"
    },
    {
        "name": "salesChannels",
        "baseName": "salesChannels",
        "type": "Array<string>"
    },
    {
        "name": "service",
        "baseName": "service",
        "type": "BusinessLineInfoUpdate.ServiceEnum"
    },
    {
        "name": "sourceOfFunds",
        "baseName": "sourceOfFunds",
        "type": "SourceOfFunds"
    },
    {
        "name": "webData",
        "baseName": "webData",
        "type": "Array<WebData>"
    },
    {
        "name": "webDataExemption",
        "baseName": "webDataExemption",
        "type": "WebDataExemption"
    }
];
(function (BusinessLineInfoUpdate) {
    let CapabilityEnum;
    (function (CapabilityEnum) {
        CapabilityEnum["ReceivePayments"] = "receivePayments";
        CapabilityEnum["ReceiveFromPlatformPayments"] = "receiveFromPlatformPayments";
        CapabilityEnum["IssueBankAccount"] = "issueBankAccount";
    })(CapabilityEnum = BusinessLineInfoUpdate.CapabilityEnum || (BusinessLineInfoUpdate.CapabilityEnum = {}));
    let ServiceEnum;
    (function (ServiceEnum) {
        ServiceEnum["PaymentProcessing"] = "paymentProcessing";
        ServiceEnum["Issuing"] = "issuing";
        ServiceEnum["Banking"] = "banking";
    })(ServiceEnum = BusinessLineInfoUpdate.ServiceEnum || (BusinessLineInfoUpdate.ServiceEnum = {}));
})(BusinessLineInfoUpdate = exports.BusinessLineInfoUpdate || (exports.BusinessLineInfoUpdate = {}));
