// @adyen/api-library/lib/src/typings/legalEntityManagement/capabilityProblemEntity.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapabilityProblemEntity = void 0;
class CapabilityProblemEntity {
    static getAttributeTypeMap() {
        return CapabilityProblemEntity.attributeTypeMap;
    }
}
exports.CapabilityProblemEntity = CapabilityProblemEntity;
CapabilityProblemEntity.discriminator = undefined;
CapabilityProblemEntity.attributeTypeMap = [
    {
        "name": "documents",
        "baseName": "documents",
        "type": "Array<string>"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "owner",
        "baseName": "owner",
        "type": "CapabilityProblemEntityRecursive"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "CapabilityProblemEntity.TypeEnum"
    }
];
(function (CapabilityProblemEntity) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BankAccount"] = "BankAccount";
        TypeEnum["Document"] = "Document";
        TypeEnum["LegalEntity"] = "LegalEntity";
        TypeEnum["Product"] = "product";
    })(TypeEnum = CapabilityProblemEntity.TypeEnum || (CapabilityProblemEntity.TypeEnum = {}));
})(CapabilityProblemEntity = exports.CapabilityProblemEntity || (exports.CapabilityProblemEntity = {}));
