// @adyen/api-library/lib/src/typings/legalEntityManagement/taxReportingClassification.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxReportingClassification = void 0;
class TaxReportingClassification {
    static getAttributeTypeMap() {
        return TaxReportingClassification.attributeTypeMap;
    }
}
exports.TaxReportingClassification = TaxReportingClassification;
TaxReportingClassification.discriminator = undefined;
TaxReportingClassification.attributeTypeMap = [
    {
        "name": "businessType",
        "baseName": "businessType",
        "type": "TaxReportingClassification.BusinessTypeEnum"
    },
    {
        "name": "financialInstitutionNumber",
        "baseName": "financialInstitutionNumber",
        "type": "string"
    },
    {
        "name": "mainSourceOfIncome",
        "baseName": "mainSourceOfIncome",
        "type": "TaxReportingClassification.MainSourceOfIncomeEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TaxReportingClassification.TypeEnum"
    }
];
(function (TaxReportingClassification) {
    let BusinessTypeEnum;
    (function (BusinessTypeEnum) {
        BusinessTypeEnum["Other"] = "other";
        BusinessTypeEnum["ListedPublicCompany"] = "listedPublicCompany";
        BusinessTypeEnum["SubsidiaryOfListedPublicCompany"] = "subsidiaryOfListedPublicCompany";
        BusinessTypeEnum["GovernmentalOrganization"] = "governmentalOrganization";
        BusinessTypeEnum["InternationalOrganization"] = "internationalOrganization";
        BusinessTypeEnum["FinancialInstitution"] = "financialInstitution.";
    })(BusinessTypeEnum = TaxReportingClassification.BusinessTypeEnum || (TaxReportingClassification.BusinessTypeEnum = {}));
    let MainSourceOfIncomeEnum;
    (function (MainSourceOfIncomeEnum) {
        MainSourceOfIncomeEnum["BusinessOperation"] = "businessOperation";
        MainSourceOfIncomeEnum["RealEstateSales"] = "realEstateSales";
        MainSourceOfIncomeEnum["InvestmentInterestOrRoyalty"] = "investmentInterestOrRoyalty";
        MainSourceOfIncomeEnum["PropertyRental"] = "propertyRental";
        MainSourceOfIncomeEnum["Other"] = "other";
    })(MainSourceOfIncomeEnum = TaxReportingClassification.MainSourceOfIncomeEnum || (TaxReportingClassification.MainSourceOfIncomeEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["NonFinancialNonReportable"] = "nonFinancialNonReportable";
        TypeEnum["FinancialNonReportable"] = "financialNonReportable";
        TypeEnum["NonFinancialActive"] = "nonFinancialActive";
        TypeEnum["NonFinancialPassive"] = "nonFinancialPassive";
    })(TypeEnum = TaxReportingClassification.TypeEnum || (TaxReportingClassification.TypeEnum = {}));
})(TaxReportingClassification = exports.TaxReportingClassification || (exports.TaxReportingClassification = {}));
