// @adyen/api-library/lib/src/typings/managementWebhooks/capabilityProblemEntityRecursive.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapabilityProblemEntityRecursive = void 0;
class CapabilityProblemEntityRecursive {
    static getAttributeTypeMap() {
        return CapabilityProblemEntityRecursive.attributeTypeMap;
    }
}
exports.CapabilityProblemEntityRecursive = CapabilityProblemEntityRecursive;
CapabilityProblemEntityRecursive.discriminator = undefined;
CapabilityProblemEntityRecursive.attributeTypeMap = [
    {
        "name": "documents",
        "baseName": "documents",
        "type": "Array<string>"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "CapabilityProblemEntityRecursive.TypeEnum"
    }
];
(function (CapabilityProblemEntityRecursive) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BankAccount"] = "BankAccount";
        TypeEnum["Document"] = "Document";
        TypeEnum["LegalEntity"] = "LegalEntity";
    })(TypeEnum = CapabilityProblemEntityRecursive.TypeEnum || (CapabilityProblemEntityRecursive.TypeEnum = {}));
})(CapabilityProblemEntityRecursive = exports.CapabilityProblemEntityRecursive || (exports.CapabilityProblemEntityRecursive = {}));
